

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Reclassification to Current Clinical Guidelines &#8212; MethylScoreAML</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'data-processing/reclassification_strategy';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Epigenome Wide Association Study" href="../models/2_Univariate_CoxPH-EWAS_v3.html" />
    <link rel="prev" title="Validation Dataset" href="test_dataset.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="../index.html">
  
  
  
  
  
    <p class="title logo__title">MethylScoreAML</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../index.html">
                    The DNA methylome refines clinical diagnosis and prognosis of Acute Myeloid Leukemia
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Data Processing</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="discovery_dataset.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="test_dataset.html">Validation Dataset</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Reclassification to Current Clinical Guidelines</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">AML Prognostic Model</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../models/2_Univariate_CoxPH-EWAS_v3.html">Epigenome Wide Association Study</a></li>
<li class="toctree-l1"><a class="reference internal" href="../models/3.1_MethylScoreAML_Px_Development_v5.html">MethylScoreAML Px Development</a></li>
<li class="toctree-l1"><a class="reference internal" href="../models/3.2_MethylScoreAML_Px_Validation_Peds_v2.html">MethylScoreAML Px Validation</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">AML Diagnostic Model</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../models/dx-score_new.html">Dx score</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Nanopore Sequencing Testing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../tests/8_Nanopore_Sample.html">Applying models to Nanopore data</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Benchmarks</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-n_components-pacmap.html">Dx score - various components</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-pedsAML-only.html">Dx score - pedsAML only</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-with-looser-filtering.html">Dx score - less strict pre-processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-without-ALL-relapse.html">Dx score - without relapse and ALL</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-without-relapse.html">Dx score - without relapse</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/data-processing/reclassification_strategy.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Reclassification to Current Clinical Guidelines</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#guidelines">Guidelines</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#who-2022-aml">WHO 2022 AML</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-process-discovery-clinical-data">Load and process discovery clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#reclassification-strategy-for-who-2022">Reclassification strategy for WHO 2022</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#functions">Functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#execution">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#categorize-eln-2022-hematopoietic-entity-age-and-trisomy-8">Categorize ELN 2022, hematopoietic entity, age, and trisomy 8</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-classification-to-test-dataset">Apply Classification to Test Dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#save-clinical-datasets">Save Clinical Datasets</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="reclassification-to-current-clinical-guidelines">
<h1><a class="toc-backref" href="#id3" role="doc-backlink">Reclassification to Current Clinical Guidelines</a><a class="headerlink" href="#reclassification-to-current-clinical-guidelines" title="Permalink to this heading">#</a></h1>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#reclassification-to-current-clinical-guidelines" id="id3">Reclassification to Current Clinical Guidelines</a></p>
<ul>
<li><p><a class="reference internal" href="#guidelines" id="id4">Guidelines</a></p>
<ul>
<li><p><a class="reference internal" href="#who-2022-aml" id="id5">WHO 2022 AML</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#load-and-process-discovery-clinical-data" id="id6">Load and process discovery clinical data</a></p></li>
<li><p><a class="reference internal" href="#reclassification-strategy-for-who-2022" id="id7">Reclassification strategy for WHO 2022</a></p>
<ul>
<li><p><a class="reference internal" href="#functions" id="id8">Functions</a></p></li>
<li><p><a class="reference internal" href="#execution" id="id9">Execution</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#categorize-eln-2022-hematopoietic-entity-age-and-trisomy-8" id="id10">Categorize ELN 2022, hematopoietic entity, age, and trisomy 8</a></p>
<ul>
<li><p><a class="reference internal" href="#id1" id="id11">Functions</a></p></li>
<li><p><a class="reference internal" href="#id2" id="id12">Execution</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#apply-classification-to-test-dataset" id="id13">Apply Classification to Test Dataset</a></p></li>
<li><p><a class="reference internal" href="#save-clinical-datasets" id="id14">Save Clinical Datasets</a></p></li>
</ul>
</li>
</ul>
</nav>
<p>Current thinking:</p>
<p>14 WHO subtypes of AML
+1 otherwise-normal control</p>
<p>May need to see how to think of mds-like and secondary neoplasms</p>
<section id="guidelines">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">Guidelines</a><a class="headerlink" href="#guidelines" title="Permalink to this heading">#</a></h2>
<section id="who-2022-aml">
<h3><a class="toc-backref" href="#id5" role="doc-backlink">WHO 2022 AML</a><a class="headerlink" href="#who-2022-aml" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">itables</span>

<span class="kn">from</span> <span class="nn">itables</span> <span class="kn">import</span> <span class="n">show</span>

<span class="c1"># Read the CSV file</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../data/who2022_aml_classification.csv&#39;</span><span class="p">)</span>

<span class="c1"># Display the dataframe using itables</span>
<span class="n">itables</span><span class="o">.</span><span class="n">show</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><style>.itables table td {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables table th {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables thead input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}

.itables tfoot input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}
</style>
<div class="itables">
<table id="a261abb0-fb8a-4975-a2aa-cf621bebcd08" class="display nowrap"style="table-layout:auto;width:auto;margin:auto;caption-side:bottom"><thead>
    <tr style="text-align: right;">
      
      <th>Subtype</th>
      <th>Frequency in paediatric AML</th>
      <th>Morphology</th>
      <th>Immunophenotype</th>
      <th>Common co-occurring genetic aberrations</th>
      <th>Prognosis</th>
    </tr>
  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css">
<script type="module">
    // Import jquery and DataTable
    import 'https://code.jquery.com/jquery-3.6.0.min.js';
    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';
    dt($);

    // Define the table data
    const data = [["AML with RUNX1::RUNXT1 fusion", "13\u201319% of cases", "Neutrophilic maturation, Auer rods; often increased basophils and/or mast cells", "Positive: CD34, HLA-DR, KIT (CD117), CD99, myeloperoxidase\\nUsually positive but can be heterogeneous: CD13, CD33 and CD38, CD19, PAX5, cytoplasmic CD79a, CD56\\nNegative: CD7", "Mutations in KIT, FLT3, CSF3R, EZH2, ASXL1, ASXL2, CBL, WT1, cohesion genes (SMC1A, SMC3, RAD21), NRAS, KRAS", "Good"], ["AML with CBFB::MYH11 fusion", "7\u201312% of cases", "Myelomonocytic; often increased eosinophils at all maturation stages", "Often multiple blast populations with signs of differentiation towards both granulocytic and monocytic lineages; may express CD2 and/or NG2 in a small subpopulation of cells", "Mutations in KIT, KRAS, NRAS, CBL, FLT3, WT1", "Good"], ["APL with PML::RARA fusion", "5\u201310% of cases but wide geographical variation (2\u201350%)", "Exclusively FAB M3; may be hypogranular", "Positive: CD33, often KIT (CD117), sometimes CD2\\nNegative: CD34 (except hypogranular variant), HLA-DR\\nOften high autofluorescence", "Mutations in FLT3", "Good (but treated differently)"], ["AML with KMT2A rearrangement", "20% of cases", "All FAB types except M3 and M6, most commonly M5", "Positive: CD33, CD65, CD4, HLA-DR (in monocytic cases), CD15, NG2; platelet antigens (in megakaryocytic cases)\\nNegative/low: CD13, CD34, CD14", "Mutations in the tyrosine kinase, PI3K, and RAS pathways; FLT3-TKD, NRAS, KRAS", "Depends on fusion partner"], ["AML with DEK::NUP214 fusion", "1\u20132% of cases", "All FAB types except M3 and M7, most commonly M2 and M4; Auer rods; basophilia in blood or bone marrow; erythroid and granulocytic dysplasia are common", "Positive: myeloperoxidase, CD13, CD33, CD38, CD123, HLA-DR; often CD34, KIT (CD117), CD15; sometimes CD64, TdT", "FLT3 ITD in two thirds of cases; complex karyotype", "Poor"], ["AML with MECOM rearrangement", "< 1% of cases", "Commonly FAB M1, M4, and M7; multilineage dysplasia is common, especially in megakaryopoiesis", "Positive: CD34, CD33, CD13, KIT (CD117), HLA-DR, CD7\\nNegative: lymphoid markers other than CD7", "NaN", "Poor"], ["AML with ETV6 fusion", "1% of all cases", "All FAB types, most commonly M0 and M1", "For MNX1::ETV6:\\nPositive: CD34, HLA-DR, often KIT (CD117), T-cell markers CD7 and CD4", "Almost always +19, sometimes +8", "Poor, but may be protocol dependent"], ["AML with KAT6A::CREBBP fusion", "0.5% of cases", "FAB M5", "Positive: HLA-DR, CD15, myeloperoxidase, CD13, CD33, sometimes CD56, CD14\\nNegative: CD34, KIT (CD117)", "NaN", "OS similar to paediatric AML in general"], ["AML with RBM15::MRTFA fusion", "0.8% of all cases", "Exclusively FAB M7", "Positive: > 1 platelet antigen, some CD34 and/or HLA-DR\\nNegative: CD45", "del(1p), +4, +7, +8, +9, +10, +14, +15, +17, +18, +20, +21(21p), complex hyperdiploid karyotype", "Lower CR rate but similar EFS/OS vs other AMKLs"], ["AML with CBFA2T3::GLIS2 fusion", "3% of all cases", "All FAB types, most commonly M7", "> 50% show the RAM phenotype; overexpression of CD56 and underexpression of HLA-DR and CD38", "Mutations in FLT3, GATA1, JAK/STAT pathway", "Poor"], ["AML with NUP98 rearrangement", "5\u20139% of cases", "NUP98::NSD1: often FAB M4 and M5\\nNUP98::KDM5A: mostly FAB M7 but also M5 and M6", "NaN", "For NUP98::NSD1: FLT3 ITD in > 80%, mutation in WT1 in 50%", "Poor"], ["AML with FUS::ERG fusion", "0.3\u20130.5% of cases", "All FAB types except M6; Auer rods", "NaN", "+8, +10, complex karyotype", "Poor"], ["AML with NPM1 mutation", "6\u20138% of cases", "FAB M1, M2, M4, and M5", "Positive: CD33, sometimes KIT (CD117), often CD36, CD64\\nNegative: CD34, sometimes HLA-DR\\nOften weak CD13", "FLT3 ITD, mutations in NRAS, WT1", "Good, also when FLT3 ITD is present"], ["AML with CEBPA mutation", "5\u20136% of cases", "Most commonly FAB M1 and M2", "Positive: CD34, HLA-DR, often CD7\\nNegative: other lymphoid antigens", "Mutations in CEBPA-TAD, CSF3R, GATA2, RAS genes, FLT3", "Good, except for CSF3R-mutated cases"]];

    // Define the dt_args
    let dt_args = {"order": []};
    dt_args["data"] = data;

    $(document).ready(function () {
        
        $('#a261abb0-fb8a-4975-a2aa-cf621bebcd08').DataTable(dt_args);
    });
</script>
</div>
</div></div>
</div>
<div class="tip admonition">
<p class="admonition-title">Source</p>
<p>Hasle H, Meshinchi S, Fogelstrand L, Alaggio R, et al. Acute myeloid leukaemias (AMLs) with defining genetic abnormalities. In: WHO Classification of Tumours Editorial Board. Paediatric tumours [Internet]. Lyon (France): International Agency for Research on Cancer; 2022 [cited 2024 Jan 1]. (WHO classification of tumours series, 5th ed.; vol. 7). Available from: <a class="reference external" href="https://tumourclassification.iarc.who.int/chapters/44">https://tumourclassification.iarc.who.int/chapters/44</a>.</p>
</div>
</section>
</section>
<section id="load-and-process-discovery-clinical-data">
<h2><a class="toc-backref" href="#id6" role="doc-backlink">Load and process discovery clinical data</a><a class="headerlink" href="#load-and-process-discovery-clinical-data" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import functions to clean up clinical data</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;../&#39;</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">source.clinical_data_cleanup_functions</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Call functions to merge, index and clean clinical data files</span>
<span class="n">labels_0531</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_0531</span><span class="p">())</span>
<span class="n">labels_1031</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_1031</span><span class="p">())</span>
<span class="n">labels_aml05</span>        <span class="o">=</span> <span class="n">clean_aml05</span>     <span class="p">(</span><span class="n">merge_index_aml05</span><span class="p">())</span>
<span class="n">labels_beataml</span>      <span class="o">=</span> <span class="n">clean_beataml</span>   <span class="p">(</span><span class="n">merge_index_beataml</span><span class="p">())</span>
<span class="n">labels_amltcga</span>      <span class="o">=</span> <span class="n">clean_amltcga</span>   <span class="p">(</span><span class="n">merge_index_amltcga</span><span class="p">())</span>
<span class="n">labels_nordic_all</span>   <span class="o">=</span> <span class="n">clean_nordic_all</span><span class="p">(</span><span class="n">merge_index_nordic_all</span><span class="p">())</span>
<span class="n">labels_mds_taml</span>     <span class="o">=</span> <span class="n">clean_mds_taml</span>  <span class="p">(</span><span class="n">merge_index_mds_taml</span><span class="p">())</span>
<span class="n">labels_all_graal</span>    <span class="o">=</span> <span class="n">clean_all_graal</span> <span class="p">(</span><span class="n">merge_index_all_graal</span><span class="p">())</span>
<span class="n">labels_target_all</span>   <span class="o">=</span> <span class="n">clean_target_all</span><span class="p">(</span><span class="n">merge_index_target_all</span><span class="p">())</span>

<span class="c1"># Combine all clinical data labels into one dataframe</span>
<span class="n">labels_combined</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">labels_aml05</span><span class="p">,</span> <span class="n">labels_beataml</span><span class="p">,</span>
                        <span class="n">labels_0531</span><span class="p">,</span> <span class="n">labels_amltcga</span><span class="p">,</span> <span class="n">labels_1031</span><span class="p">,</span>
                        <span class="n">labels_nordic_all</span><span class="p">,</span> <span class="n">labels_mds_taml</span><span class="p">,</span>
                        <span class="n">labels_all_graal</span><span class="p">,</span><span class="n">labels_target_all</span><span class="p">],</span>
                        <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;outer&#39;</span><span class="p">)</span>


<span class="c1"># Redefine output path (for troubleshooting purposes in case only this cell is run)</span>
<span class="c1"># mount = &#39;../../../&#39;</span>
<span class="n">mount</span> <span class="o">=</span> <span class="s1">&#39;/mnt/d/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;MethylScore/Intermediate_Files/&#39;</span>

<span class="c1"># Read df</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;3308samples_333059cpgs_withbatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span>

<span class="c1"># Remove samples that are not in the methyl dataset</span>
<span class="n">df_labels</span> <span class="o">=</span> <span class="n">labels_combined</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">labels_combined</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The clinical data has been indexed and cleaned.</span><span class="se">\n\</span>
<span class="s1">Exclusion of samples may be applied depending on the analysis.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The clinical data has been indexed and cleaned.
Exclusion of samples may be applied depending on the analysis.
</pre></div>
</div>
</div>
</div>
</section>
<section id="reclassification-strategy-for-who-2022">
<h2><a class="toc-backref" href="#id7" role="doc-backlink">Reclassification strategy for WHO 2022</a><a class="headerlink" href="#reclassification-strategy-for-who-2022" title="Permalink to this heading">#</a></h2>
<section id="functions">
<h3><a class="toc-backref" href="#id8" role="doc-backlink">Functions</a><a class="headerlink" href="#functions" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">classify_controls</span><span class="p">(</span><span class="n">normal_samples</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Bone Marrow Normal&#39;</span>  <span class="p">:</span> <span class="s1">&#39;Otherwise-Normal Control&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Blood Derived Normal&#39;</span><span class="p">:</span> <span class="s1">&#39;Otherwise-Normal Control&#39;</span><span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">normal_samples</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_fusion</span><span class="p">(</span><span class="n">gene_fusion</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;RUNX1-RUNX1T1&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span><span class="p">,</span>
    <span class="s1">&#39;CBFB-MYH11&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span><span class="p">,</span>
    <span class="s1">&#39;KMT2A&#39;</span><span class="p">:</span>         <span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>
    <span class="s1">&#39;MLL&#39;</span><span class="p">:</span>           <span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>
    <span class="s1">&#39;PML-RARA&#39;</span><span class="p">:</span>      <span class="s1">&#39;APL with t(15;17); PML::RARA&#39;</span><span class="p">,</span>
    <span class="s1">&#39;DEK-NUP214&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span><span class="p">,</span>
    <span class="s1">&#39;MECOM&#39;</span><span class="p">:</span>         <span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span><span class="p">,</span>
    <span class="s1">&#39;ETV6&#39;</span><span class="p">:</span>          <span class="s1">&#39;AML with ETV6 fusion&#39;</span><span class="p">,</span>
    <span class="s1">&#39;NPM1&#39;</span><span class="p">:</span>          <span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
    <span class="s1">&#39;RBM15-MKL1&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(1;22); RBM15::MKL1&#39;</span><span class="p">,</span>
    <span class="s1">&#39;NUP98&#39;</span><span class="p">:</span>         <span class="s1">&#39;AML with NUP98-fusion&#39;</span><span class="p">,</span>
    <span class="s1">&#39;KAT6A-CREBBP&#39;</span><span class="p">:</span>  <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span><span class="p">,</span>
    <span class="s1">&#39;FUS-ERG&#39;</span><span class="p">:</span>       <span class="s1">&#39;AML with t(16;21); FUS::ERG&#39;</span><span class="p">,</span>
    <span class="s1">&#39;CBFA2T3-GLIS2&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with CBFA2T3::GLIS2&#39;</span><span class="p">,</span>
    <span class="s1">&#39;BCR-ABL1&#39;</span><span class="p">:</span>       <span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>

    <span class="c1"># Other uncharacterized abdnormalities present in the dataset but not specifically in guidelines</span>

    <span class="c1"># &#39;RUNX1-CBFA2T3&#39;: &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;PRDM16-RPN1&#39;:   &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;RBM15-MRTFA&#39;:   &#39;AML with t(1;22); RBM15::MKL1&#39;,</span>
    <span class="c1"># &#39;CBFA2T3-GLIS3&#39;: &#39;AML with CBFA2T3::GLIS2&#39;,</span>
    <span class="c1"># &#39;PSIP1-NUP214&#39;:  &#39;AML with t(6;9); DEK::NUP214&#39;,</span>
    <span class="c1"># &#39;XPO1-TNRC18&#39;:   &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;HNRNPH1-ERG&#39;:   &#39;AML with t(16;21); FUS::ERG&#39;,</span>
    <span class="c1"># &#39;NIPBL-HOXB9&#39;:   &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;SET-NUP214&#39;:    &#39;AML with t(6;9); DEK::NUP214&#39;, </span>
    <span class="c1"># &#39;FLI1-IFIT2&#39;:    &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;TCF4-ZEB2&#39;:     &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;MBTD1-ZMYND11&#39;: &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;FOSB-KLF6&#39;:     &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;SFPQ-ZFP36L2&#39;:  &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;RUNX1-LINC00478&#39;:&#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;RUNX1-EVX1&#39;:     &#39;AML NOS&#39;,  </span>
    <span class="c1"># &#39;PSPC1-ZFP36L1&#39;:  &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;EWSR1-FEV&#39;:      &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;STAG2-AFF2&#39;:     &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;MYB-GATA1&#39;:      &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;RUNX1-ZFPM2&#39;:    &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;RUNX1-CBFA2T2&#39;:  &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;PIM3-BRD1&#39;:      &#39;AML NOS&#39;,</span>
    <span class="s1">&#39;KAT6A-EP300&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span><span class="p">,</span>
    <span class="c1"># &#39;DOT1L-RPS15&#39;:    &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;FUS-FEV&#39;:        &#39;AML with t(16;21); FUS::ERG&#39;,</span>
    <span class="s1">&#39;KAT6A-NCOA2&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span><span class="p">,</span>
    <span class="c1"># &#39;JARID2-PTP4A1&#39;:  &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;FUS-FLI1&#39;:       &#39;AML with t(16;21); FUS::ERG&#39;,</span>
    <span class="p">}</span>    
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">gene_fusion</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_cebpa</span><span class="p">(</span><span class="n">cebpa_mutation</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Yes&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Positive&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">cebpa_mutation</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_npm</span><span class="p">(</span><span class="n">npm_mutation</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Yes&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Positive&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated NPM1&#39;</span>
    <span class="p">}</span>

    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">npm_mutation</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_karyotype</span><span class="p">(</span><span class="n">structural_variation</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="c1"># &#39;t(8;16)&#39;: &#39;AML with t(8;16); KAT6A::CREBBP&#39;,</span>
        <span class="c1"># &#39;t(16;21)&#39;: &#39;AML with t(16;21); FUS::ERG&#39;,</span>
        <span class="c1"># &#39;t(6;9)&#39;: &#39;AML with t(6;9); DEK::NUP214&#39;,</span>
        <span class="c1"># &#39;t(1;22)&#39;: &#39;AML with t(1;22); RBM15::MKL1&#39;,</span>
        <span class="c1"># &#39;inv(3)&#39;: &#39;AML with inv(3); t(3;3); MECOM-r&#39;,</span>
        <span class="c1"># &#39;t(3;3)&#39;: &#39;AML with inv(3); t(3;3); MECOM-r&#39;,</span>
        <span class="c1"># &#39;t(6;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;,</span>
        <span class="c1"># &#39;t(1;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;,</span>
        <span class="c1"># &#39;t(4;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;,</span>
        <span class="c1"># &#39;ins(6;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;,</span>

        <span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">structural_variation</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>
            
<span class="k">def</span> <span class="nf">classify_annotated_diagnosis</span><span class="p">(</span><span class="n">diagnosis</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;mutated NPM1&#39;</span>                                                  <span class="p">:</span> <span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;mutated CEBPA&#39;</span>                                                 <span class="p">:</span> <span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">,</span>
        <span class="s1">&#39;myelodysplasia-related changes&#39;</span>                                <span class="p">:</span> <span class="s1">&#39;MDS-related; secondary myeloid&#39;</span><span class="p">,</span>
        <span class="s2">&quot;AML with mutated NPM1&quot;</span>                                         <span class="p">:</span> <span class="s2">&quot;AML with mutated NPM1&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with myelodysplasia-related changes&quot;</span>                       <span class="p">:</span> <span class="s2">&quot;MDS-related; secondary myeloid&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22); CBFB-MYH11&quot;</span> <span class="p">:</span> <span class="s2">&quot;AML with inv(16); t(16;16); CBFB::MYH11&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with mutated CEBPA&quot;</span>                                        <span class="p">:</span> <span class="s2">&quot;AML with mutated CEBPA&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Therapy-related myeloid neoplasms&quot;</span>                             <span class="p">:</span> <span class="s2">&quot;MDS-related; secondary myeloid&quot;</span><span class="p">,</span>
        <span class="s2">&quot;PML-RARA&quot;</span>                                                      <span class="p">:</span> <span class="s2">&quot;APL with t(15;17); PML::RARA&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with t(9;11)(p22;q23); MLLT3-MLL&quot;</span>                          <span class="p">:</span> <span class="s2">&quot;AML with t(v;11q23); KMT2A-r&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with t(8;21)(q22;q22.1); RUNX1-RUNX1T1&quot;</span>                    <span class="p">:</span> <span class="s2">&quot;AML with t(8;21); RUNX1::RUNX1T1&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with inv(3)(q21q26.2) or t(3;3)(q21;q26.2); RPN1-EVI1&quot;</span>     <span class="p">:</span> <span class="s2">&quot;AML with inv(3); t(3;3); MECOM-r&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Mixed phenotype acute leukaemia, T/myeloid&quot;</span>                    <span class="p">:</span> <span class="s2">&quot;MPAL T-Lymphoblastic/Myeloid&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Myeloid leukaemia associated with Down syndrome&quot;</span>               <span class="p">:</span> <span class="s2">&quot;AML with Down syndrome&quot;</span><span class="p">,</span>
        <span class="s1">&#39;AML with CBFB-MYH11&#39;</span>                                           <span class="p">:</span> <span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span><span class="p">,</span>
        <span class="s1">&#39;AML with RUNX1-RUNX1T1&#39;</span>                                        <span class="p">:</span> <span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;AML with BCR-ABL1&#39;</span>                                             <span class="p">:</span> <span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;MDS-related or secondary myeloid neoplasms&#39;</span>                    <span class="p">:</span> <span class="s1">&#39;MDS-related; secondary myeloid&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with t(8;21) (q22;q22) [AML1/ETO]&#39;</span>            <span class="p">:</span> <span class="s2">&quot;AML with t(8;21); RUNX1::RUNX1T1&quot;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with 11q23/MLL Abnormalities&#39;</span>                 <span class="p">:</span> <span class="s2">&quot;AML with t(v;11q23); KMT2A-r&quot;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with t(6;9)(p23;q34) DEK-NUP214&#39;</span>              <span class="p">:</span> <span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with inv(16) or t(16;16) [CBFbeta/MYH11]&#39;</span>     <span class="p">:</span> <span class="s2">&quot;AML with inv(16); t(16;16); CBFB::MYH11&quot;</span><span class="p">,</span>
        
        <span class="c1"># ALL</span>
        <span class="s1">&#39;T-ALL&#39;</span>                                                         <span class="p">:</span><span class="s1">&#39;T-ALL NOS&#39;</span><span class="p">,</span>
        <span class="s1">&#39;BCP-ALL HeH&#39;</span>                                                   <span class="p">:</span><span class="s1">&#39;B-ALL with hyperdiploidy&#39;</span><span class="p">,</span>
        <span class="s1">&#39;BCP-ALL t(12;21)&#39;</span>                                              <span class="p">:</span><span class="s1">&#39;B-ALL with t(12;21); ETV6::RUNX1&#39;</span><span class="p">,</span>         
        <span class="s1">&#39;BCP-ALL 11q23/MLL&#39;</span>                                             <span class="p">:</span><span class="s1">&#39;B-ALL with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL t(1;19)&#39;</span>                                               <span class="p">:</span><span class="s1">&#39;B-ALL with t(1;19); TCF3::PBX1&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL dic(9;20)&#39;</span>                                             <span class="p">:</span><span class="s1">&#39;B-ALL dic(9;20)&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL t(9;22)&#39;</span>                                               <span class="p">:</span><span class="s1">&#39;B-ALL with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL iAMP21&#39;</span>                                                <span class="p">:</span><span class="s1">&#39;B-ALL with iAMP21&#39;</span><span class="p">,</span>        
        <span class="s1">&#39;BCP-ALL &lt;45chr&#39;</span>                                                <span class="p">:</span><span class="s1">&#39;B-ALL with hypodiploidy&#39;</span><span class="p">,</span>        
        <span class="s1">&#39;BCP-ALL &gt;67chr&#39;</span>                                                <span class="p">:</span><span class="s1">&#39;B-ALL with hyperdiploidy&#39;</span><span class="p">,</span>

        <span class="c1"># MPAL</span>
        <span class="s1">&#39;MPAL with MLL&#39;</span>                                                 <span class="p">:</span><span class="s1">&#39;MPAL with t(v;11q23.3)/KMT2A-r&#39;</span><span class="p">,</span>
        <span class="s1">&#39;MPAL with T/M&#39;</span>                                                 <span class="p">:</span><span class="s1">&#39;MPAL T-Lymphoblastic/Myeloid&#39;</span><span class="p">,</span>
        <span class="s1">&#39;MPAL with B/M&#39;</span>                                                 <span class="p">:</span><span class="s1">&#39;MPAL B-Lymphoblastic/Myeloid&#39;</span><span class="p">,</span>

        <span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">diagnosis</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">process_labels_who22</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
    <span class="c1"># Merge processing functions with source columns into a single dictionary</span>
    <span class="n">column_mappings</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Annotated Controls&#39;</span>            <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">,</span> <span class="n">classify_controls</span><span class="p">),</span>
        <span class="s1">&#39;Annotated Gene Fusion&#39;</span>         <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Gene Fusion&#39;</span><span class="p">,</span> <span class="n">classify_fusion</span><span class="p">),</span>
        <span class="s1">&#39;Annotated CEBPA&#39;</span>               <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;CEBPA mutation&#39;</span><span class="p">,</span> <span class="n">classify_cebpa</span><span class="p">),</span>
        <span class="s1">&#39;Annotated NPM1&#39;</span>                <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;NPM mutation&#39;</span><span class="p">,</span> <span class="n">classify_npm</span><span class="p">),</span>
        <span class="s1">&#39;Annotated Karyotype&#39;</span>           <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Karyotype&#39;</span><span class="p">,</span> <span class="n">classify_karyotype</span><span class="p">),</span>
        <span class="s1">&#39;Annotated Dx at Acquisition&#39;</span>   <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Dx at Acquisition&#39;</span><span class="p">,</span> <span class="n">classify_annotated_diagnosis</span><span class="p">),</span>
    <span class="p">}</span>

    <span class="c1"># Looping through the mapping to process and create new columns</span>
    <span class="k">for</span> <span class="n">new_col</span><span class="p">,</span> <span class="p">(</span><span class="n">old_col</span><span class="p">,</span> <span class="n">processing_func</span><span class="p">)</span> <span class="ow">in</span> <span class="n">column_mappings</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="c1"># Apply the corresponding classification function to each column</span>
        <span class="n">df</span><span class="p">[</span><span class="n">new_col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">old_col</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">processing_func</span><span class="p">)</span>

    <span class="c1"># Combining processed columns into a single column, separating values with commas</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Combined Diagnoses&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="n">column_mappings</span><span class="o">.</span><span class="n">keys</span><span class="p">())]</span>\
        <span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">dropna</span><span class="p">()),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">)</span>

    <span class="c1"># Extracting the first diagnosis from the combined diagnoses as the final diagnosis</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Combined Diagnoses&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">str</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>

    <span class="c1"># Dropping intermediate columns used for processing</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">column_mappings</span><span class="o">.</span><span class="n">keys</span><span class="p">()),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">df</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="execution">
<h3><a class="toc-backref" href="#id9" role="doc-backlink">Execution</a><a class="headerlink" href="#execution" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Add age categorization and main disease classification to the clinical data</span>
<span class="n">df_labels2</span> <span class="o">=</span> <span class="n">process_labels_who22</span><span class="p">(</span><span class="n">df_labels</span><span class="p">)</span>
<span class="n">df_labels2</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>WHO 2022 Diagnosis
NaN                                        866
AML with t(v;11q23); KMT2A-r               317
Otherwise-Normal Control                   251
T-ALL NOS                                  242
AML with mutated NPM1                      226
MDS-related; secondary myeloid             225
B-ALL with hyperdiploidy                   190
AML with inv(16); t(16;16); CBFB::MYH11    178
AML with t(8;21); RUNX1::RUNX1T1           176
B-ALL with t(12;21); ETV6::RUNX1           163
AML with NUP98-fusion                       95
AML with mutated CEBPA                      83
APL with t(15;17); PML::RARA                31
AML with CBFA2T3::GLIS2                     30
B-ALL with t(v;11q23); KMT2A-r              28
AML with t(6;9); DEK::NUP214                26
MPAL T-Lymphoblastic/Myeloid                25
B-ALL with t(1;19); TCF3::PBX1              23
B-ALL dic(9;20)                             20
MPAL B-Lymphoblastic/Myeloid                19
B-ALL with t(9;22); BCR::ABL1               19
AML with ETV6 fusion                        16
AML with inv(3); t(3;3); MECOM-r            12
AML with t(8;16); KAT6A::CREBBP             10
B-ALL with iAMP21                           10
AML with t(1;22); RBM15::MKL1                9
MPAL with t(v;11q23.3)/KMT2A-r               6
B-ALL with hypodiploidy                      5
AML with t(16;21); FUS::ERG                  4
AML with t(9;22); BCR::ABL1                  3
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="categorize-eln-2022-hematopoietic-entity-age-and-trisomy-8">
<h2><a class="toc-backref" href="#id10" role="doc-backlink">Categorize ELN 2022, hematopoietic entity, age, and trisomy 8</a><a class="headerlink" href="#categorize-eln-2022-hematopoietic-entity-age-and-trisomy-8" title="Permalink to this heading">#</a></h2>
<section id="id1">
<h3><a class="toc-backref" href="#id11" role="doc-backlink">Functions</a><a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">process_df_labels</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Function to process a pandas dataframe, performing age categorization </span>
<span class="sd">    and main disease classification.</span>

<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">def</span> <span class="nf">categorize_age</span><span class="p">(</span><span class="n">age</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to categorize age into a specific range.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isnull</span><span class="p">(</span><span class="n">age</span><span class="p">):</span>
            <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">5</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;0-5&#39;</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">13</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;5-13&#39;</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">39</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;13-39&#39;</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">60</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;39-60&#39;</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;60+&#39;</span>

    <span class="c1"># Convert &#39;Age (years)&#39; to numeric, errors=&#39;coerce&#39; will turn non-numeric data to NaN</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">to_numeric</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">],</span> <span class="n">errors</span><span class="o">=</span><span class="s1">&#39;coerce&#39;</span><span class="p">)</span>

    <span class="c1"># Then apply your function</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (group years)&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_age</span><span class="p">)</span>
    
    <span class="k">def</span> <span class="nf">categorize_subtypes</span><span class="p">(</span><span class="n">subtype</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to classify the main disease based on a given subtype.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s1">&#39;AML&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Acute myeloid leukemia (AML)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;ALL&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Acute lymphoblastic leukemia (ALL)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;MDS&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Myelodysplastic syndrome (MDS or MDS-like)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;MPAL&#39;</span>                          <span class="p">:</span><span class="s1">&#39;Mixed phenotype acute leukemia (MPAL)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;APL&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Acute promyelocytic leukemia (APL)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Otherwise-Normal Control&#39;</span>      <span class="p">:</span><span class="s1">&#39;Otherwise-Normal (Control)&#39;</span><span class="p">,</span>
        <span class="p">}</span>

        <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">subtype</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">value</span>
    
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Hematopoietic Entity&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_subtypes</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">categorize_trisomy8</span><span class="p">(</span><span class="n">karyotype</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to classify trisomy 8 by karyotype as a separate column.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s1">&#39;+8&#39;</span><span class="p">:</span> <span class="s1">&#39;Positive&#39;</span><span class="p">,</span>
            <span class="s1">&#39;trisomy 8&#39;</span><span class="p">:</span> <span class="s1">&#39;Positive&#39;</span><span class="p">,</span>
        <span class="p">}</span>

        <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">karyotype</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">value</span>
    
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Trisomy 8 Status&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Karyotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_trisomy8</span><span class="p">)</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s1">&#39;Negative or Unknown&#39;</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">categorize_ELN2022</span><span class="p">(</span><span class="n">subtype</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to reclassify samples according to ELN AML 2022.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span>                  <span class="p">:</span><span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Otherwise-Normal Control&#39;</span>                      <span class="p">:</span><span class="s1">&#39;Otherwise-Normal Control&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with mutated CEBPA&#39;</span>                        <span class="p">:</span><span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">,</span>
            <span class="s1">&#39;MDS-related; secondary myeloid&#39;</span>                <span class="p">:</span><span class="s1">&#39;MDS-related; secondary myeloid&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span>              <span class="p">:</span><span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with mutated NPM1&#39;</span>                         <span class="p">:</span><span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span>       <span class="p">:</span><span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span>              <span class="p">:</span><span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span>                   <span class="p">:</span><span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span>                  <span class="p">:</span><span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span><span class="p">,</span>
            <span class="s1">&#39;APL with t(15;17); PML::RARA&#39;</span>                  <span class="p">:</span><span class="s1">&#39;APL with t(15;17); PML::RARA&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(1;22); RBM15::MKL1&#39;</span>                 <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(16;21); FUS::ERG&#39;</span>                   <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span>               <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with CBFA2T3::GLIS2&#39;</span>                       <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with NUP98-fusion&#39;</span>                         <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with ETV6 fusion&#39;</span>                          <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>   
        <span class="p">}</span>

        <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">subtype</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">value</span>
    
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;ELN 2022 Diagnosis&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_ELN2022</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">df</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="id2">
<h3><a class="toc-backref" href="#id12" role="doc-backlink">Execution</a><a class="headerlink" href="#id2" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_labels3</span> <span class="o">=</span> <span class="n">process_df_labels</span><span class="p">(</span><span class="n">df_labels2</span><span class="p">)</span>

<span class="n">df_labels3</span><span class="p">[</span><span class="s1">&#39;Hematopoietic Entity&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Hematopoietic Entity
Acute myeloid leukemia (AML)                  1185
Acute lymphoblastic leukemia (ALL)             700
Otherwise-Normal (Control)                     251
Myelodysplastic syndrome (MDS or MDS-like)     225
Mixed phenotype acute leukemia (MPAL)           50
Acute promyelocytic leukemia (APL)              31
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="apply-classification-to-test-dataset">
<h2><a class="toc-backref" href="#id13" role="doc-backlink">Apply Classification to Test Dataset</a><a class="headerlink" href="#apply-classification-to-test-dataset" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import functions to clean up clinical data</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;../&#39;</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">source.clinical_data_cleanup_functions</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Call functions to merge, index and clean clinical data files</span>
<span class="n">labels_aml02</span>         <span class="o">=</span> <span class="n">clean_aml02</span>       <span class="p">(</span><span class="n">merge_index_aml02</span><span class="p">())</span>
<span class="n">labels_aml08</span>         <span class="o">=</span> <span class="n">clean_aml08</span>       <span class="p">(</span><span class="n">merge_index_aml08</span><span class="p">())</span>

<span class="c1"># Combine all clinical data labels into one dataframe</span>
<span class="n">labels_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">labels_aml02</span><span class="p">,</span> <span class="n">labels_aml08</span><span class="p">]</span> <span class="p">,</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;outer&#39;</span><span class="p">)</span>

<span class="c1"># Redefine output path (for troubleshooting purposes in case only this cell is run)</span>
<span class="c1"># mount = &#39;../../../&#39;</span>
<span class="s1">&#39;/mnt/d/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;/Intermediate_Files/&#39;</span>

<span class="c1"># Read df_test</span>
<span class="n">df_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;201samples_357839cpgs_withbatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span>

<span class="c1"># Remove samples that are not in the methyl dataset</span>
<span class="n">df_labels_test</span> <span class="o">=</span> <span class="n">labels_test</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">labels_test</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_test</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The clinical data has been indexed and cleaned.</span><span class="se">\n\</span>
<span class="s1">Exclusion of samples may be applied depending on the analysis.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The clinical data has been indexed and cleaned.
Exclusion of samples may be applied depending on the analysis.
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_labels_test2</span> <span class="o">=</span> <span class="n">process_labels_who22</span><span class="p">(</span><span class="n">df_labels_test</span><span class="p">)</span>
<span class="n">df_labels_test2</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>WHO 2022 Diagnosis
NaN                                        99
AML with t(v;11q23); KMT2A-r               47
AML with t(8;21); RUNX1::RUNX1T1           29
AML with inv(16); t(16;16); CBFB::MYH11    23
AML with mutated CEBPA                      2
AML with t(6;9); DEK::NUP214                1
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_labels_test3</span> <span class="o">=</span> <span class="n">process_df_labels</span><span class="p">(</span><span class="n">df_labels_test2</span><span class="p">)</span>

<span class="n">df_labels_test3</span><span class="p">[</span><span class="s1">&#39;Hematopoietic Entity&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Hematopoietic Entity
Acute myeloid leukemia (AML)    102
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_labels3</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>WHO 2022 Diagnosis
NaN                                        866
AML with t(v;11q23); KMT2A-r               317
Otherwise-Normal Control                   251
T-ALL NOS                                  242
AML with mutated NPM1                      226
MDS-related; secondary myeloid             225
B-ALL with hyperdiploidy                   190
AML with inv(16); t(16;16); CBFB::MYH11    178
AML with t(8;21); RUNX1::RUNX1T1           176
B-ALL with t(12;21); ETV6::RUNX1           163
AML with NUP98-fusion                       95
AML with mutated CEBPA                      83
APL with t(15;17); PML::RARA                31
AML with CBFA2T3::GLIS2                     30
B-ALL with t(v;11q23); KMT2A-r              28
AML with t(6;9); DEK::NUP214                26
MPAL T-Lymphoblastic/Myeloid                25
B-ALL with t(1;19); TCF3::PBX1              23
B-ALL dic(9;20)                             20
MPAL B-Lymphoblastic/Myeloid                19
B-ALL with t(9;22); BCR::ABL1               19
AML with ETV6 fusion                        16
AML with inv(3); t(3;3); MECOM-r            12
AML with t(8;16); KAT6A::CREBBP             10
B-ALL with iAMP21                           10
AML with t(1;22); RBM15::MKL1                9
MPAL with t(v;11q23.3)/KMT2A-r               6
B-ALL with hypodiploidy                      5
AML with t(16;21); FUS::ERG                  4
AML with t(9;22); BCR::ABL1                  3
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_labels3</span><span class="p">[</span><span class="s1">&#39;ELN 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>ELN 2022 Diagnosis
None                                       1616
AML with t(v;11q23); KMT2A-r                317
Otherwise-Normal Control                    251
AML with mutated NPM1                       226
MDS-related; secondary myeloid              225
AML with inv(16); t(16;16); CBFB::MYH11     178
AML with t(8;21); RUNX1::RUNX1T1            176
AML with other recurring translocations     164
AML with mutated CEBPA                       83
APL with t(15;17); PML::RARA                 31
AML with t(6;9); DEK::NUP214                 26
AML with inv(3); t(3;3); MECOM-r             12
AML with t(9;22); BCR::ABL1                   3
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
<section id="save-clinical-datasets">
<h2><a class="toc-backref" href="#id14" role="doc-backlink">Save Clinical Datasets</a><a class="headerlink" href="#save-clinical-datasets" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Save the clinical data labels with WHO 2022 classification</span>
<span class="n">df_labels3</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;discovery_clinical_data.csv&#39;</span><span class="p">)</span>

<span class="c1"># Save the clinical data labels with WHO 2022 classification</span>
<span class="n">df_labels3</span><span class="p">[[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">,</span><span class="s1">&#39;ELN 2022 Diagnosis&#39;</span><span class="p">,</span> <span class="s1">&#39;Clinical Trial&#39;</span><span class="p">,</span> <span class="s1">&#39;Vital Status&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;discovery_clinical_data_short.xlsx&#39;</span><span class="p">)</span>

<span class="c1"># savel labels_test to excel</span>
<span class="n">df_labels_test3</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;validation_clinical_data.csv&#39;</span><span class="p">)</span>

<span class="c1"># Save individual datasets</span>

<span class="c1"># labels_aml05.to_excel(output_path + &#39;aml05_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_beataml.to_excel(output_path + &#39;beataml_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_0531.to_excel(output_path + &#39;0531_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_amltcga.to_excel(output_path + &#39;amltcga_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_1031.to_excel(output_path + &#39;1031_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_nordic_all.to_excel(output_path + &#39;nordic_all_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_mds_taml.to_excel(output_path + &#39;mds_taml_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_all_graal.to_excel(output_path + &#39;all_graal_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_target_all.to_excel(output_path + &#39;target_all_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_aml02.to_excel(output_path + &#39;aml02_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_aml08.to_excel(output_path + &#39;aml08_clinical_data.xlsx&#39;)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./data-processing"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="test_dataset.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Validation Dataset</p>
      </div>
    </a>
    <a class="right-next"
       href="../models/2_Univariate_CoxPH-EWAS_v3.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Epigenome Wide Association Study</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#guidelines">Guidelines</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#who-2022-aml">WHO 2022 AML</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-process-discovery-clinical-data">Load and process discovery clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#reclassification-strategy-for-who-2022">Reclassification strategy for WHO 2022</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#functions">Functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#execution">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#categorize-eln-2022-hematopoietic-entity-age-and-trisomy-8">Categorize ELN 2022, hematopoietic entity, age, and trisomy 8</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-classification-to-test-dataset">Apply Classification to Test Dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#save-clinical-datasets">Save Clinical Datasets</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Marchi et al., 2023. Lamba Lab @ University of Florida. Please do not share this notebook publicly.
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>