{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Learning Algorithms\n",
    "\n",
    "```{contents}\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Where the data at"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import sys\n",
    "sys.path.append('../')\n",
    "# from source.pacmap_functions import *\n",
    "\n",
    "mount = '/mnt/h/'\n",
    "input_path = mount + 'MethylScore/Intermediate_Files/'\n",
    "output_path = mount + 'MethylScore/Processed_Data/'"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Discovery dataset (df_discovery) contains 319558 columns (5mC nucleotides/probes) and 3308 rows (samples).\n",
      " Validation dataset (df_validation) contains 319558 columns (5mC nucleotides/probes) and 201 rows (samples).\n"
     ]
    }
   ],
   "source": [
    "# read df_discovery and df_validation\n",
    "df_discovery = pd.read_pickle(\n",
    "    input_path+'3308samples_333059cpgs_withbatchcorrection_bvalues.pkl').sort_index()\n",
    "\n",
    "df_validation = pd.read_pickle(\n",
    "    input_path+'201samples_357839cpgs_withbatchcorrection_bvalues.pkl').sort_index()\n",
    "\n",
    "# use overlapping features between df_discovery and df_validation\n",
    "common_features = [x for x in df_discovery.columns if x in df_validation.columns]\n",
    "\n",
    "# apply `common_features` to both df_discovery and df_validation\n",
    "df_discovery = df_discovery[common_features]\n",
    "df_validation = df_validation[common_features]\n",
    "\n",
    "print(\n",
    "f' Discovery dataset (df_discovery) contains {df_discovery.shape[1]} \\\n",
    "columns (5mC nucleotides/probes) and {df_discovery.shape[0]} rows (samples).')\n",
    "\n",
    "print(\n",
    "f' Validation dataset (df_validation) contains {df_validation.shape[1]} \\\n",
    "columns (5mC nucleotides/probes) and {df_validation.shape[0]} rows (samples).')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reduce dimensionality with PaCMAP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PaCMAP version: 0.7.1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/fmarchi/projects/AML-atlas-notebooks/.venv/lib/python3.8/site-packages/sklearn/base.py:348: InconsistentVersionWarning: Trying to unpickle estimator TruncatedSVD from version 1.2.2 when using version 1.3.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "import pacmap\n",
    "print('PaCMAP version:', pacmap.__version__)\n",
    "\n",
    "# components_list = [2, 5]\n",
    "# for components in components_list:\n",
    "\n",
    "#     # Initialize the instance\n",
    "#     reducer = pacmap.PaCMAP(n_components=components, n_neighbors=15, MN_ratio=0.4, FP_ratio=16.0, \n",
    "#                             random_state=42, lr=0.1, num_iters=5000, save_tree=True)\n",
    "\n",
    "#     # Project the high dimensional dataset into a low-dimensional embedding\n",
    "#     embedding_training = reducer.fit(df_discovery.to_numpy(dtype='float16'))\n",
    "\n",
    "#     # Save reducer\n",
    "#     pacmap.save(reducer, f'../models/pacmap_{components}d_model_al_atlas')\n",
    "\n",
    "# Load reducer\n",
    "reducer = pacmap.load('../models/pacmap_5d_model_al_atlas')\n",
    "\n",
    "# Project the high dimensional dataset into existing embedding space and return the embedding.\n",
    "embedding_training = reducer.transform(df_discovery.to_numpy(dtype='float16'))\n",
    "embedding_validation = reducer.transform(df_validation.to_numpy(dtype='float16'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load clinical data\n",
    "discovery_clinical_data = pd.read_csv(input_path+'discovery_clinical_data.csv',\n",
    "                                      low_memory=False, index_col=0)\n",
    "\n",
    "# Load clinical data\n",
    "validation_clinical_data = pd.read_csv(input_path+'validation_clinical_data.csv',\n",
    "                                        low_memory=False, index_col=0)\n",
    "\n",
    "# Concatenate clinical data files and keep overlapping columns\n",
    "common_columns = [x for x in discovery_clinical_data.columns if x in validation_clinical_data.columns]\n",
    "\n",
    "# Apply `common_columns` to both discovery_clinical_data and validation_clinical_data\n",
    "discovery_clinical_data = discovery_clinical_data[common_columns]\n",
    "validation_clinical_data = validation_clinical_data[common_columns]\n",
    "\n",
    "# Concatenate clinical data files and keep overlapping columns\n",
    "clinical_data = pd.concat([discovery_clinical_data, validation_clinical_data], axis=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preprocess data for classifiers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load output from PaCMAP\n",
    "\n",
    "Reloading it here to allow for running the cells below without running the PaCMAP code above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "bad operand type for unary ~: 'list'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[33], line 7\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;66;03m# Concatenate clinical data files and keep overlapping columns\u001b[39;00m\n\u001b[1;32m      5\u001b[0m common_columns \u001b[38;5;241m=\u001b[39m [x \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m discovery_clinical_data\u001b[38;5;241m.\u001b[39mcolumns \u001b[38;5;28;01mif\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m validation_clinical_data\u001b[38;5;241m.\u001b[39mcolumns]\n\u001b[0;32m----> 7\u001b[0m discovery_clinical_data \u001b[38;5;241m=\u001b[39m discovery_clinical_data[\u001b[38;5;241;43m~\u001b[39;49m\u001b[43mcommon_columns\u001b[49m]\n\u001b[1;32m      8\u001b[0m validation_clinical_data \u001b[38;5;241m=\u001b[39m validation_clinical_data[\u001b[38;5;241m~\u001b[39mcommon_columns]\n\u001b[1;32m     10\u001b[0m \u001b[38;5;66;03m# Concatenate clinical data files and keep overlapping columns\u001b[39;00m\n",
      "\u001b[0;31mTypeError\u001b[0m: bad operand type for unary ~: 'list'"
     ]
    }
   ],
   "source": [
    "discovery_clinical_data = pd.read_excel(input_path+'discovery_clinical_data_shortened.xlsx', index_col=0).sort_index()\n",
    "validation_clinical_data = pd.read_csv(input_path+'validation_clinical_data.csv', index_col=0).sort_index()\n",
    "\n",
    "# Concatenate clinical data files and keep overlapping columns\n",
    "common_columns = [x for x in discovery_clinical_data.columns if x in validation_clinical_data.columns]\n",
    "\n",
    "discovery_clinical_data = discovery_clinical_data[common_columns]\n",
    "validation_clinical_data = validation_clinical_data[common_columns]\n",
    "\n",
    "# Concatenate clinical data files and keep overlapping columns\n",
    "clinical_data = pd.concat([discovery_clinical_data, validation_clinical_data], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample Type</th>\n",
       "      <th>Clinical Trial</th>\n",
       "      <th>Batch</th>\n",
       "      <th>WHO 2022 Diagnosis</th>\n",
       "      <th>ELN 2022 Diagnosis</th>\n",
       "      <th>Hematopoietic Entity</th>\n",
       "      <th>FAB</th>\n",
       "      <th>FLT3 ITD</th>\n",
       "      <th>Age (group years)</th>\n",
       "      <th>Age (years)</th>\n",
       "      <th>...</th>\n",
       "      <th>os.evnt</th>\n",
       "      <th>os.time</th>\n",
       "      <th>efs.evnt</th>\n",
       "      <th>efs.time</th>\n",
       "      <th>Gene Fusion</th>\n",
       "      <th>CEBPA mutation</th>\n",
       "      <th>NPM mutation</th>\n",
       "      <th>Train-Test</th>\n",
       "      <th>WHO 2022 Combined Diagnoses</th>\n",
       "      <th>Karyotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid</th>\n",
       "      <td>Primary Blood Derived Cancer - Bone Marrow</td>\n",
       "      <td>AAML03P1</td>\n",
       "      <td>GDC_TARGET-AML</td>\n",
       "      <td>AML with t(v;11q23); KMT2A-r</td>\n",
       "      <td>AML with t(v;11q23); KMT2A-r</td>\n",
       "      <td>Acute myeloid leukemia (AML)</td>\n",
       "      <td>M5</td>\n",
       "      <td>No</td>\n",
       "      <td>13-39</td>\n",
       "      <td>14.624658</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.153425</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.210959</td>\n",
       "      <td>MLLT10-PSD3</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Train Sample</td>\n",
       "      <td>AML with t(v;11q23); KMT2A-r</td>\n",
       "      <td>47,XY,+10[14]/46,XY[6]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0031efba-f564-4fff-bd7b-2b97f37218c1_noid</th>\n",
       "      <td>Bone Marrow Normal</td>\n",
       "      <td>AAML0531</td>\n",
       "      <td>GDC_TARGET-AML</td>\n",
       "      <td>Otherwise-Normal Control</td>\n",
       "      <td>Otherwise-Normal Control</td>\n",
       "      <td>Otherwise-Normal (Control)</td>\n",
       "      <td>M4</td>\n",
       "      <td>Yes</td>\n",
       "      <td>5-13</td>\n",
       "      <td>5.136986</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.515068</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.515068</td>\n",
       "      <td>NUP98-NSD1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Train Sample</td>\n",
       "      <td>Otherwise-Normal Control,AML with NUP98-fusion</td>\n",
       "      <td>46,XX[20]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid</th>\n",
       "      <td>Primary Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>AAML0531</td>\n",
       "      <td>GDC_TARGET-AML</td>\n",
       "      <td>AML with NUP98-fusion</td>\n",
       "      <td>AML with other recurring translocations</td>\n",
       "      <td>Acute myeloid leukemia (AML)</td>\n",
       "      <td>M4</td>\n",
       "      <td>Yes</td>\n",
       "      <td>5-13</td>\n",
       "      <td>5.136986</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.515068</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.515068</td>\n",
       "      <td>NUP98-NSD1,NUP98-NSD1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Train Sample</td>\n",
       "      <td>AML with NUP98-fusion</td>\n",
       "      <td>46,XX[20]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>003d9efe-90a1-42cf-84c4-03fbabefe60b_noid</th>\n",
       "      <td>Bone Marrow Normal</td>\n",
       "      <td>TARGET ALL</td>\n",
       "      <td>GDC_TARGET-ALL</td>\n",
       "      <td>Otherwise-Normal Control</td>\n",
       "      <td>Otherwise-Normal Control</td>\n",
       "      <td>Otherwise-Normal (Control)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Train Sample</td>\n",
       "      <td>Otherwise-Normal Control</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>004c953f-d999-4d82-898d-c091e692df3c_noid</th>\n",
       "      <td>Primary Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>AAML0531</td>\n",
       "      <td>GDC_TARGET-AML</td>\n",
       "      <td>AML with t(1;22); RBM15::MKL1</td>\n",
       "      <td>AML with other recurring translocations</td>\n",
       "      <td>Acute myeloid leukemia (AML)</td>\n",
       "      <td>M7</td>\n",
       "      <td>No</td>\n",
       "      <td>0-5</td>\n",
       "      <td>1.542466</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.410959</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.410959</td>\n",
       "      <td>RBM15-MKL1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Train Sample</td>\n",
       "      <td>AML with t(1;22); RBM15::MKL1</td>\n",
       "      <td>53,XX,t(1;22)(p13;q13),+der(1)t(1;22)(p13;q13)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9426021149_R04C01</th>\n",
       "      <td>Diagnosis</td>\n",
       "      <td>AML08</td>\n",
       "      <td>Multicenter AML02,08 Trials</td>\n",
       "      <td>AML with t(v;11q23); KMT2A-r</td>\n",
       "      <td>AML with t(v;11q23); KMT2A-r</td>\n",
       "      <td>Acute myeloid leukemia (AML)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>0-5</td>\n",
       "      <td>0.427397</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.153425</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.153425</td>\n",
       "      <td>KMT2A</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Test Sample</td>\n",
       "      <td>AML with t(v;11q23); KMT2A-r,AML with t(v;11q2...</td>\n",
       "      <td>46,XY,t(4;10;11)(q31;p11.2;q23)[22]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9426021149_R04C02</th>\n",
       "      <td>Diagnosis</td>\n",
       "      <td>AML08</td>\n",
       "      <td>Multicenter AML02,08 Trials</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1</td>\n",
       "      <td>Acute myeloid leukemia (AML)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>13-39</td>\n",
       "      <td>14.709589</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.238356</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.238356</td>\n",
       "      <td>RUNX1-RUNX1T1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Test Sample</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1,AML with muta...</td>\n",
       "      <td>46,XX,t(6;21;8;11)(q21;q22;q22;q13)[20]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9426021149_R05C01</th>\n",
       "      <td>Diagnosis</td>\n",
       "      <td>AML08</td>\n",
       "      <td>Multicenter AML02,08 Trials</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1</td>\n",
       "      <td>Acute myeloid leukemia (AML)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>13-39</td>\n",
       "      <td>15.158904</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.336986</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.336986</td>\n",
       "      <td>RUNX1-RUNX1T1</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Test Sample</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1,AML with t(8;...</td>\n",
       "      <td>46,XY,t(8;21)(q22;q22)[19]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9426021149_R05C02</th>\n",
       "      <td>Diagnosis</td>\n",
       "      <td>AML08</td>\n",
       "      <td>Multicenter AML02,08 Trials</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>5-13</td>\n",
       "      <td>7.536986</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.769863</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.156164</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Test Sample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46,XX[43]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9426021149_R06C01</th>\n",
       "      <td>Diagnosis</td>\n",
       "      <td>AML08</td>\n",
       "      <td>Multicenter AML02,08 Trials</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1</td>\n",
       "      <td>Acute myeloid leukemia (AML)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>13-39</td>\n",
       "      <td>15.868493</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.142466</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.142466</td>\n",
       "      <td>RUNX1-RUNX1T1</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Test Sample</td>\n",
       "      <td>AML with t(8;21); RUNX1::RUNX1T1,AML with t(8;...</td>\n",
       "      <td>46,X,-Y,+8,t(8;21)(q22;q22)[19]/46,XY[1]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3509 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                               Sample Type  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid       Primary Blood Derived Cancer - Bone Marrow   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid                               Bone Marrow Normal   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid  Primary Blood Derived Cancer - Peripheral Blood   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid                               Bone Marrow Normal   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid  Primary Blood Derived Cancer - Peripheral Blood   \n",
       "...                                                                                    ...   \n",
       "9426021149_R04C01                                                                Diagnosis   \n",
       "9426021149_R04C02                                                                Diagnosis   \n",
       "9426021149_R05C01                                                                Diagnosis   \n",
       "9426021149_R05C02                                                                Diagnosis   \n",
       "9426021149_R06C01                                                                Diagnosis   \n",
       "\n",
       "                                          Clinical Trial  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid       AAML03P1   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid       AAML0531   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid       AAML0531   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid     TARGET ALL   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid       AAML0531   \n",
       "...                                                  ...   \n",
       "9426021149_R04C01                                  AML08   \n",
       "9426021149_R04C02                                  AML08   \n",
       "9426021149_R05C01                                  AML08   \n",
       "9426021149_R05C02                                  AML08   \n",
       "9426021149_R06C01                                  AML08   \n",
       "\n",
       "                                                                 Batch  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid               GDC_TARGET-AML   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid               GDC_TARGET-AML   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid               GDC_TARGET-AML   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid               GDC_TARGET-ALL   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid               GDC_TARGET-AML   \n",
       "...                                                                ...   \n",
       "9426021149_R04C01                          Multicenter AML02,08 Trials   \n",
       "9426021149_R04C02                          Multicenter AML02,08 Trials   \n",
       "9426021149_R05C01                          Multicenter AML02,08 Trials   \n",
       "9426021149_R05C02                          Multicenter AML02,08 Trials   \n",
       "9426021149_R06C01                          Multicenter AML02,08 Trials   \n",
       "\n",
       "                                                         WHO 2022 Diagnosis  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid      AML with t(v;11q23); KMT2A-r   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid          Otherwise-Normal Control   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid             AML with NUP98-fusion   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid          Otherwise-Normal Control   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid     AML with t(1;22); RBM15::MKL1   \n",
       "...                                                                     ...   \n",
       "9426021149_R04C01                              AML with t(v;11q23); KMT2A-r   \n",
       "9426021149_R04C02                          AML with t(8;21); RUNX1::RUNX1T1   \n",
       "9426021149_R05C01                          AML with t(8;21); RUNX1::RUNX1T1   \n",
       "9426021149_R05C02                                                       NaN   \n",
       "9426021149_R06C01                          AML with t(8;21); RUNX1::RUNX1T1   \n",
       "\n",
       "                                                                ELN 2022 Diagnosis  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid             AML with t(v;11q23); KMT2A-r   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid                 Otherwise-Normal Control   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid  AML with other recurring translocations   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid                 Otherwise-Normal Control   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid  AML with other recurring translocations   \n",
       "...                                                                            ...   \n",
       "9426021149_R04C01                                     AML with t(v;11q23); KMT2A-r   \n",
       "9426021149_R04C02                                 AML with t(8;21); RUNX1::RUNX1T1   \n",
       "9426021149_R05C01                                 AML with t(8;21); RUNX1::RUNX1T1   \n",
       "9426021149_R05C02                                                              NaN   \n",
       "9426021149_R06C01                                 AML with t(8;21); RUNX1::RUNX1T1   \n",
       "\n",
       "                                                   Hematopoietic Entity  FAB  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid  Acute myeloid leukemia (AML)   M5   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid    Otherwise-Normal (Control)   M4   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid  Acute myeloid leukemia (AML)   M4   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid    Otherwise-Normal (Control)  NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid  Acute myeloid leukemia (AML)   M7   \n",
       "...                                                                 ...  ...   \n",
       "9426021149_R04C01                          Acute myeloid leukemia (AML)  NaN   \n",
       "9426021149_R04C02                          Acute myeloid leukemia (AML)  NaN   \n",
       "9426021149_R05C01                          Acute myeloid leukemia (AML)  NaN   \n",
       "9426021149_R05C02                                                   NaN  NaN   \n",
       "9426021149_R06C01                          Acute myeloid leukemia (AML)  NaN   \n",
       "\n",
       "                                          FLT3 ITD Age (group years)  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid       No             13-39   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid      Yes              5-13   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid      Yes              5-13   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid      NaN               NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid       No               0-5   \n",
       "...                                            ...               ...   \n",
       "9426021149_R04C01                               No               0-5   \n",
       "9426021149_R04C02                               No             13-39   \n",
       "9426021149_R05C01                              Yes             13-39   \n",
       "9426021149_R05C02                               No              5-13   \n",
       "9426021149_R06C01                               No             13-39   \n",
       "\n",
       "                                           Age (years)  ... os.evnt  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid    14.624658  ...     1.0   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid     5.136986  ...     1.0   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid     5.136986  ...     1.0   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid          NaN  ...     NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid     1.542466  ...     0.0   \n",
       "...                                                ...  ...     ...   \n",
       "9426021149_R04C01                             0.427397  ...     0.0   \n",
       "9426021149_R04C02                            14.709589  ...     1.0   \n",
       "9426021149_R05C01                            15.158904  ...     1.0   \n",
       "9426021149_R05C02                             7.536986  ...     1.0   \n",
       "9426021149_R06C01                            15.868493  ...     0.0   \n",
       "\n",
       "                                             os.time efs.evnt   efs.time  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid   1.153425      1.0   0.210959   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid   0.515068      1.0   0.515068   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid   0.515068      1.0   0.515068   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid        NaN      0.0        NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid   6.410959      0.0   6.410959   \n",
       "...                                              ...      ...        ...   \n",
       "9426021149_R04C01                           7.153425      0.0   7.153425   \n",
       "9426021149_R04C02                           1.238356      1.0   1.238356   \n",
       "9426021149_R05C01                           0.336986      1.0   0.336986   \n",
       "9426021149_R05C02                           3.769863      1.0   0.156164   \n",
       "9426021149_R06C01                          11.142466      0.0  11.142466   \n",
       "\n",
       "                                                     Gene Fusion  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid            MLLT10-PSD3   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid             NUP98-NSD1   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid  NUP98-NSD1,NUP98-NSD1   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid                    NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid             RBM15-MKL1   \n",
       "...                                                          ...   \n",
       "9426021149_R04C01                                          KMT2A   \n",
       "9426021149_R04C02                                  RUNX1-RUNX1T1   \n",
       "9426021149_R05C01                                  RUNX1-RUNX1T1   \n",
       "9426021149_R05C02                                            NaN   \n",
       "9426021149_R06C01                                  RUNX1-RUNX1T1   \n",
       "\n",
       "                                          CEBPA mutation NPM mutation  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid             No           No   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid             No           No   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid             No           No   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid            NaN          NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid             No           No   \n",
       "...                                                  ...          ...   \n",
       "9426021149_R04C01                                     No          NaN   \n",
       "9426021149_R04C02                                    Yes          NaN   \n",
       "9426021149_R05C01                                     No          NaN   \n",
       "9426021149_R05C02                                     No          NaN   \n",
       "9426021149_R06C01                                     No          NaN   \n",
       "\n",
       "                                             Train-Test  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid  Train Sample   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid  Train Sample   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid  Train Sample   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid  Train Sample   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid  Train Sample   \n",
       "...                                                 ...   \n",
       "9426021149_R04C01                           Test Sample   \n",
       "9426021149_R04C02                           Test Sample   \n",
       "9426021149_R05C01                           Test Sample   \n",
       "9426021149_R05C02                           Test Sample   \n",
       "9426021149_R06C01                           Test Sample   \n",
       "\n",
       "                                                                 WHO 2022 Combined Diagnoses  \\\n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid                       AML with t(v;11q23); KMT2A-r   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid     Otherwise-Normal Control,AML with NUP98-fusion   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid                              AML with NUP98-fusion   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid                           Otherwise-Normal Control   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid                      AML with t(1;22); RBM15::MKL1   \n",
       "...                                                                                      ...   \n",
       "9426021149_R04C01                          AML with t(v;11q23); KMT2A-r,AML with t(v;11q2...   \n",
       "9426021149_R04C02                          AML with t(8;21); RUNX1::RUNX1T1,AML with muta...   \n",
       "9426021149_R05C01                          AML with t(8;21); RUNX1::RUNX1T1,AML with t(8;...   \n",
       "9426021149_R05C02                                                                        NaN   \n",
       "9426021149_R06C01                          AML with t(8;21); RUNX1::RUNX1T1,AML with t(8;...   \n",
       "\n",
       "                                                                                   Karyotype  \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid                            47,XY,+10[14]/46,XY[6]   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid                                         46,XX[20]   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid                                         46,XX[20]   \n",
       "003d9efe-90a1-42cf-84c4-03fbabefe60b_noid                                                NaN  \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid  53,XX,t(1;22)(p13;q13),+der(1)t(1;22)(p13;q13)...  \n",
       "...                                                                                      ...  \n",
       "9426021149_R04C01                                        46,XY,t(4;10;11)(q31;p11.2;q23)[22]  \n",
       "9426021149_R04C02                                    46,XX,t(6;21;8;11)(q21;q22;q22;q13)[20]  \n",
       "9426021149_R05C01                                                 46,XY,t(8;21)(q22;q22)[19]  \n",
       "9426021149_R05C02                                                                  46,XX[43]  \n",
       "9426021149_R06C01                                   46,X,-Y,+8,t(8;21)(q22;q22)[19]/46,XY[1]  \n",
       "\n",
       "[3509 rows x 27 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "\n",
    "\n",
    "n_components = 5\n",
    "\n",
    "# Create a list of strings containing \"PaCMAP \" + the number of components\n",
    "components = [f'PaCMAP {i}' for i in range(1, n_components+1)]\n",
    "\n",
    "# Load pacmap output data\n",
    "df = pd.read_csv(output_path+f'pacmap_output/pacmap_{n_components}d_model_dx_al.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select samples Px"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Drop the samples with missing labels for the selected column\n",
    "df_px = df[~df['Vital Status'].isna()]\n",
    "\n",
    "df_px2 = df_px.copy()\n",
    "\n",
    "# # Exclude the `Blood Derived Normal`and `Bone Marrow Normal` from `Sample Type`\n",
    "# df_px2 = df_px[~df_px['Sample Type'].isin(['Relapse', 'Recurrent Blood Derived Cancer - Bone Marrow',\n",
    "#                                             'Recurrent Blood Derived Cancer - Peripheral Blood',\n",
    "#                                             'Blood Derived Normal', 'Bone Marrow Normal'])]\n",
    "\n",
    "# print the number of samples dropped and the amount remaining\n",
    "print(df.shape[0]-df_px2.shape[0], 'samples removed.'\\\n",
    ", df_px2.shape[0], 'samples remaining.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select samples Dx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "# drop the samples with missing labels for the ELN AML 2022 Diagnosis\n",
    "df_dx = df[~df['WHO 2022 Diagnosis'].isna()]\n",
    "\n",
    "# exclude the classes with fewer than 10 samples\n",
    "df_dx2 = df_dx[~df_dx['WHO 2022 Diagnosis'].isin([\n",
    "                                    'AML with t(1;22); RBM15::MKL1',\n",
    "                                       'MPAL with t(v;11q23.3)/KMT2A-r',\n",
    "                                       'B-ALL with hypodiploidy',\n",
    "                                       'AML with t(16;21); FUS::ERG',\n",
    "                                       'AML with t(9;22); BCR::ABL1'\n",
    "                                       ])]\n",
    "\n",
    "# print the number of samples dropped and the amount remaining\n",
    "print(df.shape[0]-df_dx2.shape[0], 'samples removed.'\\\n",
    ", df_dx2.shape[0], 'samples remaining.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define X and y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def custom_train_test_split(df, feature_columns, target_column, split_column):\n",
    "\n",
    "    X = df[feature_columns].to_numpy()\n",
    "    y = df[target_column].to_numpy()\n",
    "\n",
    "    train_mask = df[split_column] == 'Train Sample'\n",
    "    test_mask = df[split_column] == 'Test Sample'\n",
    "\n",
    "    X_train, X_test = X[train_mask], X[test_mask]\n",
    "    y_train, y_test = y[train_mask], y[test_mask]\n",
    "\n",
    "    print('X_train shape:', X_train.shape, 'X_test shape:', X_test.shape)\n",
    "\n",
    "    return X_train, X_test, y_train, y_test\n",
    "\n",
    "# Execution\n",
    "X_train_dx, X_test_dx, y_train_dx, y_test_dx = custom_train_test_split(df_dx2, components,'WHO 2022 Diagnosis', 'Train-Test')\n",
    "X_train_px, X_test_px, y_train_px, y_test_px = custom_train_test_split(df_px2, components,'Vital Status', 'Train-Test')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Benchmark classifiers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define confusion matrix plot function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_confusion_matrix_stacked(clf, x_train, y_train, x_test, y_test, \n",
    "                                  title='', \n",
    "                                  tick_fontsize=10, label_fontsize=10,\n",
    "                                  figsize=(10, 5)):\n",
    "\n",
    "    sns.set_theme(style='white')\n",
    "\n",
    "    fig, axs = plt.subplots(1, 2, figsize=figsize, sharey=True)\n",
    "\n",
    "    fig.subplots_adjust(wspace=0.05)  # Adjust the width space\n",
    "\n",
    "    for i, (ax, x, y, subset) in enumerate(zip(axs, [x_train, x_test], [y_train, y_test],\n",
    "                                                ['Train with 10-fold CV', 'Validation'])):\n",
    "        predictions = clf.predict(x)\n",
    "        print(f'Overall accuracy score in {subset}: {accuracy_score(y, predictions):.3f}')\n",
    "        cm = confusion_matrix(y, predictions, labels=clf.classes_, normalize='true')\n",
    "        disp = ConfusionMatrixDisplay(confusion_matrix=cm, display_labels=clf.classes_)\n",
    "        disp.plot(cmap='Blues', values_format='.2f', xticks_rotation='vertical', colorbar=False, ax=ax)\n",
    "    \n",
    "\n",
    "        # Set font size of the numbers inside the confusion matrix\n",
    "        for texts in disp.text_:\n",
    "            for text in texts:\n",
    "                text.set_fontsize(label_fontsize)\n",
    "\n",
    "        ax.tick_params(axis='both', which='major', labelsize=tick_fontsize)\n",
    "        ax.set_title(subset + ', n=' + str(len(x)), fontsize=label_fontsize+1, pad=10)\n",
    "        ax.set_xlabel('Predicted label', fontsize=label_fontsize+1)\n",
    "\n",
    "        # Add y labels\n",
    "        if i == 0:\n",
    "            ax.set_ylabel('True label', fontsize=label_fontsize+1)\n",
    "        else:\n",
    "            ax.set_ylabel('')\n",
    "\n",
    "        # remove x tick labels\n",
    "        ax.xaxis.set_ticklabels([])\n",
    "        \n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Execute "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import GridSearchCV\n",
    "from lightgbm import LGBMClassifier\n",
    "from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay, accuracy_score\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import joblib\n",
    "\n",
    "\n",
    "def benchmark_classifier(X_train, y_train, X_test, y_test, figsize=(16,8)):\n",
    "\n",
    "    param_grid = {\n",
    "    'num_leaves': [2, 3, 4, 5, 6, 7, 8, 9, 10, 15, 20], # number of leaves in full tree\n",
    "    'max_depth': [2, 3, 4, 5, 6, 7, 8, 9, 10],  # maximum depth of a tree\n",
    "    # 'learning_rate': [0.001, 0.01, 0.015, 0.02, 0.1],  # learning rate\n",
    "    # 'n_estimators': [10, 50, 100, 200],  # number of trees (or rounds)\n",
    "    }\n",
    "\n",
    "    # Initialize the LGBM Classifier with regularization\n",
    "    lgbm = LGBMClassifier(random_state=42, n_jobs=-1, reg_alpha=1.0, reg_lambda=1.0)\n",
    "\n",
    "    # Perform grid search with stratified cross-validation\n",
    "    grid_search = GridSearchCV(lgbm, param_grid, cv=10, n_jobs=-1, scoring='roc_auc_ovr_weighted')\n",
    "\n",
    "    # Fit the grid search to the data\n",
    "    grid_search.fit(X_train, y_train)\n",
    "    print(f\"Best parameters: {grid_search.best_params_}\")\n",
    "\n",
    "    # Get the best model\n",
    "    clf = grid_search.best_estimator_\n",
    "\n",
    "    plot = plot_confusion_matrix_stacked(clf, X_train, y_train, X_test, y_test,\n",
    "                                  tick_fontsize=7, label_fontsize=7, figsize=figsize)\n",
    "    return clf, plot\n",
    "\n",
    "    # Plot the confusion matrix\n",
    "\n",
    "# Benchmark, train\n",
    "lgbm_dx_model, plot = benchmark_classifier(X_train_dx, y_train_dx, X_test_dx, y_test_dx)\n",
    "lgbm_px_model, plot = benchmark_classifier(X_train_px, y_train_px, X_test_px, y_test_px, figsize=(5,3))\n",
    "\n",
    "# Save models\n",
    "joblib.dump(lgbm_px_model,'../models/lgbm_px_model.pkl')\n",
    "joblib.dump(lgbm_dx_model,'../models/lgbm_dx_model.pkl')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_predictions(df, classifier, model_name):\n",
    "\n",
    "    # ignore sklearn warnings\n",
    "    import warnings\n",
    "    warnings.filterwarnings('ignore')\n",
    "\n",
    "    # Select necessary columns and set the index\n",
    "    df_features = df.set_index('index')[['PaCMAP 1', 'PaCMAP 2', 'PaCMAP 3', 'PaCMAP 4', 'PaCMAP 5']]\n",
    "\n",
    "    # Predict using the selected columns\n",
    "    predictions = classifier.predict(df_features)\n",
    "\n",
    "    # Predict probabilities using the selected columns\n",
    "    probabilities = classifier.predict_proba(df_features)\n",
    "\n",
    "    # Convert predictions to a Series with the same index as df_features\n",
    "    predictions_series = pd.Series(predictions, index=df_features.index, name=model_name)\n",
    "\n",
    "    # Convert probabilities to a DataFrame with the same index as df_features and the same columns as the classes\n",
    "    probabilities_df = pd.DataFrame(probabilities, index=df_features.index, columns=classifier.classes_).round(3)\n",
    "\n",
    "    # Add \" - predict_proba\" to the column names\n",
    "    probabilities_df.columns ='P(' + probabilities_df.columns + ')'\n",
    "\n",
    "    # # Create a new column named \"Confidence\" and set it to True if the max probability is >= confidence_threshold\n",
    "    # probabilities_df['Confidence >=' + str(confidence_threshold)] = probabilities_df.max(axis=1) >= confidence_threshold\n",
    "\n",
    "    # Transform classes of the predictions into integers based on unique values in the classes\n",
    "    probabilities_df[model_name + '_int'] = predictions_series.map({c: i for i, c in enumerate(classifier.classes_)})\n",
    "\n",
    "    # Join predictions with the original DataFrame (already indexed)\n",
    "    df_joined = predictions_series.to_frame().join(probabilities_df)\n",
    "\n",
    "    return df_joined\n",
    "\n",
    "# Execution\n",
    "df_pred_px = save_predictions(df=df, classifier=lgbm_px_model, model_name='AML Epigenomic Risk')\n",
    "df_pred_dx = save_predictions(df=df, classifier=lgbm_dx_model, model_name='AL Epigenomic Phenotype')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_combined = df.join(df_pred_px).join(df_pred_dx)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Other benchmarked classifiers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gaussian Process Classifier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from sklearn.gaussian_process import GaussianProcessClassifier\n",
    "\n",
    "# # Define the parameter grid\n",
    "# param_grid = {\n",
    "#     'multi_class': ['one_vs_rest'], \n",
    "# }\n",
    "\n",
    "# # Initialize the Gaussian Process Classifier\n",
    "# gpc = GaussianProcessClassifier(random_state=42, n_jobs=-1)\n",
    "\n",
    "# # Perform grid search with cross-validation\n",
    "# grid_search = GridSearchCV(gpc, param_grid, cv=10, n_jobs=-1, scoring='roc_auc_ovr_weighted')\n",
    "# grid_search.fit(X_train, y_train)\n",
    "\n",
    "# # Fit and predict using the best estimator\n",
    "# gpc_px_model = grid_search.best_estimator_\n",
    "\n",
    "# y_pred = gpc_px_model.predict(X_test)  # Predict using the best model\n",
    "\n",
    "# # Print the best parameters and the best score\n",
    "# print(f\"Best parameters: {grid_search.best_params_}\")\n",
    "\n",
    "# # Fit and predict using the best estimator\n",
    "# gpc_px_model = grid_search.best_estimator_\n",
    "\n",
    "# y_pred_train = gpc_px_model.predict(X_train)  # Predict using the best model\n",
    "# y_pred_test = gpc_px_model.predict(X_test)  # Predict using the best model\n",
    "\n",
    "\n",
    "# print(f'Overall accuracy score with best estimator in train: {accuracy_score(y_train, y_pred_train):.3f}')\n",
    "# print(f'Overall accuracy score with best estimator in test: {accuracy_score(y_test, y_pred_test):.3f}')\n",
    "\n",
    "# # plot the confusion matrix\n",
    "# plot_confusion_matrix_stacked(gpc_px_model, X_train, y_train, X_test, y_test,\n",
    "#                               tick_fontsize=7, label_fontsize=7, figsize = (5,3))\n",
    "\n",
    "# # you shall save the model\n",
    "# pickle.dump(gpc_px_model, open('../models/gpc_px_model.pkl', 'wb'))\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Watermark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [],
   "source": [
    "%load_ext watermark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "# watermark with all libraries used in this notebook\n",
    "%watermark -v -p numpy,pandas,bokeh,pacmap,itables -a Francisco_Marchi@Lamba_Lab_UF -d -m"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv_py38",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
