{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Harmonize ONT reads with array reference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Converting CRAM to FASTQ (or basecall straight into fastq and skip this step):\n",
    "\n",
    "    `samtools fastq -@ <num_threads> input.cram > output.fastq`\n",
    "\n",
    "2. Minimap2 aligment:\n",
    "\n",
    "    `minimap2 -ax lr:hq -t <num_threads> ../ref/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna input.fastq.gz > output.sam`\n",
    "\n",
    "3. Converting SAM to BAM, sorting, and indexing:\n",
    "\n",
    "    a. Convert SAM to BAM:\n",
    "\n",
    "    `samtools view -@ <num_threads> -bS output.sam > output.bam`\n",
    "\n",
    "    b. Sort BAM:\n",
    "\n",
    "    `samtools sort -@ <num_threads> output.bam -o sorted_output.bam`\n",
    "\n",
    "    c. Create BAM Index:\n",
    "\n",
    "    `samtools index sorted_output.bam`\n",
    "\n",
    "4. Modkit\n",
    "    \n",
    "    `modkit pileup uf_hembank_1852.bam uf_hembank_1852_pacmap.bed --combine-mods --no-filtering -t 32 --combine-strands --cpg --ref GCA_000001405.15_GRCh38_no_alt_analysis_set.fna --include-bed pacmap_reference.bed `\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Where data at?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "mount = '/mnt/d/'\n",
    "\n",
    "reference_path = mount + 'genome_references/Illumina_methylation_arrays/EPIC.hg38.manifest.tsv.gz'\n",
    "output_path = mount + 'MethylScore_v2/Processed_Data/'\n",
    "\n",
    "sample_name = 'uf_hembank_1852'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create BED6 file to harmonize probes from EPIC array with nanopore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read df_discovery and df_validation\n",
    "df_discovery = pd.read_pickle(mount+'MethylScore_v2/Intermediate_Files/'+'3308samples_333059cpgs_withbatchcorrection_bvalues.pkl').sort_index().iloc[:,1:]\n",
    "\n",
    "array_reference = pd.read_csv(\"/mnt/c/Users/fmarc/OneDrive/Desktop/nanopore_processed/ref/EPIC.hg38.manifest.gencode.v36.tsv.gz\", sep='\\t', compression='gzip',\n",
    "                              usecols=['CpG_chrm','CpG_beg','CpG_end','probe_strand','probeID']\n",
    "                              ).set_index('probeID').sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "pacmap_reference = array_reference.loc[df_discovery.columns].reset_index()\n",
    "\n",
    "# remove `.0` from `CpG_beg` and `CpG_end` and coordinate\n",
    "pacmap_reference['CpG_beg'] = pacmap_reference['CpG_beg'].astype(int)\n",
    "pacmap_reference['CpG_end'] = pacmap_reference['CpG_end'].astype(int)\n",
    "pacmap_reference['score'] = 0\n",
    "\n",
    "# make `pacamap_reference` a BED6 file\n",
    "# pacmap_reference['CpG_beg'] = pacmap_reference['CpG_beg'] - 1\n",
    "# pacmap_reference['CpG_end'] = pacmap_reference['CpG_end'] - 1\n",
    "# pacmap_reference['CpG_chrm'] = pacmap_reference['CpG_chrm'].str.replace('chr','')\n",
    "# pacmap_reference['CpG_chrm'] = pacmap_reference['CpG_chrm'].str.replace('X','23')\n",
    "# pacmap_reference['CpG_chrm'] = pacmap_reference['CpG_chrm'].str.replace('Y','24')\n",
    "# pacmap_reference['CpG_chrm'] = pacmap_reference['CpG_chrm'].astype(str)\n",
    "\n",
    "pacmap_reference = pacmap_reference[['CpG_chrm','CpG_beg','CpG_end','IlmnID','score','probe_strand']]\n",
    "pacmap_reference = pacmap_reference.rename(columns={'CpG_chrm':'chrom','CpG_beg':'chromStart','CpG_end':'chromEnd','IlmnID':'name','probe_strand':'strand'})\n",
    "\n",
    "# sort by chromosome and start\n",
    "pacmap_reference = pacmap_reference.sort_values(by=['chrom','chromStart'])\n",
    "\n",
    "pacmap_reference.to_csv('/mnt/c/Users/fmarc/OneDrive/Desktop/nanopore_processed/ref/pacmap_reference.bed', sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>coordinate</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1:69590</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69590</td>\n",
       "      <td>69592</td>\n",
       "      <td>cg21870274</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:864702</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864702</td>\n",
       "      <td>864704</td>\n",
       "      <td>cg08258224</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:870160</th>\n",
       "      <td>chr1</td>\n",
       "      <td>870160</td>\n",
       "      <td>870162</td>\n",
       "      <td>cg16619049</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:877158</th>\n",
       "      <td>chr1</td>\n",
       "      <td>877158</td>\n",
       "      <td>877160</td>\n",
       "      <td>cg18147296</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:898802</th>\n",
       "      <td>chr1</td>\n",
       "      <td>898802</td>\n",
       "      <td>898804</td>\n",
       "      <td>cg13938959</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:138119084</th>\n",
       "      <td>chr9</td>\n",
       "      <td>138119084</td>\n",
       "      <td>138119086</td>\n",
       "      <td>cg00378292</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:138120221</th>\n",
       "      <td>chr9</td>\n",
       "      <td>138120221</td>\n",
       "      <td>138120223</td>\n",
       "      <td>cg07982825</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:138122338</th>\n",
       "      <td>chr9</td>\n",
       "      <td>138122338</td>\n",
       "      <td>138122340</td>\n",
       "      <td>cg14491707</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:138122548</th>\n",
       "      <td>chr9</td>\n",
       "      <td>138122548</td>\n",
       "      <td>138122550</td>\n",
       "      <td>cg13811936</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:138123004</th>\n",
       "      <td>chr9</td>\n",
       "      <td>138123004</td>\n",
       "      <td>138123006</td>\n",
       "      <td>cg14022794</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333058 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               chrom  chromStart   chromEnd        name  score strand\n",
       "coordinate                                                           \n",
       "chr1:69590      chr1       69590      69592  cg21870274      0      +\n",
       "chr1:864702     chr1      864702     864704  cg08258224      0      -\n",
       "chr1:870160     chr1      870160     870162  cg16619049      0      -\n",
       "chr1:877158     chr1      877158     877160  cg18147296      0      -\n",
       "chr1:898802     chr1      898802     898804  cg13938959      0      +\n",
       "...              ...         ...        ...         ...    ...    ...\n",
       "chr9:138119084  chr9   138119084  138119086  cg00378292      0      +\n",
       "chr9:138120221  chr9   138120221  138120223  cg07982825      0      -\n",
       "chr9:138122338  chr9   138122338  138122340  cg14491707      0      +\n",
       "chr9:138122548  chr9   138122548  138122550  cg13811936      0      -\n",
       "chr9:138123004  chr9   138123004  138123006  cg14022794      0      +\n",
       "\n",
       "[333058 rows x 6 columns]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pacmap_reference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load modkit processed file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Define columns to be used for the input data\n",
    "usecols = [0, 1, 4, 10]\n",
    "column_names = [\"chrom\", \"start_position\", \"score\", \"fraction_modified\"]\n",
    "\n",
    "# Read the input data, skipping the first row if it's a header or irrelevant\n",
    "df = pd.read_csv('/mnt/c/Users/fmarc/OneDrive/Desktop/nanopore_processed/bed/uf_hembank_1852_pacmap.bed', sep='\\s+', skiprows=1, usecols=usecols, names=column_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Create 'coordinate' column for merging\n",
    "df['coordinate'] = df['chrom'].astype(str) + ':' + df['start_position'].astype(str)\n",
    "\n",
    "df_filtered = df.set_index('coordinate')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start_position</th>\n",
       "      <th>score</th>\n",
       "      <th>fraction_modified</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>coordinate</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1:864703</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864703</td>\n",
       "      <td>9</td>\n",
       "      <td>88.89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:870161</th>\n",
       "      <td>chr1</td>\n",
       "      <td>870161</td>\n",
       "      <td>6</td>\n",
       "      <td>66.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:877159</th>\n",
       "      <td>chr1</td>\n",
       "      <td>877159</td>\n",
       "      <td>9</td>\n",
       "      <td>77.78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:898802</th>\n",
       "      <td>chr1</td>\n",
       "      <td>898802</td>\n",
       "      <td>13</td>\n",
       "      <td>38.46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:898803</th>\n",
       "      <td>chr1</td>\n",
       "      <td>898803</td>\n",
       "      <td>1</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22:50737087</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50737087</td>\n",
       "      <td>11</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22:50737978</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50737978</td>\n",
       "      <td>15</td>\n",
       "      <td>93.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22:50738282</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50738282</td>\n",
       "      <td>13</td>\n",
       "      <td>61.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22:50739553</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50739553</td>\n",
       "      <td>15</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22:50783583</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50783583</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>337649 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chrom  start_position  score  fraction_modified\n",
       "coordinate                                                     \n",
       "chr1:864703      chr1          864703      9              88.89\n",
       "chr1:870161      chr1          870161      6              66.67\n",
       "chr1:877159      chr1          877159      9              77.78\n",
       "chr1:898802      chr1          898802     13              38.46\n",
       "chr1:898803      chr1          898803      1             100.00\n",
       "...               ...             ...    ...                ...\n",
       "chr22:50737087  chr22        50737087     11             100.00\n",
       "chr22:50737978  chr22        50737978     15              93.33\n",
       "chr22:50738282  chr22        50738282     13              61.54\n",
       "chr22:50739553  chr22        50739553     15             100.00\n",
       "chr22:50783583  chr22        50783583     10               0.00\n",
       "\n",
       "[337649 rows x 4 columns]"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "pacmap_reference['coordinate'] = pacmap_reference['chrom'].astype(str) + ':' + pacmap_reference['chromStart'].astype(str)\n",
    "pacmap_reference = pacmap_reference.set_index('coordinate')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Join with reference data on 'coordinate'\n",
    "df_merged = df_filtered.join(pacmap_reference[['name']], how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start_position</th>\n",
       "      <th>score</th>\n",
       "      <th>fraction_modified</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>coordinate</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr10:100008943</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100008943</td>\n",
       "      <td>6</td>\n",
       "      <td>50.00</td>\n",
       "      <td>cg14405603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:100020168</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020168</td>\n",
       "      <td>14</td>\n",
       "      <td>85.71</td>\n",
       "      <td>cg01099300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:100062255</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100062255</td>\n",
       "      <td>8</td>\n",
       "      <td>87.50</td>\n",
       "      <td>cg25833003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:100065360</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100065360</td>\n",
       "      <td>6</td>\n",
       "      <td>83.33</td>\n",
       "      <td>cg19754520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:100066954</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100066954</td>\n",
       "      <td>10</td>\n",
       "      <td>80.00</td>\n",
       "      <td>cg22534585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:99906442</th>\n",
       "      <td>chr9</td>\n",
       "      <td>99906442</td>\n",
       "      <td>9</td>\n",
       "      <td>0.00</td>\n",
       "      <td>cg21606237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:99906832</th>\n",
       "      <td>chr9</td>\n",
       "      <td>99906832</td>\n",
       "      <td>11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>cg03950009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:99907330</th>\n",
       "      <td>chr9</td>\n",
       "      <td>99907330</td>\n",
       "      <td>11</td>\n",
       "      <td>9.09</td>\n",
       "      <td>cg14153061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:99967624</th>\n",
       "      <td>chr9</td>\n",
       "      <td>99967624</td>\n",
       "      <td>9</td>\n",
       "      <td>100.00</td>\n",
       "      <td>cg14233861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:99998980</th>\n",
       "      <td>chr9</td>\n",
       "      <td>99998980</td>\n",
       "      <td>8</td>\n",
       "      <td>100.00</td>\n",
       "      <td>cg05469304</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>168768 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chrom  start_position  score  fraction_modified        name\n",
       "coordinate                                                                  \n",
       "chr10:100008943  chr10       100008943      6              50.00  cg14405603\n",
       "chr10:100020168  chr10       100020168     14              85.71  cg01099300\n",
       "chr10:100062255  chr10       100062255      8              87.50  cg25833003\n",
       "chr10:100065360  chr10       100065360      6              83.33  cg19754520\n",
       "chr10:100066954  chr10       100066954     10              80.00  cg22534585\n",
       "...                ...             ...    ...                ...         ...\n",
       "chr9:99906442     chr9        99906442      9               0.00  cg21606237\n",
       "chr9:99906832     chr9        99906832     11               0.00  cg03950009\n",
       "chr9:99907330     chr9        99907330     11               9.09  cg14153061\n",
       "chr9:99967624     chr9        99967624      9             100.00  cg14233861\n",
       "chr9:99998980     chr9        99998980      8             100.00  cg05469304\n",
       "\n",
       "[168768 rows x 5 columns]"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Interpolate missing values in 'fraction_modified' column linearly\n",
    "df_merged['fraction_modified'] = df_merged['fraction_modified'].astype(float).interpolate(method='linear')\n",
    "\n",
    "\n",
    "df_merged = df_merged[['fraction_modified']].join(pacmap_reference, how='inner').fillna(50)\n",
    "\n",
    "df_merged = df_merged.drop_duplicates(subset='name')\n",
    "\n",
    "# Calculate the fraction_modified and prepare the final DataFrame\n",
    "df_merged.loc[:, sample_name] = (df_merged['fraction_modified'] / 100).round(3)\n",
    "\n",
    "df_processed = df_merged[['name', sample_name]].set_index('name').T"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Apply PaCMAP model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "import pacmap\n",
    "\n",
    "def apply_pacmap_model_to_new_data(df, components):\n",
    "\n",
    "    # Load reducer\n",
    "    reducer = pacmap.load(f'../models/pacmap_{components}d_model_al_atlas')\n",
    "\n",
    "    # Project the high dimensional dataset into existing embedding space and return the embedding.\n",
    "    embedding = reducer.transform(df.to_numpy(dtype='float16'))\n",
    "\n",
    "    # Create column names\n",
    "    cols = ['PaCMAP '+ str(i+1) + f' of {components}' for i in range(components)]\n",
    "\n",
    "    # Turn embedding into dataframe\n",
    "    df_embedding = pd.DataFrame(embedding, columns=cols, index=df.index)\n",
    "\n",
    "    return df_embedding\n",
    "\n",
    "df_embedding_2d = apply_pacmap_model_to_new_data(df_processed, 2)\n",
    "df_embedding_5d = apply_pacmap_model_to_new_data(df_processed, 5)\n",
    "\n",
    "df_embedding_2d.to_pickle(output_path + sample_name + '_pacmap_2d.pkl')\n",
    "df_embedding_5d.to_pickle(output_path + sample_name + '_pacmap_5d.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Apply supervised models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AML Epigenomic Risk</th>\n",
       "      <th>AML Epigenomic Risk P(High Risk)</th>\n",
       "      <th>AL Epigenomic Phenotype</th>\n",
       "      <th>P(AML with inv(16); t(16;16); CBFB::MYH11)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>uf_hembank_1852</th>\n",
       "      <td>High</td>\n",
       "      <td>0.501</td>\n",
       "      <td>AML with inv(16); t(16;16); CBFB::MYH11</td>\n",
       "      <td>0.636</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                AML Epigenomic Risk  AML Epigenomic Risk P(High Risk)  \\\n",
       "uf_hembank_1852                High                             0.501   \n",
       "\n",
       "                                 AL Epigenomic Phenotype  \\\n",
       "uf_hembank_1852  AML with inv(16); t(16;16); CBFB::MYH11   \n",
       "\n",
       "                 P(AML with inv(16); t(16;16); CBFB::MYH11)  \n",
       "uf_hembank_1852                                       0.636  "
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import joblib\n",
    "\n",
    "# Load models\n",
    "lgbm_px_model = joblib.load('../models/lgbm_px_model.pkl')\n",
    "lgbm_dx_model = joblib.load('../models/lgbm_dx_model.pkl')\n",
    "\n",
    "# load `df_embedding_5d` from the previous step\n",
    "df_embedding_5d = pd.read_pickle(output_path + sample_name + '_pacmap_5d.pkl')\n",
    "\n",
    "def save_predictions(df, classifier, model_name):\n",
    "\n",
    "    # ignore sklearn warnings\n",
    "    import warnings\n",
    "    warnings.filterwarnings('ignore')\n",
    "\n",
    "    # Select necessary columns\n",
    "    df_features = df.copy()\n",
    "\n",
    "    # Predict using the selected columns\n",
    "    predictions = classifier.predict(df_features)\n",
    "\n",
    "    # Predict probabilities using the selected columns\n",
    "    probabilities = classifier.predict_proba(df_features)\n",
    "\n",
    "    # Convert predictions to a Series with the same index as df_features\n",
    "    predictions_series = pd.Series(predictions, index=df_features.index, name=model_name)\n",
    "\n",
    "    # Convert probabilities to a DataFrame with the same index as df_features and the same columns as the classes\n",
    "    probabilities_df = pd.DataFrame(probabilities, index=df_features.index, columns=classifier.classes_).round(3)\n",
    "\n",
    "    # Add \" - predict_proba\" to the column names\n",
    "    probabilities_df.columns ='P(' + probabilities_df.columns + ')'\n",
    "\n",
    "    # Transform classes of the predictions into integers based on unique values in the classes\n",
    "    probabilities_df[model_name + '_int'] = predictions_series.map({c: i for i, c in enumerate(classifier.classes_)})\n",
    "\n",
    "    # Join predictions with the original DataFrame (already indexed)\n",
    "    df_joined = predictions_series.to_frame().join(probabilities_df)\n",
    "\n",
    "    return df_joined\n",
    "\n",
    "# Execution\n",
    "df_pred_px = save_predictions(df=df_embedding_5d, classifier=lgbm_px_model, model_name='AML Epigenomic Risk')\n",
    "df_pred_dx = save_predictions(df=df_embedding_5d, classifier=lgbm_dx_model, model_name='AL Epigenomic Phenotype')\n",
    "\n",
    "# Map the classes to more desirable labels (low and high risk)\n",
    "df_pred_px['AML Epigenomic Risk'] = df_pred_px['AML Epigenomic Risk'].map({'Alive': 'Low', 'Dead': 'High'})\n",
    "df_pred_px = df_pred_px.rename(columns={'P(Alive)': 'AML Epigenomic Risk P(Low Risk)', 'P(Dead)': 'AML Epigenomic Risk P(High Risk)'})\n",
    "\n",
    "# Join predictions with clinical data\n",
    "df_combined = df_embedding_2d.join(df_embedding_5d).join(df_pred_px).join(df_pred_dx)\n",
    "\n",
    "df_combined[['AML Epigenomic Risk', 'AML Epigenomic Risk P(High Risk)', 'AL Epigenomic Phenotype', f'P({df_combined[\"AL Epigenomic Phenotype\"].item()})']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## EWASCox-Lasso"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Continuous score cut at the value of 0.4934\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>EWASCox_OS_48CpGs</th>\n",
       "      <th>EWASCox_OS_48CpGs Categorical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>uf_hembank_1852</th>\n",
       "      <td>1.451607</td>\n",
       "      <td>High</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "name             EWASCox_OS_48CpGs EWASCox_OS_48CpGs Categorical\n",
       "uf_hembank_1852           1.451607                          High"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import math\n",
    "import sys\n",
    "sys.path.append('../')\n",
    "from source.cox_lasso import *\n",
    "\n",
    "raw_coefs = pd.read_csv(output_path + 'multivariate_cox_lasso/ewas_cog_os_raw_coefs_newrisk.csv', index_col=0)\n",
    "\n",
    "mean_coefs = set_cutoff(coefs=raw_coefs,threshold=0.99)\n",
    "\n",
    "df_validation = df_processed[mean_coefs.index]\n",
    "\n",
    "df_validation_transformed = df_validation.replace(1, 0.999).replace(0, 0.001)\n",
    "\n",
    "def beta2m(val):\n",
    "    '''Transfrom beta-values into m-values'''\n",
    "    return math.log2(val/(1-val))\n",
    "\n",
    "x_test_m = df_validation_transformed.apply(np.vectorize(beta2m))\n",
    "\n",
    "def standardize_data(df, reference_df):\n",
    "    \"\"\"Standardize data using mean and standard deviation of reference dataset\"\"\"\n",
    "\n",
    "    # Keep only columns that are in both datasets\n",
    "    reference_df = reference_df.loc[:, df.columns]\n",
    "\n",
    "    # Standardize data\n",
    "    df_z = (df - reference_df.mean()) / reference_df.std()\n",
    "\n",
    "    return df_z\n",
    "\n",
    "# Read top CpGs selected from previous code file (univariate cox-ph EWAS)\n",
    "ewas_top_cpgs = pd.read_csv(output_path+'ewas_dmr/ewas_top_cpgs_os.csv', index_col=0)\n",
    "\n",
    "# Standardize data\n",
    "x_test_m_z = standardize_data(df= x_test_m, reference_df= ewas_top_cpgs)\n",
    "\n",
    "score_name = 'EWASCox_OS_48CpGs'\n",
    "\n",
    "df_test, threshold = generate_coxph_score(coef_mean=mean_coefs,\n",
    "                                        x=x_test_m_z,\n",
    "                                        df=df_validation_transformed,\n",
    "                                        score_name=score_name,\n",
    "                                        train_test=0.4934,\n",
    "                                        rpart_outcome='os.time')\n",
    "\n",
    "df_validation_transformed[['EWASCox_OS_48CpGs','EWASCox_OS_48CpGs Categorical']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed data for uf_hembank_1852 saved as /mnt/d/MethylScore_v2/Processed_Data/uf_hembank_1852_processed.xlsx\n"
     ]
    }
   ],
   "source": [
    "df_nanopore = df_combined.join(df_validation_transformed[['EWASCox_OS_48CpGs','EWASCox_OS_48CpGs Categorical']])\n",
    "\n",
    "df_nanopore['Train-Test'] = 'Long-read Nanopore sequencing'\n",
    "df_nanopore['Clinical Trial'] = 'UF Hem Bank'\n",
    "df_nanopore['Patient_ID'] = sample_name\n",
    "df_nanopore['Hematopoietic Entity'] = np.nan\n",
    "df_nanopore['WHO 2022 Diagnosis'] =  np.nan\n",
    "df_nanopore['Vital Status'] = np.nan\n",
    "df_nanopore['Risk Group AAML1831'] = np.nan\n",
    "\n",
    "df_nanopore.to_excel(output_path + sample_name + '_processed.xlsx')\n",
    "\n",
    "# print save message\n",
    "print(f'Processed data for {sample_name} saved as {output_path + sample_name + \"_processed.xlsx\"}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Watermark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [],
   "source": [
    "%load_ext watermark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Author: Francisco_Marchi@Lamba_Lab_UF\n",
      "\n",
      "Python implementation: CPython\n",
      "Python version       : 3.8.16\n",
      "IPython version      : 8.12.3\n",
      "\n",
      "numpy   : 1.24.4\n",
      "pandas  : 2.0.3\n",
      "pacmap  : 0.7.0\n",
      "sklearn : 1.2.2\n",
      "lightgbm: 3.3.5\n",
      "\n",
      "Compiler    : GCC 11.3.0\n",
      "OS          : Linux\n",
      "Release     : 5.15.133.1-microsoft-standard-WSL2\n",
      "Machine     : x86_64\n",
      "Processor   : x86_64\n",
      "CPU cores   : 6\n",
      "Architecture: 64bit\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# watermark with all libraries used in this notebook\n",
    "%watermark -v -p numpy,pandas,pacmap,sklearn,lightgbm -a Francisco_Marchi@Lamba_Lab_UF -d -m"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```{note}\n",
    "Please only use the following versions:\n",
    "`python`: 3.8.16\n",
    "`pacmap`: 0.7.0\n",
    "`lightgbm`: 3.3.5\n",
    "`scikit-learn`: 1.2.2\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv_py38",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
