
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Data Download and Initial Processing &#8212; Lamba Lab</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script async="async" kind="hypothesis" src="https://hypothes.is/embed.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Methyl Array Data Processing" href="2_MethylData_Processing.html" />
    <link rel="prev" title="Introduction" href="docs/Intro_to_Methylation.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="_static/dna1.gif" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Lamba Lab</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="index.html">
                    FM’s Electronic Notebook Instructions
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Specific Aims
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="docs/Specific_Aims.html">
   Specific Aims
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Manuscripts
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="docs/Intro_to_Methylation.html">
   Introduction
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Code
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Data Download and Initial Processing
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="2_MethylData_Processing.html">
   Methyl Array Data Processing
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="3_ClinicalData_Processing.html">
   Clinical Data Processing
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="4_Dimension_Reduction.html">
   Dimension Reduction with PaCMAP
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="5_Methylome_Visualization.html">
   PaCMAP Visualization
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="5.1_Methylome_Bokeh.html">
   PaCMAP Interactive Visualization
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="6_Survival_Analysis.html">
   Survival Analysis FM
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="6.1_Survival_Analysis_FS.html">
   Survival Analysis FS
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="7_Score_Visualization.html">
   Kaplan Meiers, Forest Plots, Patient Characteristics
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/1_MethylData_PreProcessing.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="_sources/1_MethylData_PreProcessing.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#overview">
   Overview
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#download-and-pre-process-raw-data-from-gene-expression-omnibus-geo">
   Download and Pre-Process Raw Data from Gene Expression Omnibus (GEO)
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#download-and-pre-process-raw-data-from-genomic-data-commons-gdc">
   Download and Pre-Process Raw Data from Genomic Data Commons (GDC)
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Data Download and Initial Processing</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#overview">
   Overview
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#download-and-pre-process-raw-data-from-gene-expression-omnibus-geo">
   Download and Pre-Process Raw Data from Gene Expression Omnibus (GEO)
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#download-and-pre-process-raw-data-from-genomic-data-commons-gdc">
   Download and Pre-Process Raw Data from Genomic Data Commons (GDC)
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="data-download-and-initial-processing">
<h1>Data Download and Initial Processing<a class="headerlink" href="#data-download-and-initial-processing" title="Permalink to this headline">#</a></h1>
<section id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">#</a></h2>
<p>This is a comprehensive chapter on the pipeline utilized to download and process raw methylation array data from <em>Infinium MethylationEPIC</em> and <em>Infinium HumanMethylation450</em>:</p>
<ol class="simple">
<li><p>Download raw datasets from GEO or GDC using <code class="docutils literal notranslate"><span class="pre">methylprep</span></code></p></li>
<li><p>Pre-process raw data using <code class="docutils literal notranslate"><span class="pre">methylprep</span></code>,
which is a python library that offers R’s <a class="reference external" href="https://www.bioconductor.org/packages/release/bioc/html/sesame.html">SeSAMe</a> as the backend analysis suite.</p></li>
<li><p>Process data exclusion criteria using <code class="docutils literal notranslate"><span class="pre">methylcheck</span></code></p></li>
<li><p>Run Illumina’s Quality Control</p></li>
<li><p>Prepare data for analysis</p></li>
</ol>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This pipeline has been adapted from python’s <a class="reference external" href="https://pypi.org/project/methylsuite/">methylsuite</a>.</p>
</div>
</section>
<section id="download-and-pre-process-raw-data-from-gene-expression-omnibus-geo">
<h2>Download and Pre-Process Raw Data from Gene Expression Omnibus (GEO)<a class="headerlink" href="#download-and-pre-process-raw-data-from-gene-expression-omnibus-geo" title="Permalink to this headline">#</a></h2>
<p>Below is the implementation of the above diagram to this project:</p>
<ol class="simple">
<li><p>Create a virtual environmnet with python 3.7:</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>create<span class="w"> </span>--n<span class="w"> </span>&lt;name&gt;<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">3</span>.7
</pre></div>
</div>
<ol class="simple">
<li><p>In a linux shell, activate environment:</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">source</span><span class="w"> </span>activate<span class="w"> </span>methylsuite
</pre></div>
</div>
<ol class="simple">
<li><p>Install methylsuite and dependencies:</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>install<span class="w"> </span>statsmodels
pip<span class="w"> </span>install<span class="w"> </span>methylsuite
</pre></div>
</div>
<ol class="simple">
<li><p>Download dataset directly from GEO:</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python<span class="w"> </span>-m<span class="w"> </span>methylprep<span class="w"> </span>beta_bake<span class="w"> </span>-i<span class="w"> </span>&lt;GEO<span class="w"> </span>series&gt;<span class="w"> </span>-d<span class="w"> </span>&lt;directory&gt;<span class="w"> </span>
</pre></div>
</div>
<ol class="simple">
<li><p>Process dataset:</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python<span class="w"> </span>-m<span class="w"> </span>methylprep<span class="w"> </span>-v<span class="w"> </span>process<span class="w"> </span>--array_type<span class="w"> </span>&lt;epic<span class="w"> </span>or<span class="w"> </span>450k&gt;<span class="w"> </span>-d<span class="w"> </span>&lt;directory&gt;<span class="w"> </span>--all<span class="w"> </span>--no_sample_sheet
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If files are larger than RAM, specify batches
Full <code class="docutils literal notranslate"><span class="pre">methylprep</span></code> API can be found <a class="reference external" href="https://life-epigenetics-methylprep.readthedocs-hosted.com/en/latest/docs/cli.html">here</a></p>
</div>
</section>
<section id="download-and-pre-process-raw-data-from-genomic-data-commons-gdc">
<h2>Download and Pre-Process Raw Data from Genomic Data Commons (GDC)<a class="headerlink" href="#download-and-pre-process-raw-data-from-genomic-data-commons-gdc" title="Permalink to this headline">#</a></h2>
<ul class="simple">
<li><p>Go to <a class="reference external" href="https://portal.gdc.cancer.gov/">GDC</a></p></li>
<li><p>Click on <em>Repository</em></p></li>
<li><p>In <em>Files</em> on the left side of the screen, find <em>Experimental Strategy</em></p></li>
<li><p>Select <em>Methylation Array</em>, which will prompt you to see all cases and files containing methylation array data available</p></li>
<li><p>For our project, we will select the following filter/search strategy, which you can copy and paste into <em>advanced search</em> (or search manually):</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>    <span class="n">cases</span><span class="o">.</span><span class="n">project</span><span class="o">.</span><span class="n">program</span><span class="o">.</span><span class="n">name</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;TARGET&quot;</span><span class="p">]</span> <span class="ow">and</span> <span class="n">cases</span><span class="o">.</span><span class="n">project</span><span class="o">.</span><span class="n">project_id</span> 
    <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;TARGET-AML&quot;</span><span class="p">]</span> <span class="ow">and</span> <span class="n">files</span><span class="o">.</span><span class="n">data_format</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;idat&quot;</span><span class="p">]</span> <span class="ow">and</span> 
    <span class="n">files</span><span class="o">.</span><span class="n">experimental_strategy</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;Methylation Array&quot;</span><span class="p">]</span> <span class="ow">and</span> 
    <span class="n">files</span><span class="o">.</span><span class="n">platform</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;illumina human methylation 450&quot;</span><span class="p">]</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Most (if not all?) methylation data available through GDC is publicly available, so no access to controlled data is necessary!</p>
</div>
<ul class="simple">
<li><p>Once you have a selection of your samples, add them all to <em>cart</em>, which you can do by clicking on the <em>cart</em> symbol in the header of the first column</p></li>
<li><p>Then go to the top right corner of the screen and click on <em>cart</em></p></li>
<li><p>Download the <em>Manisfest</em> file (under the <em>Download</em> tab), as well as the <em>Sample Sheet</em>, and <em>Clinical</em>. Do not download the <em>Cart</em> - files are too large.</p></li>
<li><p>To download the data, you will need the GDC Data Transfer Tool (command-line) that you can find <a class="reference external" href="https://gdc.cancer.gov/access-data/gdc-data-transfer-tool">here</a></p></li>
<li><p>Download the software, unzip it, and place the .exe in a directory of easy access</p></li>
<li><p>In your terminal, go to the directory containing the .exe file and type the following:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">.</span>\<span class="n">gdc</span><span class="o">-</span><span class="n">client</span> <span class="n">download</span> <span class="o">-</span><span class="n">m</span> <span class="o">&lt;</span><span class="n">path</span> <span class="n">to</span> <span class="n">the</span> <span class="n">manisfest</span> <span class="n">file</span> <span class="n">you</span> <span class="n">downloaded</span><span class="o">&gt;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>When the .idat files are downloaded, open the sample sheet file you downloaded previously. It should be a .tsv file readable in excell.</p></li>
<li><p>Save the sample sheet file as <em>samplesheet.csv</em> in the directory where the data are located</p></li>
<li><p>Congrats! You are ready to process the files using <code class="docutils literal notranslate"><span class="pre">methylprep</span></code> (see chapter above for how to install methylprep and dependencies).</p></li>
<li><p>Like before, we will need to open methylprep on the terminal:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="n">activate</span> <span class="n">methylsuite</span>
</pre></div>
</div>
<ul class="simple">
<li><p>And run the command to process the files:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="o">-</span><span class="n">m</span> <span class="n">methylprep</span> <span class="o">-</span><span class="n">v</span> <span class="n">process</span> <span class="o">--</span><span class="n">array_type</span> <span class="mi">450</span><span class="n">k</span> <span class="o">-</span><span class="n">d</span> <span class="o">&lt;</span><span class="n">path</span> <span class="n">to</span> <span class="n">data</span> <span class="ow">and</span> <span class="n">samplesheet</span><span class="o">.</span><span class="n">csv</span><span class="o">&gt;</span> <span class="o">--</span><span class="nb">all</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If files are larger than your RAM memory, you will need to run in batches (check methylprep’s documentation).
Also, be mindful of \ and / differences if working in a windows machine through git bash. Some of methylprep code may break on Windows. Use a linux, Mac, or WSL if possible.</p>
</div>
<ul class="simple">
<li><p>Fantastic, files are now processed and ready for <code class="docutils literal notranslate"><span class="pre">methylcheck</span></code>!</p></li>
</ul>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="docs/Intro_to_Methylation.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Introduction</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="2_MethylData_Processing.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Methyl Array Data Processing</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Lamba Lab @ University of Florida<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>