

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>CoxPH-EWAS unadjusted with overall survival as outcome &#8212; Lamba Lab at UF</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '2_ClinicalData_Processing';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Specific Aims" href="docs/Specific_Aims.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="#">
  
  
  
  
  
    <p class="title logo__title">Lamba Lab at UF</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1 current active">
                <a class="reference internal" href="#">
                    CoxPH-EWAS unadjusted with overall survival as outcome
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Specific Aims</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="docs/Specific_Aims.html">Specific Aims</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="docs/Intro_to_Methylation.html">Understanding DNA Methylation</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Data Preparation</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="1_Data_Download_Process.html">Methylation-based Classification and Prognosis of AML: Methods</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/2_ClinicalData_Processing.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>CoxPH-EWAS unadjusted with overall survival as outcome</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-process-clinical-data">Load and process clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove Samples based on Certain Clinical Features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-bone-marrow-samples-only">Select diagnostic bone marrow samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas-unadjusted-with-overall-survival-as-outcome">Run CoxPH-EWAS unadjusted with overall survival as outcome</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#table-top-results-by-p-value">Table top results by p-value</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-genes-from-top-cpgs">Fetch genes from top CpGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="coxph-ewas-unadjusted-with-overall-survival-as-outcome">
<h1><a class="toc-backref" href="#id1" role="doc-backlink">CoxPH-EWAS unadjusted with overall survival as outcome</a><a class="headerlink" href="#coxph-ewas-unadjusted-with-overall-survival-as-outcome" title="Permalink to this heading">#</a></h1>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#coxph-ewas-unadjusted-with-overall-survival-as-outcome" id="id1">CoxPH-EWAS unadjusted with overall survival as outcome</a></p>
<ul>
<li><p><a class="reference internal" href="#load-and-process-clinical-data" id="id2">Load and process clinical data</a></p></li>
<li><p><a class="reference internal" href="#remove-samples-based-on-certain-clinical-features" id="id3">Remove Samples based on Certain Clinical Features</a></p>
<ul>
<li><p><a class="reference internal" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" id="id4">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></p></li>
<li><p><a class="reference internal" href="#select-diagnostic-bone-marrow-samples-only" id="id5">Select diagnostic bone marrow samples only</a></p></li>
<li><p><a class="reference internal" href="#remove-duplicate-samples" id="id6">Remove duplicate samples</a></p></li>
<li><p><a class="reference internal" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" id="id7">Match samples in clinical data to samples in methylation data</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#transform-to-m-values" id="id8">Transform to M-values</a></p></li>
<li><p><a class="reference internal" href="#run-coxph-ewas-unadjusted-with-overall-survival-as-outcome" id="id9">Run CoxPH-EWAS unadjusted with overall survival as outcome</a></p></li>
<li><p><a class="reference internal" href="#table-top-results-by-p-value" id="id10">Table top results by p-value</a></p></li>
<li><p><a class="reference internal" href="#fetch-genes-from-top-cpgs" id="id11">Fetch genes from top CpGs</a></p></li>
<li><p><a class="reference internal" href="#watermark" id="id12">Watermark</a></p></li>
</ul>
</li>
</ul>
</nav>
<section id="load-and-process-clinical-data">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Load and process clinical data</a><a class="headerlink" href="#load-and-process-clinical-data" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">input_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Intermediate_Files/&#39;</span>
<span class="n">clinical_data_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Raw_Data/Clinical_Data/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Processed_Data/&#39;</span>


<span class="n">df_methyl</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;2_MethylData_Processing_Output.pkl&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39; Dataset (df) contains </span><span class="si">{</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> columns (5mC nucleotides/probes) and </span><span class="si">{</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> rows (samples).&#39;</span><span class="p">)</span>

<span class="c1"># Import functions to clean up clinical data</span>
<span class="kn">from</span> <span class="nn">FM_Functions.Clinical_Data_CleanUp</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Call functions to merge, index and clean clinical data files</span>
<span class="n">labels_0531</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_0531</span><span class="p">())</span>
<span class="n">labels_1031</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_1031</span><span class="p">())</span>
<span class="n">labels_aml05</span>        <span class="o">=</span> <span class="n">clean_aml05</span>     <span class="p">(</span><span class="n">merge_index_aml05</span><span class="p">())</span>
<span class="n">labels_beataml</span>      <span class="o">=</span> <span class="n">clean_beataml</span>   <span class="p">(</span><span class="n">merge_index_beataml</span><span class="p">())</span>
<span class="n">labels_amltcga</span>      <span class="o">=</span> <span class="n">clean_amltcga</span>   <span class="p">(</span><span class="n">merge_index_amltcga</span><span class="p">())</span>
<span class="n">labels_nordic_all</span>   <span class="o">=</span> <span class="n">clean_nordic_all</span><span class="p">(</span><span class="n">merge_index_nordic_all</span><span class="p">())</span>
<span class="n">labels_mds_taml</span>     <span class="o">=</span> <span class="n">clean_mds_taml</span>  <span class="p">(</span><span class="n">merge_index_mds_taml</span><span class="p">())</span>
<span class="n">labels_all_graal</span>    <span class="o">=</span> <span class="n">clean_all_graal</span> <span class="p">(</span><span class="n">merge_index_all_graal</span><span class="p">())</span>
<span class="n">labels_target_all</span>   <span class="o">=</span> <span class="n">clean_target_all</span><span class="p">(</span><span class="n">merge_index_target_all</span><span class="p">())</span>

<span class="c1"># Combine all clinical data labels into one dataframe</span>
<span class="n">labels_combined</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">labels_aml05</span><span class="p">,</span> <span class="n">labels_beataml</span><span class="p">,</span>
                         <span class="n">labels_0531</span><span class="p">,</span> <span class="n">labels_amltcga</span><span class="p">,</span> <span class="n">labels_1031</span><span class="p">,</span>
                        <span class="n">labels_nordic_all</span><span class="p">,</span> <span class="n">labels_mds_taml</span><span class="p">,</span>
                        <span class="n">labels_all_graal</span><span class="p">,</span><span class="n">labels_target_all</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;outer&#39;</span><span class="p">)</span>

<span class="c1"># Remove samples that are not in the methyl dataset</span>
<span class="n">df_labels</span> <span class="o">=</span> <span class="n">labels_combined</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">labels_combined</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> Dataset (df) contains 333352 columns (5mC nucleotides/probes) and 3330 rows (samples).
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-samples-based-on-certain-clinical-features">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Remove Samples based on Certain Clinical Features</a><a class="headerlink" href="#remove-samples-based-on-certain-clinical-features" title="Permalink to this heading">#</a></h2>
<section id="select-samples-from-aaml1031-0531-and-03p1-clinical-trials">
<h3><a class="toc-backref" href="#id4" role="doc-backlink">Select samples from AAML1031, 0531, and 03P1 clinical trials</a><a class="headerlink" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df1</span> <span class="o">=</span> <span class="n">df_labels</span><span class="p">[</span><span class="n">df_labels</span><span class="p">[</span><span class="s1">&#39;Clinical Trial&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;AAML0531&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML1031&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML03P1&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df_labels</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2049 samples were removed. 1281 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="select-diagnostic-bone-marrow-samples-only">
<h3><a class="toc-backref" href="#id5" role="doc-backlink">Select diagnostic bone marrow samples only</a><a class="headerlink" href="#select-diagnostic-bone-marrow-samples-only" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df2</span> <span class="o">=</span> <span class="n">df1</span><span class="p">[</span><span class="n">df1</span><span class="p">[</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Diagnosis&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Bone Marrow&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>350 samples were removed. 931 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-duplicate-samples">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Remove duplicate samples</a><a class="headerlink" href="#remove-duplicate-samples" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df3</span> <span class="o">=</span> <span class="n">df2</span><span class="p">[</span><span class="o">~</span><span class="n">df2</span><span class="p">[</span><span class="s1">&#39;Patient_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="s1">&#39;last&#39;</span><span class="p">)]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>7 samples were removed. 924 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="match-samples-in-clinical-data-to-samples-in-methylation-data">
<h3><a class="toc-backref" href="#id7" role="doc-backlink">Match samples in clinical data to samples in methylation data</a><a class="headerlink" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Match samples in clinical data to samples in methylation data</span>
<span class="n">df_methyl_filtered</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">:]</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Samples in clinical data matched to samples in methylation data.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Samples in clinical data matched to samples in methylation data.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="transform-to-m-values">
<h2><a class="toc-backref" href="#id8" role="doc-backlink">Transform to M-values</a><a class="headerlink" href="#transform-to-m-values" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_methyl_clipped</span> <span class="o">=</span> <span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">clip</span><span class="p">(</span><span class="mf">0.001</span><span class="p">,</span> <span class="mf">0.999</span><span class="p">)</span>

<span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>


<span class="k">def</span> <span class="nf">beta2m</span><span class="p">(</span><span class="n">val</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Transfrom beta-values into m-values&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">math</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">val</span><span class="o">/</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">val</span><span class="p">))</span>

<span class="n">x_train_m</span> <span class="o">=</span> <span class="n">df_methyl_clipped</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>

<span class="c1"># Get correlation between Beta and M Values</span>
<span class="n">corr</span> <span class="o">=</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">corrwith</span><span class="p">(</span><span class="n">df_methyl_filtered</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Glue them into same dataframe for vizualization purposes</span>
<span class="n">beta_to_m_values</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">x_train_m</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">corr</span><span class="p">],</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                             <span class="n">keys</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="s1">&#39;Correlation&#39;</span><span class="p">],</span>
                             <span class="n">join</span><span class="o">=</span> <span class="s1">&#39;inner&#39;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="n">hue</span><span class="p">,</span><span class="n">s</span><span class="p">):</span>
    
    <span class="c1"># Define theme</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>
    <span class="n">f</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">despine</span><span class="p">(</span><span class="n">f</span><span class="p">,</span> <span class="n">left</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">bottom</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

    <span class="c1"># Define scatterplot</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
                    <span class="n">hue</span><span class="o">=</span><span class="n">hue</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;flare&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">s</span><span class="p">,</span>
                    <span class="n">sizes</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">8</span><span class="p">),</span> <span class="n">linewidth</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span>
                    <span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1"># Calculate Pearson’s correlation coefficient (r)</span>
    <span class="c1"># and its two-tailed p-value (p)</span>

    <span class="n">r</span><span class="p">,</span><span class="n">p</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">x</span><span class="p">],</span><span class="n">df</span><span class="p">[</span><span class="n">y</span><span class="p">])</span>

    <span class="c1"># Annotate r and p into the plot</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="sa">fr</span><span class="s1">&#39;$\rho$: </span><span class="si">{</span><span class="n">r</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">, p-value: </span><span class="si">{</span><span class="n">p</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span>
                <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mf">6.0</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">),</span> <span class="n">xycoords</span><span class="o">=</span><span class="s1">&#39;data&#39;</span><span class="p">,</span>
                <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span>
                <span class="n">bbox</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;boxstyle&#39;</span><span class="p">:</span> <span class="s1">&#39;round&#39;</span><span class="p">,</span> <span class="s1">&#39;facecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;none&#39;</span><span class="p">,</span>
                <span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;lightgray&#39;</span><span class="p">})</span>

    <span class="c1"># Define plot specs</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="n">y</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;Concordance by Pearson&#39;s correlation coefficient ($\rho$) in &quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">))</span> <span class="o">+</span> <span class="s1">&#39; CpG probes&#39;</span><span class="p">,</span>
               <span class="n">fontsize</span> <span class="o">=</span> <span class="mi">12</span><span class="p">)</span> 

    <span class="c1"># Save plot figure</span>
    
    <span class="n">plt</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../Figures/Beta_to_M_Values_&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">))</span> <span class="o">+</span> <span class="s1">&#39;_scatterplot&#39;</span> <span class="o">+</span> <span class="s1">&#39;.png&#39;</span><span class="p">,</span>
                 <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>

    <span class="k">return</span><span class="p">(</span><span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">())</span>

<span class="n">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">beta_to_m_values</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="n">hue</span><span class="o">=</span><span class="s1">&#39;Correlation&#39;</span><span class="p">,</span><span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/0602565a0a9d4e01800361f1d89f245e26e19bbda23232083e742e4c3ab6f490.png" src="_images/0602565a0a9d4e01800361f1d89f245e26e19bbda23232083e742e4c3ab6f490.png" />
</div>
</div>
</section>
<section id="run-coxph-ewas-unadjusted-with-overall-survival-as-outcome">
<h2><a class="toc-backref" href="#id9" role="doc-backlink">Run CoxPH-EWAS unadjusted with overall survival as outcome</a><a class="headerlink" href="#run-coxph-ewas-unadjusted-with-overall-survival-as-outcome" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">methylize</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>

<span class="c1"># ewasOS_un = methylize.diff_meth_pos(meth_data = x_train_m,</span>
<span class="c1">#                                     duration = df3[&#39;os.time&#39;],</span>
<span class="c1">#                                     event_observed = df3[&#39;os.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = df3[&#39;os.time&#39;], # not used</span>
<span class="c1">#                                     covariates = False,</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = True,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_cog_os_unadjusted&#39;)</span>

<span class="n">ewasOS_un</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_cog_os_unadjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">methylize</span><span class="o">.</span><span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewasOS_un</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; Children&#39;s Oncology Group AML clinical trials&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/a0fcc94918b58efcf1f9099ec1cc65e71d963987033bf2b508324972ad5ec75c.png" src="_images/a0fcc94918b58efcf1f9099ec1cc65e71d963987033bf2b508324972ad5ec75c.png" />
</div>
</div>
</section>
<section id="table-top-results-by-p-value">
<h2><a class="toc-backref" href="#id10" role="doc-backlink">Table top results by p-value</a><a class="headerlink" href="#table-top-results-by-p-value" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ewas_top</span> <span class="o">=</span> <span class="n">ewasOS_un</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">10</span><span class="p">)</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="o">-</span><span class="mi">3</span><span class="p">]</span>
<span class="n">ewas_top</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Coefficient</th>
      <th>StandardError</th>
      <th>HazardsRatio</th>
      <th>95%CI_lower</th>
      <th>95%CI_upper</th>
      <th>ZScore</th>
      <th>PValue</th>
      <th>FDR_QValue</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>cg10758466</th>
      <td>-0.240239</td>
      <td>0.044954</td>
      <td>0.786440</td>
      <td>0.720114</td>
      <td>0.858876</td>
      <td>-5.344147</td>
      <td>9.084404e-08</td>
      <td>0.026436</td>
    </tr>
    <tr>
      <th>cg02959551</th>
      <td>-0.292930</td>
      <td>0.055878</td>
      <td>0.746074</td>
      <td>0.668680</td>
      <td>0.832426</td>
      <td>-5.242278</td>
      <td>1.586062e-07</td>
      <td>0.026436</td>
    </tr>
    <tr>
      <th>cg06453088</th>
      <td>-0.147930</td>
      <td>0.028994</td>
      <td>0.862492</td>
      <td>0.814845</td>
      <td>0.912924</td>
      <td>-5.102090</td>
      <td>3.359236e-07</td>
      <td>0.037327</td>
    </tr>
    <tr>
      <th>cg08122885</th>
      <td>0.121177</td>
      <td>0.025894</td>
      <td>1.128825</td>
      <td>1.072965</td>
      <td>1.187593</td>
      <td>4.679711</td>
      <td>2.872789e-06</td>
      <td>0.239412</td>
    </tr>
    <tr>
      <th>cg22272840</th>
      <td>0.091995</td>
      <td>0.020261</td>
      <td>1.096360</td>
      <td>1.053675</td>
      <td>1.140773</td>
      <td>4.540507</td>
      <td>5.611907e-06</td>
      <td>0.302352</td>
    </tr>
    <tr>
      <th>cg14125846</th>
      <td>-0.370650</td>
      <td>0.082376</td>
      <td>0.690286</td>
      <td>0.587368</td>
      <td>0.811237</td>
      <td>-4.499463</td>
      <td>6.812542e-06</td>
      <td>0.302352</td>
    </tr>
    <tr>
      <th>cg10294820</th>
      <td>-0.162761</td>
      <td>0.036314</td>
      <td>0.849794</td>
      <td>0.791413</td>
      <td>0.912482</td>
      <td>-4.482065</td>
      <td>7.392431e-06</td>
      <td>0.302352</td>
    </tr>
    <tr>
      <th>cg18060030</th>
      <td>-0.173763</td>
      <td>0.038999</td>
      <td>0.840496</td>
      <td>0.778646</td>
      <td>0.907259</td>
      <td>-4.455637</td>
      <td>8.364463e-06</td>
      <td>0.302352</td>
    </tr>
    <tr>
      <th>cg09755939</th>
      <td>-0.376726</td>
      <td>0.084731</td>
      <td>0.686104</td>
      <td>0.581122</td>
      <td>0.810052</td>
      <td>-4.446153</td>
      <td>8.742158e-06</td>
      <td>0.302352</td>
    </tr>
    <tr>
      <th>cg08259755</th>
      <td>-0.250416</td>
      <td>0.057035</td>
      <td>0.778477</td>
      <td>0.696141</td>
      <td>0.870551</td>
      <td>-4.390562</td>
      <td>1.130581e-05</td>
      <td>0.302352</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="fetch-genes-from-top-cpgs">
<h2><a class="toc-backref" href="#id11" role="doc-backlink">Fetch genes from top CpGs</a><a class="headerlink" href="#fetch-genes-from-top-cpgs" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../Data/UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span><span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span> <span class="s1">&#39;TranscriptIDs&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">ewasOS_un_genes</span> <span class="o">=</span> <span class="n">zhou2016_probes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">ewas_top</span><span class="p">[[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]],</span><span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span>
<span class="n">ewasOS_un_genes</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>TranscriptIDs</th>
      <th>CGIPosition</th>
      <th>PValue</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>cg10758466</th>
      <td>chr17</td>
      <td>43896718</td>
      <td>43896720</td>
      <td>MPP2;MPP2;MPP2;MPP2;MPP2;MPP2;MPP2;MPP2;MPP2;M...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000269095.7;ENST00000377184.6;ENST000004...</td>
      <td>N_Shelf</td>
      <td>9.084404e-08</td>
    </tr>
    <tr>
      <th>cg02959551</th>
      <td>chr6</td>
      <td>31659208</td>
      <td>31659210</td>
      <td>C6orf47;C6orf47-AS1</td>
      <td>protein_coding;antisense</td>
      <td>ENST00000375911.1;ENST00000422049.1</td>
      <td>N_Shore</td>
      <td>1.586062e-07</td>
    </tr>
    <tr>
      <th>cg06453088</th>
      <td>chr4</td>
      <td>5885997</td>
      <td>5885999</td>
      <td>CRMP1;CRMP1;CRMP1;CRMP1;CRMP1</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000324989.10;ENST00000397890.5;ENST00000...</td>
      <td>N_Shore</td>
      <td>3.359236e-07</td>
    </tr>
    <tr>
      <th>cg08122885</th>
      <td>chr7</td>
      <td>38976449</td>
      <td>38976451</td>
      <td>.</td>
      <td>.</td>
      <td>.</td>
      <td>S_Shore</td>
      <td>2.872789e-06</td>
    </tr>
    <tr>
      <th>cg22272840</th>
      <td>chr8</td>
      <td>10729371</td>
      <td>10729373</td>
      <td>CTD-2135J3.3;CTD-2135J3.3;CTD-2135J3.4;SOX7</td>
      <td>antisense;antisense;protein_coding;protein_coding</td>
      <td>ENST00000506149.2;ENST00000519568.1;ENST000005...</td>
      <td>Island</td>
      <td>5.611907e-06</td>
    </tr>
    <tr>
      <th>cg14125846</th>
      <td>chr19</td>
      <td>38948540</td>
      <td>38948542</td>
      <td>CTC-360G5.8;FBXO17;FBXO17;FBXO17;FBXO17;FBXO17...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000599996.1;ENST00000292852.7;ENST000005...</td>
      <td>N_Shore</td>
      <td>6.812542e-06</td>
    </tr>
    <tr>
      <th>cg10294820</th>
      <td>chr16</td>
      <td>86196091</td>
      <td>86196093</td>
      <td>LINC01082;RP11-805I24.3</td>
      <td>lincRNA;lincRNA</td>
      <td>ENST00000601250.1;ENST00000599841.1</td>
      <td>.</td>
      <td>7.392431e-06</td>
    </tr>
    <tr>
      <th>cg18060030</th>
      <td>chr3</td>
      <td>169249449</td>
      <td>169249451</td>
      <td>MECOM;MECOM;MECOM</td>
      <td>protein_coding;protein_coding;protein_coding</td>
      <td>ENST00000481315.1;ENST00000485957.1;ENST000004...</td>
      <td>.</td>
      <td>8.364463e-06</td>
    </tr>
    <tr>
      <th>cg09755939</th>
      <td>chr11</td>
      <td>15115141</td>
      <td>15115143</td>
      <td>INSC;INSC</td>
      <td>protein_coding;protein_coding</td>
      <td>ENST00000379554.6;ENST00000379556.6</td>
      <td>S_Shore</td>
      <td>8.742158e-06</td>
    </tr>
    <tr>
      <th>cg08259755</th>
      <td>chr8</td>
      <td>55525160</td>
      <td>55525162</td>
      <td>XKR4;XKR4;XKR4</td>
      <td>protein_coding;protein_coding;protein_coding</td>
      <td>ENST00000327381.6;ENST00000518261.1;ENST000006...</td>
      <td>.</td>
      <td>1.130581e-05</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="watermark">
<h2><a class="toc-backref" href="#id12" role="doc-backlink">Watermark</a><a class="headerlink" href="#watermark" title="Permalink to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Python implementation: CPython
Python version       : 3.8.16
IPython version      : 8.12.2

methylize: 1.1.1
pandas   : 2.0.1

Compiler    : GCC 11.3.0
OS          : Linux
Release     : 5.15.90.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 20
Architecture: 64bit
</pre></div>
</div>
</div>
</div>
</section>
<div class="toctree-wrapper compound">
</div>
<div class="toctree-wrapper compound">
</div>
<div class="toctree-wrapper compound">
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="right-next"
       href="docs/Specific_Aims.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Specific Aims</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-process-clinical-data">Load and process clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove Samples based on Certain Clinical Features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-bone-marrow-samples-only">Select diagnostic bone marrow samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas-unadjusted-with-overall-survival-as-outcome">Run CoxPH-EWAS unadjusted with overall survival as outcome</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#table-top-results-by-p-value">Table top results by p-value</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-genes-from-top-cpgs">Fetch genes from top CpGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Francisco Marchi
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>