

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome &#8212; Lamba Lab at UF</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '2_Univariate_CoxPH-EWAS_v2';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="CoxPH-Lasso on top EWAS results" href="3_Multivariate_CoxPH-Lasso_v2.html" />
    <link rel="prev" title="Data Download and Processing" href="1_Data_Download_Process.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
  
    <p class="title logo__title">Lamba Lab at UF</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="index.html">
                    DNA methylation-based diagnosis and prognosis of pediatric AML
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Specific Aims</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="docs/Specific_Aims.html">Specific Aims</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="docs/Intro_to_Methylation.html">Understanding DNA Methylation</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Data Preparation</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="1_Data_Download_Process.html">Data Download and Processing</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_Multivariate_CoxPH-Lasso_v2.html">CoxPH-Lasso on top EWAS results</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/2_Univariate_CoxPH-EWAS_v2.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-bone-marrow-samples-only">Select diagnostic bone marrow samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas-os">Run CoxPH-EWAS OS</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#unadjusted">Unadjusted</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#risk-group-adjusted">Risk group adjusted</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas-efs">Run CoxPH-EWAS EFS</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Unadjusted</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Risk group adjusted</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-statistically-significant-cpgs">Select statistically significant CpGs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#threshold-based-on-suggested-significance-p-values-1-10e-6">Threshold based on suggested-significance p-values (1*10e-6)</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-genes">Fetch genes</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#os-suggested-significance-probes">OS suggested-significance probes:</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#efs-suggested-significance-probes">EFS suggested-significance probes:</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#standardize-and-save-top-probes-risk-adjusted-os-at-1-10e-6">Standardize and save top probes (risk-adjusted OS at 1*10e-6)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="coxph-ewas-unadjusted-in-cog-trials-with-os-and-efs-as-outcome">
<h1><a class="toc-backref" href="#id3" role="doc-backlink">CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</a><a class="headerlink" href="#coxph-ewas-unadjusted-in-cog-trials-with-os-and-efs-as-outcome" title="Permalink to this heading">#</a></h1>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#coxph-ewas-unadjusted-in-cog-trials-with-os-and-efs-as-outcome" id="id3">CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</a></p>
<ul>
<li><p><a class="reference internal" href="#load-data" id="id4">Load data</a></p></li>
<li><p><a class="reference internal" href="#remove-samples-based-on-certain-clinical-features" id="id5">Remove samples based on certain clinical features</a></p>
<ul>
<li><p><a class="reference internal" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" id="id6">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></p></li>
<li><p><a class="reference internal" href="#select-diagnostic-bone-marrow-samples-only" id="id7">Select diagnostic bone marrow samples only</a></p></li>
<li><p><a class="reference internal" href="#remove-duplicate-samples" id="id8">Remove duplicate samples</a></p></li>
<li><p><a class="reference internal" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" id="id9">Match samples in clinical data to samples in methylation data</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#transform-to-m-values" id="id10">Transform to M-values</a></p></li>
<li><p><a class="reference internal" href="#perform-batch-correction" id="id11">Perform batch correction</a></p>
<ul>
<li><p><a class="reference internal" href="#evaluate-batch-effects" id="id12">Evaluate batch effects</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#run-coxph-ewas-os" id="id13">Run CoxPH-EWAS OS</a></p>
<ul>
<li><p><a class="reference internal" href="#unadjusted" id="id14">Unadjusted</a></p></li>
<li><p><a class="reference internal" href="#risk-group-adjusted" id="id15">Risk group adjusted</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#run-coxph-ewas-efs" id="id16">Run CoxPH-EWAS EFS</a></p>
<ul>
<li><p><a class="reference internal" href="#id1" id="id17">Unadjusted</a></p></li>
<li><p><a class="reference internal" href="#id2" id="id18">Risk group adjusted</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#select-statistically-significant-cpgs" id="id19">Select statistically significant CpGs</a></p>
<ul>
<li><p><a class="reference internal" href="#threshold-based-on-suggested-significance-p-values-1-10e-6" id="id20">Threshold based on suggested-significance p-values (1*10e-6)</a></p>
<ul>
<li><p><a class="reference internal" href="#fetch-genes" id="id21">Fetch genes</a></p></li>
<li><p><a class="reference internal" href="#os-suggested-significance-probes" id="id22">OS suggested-significance probes:</a></p></li>
<li><p><a class="reference internal" href="#efs-suggested-significance-probes" id="id23">EFS suggested-significance probes:</a></p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference internal" href="#standardize-and-save-top-probes-risk-adjusted-os-at-1-10e-6" id="id24">Standardize and save top probes (risk-adjusted OS at 1*10e-6)</a></p></li>
<li><p><a class="reference internal" href="#watermark" id="id25">Watermark</a></p></li>
</ul>
</li>
</ul>
</nav>
<section id="load-data">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">Load data</a><a class="headerlink" href="#load-data" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">input_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Intermediate_Files/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Processed_Data/&#39;</span>


<span class="n">df_methyl</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;df_batch_uncorrected.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="n">df_labels</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;clinical_data.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">low_memory</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39; Dataset (df) contains </span><span class="si">{</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> columns (5mC nucleotides/probes) and </span><span class="si">{</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> rows (samples).&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> Dataset (df) contains 333352 columns (5mC nucleotides/probes) and 3330 rows (samples).
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-samples-based-on-certain-clinical-features">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Remove samples based on certain clinical features</a><a class="headerlink" href="#remove-samples-based-on-certain-clinical-features" title="Permalink to this heading">#</a></h2>
<section id="select-samples-from-aaml1031-0531-and-03p1-clinical-trials">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Select samples from AAML1031, 0531, and 03P1 clinical trials</a><a class="headerlink" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df1</span> <span class="o">=</span> <span class="n">df_labels</span><span class="p">[</span><span class="n">df_labels</span><span class="p">[</span><span class="s1">&#39;Clinical Trial&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;AAML0531&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML1031&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML03P1&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df_labels</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2049 samples were removed. 1281 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="select-diagnostic-bone-marrow-samples-only">
<h3><a class="toc-backref" href="#id7" role="doc-backlink">Select diagnostic bone marrow samples only</a><a class="headerlink" href="#select-diagnostic-bone-marrow-samples-only" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df2</span> <span class="o">=</span> <span class="n">df1</span><span class="p">[</span><span class="n">df1</span><span class="p">[</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Diagnosis&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Bone Marrow&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>350 samples were removed. 931 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-duplicate-samples">
<h3><a class="toc-backref" href="#id8" role="doc-backlink">Remove duplicate samples</a><a class="headerlink" href="#remove-duplicate-samples" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df3</span> <span class="o">=</span> <span class="n">df2</span><span class="p">[</span><span class="o">~</span><span class="n">df2</span><span class="p">[</span><span class="s1">&#39;Patient_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="s1">&#39;last&#39;</span><span class="p">)]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>7 samples were removed. 924 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="match-samples-in-clinical-data-to-samples-in-methylation-data">
<h3><a class="toc-backref" href="#id9" role="doc-backlink">Match samples in clinical data to samples in methylation data</a><a class="headerlink" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Match samples in clinical data to samples in methylation data</span>
<span class="n">df_methyl_filtered</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">:]</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Samples in clinical data matched to samples in methylation data.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Samples in clinical data matched to samples in methylation data.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="transform-to-m-values">
<h2><a class="toc-backref" href="#id10" role="doc-backlink">Transform to M-values</a><a class="headerlink" href="#transform-to-m-values" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>


<span class="k">def</span> <span class="nf">beta2m</span><span class="p">(</span><span class="n">val</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Transfrom beta-values into m-values&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">math</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">val</span><span class="o">/</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">val</span><span class="p">))</span>

<span class="n">x_train_m</span> <span class="o">=</span> <span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>

<span class="c1"># Get correlation between Beta and M Values</span>
<span class="n">corr</span> <span class="o">=</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">corrwith</span><span class="p">(</span><span class="n">df_methyl_filtered</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Glue them into same dataframe for vizualization purposes</span>
<span class="n">beta_to_m_values</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">x_train_m</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">corr</span><span class="p">],</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                             <span class="n">keys</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="s1">&#39;Correlation&#39;</span><span class="p">],</span>
                             <span class="n">join</span><span class="o">=</span> <span class="s1">&#39;inner&#39;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="n">hue</span><span class="p">,</span><span class="n">s</span><span class="p">):</span>
    
    <span class="c1"># Define theme</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>
    <span class="n">f</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">despine</span><span class="p">(</span><span class="n">f</span><span class="p">,</span> <span class="n">left</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">bottom</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

    <span class="c1"># Define scatterplot</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
                    <span class="n">hue</span><span class="o">=</span><span class="n">hue</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;flare&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">s</span><span class="p">,</span>
                    <span class="n">sizes</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">8</span><span class="p">),</span> <span class="n">linewidth</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span>
                    <span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1"># Calculate Pearsonâ€™s correlation coefficient (r)</span>
    <span class="c1"># and its two-tailed p-value (p)</span>

    <span class="n">r</span><span class="p">,</span><span class="n">p</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">x</span><span class="p">],</span><span class="n">df</span><span class="p">[</span><span class="n">y</span><span class="p">])</span>

    <span class="c1"># Annotate r and p into the plot</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="sa">fr</span><span class="s1">&#39;$\rho$: </span><span class="si">{</span><span class="n">r</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">, p-value: </span><span class="si">{</span><span class="n">p</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span>
                <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mf">6.0</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">),</span> <span class="n">xycoords</span><span class="o">=</span><span class="s1">&#39;data&#39;</span><span class="p">,</span>
                <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span>
                <span class="n">bbox</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;boxstyle&#39;</span><span class="p">:</span> <span class="s1">&#39;round&#39;</span><span class="p">,</span> <span class="s1">&#39;facecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;none&#39;</span><span class="p">,</span>
                <span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;lightgray&#39;</span><span class="p">})</span>

    <span class="c1"># Define plot specs</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="n">y</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;Concordance by Pearson&#39;s correlation coefficient ($\rho$) in &quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">))</span> <span class="o">+</span> <span class="s1">&#39; CpG probes&#39;</span><span class="p">,</span>
               <span class="n">fontsize</span> <span class="o">=</span> <span class="mi">12</span><span class="p">)</span> 

    <span class="c1"># Save plot figure</span>
    
    <span class="c1"># plt.savefig(&#39;../Figures/Beta_to_M_Values_&#39; + str(len(df)) + &#39;_scatterplot&#39; + &#39;.png&#39;,</span>
    <span class="c1">#              bbox_inches=&#39;tight&#39;, dpi=300)</span>

    <span class="k">return</span><span class="p">(</span><span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">())</span>

<span class="n">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">beta_to_m_values</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="n">hue</span><span class="o">=</span><span class="s1">&#39;Correlation&#39;</span><span class="p">,</span><span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/d427fda8c2d9324d392e13b8b2e9e0a19df3d6ea893b3a323da06059c5ed09e2.png" src="_images/d427fda8c2d9324d392e13b8b2e9e0a19df3d6ea893b3a323da06059c5ed09e2.png" />
</div>
</div>
</section>
<section id="perform-batch-correction">
<h2><a class="toc-backref" href="#id11" role="doc-backlink">Perform batch correction</a><a class="headerlink" href="#perform-batch-correction" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>pyCombat</strong> is a Python tool for batch effects correction in high-throughput molecular data using empirical Bayes methods</p></li>
<li><p><strong>Github</strong>: <a class="reference external" href="https://epigenelabs.github.io/pyComBat/">https://epigenelabs.github.io/pyComBat/</a></p></li>
<li><p><strong>Implementation Preprint</strong>: <a class="reference external" href="https://doi.org/10.1101/2020.03.17.995431">bioRxiv</a></p></li>
<li><p><strong>Original Paper</strong>: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/16632515/">Adjusting batch effects in microarray expression data using empirical Bayes methods</a></p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">combat.pycombat</span> <span class="kn">import</span> <span class="n">pycombat</span>

<span class="c1"># Correct batch effects in the training dataset</span>
<span class="n">df_batchcorrected</span> <span class="o">=</span> <span class="n">pycombat</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">batch</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully corrected batch effects in the training dataset.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Found 3 batches.
Adjusting for 0 covariate(s) or covariate level(s).
Standardizing Data across genes.
Fitting L/S model and finding priors.
Finding parametric adjustments.
Adjusting the Data
Succesfully corrected batch effects in the training dataset.
</pre></div>
</div>
</div>
</div>
<section id="evaluate-batch-effects">
<h3><a class="toc-backref" href="#id12" role="doc-backlink">Evaluate batch effects</a><a class="headerlink" href="#evaluate-batch-effects" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">sklearn.preprocessing</span> <span class="kn">import</span> <span class="n">StandardScaler</span>
<span class="kn">from</span> <span class="nn">sklearn.decomposition</span> <span class="kn">import</span> <span class="n">PCA</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="k">def</span> <span class="nf">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Standardizes the data.&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">StandardScaler</span><span class="p">()</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">perform_pca</span><span class="p">(</span><span class="n">std_data</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Performs PCA on standardized data.&#39;&#39;&#39;</span>
    <span class="n">pca</span> <span class="o">=</span> <span class="n">PCA</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="mi">42</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">pca</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">std_data</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">create_principal_df</span><span class="p">(</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Creates a DataFrame with the PCA data.&#39;&#39;&#39;</span>
    <span class="n">principal_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="s1">&#39;PC2&#39;</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span>
    <span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;hue&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">principal_df</span>

<span class="k">def</span> <span class="nf">filter_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">principal_df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Filters the DataFrame by removing outliers.&#39;&#39;&#39;</span>
    <span class="n">outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">threshold</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">outliers</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df_batchcorrected</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_batchcorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df_batchcorrected</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_uncorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Plots PCA data.&#39;&#39;&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">,</span> <span class="n">legend</span> <span class="ow">in</span> <span class="p">[(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA after batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="s1">&#39;full&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="n">non_outliers</span><span class="p">,</span> <span class="n">ax1</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA before batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">non_outliers</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="kc">False</span><span class="p">)]:</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PC2&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;hue&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="n">legend</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;PC 1&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;PC 2&#39;</span><span class="p">)</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.05</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">plot_data</span><span class="p">(</span><span class="n">pcs_batchcorrected</span><span class="p">,</span> <span class="n">pcs_uncorrected</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/c1d98325e380f09b01735341403a11342d2048df65771a3aa9a625290a974759.png" src="_images/c1d98325e380f09b01735341403a11342d2048df65771a3aa9a625290a974759.png" />
</div>
</div>
</section>
</section>
<section id="run-coxph-ewas-os">
<h2><a class="toc-backref" href="#id13" role="doc-backlink">Run CoxPH-EWAS OS</a><a class="headerlink" href="#run-coxph-ewas-os" title="Permalink to this heading">#</a></h2>
<section id="unadjusted">
<h3><a class="toc-backref" href="#id14" role="doc-backlink">Unadjusted</a><a class="headerlink" href="#unadjusted" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">methylize</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>

<span class="c1"># ewas_os_cog_unadjusted = methylize.diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = df3[&#39;os.time&#39;],</span>
<span class="c1">#                                     event_observed = df3[&#39;os.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = df3[&#39;os.time&#39;], # not used</span>
<span class="c1">#                                     covariates = False,</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = True,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected&#39;)</span>

<span class="n">ewas_os_cog_unadjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_unadjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">methylize</span><span class="o">.</span><span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_os_cog_unadjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG OS risk unadjusted&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/4a2f2f4a5c6ea5a77869ef8900ee5223a227989de1a7009024000b5d078dae2c.png" src="_images/4a2f2f4a5c6ea5a77869ef8900ee5223a227989de1a7009024000b5d078dae2c.png" />
</div>
</div>
</section>
<section id="risk-group-adjusted">
<h3><a class="toc-backref" href="#id15" role="doc-backlink">Risk group adjusted</a><a class="headerlink" href="#risk-group-adjusted" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df3</span> <span class="o">=</span> <span class="n">df3</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">Risk_Group_Bin</span><span class="o">=</span><span class="n">df3</span><span class="p">[</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">({</span><span class="s1">&#39;Low Risk&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">:</span><span class="mf">0.5</span><span class="p">,</span><span class="s1">&#39;High Risk&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">}))</span>

<span class="k">def</span> <span class="nf">impute_by_mean</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Impute missing values by computing feature average&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">mean</span><span class="p">())</span>

<span class="n">covar_os</span> <span class="o">=</span> <span class="nb">round</span><span class="p">(</span><span class="n">impute_by_mean</span><span class="p">(</span><span class="n">df3</span><span class="p">[[</span><span class="s1">&#39;os.evnt&#39;</span><span class="p">,</span><span class="s1">&#39;os.time&#39;</span><span class="p">,</span><span class="s1">&#39;Risk_Group_Bin&#39;</span><span class="p">]]),</span><span class="mi">3</span><span class="p">)</span>

<span class="c1"># ewas_os_cog_adjusted = methylize.diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = covar_os[&#39;os.time&#39;],</span>
<span class="c1">#                                     event_observed = covar_os[&#39;os.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = covar_os[&#39;os.time&#39;], # not used</span>
<span class="c1">#                                     covariates = covar_os[&#39;Risk_Group_Bin&#39;],</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = False,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted&#39;)</span>

<span class="n">ewas_os_cog_adjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">methylize</span><span class="o">.</span><span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_os_cog_adjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG OS risk adjusted&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/6e9d642dbe03cd917cc695023e083c6e98bf5ff10f1125d8417f3885f62a83f1.png" src="_images/6e9d642dbe03cd917cc695023e083c6e98bf5ff10f1125d8417f3885f62a83f1.png" />
</div>
</div>
</section>
</section>
<section id="run-coxph-ewas-efs">
<h2><a class="toc-backref" href="#id16" role="doc-backlink">Run CoxPH-EWAS EFS</a><a class="headerlink" href="#run-coxph-ewas-efs" title="Permalink to this heading">#</a></h2>
<section id="id1">
<h3><a class="toc-backref" href="#id17" role="doc-backlink">Unadjusted</a><a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">methylize</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>

<span class="c1"># ewas_efs_cog_unadjusted = methylize.diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = df3[&#39;efs.time&#39;],</span>
<span class="c1">#                                     event_observed = df3[&#39;efs.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = df3[&#39;efs.time&#39;], # not used</span>
<span class="c1">#                                     covariates = False,</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = True,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_efs_batch_corrected&#39;)</span>

<span class="n">ewas_efs_cog_unadjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_efs_batch_corrected_risk_unadjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">methylize</span><span class="o">.</span><span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_efs_cog_unadjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG EFS risk unadjusted&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_efs_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/0dd041c906815fd62db71241b8655995d0db00d93c4c081204e48d5944a2399e.png" src="_images/0dd041c906815fd62db71241b8655995d0db00d93c4c081204e48d5944a2399e.png" />
</div>
</div>
</section>
<section id="id2">
<h3><a class="toc-backref" href="#id18" role="doc-backlink">Risk group adjusted</a><a class="headerlink" href="#id2" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df3</span> <span class="o">=</span> <span class="n">df3</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">Risk_Group_Bin</span><span class="o">=</span><span class="n">df3</span><span class="p">[</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">({</span><span class="s1">&#39;Low Risk&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">:</span><span class="mf">0.5</span><span class="p">,</span><span class="s1">&#39;High Risk&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">}))</span>

<span class="k">def</span> <span class="nf">impute_by_mean</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Impute missing values by computing feature average&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">mean</span><span class="p">())</span>

<span class="n">covar_efs</span> <span class="o">=</span> <span class="nb">round</span><span class="p">(</span><span class="n">impute_by_mean</span><span class="p">(</span><span class="n">df3</span><span class="p">[[</span><span class="s1">&#39;efs.evnt&#39;</span><span class="p">,</span><span class="s1">&#39;efs.time&#39;</span><span class="p">,</span><span class="s1">&#39;Risk_Group_Bin&#39;</span><span class="p">]]),</span><span class="mi">3</span><span class="p">)</span>

<span class="c1"># ewas_efs_cog_adjusted = methylize.diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = covar_efs[&#39;efs.time&#39;],</span>
<span class="c1">#                                     event_observed = covar_efs[&#39;efs.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = covar_efs[&#39;efs.time&#39;], # not used</span>
<span class="c1">#                                     covariates = covar_efs[&#39;Risk_Group_Bin&#39;],</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = False,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_efs_batch_corrected_risk_adjusted&#39;)</span>

<span class="n">ewas_efs_cog_adjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_efs_batch_corrected_risk_adjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">methylize</span><span class="o">.</span><span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_efs_cog_adjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG EFS risk adjusted&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_efs_adjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/dd3762ae14288849435bebb2ae4b1a5c0c484178ed260f39476f23b5b205e87c.png" src="_images/dd3762ae14288849435bebb2ae4b1a5c0c484178ed260f39476f23b5b205e87c.png" />
</div>
</div>
</section>
</section>
<section id="select-statistically-significant-cpgs">
<h2><a class="toc-backref" href="#id19" role="doc-backlink">Select statistically significant CpGs</a><a class="headerlink" href="#select-statistically-significant-cpgs" title="Permalink to this heading">#</a></h2>
<section id="threshold-based-on-suggested-significance-p-values-1-10e-6">
<h3><a class="toc-backref" href="#id20" role="doc-backlink">Threshold based on suggested-significance p-values (1*10e-6)</a><a class="headerlink" href="#threshold-based-on-suggested-significance-p-values-1-10e-6" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">select_rows</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">threshold</span><span class="p">]</span>

<span class="k">def</span> <span class="nf">get_intersection</span><span class="p">(</span><span class="n">df1</span><span class="p">,</span> <span class="n">df2</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df1</span><span class="p">[</span><span class="n">df1</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df2</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># store the dataframes in a dictionary</span>
<span class="n">data_dict</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Unadjusted&#39;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s1">&#39;os&#39;</span><span class="p">:</span> <span class="n">ewas_os_cog_unadjusted</span><span class="p">,</span>
        <span class="s1">&#39;efs&#39;</span><span class="p">:</span> <span class="n">ewas_efs_cog_unadjusted</span><span class="p">,</span>
    <span class="p">},</span>
    <span class="s1">&#39;Adjusted&#39;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s1">&#39;os&#39;</span><span class="p">:</span> <span class="n">ewas_os_cog_adjusted</span><span class="p">,</span>
        <span class="s1">&#39;efs&#39;</span><span class="p">:</span> <span class="n">ewas_efs_cog_adjusted</span><span class="p">,</span>
    <span class="p">}</span>
<span class="p">}</span>

<span class="c1"># empty dictionary to store the results</span>
<span class="n">results</span> <span class="o">=</span> <span class="p">{}</span>

<span class="c1"># loop through the data_dict</span>
<span class="k">for</span> <span class="n">condition</span><span class="p">,</span> <span class="n">dfs</span> <span class="ow">in</span> <span class="n">data_dict</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">top_dfs</span> <span class="o">=</span> <span class="p">{</span><span class="n">key</span><span class="p">:</span> <span class="n">select_rows</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="mi">1</span><span class="o">*</span><span class="mf">10e-6</span><span class="p">)</span> <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">df</span> <span class="ow">in</span> <span class="n">dfs</span><span class="o">.</span><span class="n">items</span><span class="p">()}</span>
    <span class="n">intersection</span> <span class="o">=</span> <span class="n">get_intersection</span><span class="p">(</span><span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">],</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">])</span>

    <span class="n">results</span><span class="p">[</span><span class="n">condition</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;os&#39;</span><span class="p">:</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="s1">&#39;efs&#39;</span><span class="p">:</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="s1">&#39;intersection&#39;</span><span class="p">:</span> <span class="n">intersection</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
    <span class="p">}</span>

<span class="c1"># convert results to a pandas DataFrame and display</span>
<span class="n">result_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">results</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>
<span class="n">result_df</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Significant CpG probes in OS&#39;</span><span class="p">,</span> 
                    <span class="s1">&#39;Significant CpG probes in EFS&#39;</span><span class="p">,</span>
                    <span class="s1">&#39;Significant overlapping CpG probes&#39;</span><span class="p">]</span>
<span class="n">result_df</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Significant CpG probes in OS</th>
      <th>Significant CpG probes in EFS</th>
      <th>Significant overlapping CpG probes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Unadjusted</th>
      <td>3740</td>
      <td>2634</td>
      <td>1741</td>
    </tr>
    <tr>
      <th>Adjusted</th>
      <td>167</td>
      <td>55</td>
      <td>17</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<section id="fetch-genes">
<h4><a class="toc-backref" href="#id21" role="doc-backlink">Fetch genes</a><a class="headerlink" href="#fetch-genes" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Define a function to handle repetitive tasks</span>
<span class="k">def</span> <span class="nf">process_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">other_df</span><span class="p">,</span> <span class="n">column_drop</span><span class="p">):</span>
    <span class="c1"># Join and sort values</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">other_df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">column_drop</span><span class="p">),</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span>

    <span class="c1"># Handle columns with multiple genes separated by `;`</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneClasses&#39;</span><span class="p">]:</span>
        <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">))))</span>

    <span class="k">return</span> <span class="n">df</span>

<span class="c1"># Load manifest</span>
<span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../Data/UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span>
                                                     <span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">top_efs</span> <span class="o">=</span> <span class="n">process_df</span><span class="p">(</span><span class="n">zhou2016_probes</span><span class="p">,</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;chromosome&#39;</span><span class="p">,</span> <span class="s1">&#39;MAPINFO&#39;</span><span class="p">])</span>
<span class="n">top_os</span> <span class="o">=</span> <span class="n">process_df</span><span class="p">(</span><span class="n">zhou2016_probes</span><span class="p">,</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;chromosome&#39;</span><span class="p">,</span> <span class="s1">&#39;MAPINFO&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="os-suggested-significance-probes">
<h4><a class="toc-backref" href="#id22" role="doc-backlink">OS suggested-significance probes:</a><a class="headerlink" href="#os-suggested-significance-probes" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">itables</span> <span class="kn">import</span> <span class="n">show</span>

<span class="n">show</span><span class="p">(</span><span class="n">top_os</span><span class="p">,</span> <span class="n">classes</span><span class="o">=</span><span class="s2">&quot;display nowrap cell-border&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><style>.itables table td {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables table th {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables thead input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}

.itables tfoot input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}
</style>
<div class="itables">
<table id="9cc509ba-8ae9-47c2-aa29-147c140a9924" class="display nowrap cell-border"style="table-layout:auto;width:auto;margin:auto;caption-side:bottom"><thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>CGIPosition</th>
      <th>Coefficient</th>
      <th>StandardError</th>
      <th>HazardsRatio</th>
      <th>95%CI_lower</th>
      <th>95%CI_upper</th>
      <th>ZScore</th>
      <th>PValue</th>
      <th>FDR_QValue</th>
      <th>minuslog10value</th>
    </tr>
  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css">
<script type="module">
    // Import jquery and DataTable
    import 'https://code.jquery.com/jquery-3.6.0.min.js';
    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';
    dt($);

    // Define the table data
    const data = [["cg08900363", "chr7", 597816, 597818, "PRKAR1B", "protein_coding", ".", -0.331078, 0.050911, 0.718149, 0.649949, 0.793506, -6.503084, 7.868996e-11, 2.6e-05, 10.104081], ["cg14928764", "chr20", 23083970, 23083972, "CD93", "protein_coding", ".", -0.224246, 0.035119, 0.799119, 0.745964, 0.856061, -6.385291, 1.710718e-10, 2.9e-05, 9.766821], ["cg01543603", "chr16", 89403132, 89403134, "RP1-168P16.2;ANKRD11", "protein_coding;sense_intronic", ".", 0.211623, 0.037018, 1.235682, 1.149203, 1.32867, 5.716696, 1.086149e-08, 0.001207, 7.964111], ["cg00192882", "chr17", 19387806, 19387808, "MFAP4", "protein_coding", ".", 0.28153, 0.049928, 1.325155, 1.201621, 1.46139, 5.638657, 1.713814e-08, 0.001428, 7.766036], ["cg08404702", "chr16", 2083386, 2083388, "TSC2", "protein_coding", "N_Shore", 0.134977, 0.024658, 1.14451, 1.090513, 1.201181, 5.474013, 4.399567e-08, 0.002732, 7.35659], ["cg06339275", "chr3", 62948996, 62948998, "LINC00698", "lincRNA", ".", -0.187293, 0.034339, 0.829201, 0.77523, 0.886929, -5.454276, 4.917286e-08, 0.002732, 7.308275], ["cg19432033", "chr17", 7613901, 7613903, "FXR2;SHBG", "protein_coding", "N_Shore", -0.229545, 0.042333, 0.794895, 0.731604, 0.863662, -5.422394, 5.880609e-08, 0.0028, 7.230578], ["cg07660635", "chr16", 85902540, 85902542, "IRF8", "protein_coding", "S_Shelf", -0.172882, 0.032175, 0.841237, 0.789824, 0.895995, -5.373152, 7.737196e-08, 0.002848, 7.111416], ["cg05480169", "chr10", 133253903, 133253905, ".", ".", ".", -0.173415, 0.032365, 0.840789, 0.78911, 0.895852, -5.35804, 8.412983e-08, 0.002848, 7.07505], ["cg05348324", "chr11", 119797455, 119797457, ".", ".", ".", 0.189851, 0.035451, 1.20907, 1.127912, 1.296067, 5.355263, 8.543183e-08, 0.002848, 7.06838], ["cg18886274", "chr7", 85160554, 85160556, "SEMA3D", "protein_coding", ".", -0.185516, 0.03503, 0.830675, 0.775558, 0.88971, -5.296003, 1.183652e-07, 0.003467, 6.926776], ["cg10009497", "chr7", 1055779, 1055781, "GPR146;C7orf50;RP11-449P15.1", "protein_coding;antisense", "N_Shore", 0.168837, 0.031957, 1.183928, 1.112047, 1.260455, 5.283209, 1.269405e-07, 0.003467, 6.8964], ["cg20703895", "chr1", 157438011, 157438013, ".", ".", ".", -0.455202, 0.086813, 0.63432, 0.535073, 0.751976, -5.243468, 1.575859e-07, 0.003467, 6.802483], ["cg21750233", "chr11", 6483080, 6483082, "TIMM10B", "protein_coding", "S_Shore", -0.231931, 0.04425, 0.793001, 0.727123, 0.864847, -5.241377, 1.593827e-07, 0.003467, 6.797559], ["cg20018344", "chr20", 3223491, 3223493, "ITPA", "protein_coding", ".", 0.209347, 0.040028, 1.232873, 1.139846, 1.333492, 5.229986, 1.69523e-07, 0.003467, 6.770771], ["cg23128949", "chr12", 124501577, 124501579, "NCOR2", "protein_coding", ".", 0.138234, 0.026433, 1.148244, 1.090272, 1.209299, 5.229683, 1.698006e-07, 0.003467, 6.770061], ["cg11823637", "chr12", 22729886, 22729888, "RP11-114G22.1", "lincRNA", ".", -0.360643, 0.069361, 0.697228, 0.608604, 0.798757, -5.199515, 1.998096e-07, 0.003467, 6.699384], ["cg25292309", "chr2", 58762291, 58762293, "LINC01122", "lincRNA", ".", -0.250884, 0.04843, 0.778113, 0.70765, 0.855592, -5.180296, 2.215339e-07, 0.003467, 6.65456], ["cg26223555", "chr19", 51824187, 51824189, "FPR3", "protein_coding", ".", -0.32781, 0.063281, 0.7205, 0.636456, 0.815641, -5.180204, 2.21643e-07, 0.003467, 6.654346], ["cg00920892", "chr8", 37874705, 37874707, "RAB11FIP1", "protein_coding", ".", 0.21351, 0.041233, 1.238016, 1.141902, 1.342219, 5.178183, 2.240579e-07, 0.003467, 6.64964], ["cg15787807", "chr11", 64741618, 64741620, "RASGRP2", "protein_coding", "N_Shore", -0.211016, 0.04077, 0.809761, 0.747572, 0.877123, -5.175756, 2.269894e-07, 0.003467, 6.643994], ["cg13984533", "chr1", 37796839, 37796841, "MANEAL", "protein_coding", "S_Shelf", -0.201566, 0.039007, 0.81745, 0.757283, 0.882396, -5.167462, 2.37294e-07, 0.003467, 6.624713], ["cg10817441", "chr6", 32855950, 32855952, "PSMB9", "protein_coding", "S_Shore", -0.284868, 0.055212, 0.752113, 0.674974, 0.838069, -5.15953, 2.475705e-07, 0.003467, 6.606301], ["cg16721321", "chr4", 1208907, 1208909, "CTBP1-AS;SPON2", "protein_coding;antisense", "N_Shore", 0.115063, 0.022339, 1.121944, 1.07388, 1.172159, 5.150659, 2.595727e-07, 0.003467, 6.585741], ["cg13213755", "chr15", 73634112, 73634114, "NPTN", "protein_coding", "S_Shore", -0.314448, 0.061054, 0.730192, 0.64784, 0.823011, -5.150369, 2.599741e-07, 0.003467, 6.58507], ["cg10098541", "chr3", 123066802, 123066804, "PDIA5", "protein_coding", "N_Shore", -0.229958, 0.044778, 0.794567, 0.727805, 0.867452, -5.135516, 2.813709e-07, 0.003608, 6.550721], ["cg01614643", "chr6", 30156991, 30156993, "TRIM10", "protein_coding", ".", -0.217595, 0.042543, 0.804451, 0.740095, 0.874404, -5.114763, 3.141346e-07, 0.003878, 6.502884], ["cg13360395", "chr2", 123907703, 123907705, ".", ".", ".", -0.188217, 0.036959, 0.828435, 0.770546, 0.890672, -5.09258, 3.532238e-07, 0.003886, 6.45195], ["cg18586780", "chr2", 235044659, 235044661, "SH3BP4", "protein_coding", "S_Shore", -0.251658, 0.049427, 0.777511, 0.705722, 0.856602, -5.091462, 3.553122e-07, 0.003886, 6.44939], ["cg16424082", "chr12", 50929956, 50929958, "METTL7A", "protein_coding", ".", -0.384278, 0.075476, 0.680942, 0.587306, 0.789506, -5.091394, 3.554396e-07, 0.003886, 6.449234], ["cg19706516", "chr1", 151131392, 151131394, "RP11-68I18.10", "sense_overlapping", "Island", 0.219786, 0.043195, 1.24581, 1.14468, 1.355874, 5.088243, 3.613957e-07, 0.003886, 6.442017], ["cg18151487", "chr10", 102715582, 102715584, "SFXN2;ARL3", "protein_coding", "S_Shore", -0.200813, 0.039536, 0.818066, 0.757068, 0.883978, -5.079215, 3.789983e-07, 0.003948, 6.421363], ["cg17099306", "chr13", 49549508, 49549510, "RCBTB1", "protein_coding", ".", 0.225924, 0.044562, 1.25348, 1.148645, 1.367883, 5.069836, 3.981595e-07, 0.004022, 6.399943], ["cg00059652", "chr17", 69609356, 69609358, "LINC01483", "lincRNA", ".", -0.176244, 0.034833, 0.838413, 0.783084, 0.897652, -5.059691, 4.199356e-07, 0.004117, 6.376817], ["cg09411730", "chr11", 65423474, 65423476, "NEAT1_2;NEAT1_3;NEAT1", "misc_RNA;lincRNA", "Island", 0.14878, 0.029668, 1.160418, 1.094866, 1.229895, 5.014857, 5.307295e-07, 0.004944, 6.275127], ["cg26310429", "chr17", 73028309, 73028311, "SLC39A11", "protein_coding", ".", -0.134286, 0.026784, 0.87434, 0.829625, 0.921465, -5.013697, 5.339408e-07, 0.004944, 6.272507], ["cg06585893", "chr1", 230747805, 230747807, "CAPN9;RP11-99J16__A.2", "protein_coding;antisense", ".", 0.280006, 0.056023, 1.323137, 1.185545, 1.476698, 4.998048, 5.79135e-07, 0.005218, 6.23722], ["cg04205653", "chr21", 44805974, 44805976, "SUMO3", "protein_coding", "S_Shelf", -0.498789, 0.100035, 0.607266, 0.499147, 0.738804, -4.986125, 6.160238e-07, 0.005386, 6.210403], ["cg00550080", "chr6", 30157014, 30157016, "TRIM10", "protein_coding", ".", -0.213383, 0.042833, 0.807847, 0.742797, 0.878594, -4.981762, 6.300802e-07, 0.005386, 6.200604], ["cg16449402", "chr3", 9733040, 9733042, "BRPF1", "protein_coding", "S_Shore", -0.244462, 0.04912, 0.783126, 0.711247, 0.862268, -4.97684, 6.463057e-07, 0.005386, 6.189562], ["cg09890699", "chr1", 54356179, 54356181, "SSBP3", "protein_coding", "N_Shore", 0.220878, 0.044603, 1.247172, 1.142773, 1.361108, 4.952055, 7.343392e-07, 0.005828, 6.134103], ["cg08329113", "chr16", 70737238, 70737240, "VAC14", "protein_coding", ".", 0.192322, 0.038842, 1.21206, 1.123213, 1.307936, 4.951397, 7.36827e-07, 0.005828, 6.132634], ["cg12251145", "chr6", 30156966, 30156968, "TRIM10", "protein_coding", ".", -0.22664, 0.045809, 0.797208, 0.72875, 0.872096, -4.94749, 7.517673e-07, 0.005828, 6.123917], ["cg26361535", "chr8", 143494433, 143494435, "ZC3H3", "protein_coding", ".", 0.177805, 0.036246, 1.194592, 1.112672, 1.282545, 4.905487, 9.319586e-07, 0.007061, 6.030603], ["cg21785847", "chr2", 216081555, 216081557, "PECR;TMEM169", "protein_coding", "N_Shore", -0.129757, 0.026568, 0.878309, 0.833744, 0.925257, -4.883918, 1.039983e-06, 0.00748, 5.982974], ["cg15081720", "chr1", 119323994, 119323996, "RP11-418J17.3", "lincRNA", "N_Shelf", -0.188485, 0.038646, 0.828213, 0.767797, 0.893383, -4.877202, 1.076014e-06, 0.00748, 5.968182], ["cg13857210", "chr9", 136107370, 136107372, ".", ".", ".", 0.156148, 0.032026, 1.168999, 1.097877, 1.244728, 4.875703, 1.084218e-06, 0.00748, 5.964883], ["cg15022671", "chr2", 203637474, 203637476, "AC125238.2", "processed_pseudogene", ".", -0.203631, 0.041791, 0.815763, 0.751608, 0.885394, -4.872586, 1.10147e-06, 0.00748, 5.958027], ["cg15119221", "chr17", 19387441, 19387443, "MFAP4", "protein_coding", ".", 0.216043, 0.044395, 1.241156, 1.137725, 1.353989, 4.866397, 1.136513e-06, 0.00748, 5.944425], ["cg18437839", "chr1", 201476584, 201476586, ".", ".", ".", -0.282336, 0.058028, 0.75402, 0.672961, 0.844844, -4.865523, 1.141544e-06, 0.00748, 5.942507], ["cg08480739", "chr11", 67997549, 67997551, "UNC93B1", "protein_coding", "Island", -0.11367, 0.023376, 0.892552, 0.852582, 0.934397, -4.862723, 1.157818e-06, 0.00748, 5.93636], ["cg03769055", "chr6", 30157026, 30157028, "TRIM10", "protein_coding", ".", -0.220141, 0.045285, 0.802406, 0.734256, 0.876882, -4.861189, 1.166826e-06, 0.00748, 5.932994], ["cg06748884", "chr16", 11138900, 11138902, "CLEC16A", "protein_coding", ".", -0.253139, 0.052126, 0.77636, 0.70096, 0.85987, -4.856299, 1.196003e-06, 0.007522, 5.922268], ["cg01312394", "chr2", 46882240, 46882242, ".", ".", ".", -0.211438, 0.043728, 0.809419, 0.742937, 0.88185, -4.83531, 1.329384e-06, 0.008207, 5.87635], ["cg11818438", "chr8", 118904290, 118904292, ".", ".", ".", -0.245208, 0.050932, 0.782541, 0.708197, 0.864691, -4.814419, 1.476284e-06, 0.008948, 5.83083], ["cg00223135", "chr15", 43507046, 43507048, "TP53BP1", "protein_coding", "N_Shelf", -0.373804, 0.077776, 0.688112, 0.590821, 0.801423, -4.806152, 1.538626e-06, 0.009081, 5.812867], ["cg08467866", "chr16", 75208215, 75208217, "CTRB2", "protein_coding", ".", 0.190435, 0.039638, 1.209775, 1.119347, 1.307509, 4.804327, 1.552728e-06, 0.009081, 5.808905], ["cg00206490", "chr16", 89403320, 89403322, "RP1-168P16.2;ANKRD11", "protein_coding;sense_intronic", ".", 0.142392, 0.029794, 1.153028, 1.087626, 1.222363, 4.779246, 1.759539e-06, 0.010113, 5.754601], ["cg17632028", "chr4", 652752, 652754, "PDE6B", "protein_coding", "N_Shore", -0.192932, 0.040438, 0.824538, 0.761711, 0.892547, -4.771112, 1.832115e-06, 0.010351, 5.737047], ["cg00945409", "chr10", 78977907, 78977909, "ZMIZ1-AS1", "antisense", "S_Shelf", -0.276731, 0.058061, 0.758258, 0.676699, 0.849648, -4.766189, 1.877429e-06, 0.010431, 5.726436], ["cg21739530", "chr4", 173264916, 173264918, "GALNT7", "protein_coding", ".", 0.170982, 0.035904, 1.18647, 1.105848, 1.27297, 4.762247, 1.914489e-06, 0.010462, 5.717947], ["cg03596113", "chr1", 3213404, 3213406, "PRDM16", "protein_coding", ".", -0.237656, 0.049972, 0.788474, 0.71491, 0.869608, -4.755804, 1.976579e-06, 0.010592, 5.704086], ["cg13702536", "chr12", 122731005, 122731007, "HCAR1", "protein_coding", ".", -0.142431, 0.029965, 0.867247, 0.81778, 0.919706, -4.753247, 2.001759e-06, 0.010592, 5.698588], ["cg01591830", "chr16", 89599591, 89599593, ".", ".", "S_Shelf", 0.194315, 0.040948, 1.214479, 1.120818, 1.315966, 4.745456, 2.080367e-06, 0.010814, 5.68186], ["cg10057841", "chr11", 130126348, 130126350, "APLP2", "protein_coding", "S_Shelf", -0.368313, 0.077659, 0.691901, 0.594211, 0.805651, -4.742721, 2.108662e-06, 0.010814, 5.675993], ["cg21637653", "chr17", 17213452, 17213454, "FLCN;MPRIP;RP11-45M22.4", "protein_coding", ".", -0.412391, 0.08706, 0.662066, 0.558207, 0.785247, -4.73686, 2.170546e-06, 0.010963, 5.663431], ["cg01443071", "chr17", 41968231, 41968233, "CNP", "protein_coding", "Island", -0.15094, 0.031921, 0.859899, 0.80775, 0.915416, -4.728605, 2.260674e-06, 0.011222, 5.645762], ["cg04092699", "chr2", 76748093, 76748095, "LRRTM4", "protein_coding", ".", -0.22895, 0.048459, 0.795368, 0.723302, 0.874615, -4.724587, 2.305835e-06, 0.011222, 5.637172], ["cg09281220", "chr2", 219595362, 219595364, ".", ".", "N_Shelf", 0.195703, 0.041437, 1.216166, 1.121298, 1.31906, 4.722854, 2.325582e-06, 0.011222, 5.633468], ["cg21477075", "chr6", 168032755, 168032757, "KIF25", "protein_coding", "N_Shelf", -0.163515, 0.034642, 0.849154, 0.793413, 0.908811, -4.72017, 2.356477e-06, 0.011222, 5.627737], ["cg16822174", "chr12", 19483011, 19483013, "AEBP2", "protein_coding", ".", 0.164864, 0.034973, 1.179233, 1.101109, 1.262901, 4.713986, 2.429178e-06, 0.011405, 5.614541], ["cg18581405", "chr19", 3180036, 3180038, "S1PR4", "protein_coding", "S_Shore", 0.152364, 0.032352, 1.164584, 1.093031, 1.240821, 4.709528, 2.482906e-06, 0.011496, 5.60504], ["cg04407248", "chr16", 31136874, 31136876, "PRSS8", "protein_coding", ".", -0.214205, 0.045514, 0.807183, 0.738297, 0.882497, -4.706401, 2.521278e-06, 0.011513, 5.598379], ["cg19479744", "chr6", 155701180, 155701182, ".", ".", ".", -0.280302, 0.059754, 0.755556, 0.672054, 0.849433, -4.690948, 2.719423e-06, 0.011938, 5.565523], ["cg20964021", "chr7", 156641708, 156641710, "RNF32;LINC01006", "protein_coding;lincRNA", "S_Shore", -0.35657, 0.076014, 0.700073, 0.603171, 0.812544, -4.690855, 2.720653e-06, 0.011938, 5.565327], ["cg25898577", "chr22", 21937868, 21937870, "PPM1F;LL22NC03-86G7.1", "protein_coding;antisense", "N_Shore", -0.147388, 0.031424, 0.862959, 0.811413, 0.917781, -4.690229, 2.72899e-06, 0.011938, 5.563998], ["cg27018309", "chr16", 8849264, 8849266, "RP11-77H9.2;RP11-152P23.2;PMM2", "protein_coding;antisense", ".", -0.194816, 0.041563, 0.822986, 0.758602, 0.892833, -4.687301, 2.768313e-06, 0.011938, 5.557785], ["cg12762600", "chr15", 89647922, 89647924, "KIF7", "protein_coding", "N_Shore", -0.268044, 0.057208, 0.764874, 0.683746, 0.855628, -4.685459, 2.793327e-06, 0.011938, 5.553878], ["cg27658601", "chr17", 82836204, 82836206, "TBCD;ZNF750", "protein_coding", "S_Shelf", -0.155852, 0.033306, 0.855686, 0.801613, 0.913407, -4.679422, 2.876845e-06, 0.012139, 5.541084], ["cg07593405", "chr7", 118900, 118902, ".", ".", "S_Shore", -0.147751, 0.031594, 0.862646, 0.810848, 0.917753, -4.676517, 2.917881e-06, 0.012158, 5.534932], ["cg19461880", "chr11", 6483135, 6483137, "TIMM10B", "protein_coding", "S_Shore", -0.19945, 0.042695, 0.819181, 0.753421, 0.890681, -4.671491, 2.990206e-06, 0.012226, 5.524299], ["cg04400145", "chr20", 1893035, 1893037, "SIRPA;RP5-858L17.1", "protein_coding;antisense", "N_Shore", -0.205522, 0.044036, 0.814222, 0.746895, 0.887618, -4.667187, 3.053515e-06, 0.012226, 5.5152], ["cg11661187", "chr4", 89835381, 89835383, "SNCA;SNCA-AS1", "protein_coding;antisense", "N_Shore", -0.178332, 0.038252, 0.836665, 0.776231, 0.901804, -4.661978, 3.13185e-06, 0.012226, 5.504199], ["cg10805039", "chr17", 847000, 847002, "NXN", "protein_coding", ".", -0.174223, 0.037373, 0.840109, 0.780771, 0.903957, -4.661759, 3.135175e-06, 0.012226, 5.503738], ["cg18019132", "chr11", 65424235, 65424237, "NEAT1", "lincRNA", "S_Shore", 0.233758, 0.050169, 1.263339, 1.145027, 1.393875, 4.659413, 3.171131e-06, 0.012226, 5.498786], ["cg00539049", "chr2", 149077019, 149077021, "LYPD6B", "protein_coding", ".", -0.191153, 0.041032, 0.826006, 0.762179, 0.895179, -4.658634, 3.183139e-06, 0.012226, 5.497144], ["cg01771213", "chr1", 57270164, 57270166, "DAB1", "protein_coding", ".", -0.222901, 0.047868, 0.800194, 0.728535, 0.878902, -4.656605, 3.214661e-06, 0.012226, 5.492865], ["cg10280339", "chr15", 65519989, 65519991, ".", ".", "S_Shelf", -0.208292, 0.044755, 0.81197, 0.74378, 0.886412, -4.654052, 3.254754e-06, 0.012226, 5.487482], ["cg16429055", "chr17", 17051538, 17051540, "MPRIP", "protein_coding", ".", -0.312358, 0.067143, 0.731719, 0.641493, 0.834635, -4.652129, 3.28525e-06, 0.012226, 5.483432], ["cg09140274", "chr1", 220896850, 220896852, ".", ".", "S_Shore", -0.21792, 0.046855, 0.80419, 0.733628, 0.881539, -4.650963, 3.303882e-06, 0.012226, 5.480975], ["cg07253321", "chr6", 146598717, 146598719, "RP11-715G15.1;ADGB", "protein_coding;lincRNA", "N_Shore", -0.222246, 0.047828, 0.800718, 0.729069, 0.879408, -4.646827, 3.370796e-06, 0.012226, 5.472268], ["cg26130533", "chr4", 1208911, 1208913, "CTBP1-AS;SPON2", "protein_coding;antisense", "N_Shore", 0.106889, 0.023011, 1.112811, 1.063738, 1.164147, 4.645201, 3.397449e-06, 0.012226, 5.468847], ["cg11600161", "chr11", 67404113, 67404115, "TBC1D10C;PPP1CA", "protein_coding", "S_Shelf", 0.285081, 0.061382, 1.329869, 1.179127, 1.499882, 4.644387, 3.410881e-06, 0.012226, 5.467133], ["cg14978242", "chr5", 80205311, 80205313, "SERINC5", "protein_coding", ".", 0.189691, 0.040934, 1.208876, 1.115677, 1.30986, 4.634047, 3.585853e-06, 0.012633, 5.445407], ["cg17669009", "chr19", 51417630, 51417632, "SIGLEC10", "protein_coding", ".", 0.304133, 0.06565, 1.35545, 1.191795, 1.541577, 4.63261, 3.610848e-06, 0.012633, 5.442391], ["cg11840849", "chr10", 93363960, 93363962, "MYOF", "protein_coding", ".", -0.151701, 0.032757, 0.859245, 0.805812, 0.916221, -4.631047, 3.638206e-06, 0.012633, 5.439113], ["cg18062721", "chr3", 11601952, 11601954, "VGLL4", "protein_coding", ".", 0.111702, 0.024161, 1.118179, 1.066463, 1.172403, 4.623302, 3.776797e-06, 0.012662, 5.422876], ["cg03445800", "chr3", 481328, 481330, ".", ".", ".", -0.164065, 0.035495, 0.848687, 0.791652, 0.909831, -4.622225, 3.796464e-06, 0.012662, 5.420621], ["cg15219347", "chr19", 17378501, 17378503, "PLVAP", "protein_coding", ".", 0.199163, 0.043097, 1.220381, 1.12153, 1.327944, 4.621243, 3.81447e-06, 0.012662, 5.418566], ["cg04516952", "chr19", 50968785, 50968787, "KLK6;CTB-147C22.9", "protein_coding;antisense", ".", 0.253183, 0.054788, 1.288119, 1.156965, 1.43414, 4.62114, 3.816366e-06, 0.012662, 5.41835], ["cg11958234", "chr1", 117684205, 117684207, ".", ".", ".", -0.175418, 0.037969, 0.839106, 0.778929, 0.903932, -4.62006, 3.836291e-06, 0.012662, 5.416088], ["cg10187365", "chr21", 33488648, 33488650, "DNAJC28", "protein_coding", "N_Shelf", -0.437951, 0.094892, 0.645357, 0.535831, 0.777271, -4.615245, 3.926312e-06, 0.012832, 5.406015], ["cg15538767", "chr5", 150490832, 150490834, "NDST1", "protein_coding", ".", 0.192576, 0.04181, 1.212369, 1.116981, 1.315903, 4.605955, 4.105771e-06, 0.013183, 5.386605], ["cg03272932", "chr16", 87555286, 87555288, ".", ".", ".", -0.155724, 0.033822, 0.855795, 0.800903, 0.914449, -4.604182, 4.140908e-06, 0.013183, 5.382904], ["cg07143043", "chr10", 131970793, 131970795, "BNIP3;RP11-45A17.2", "protein_coding;antisense", ".", 0.210412, 0.045715, 1.234186, 1.128413, 1.349875, 4.602672, 4.171048e-06, 0.013183, 5.379755], ["cg01140244", "chr10", 132685455, 132685457, "INPP5A", "protein_coding", ".", 0.199266, 0.043303, 1.220506, 1.121192, 1.328618, 4.601623, 4.192113e-06, 0.013183, 5.377567], ["cg17078656", "chr1", 12475057, 12475059, "VPS13D", "protein_coding", ".", -0.319168, 0.069453, 0.726753, 0.634261, 0.832733, -4.595425, 4.318679e-06, 0.013455, 5.364649], ["cg24800630", "chr6", 156426519, 156426521, ".", ".", ".", -0.266332, 0.058006, 0.766185, 0.683846, 0.858437, -4.591421, 4.40239e-06, 0.013588, 5.356311], ["cg17518965", "chr19", 3178956, 3178958, "S1PR4", "protein_coding", "Island", 0.123579, 0.02695, 1.131539, 1.073322, 1.192914, 4.585534, 4.528273e-06, 0.013834, 5.344067], ["cg04339462", "chr14", 99561858, 99561860, "CCDC85C", "protein_coding", ".", 0.132871, 0.028999, 1.142103, 1.079, 1.208896, 4.581972, 4.606123e-06, 0.013834, 5.336664], ["cg25392500", "chr6", 32127129, 32127131, "ATF6B", "protein_coding", ".", -0.215389, 0.047024, 0.806228, 0.735244, 0.884065, -4.580439, 4.640012e-06, 0.013834, 5.333481], ["cg21955549", "chr15", 65026930, 65026932, "MTFMT", "protein_coding", "N_Shelf", -0.293429, 0.064066, 0.745702, 0.657707, 0.845471, -4.580084, 4.647884e-06, 0.013834, 5.332745], ["cg20460366", "chr20", 46305392, 46305394, "CDH22", "protein_coding", "N_Shore", 0.151734, 0.033197, 1.16385, 1.090535, 1.242095, 4.570657, 4.861968e-06, 0.014343, 5.313188], ["cg18964582", "chr1", 236393881, 236393883, "EDARADD", "protein_coding", "N_Shore", -0.155491, 0.034036, 0.855995, 0.800755, 0.915045, -4.568437, 4.913738e-06, 0.014368, 5.308588], ["cg23716141", "chr17", 77364394, 77364396, "SEPT9", "protein_coding", ".", 0.129057, 0.028266, 1.137755, 1.076438, 1.202565, 4.565848, 4.974792e-06, 0.0144, 5.303225], ["cg05482050", "chr5", 177982332, 177982334, "RP11-1252I4.2", "lincRNA", "N_Shelf", 0.292084, 0.064021, 1.339216, 1.181287, 1.518258, 4.562292, 5.059816e-06, 0.0144, 5.295865], ["cg10695490", "chr19", 49156041, 49156043, "HRC", "protein_coding", "N_Shore", 0.200883, 0.044042, 1.222482, 1.121382, 1.332696, 4.561132, 5.087855e-06, 0.0144, 5.293465], ["cg23458843", "chr12", 57742661, 57742663, "AGAP2;TSPAN31", "protein_coding", "S_Shelf", -0.37663, 0.082609, 0.68617, 0.583599, 0.806769, -4.559166, 5.135723e-06, 0.0144, 5.289398], ["cg20378520", "chr12", 125219264, 125219266, "TMEM132B", "protein_coding", ".", -0.216323, 0.047467, 0.805475, 0.733919, 0.884009, -4.557287, 5.181858e-06, 0.0144, 5.285514], ["cg12984636", "chr9", 104693221, 104693223, "OR13D1", "protein_coding", ".", -0.233926, 0.051331, 0.79142, 0.715672, 0.875186, -4.557211, 5.183723e-06, 0.0144, 5.285358], ["cg09797463", "chr10", 89235539, 89235541, "LIPA", "protein_coding", ".", -0.202416, 0.044485, 0.816755, 0.748559, 0.891164, -4.550201, 5.359477e-06, 0.014765, 5.270878], ["cg01052291", "chr6", 132715433, 132715435, "VNN1", "protein_coding", ".", -0.221104, 0.048649, 0.801633, 0.728729, 0.881831, -4.544921, 5.495596e-06, 0.015016, 5.259985], ["cg22162847", "chr1", 150562302, 150562304, "ADAMTSL4-AS1", "processed_transcript", "S_Shore", 0.270895, 0.059706, 1.311138, 1.166344, 1.473907, 4.537178, 5.701204e-06, 0.015451, 5.244033], ["cg13691108", "chr20", 38804603, 38804605, "PPP1R16B", "protein_coding", "N_Shore", 0.203341, 0.044863, 1.22549, 1.122335, 1.338127, 4.532491, 5.829214e-06, 0.015666, 5.23439], ["cg09744397", "chr15", 101725089, 101725091, "TARSL2", "protein_coding", "S_Shore", -0.200054, 0.044159, 0.818687, 0.750808, 0.892702, -4.530259, 5.891145e-06, 0.015666, 5.2298], ["cg26846864", "chr16", 3257325, 3257327, "MEFV", "protein_coding", "S_Shelf", -0.189924, 0.041933, 0.827022, 0.761769, 0.897865, -4.529178, 5.921369e-06, 0.015666, 5.227578], ["cg22401679", "chr1", 94614490, 94614492, ".", ".", ".", -0.158317, 0.035014, 0.853579, 0.796967, 0.914214, -4.521542, 6.139069e-06, 0.01597, 5.211897], ["cg18530324", "chr18", 48538620, 48538622, "CTIF", "protein_coding", "N_Shore", -0.147498, 0.032625, 0.862864, 0.809417, 0.91984, -4.521075, 6.152636e-06, 0.01597, 5.210939], ["cg03601372", "chr19", 36776123, 36776125, "CTD-2162K18.3;CTD-2162K18.4", "lincRNA", "Island", -0.190259, 0.042092, 0.826745, 0.761278, 0.897842, -4.520132, 6.180121e-06, 0.01597, 5.209003], ["cg07892374", "chr3", 130849277, 130849279, "ATP2C1", "protein_coding", ".", 0.256001, 0.056725, 1.291754, 1.155833, 1.443659, 4.512989, 6.39204e-06, 0.01629, 5.194361], ["cg01782668", "chr4", 148026284, 148026286, "ARHGAP10", "protein_coding", ".", -0.245573, 0.054419, 0.782256, 0.703117, 0.870302, -4.51267, 6.401648e-06, 0.01629, 5.193708], ["cg21094613", "chr11", 79189641, 79189643, "TENM4", "protein_coding", ".", -0.214585, 0.047618, 0.806876, 0.734978, 0.885808, -4.506367, 6.594691e-06, 0.016399, 5.180806], ["cg27125093", "chr19", 3179829, 3179831, "S1PR4", "protein_coding", "Island", 0.112557, 0.02499, 1.119136, 1.065643, 1.175315, 4.504134, 6.664423e-06, 0.016399, 5.176237], ["cg10089193", "chr16", 1090938, 1090940, "C1QTNF8", "protein_coding", "S_Shore", 0.164314, 0.036494, 1.178584, 1.097228, 1.265972, 4.502516, 6.715368e-06, 0.016399, 5.17293], ["cg13927938", "chr1", 204490455, 204490457, "PIK3C2B", "protein_coding", "N_Shelf", -0.280143, 0.062227, 0.755676, 0.66891, 0.853696, -4.501959, 6.732985e-06, 0.016399, 5.171792], ["cg15586054", "chr6", 165658541, 165658543, "PDE10A", "protein_coding", "N_Shore", -0.148862, 0.033079, 0.861688, 0.807594, 0.919406, -4.500147, 6.790656e-06, 0.016399, 5.168088], ["cg22571135", "chr7", 55309356, 55309358, ".", ".", ".", -0.140763, 0.031284, 0.868695, 0.817031, 0.923626, -4.49958, 6.808801e-06, 0.016399, 5.166929], ["cg18072388", "chr19", 10323084, 10323086, "RAVER1", "protein_coding", "N_Shore", 0.191606, 0.042585, 1.211193, 1.114204, 1.316625, 4.499351, 6.81613e-06, 0.016399, 5.166462], ["cg21853989", "chr1", 45339053, 45339055, "MUTYH;TOE1", "protein_coding", "N_Shore", -0.398057, 0.088483, 0.671624, 0.564689, 0.798809, -4.49867, 6.837978e-06, 0.016399, 5.165072], ["cg08742262", "chr7", 158549590, 158549592, "PTPRN2", "protein_coding", "N_Shore", 0.143855, 0.032018, 1.154716, 1.084481, 1.229501, 4.492979, 7.023382e-06, 0.016723, 5.153454], ["cg16280363", "chr12", 57489107, 57489109, "MARS;ARHGAP9", "protein_coding", "S_Shore", -0.18618, 0.041547, 0.830124, 0.765205, 0.900551, -4.481155, 7.424002e-06, 0.017552, 5.129362], ["cg09218727", "chr10", 104350553, 104350555, ".", ".", ".", -0.253133, 0.056516, 0.776365, 0.694959, 0.867306, -4.478948, 7.501195e-06, 0.017609, 5.12487], ["cg25074809", "chr2", 15945893, 15945895, "MYCN", "protein_coding", "S_Shelf", -0.121534, 0.027147, 0.885561, 0.839674, 0.933955, -4.476924, 7.572616e-06, 0.017653, 5.120754], ["cg06715885", "chr12", 57201608, 57201610, "LRP1", "protein_coding", "N_Shore", 0.238493, 0.053292, 1.269334, 1.143442, 1.409087, 4.475234, 7.63276e-06, 0.017669, 5.117318], ["cg03040821", "chr1", 212792866, 212792868, "TATDN3;NSL1", "protein_coding", "S_Shore", -0.148003, 0.033112, 0.862429, 0.808236, 0.920255, -4.46979, 7.829638e-06, 0.018, 5.106258], ["cg06142946", "chr11", 17351576, 17351578, "NCR3LG1", "protein_coding", "Island", 0.16908, 0.03784, 1.184215, 1.099566, 1.275381, 4.468321, 7.883585e-06, 0.018, 5.103276], ["cg27064178", "chr22", 49784571, 49784573, "BRD1", "protein_coding", "Island", 0.162857, 0.036476, 1.176868, 1.095668, 1.264085, 4.464738, 8.01669e-06, 0.018122, 5.096005], ["cg02918093", "chr16", 10624083, 10624085, ".", ".", "N_Shelf", -0.253056, 0.056689, 0.776424, 0.694777, 0.867666, -4.463961, 8.045832e-06, 0.018122, 5.094429], ["cg05974498", "chr19", 52096002, 52096004, "ZNF432;ZNF841", "protein_coding", "S_Shore", -0.179055, 0.040229, 0.83606, 0.772671, 0.904649, -4.450928, 8.549995e-06, 0.019035, 5.068034], ["cg26925717", "chr14", 20999156, 20999158, "RP11-84C10.4;SLC39A2", "protein_coding;antisense", ".", -0.196152, 0.044088, 0.821887, 0.753849, 0.896066, -4.449093, 8.623366e-06, 0.019035, 5.064323], ["cg27396056", "chr3", 188947203, 188947205, "TPRG1;TPRG1-AS1", "protein_coding;lincRNA", "N_Shore", -0.109317, 0.024583, 0.896446, 0.854277, 0.940696, -4.446805, 8.715673e-06, 0.019035, 5.059699], ["cg04543865", "chr16", 81313945, 81313947, "GAN;RP11-55K13.1", "protein_coding;antisense", "N_Shore", -0.178117, 0.040055, 0.836844, 0.773659, 0.90519, -4.446772, 8.717009e-06, 0.019035, 5.059633], ["cg11039729", "chr10", 35379396, 35379398, "CCNY", "protein_coding", ".", 0.194583, 0.043774, 1.214804, 1.114926, 1.323629, 4.445204, 8.780841e-06, 0.019035, 5.056464], ["cg01476568", "chr10", 5692752, 5692754, "FAM208B", "protein_coding", ".", 0.094776, 0.021329, 1.099412, 1.054399, 1.146347, 4.443464, 8.852179e-06, 0.019035, 5.05295], ["cg06605704", "chr16", 4501145, 4501147, "HMOX2", "protein_coding", ".", -0.401022, 0.090268, 0.669635, 0.56105, 0.799235, -4.442577, 8.888798e-06, 0.019035, 5.051157], ["cg23875404", "chr13", 36154731, 36154733, ".", ".", "N_Shore", -0.181785, 0.040928, 0.833781, 0.76951, 0.90342, -4.441597, 8.92936e-06, 0.019035, 5.04918], ["cg04839706", "chr16", 46886162, 46886164, "GPT2", "protein_coding", "S_Shore", -0.180264, 0.040593, 0.83505, 0.771186, 0.904202, -4.440741, 8.964975e-06, 0.019035, 5.047451], ["cg20435097", "chr10", 124632254, 124632256, "RP11-12J10.3;FAM53B", "protein_coding", ".", 0.180404, 0.040655, 1.197701, 1.105969, 1.297042, 4.437429, 9.103975e-06, 0.019109, 5.040769], ["cg17082549", "chr3", 128620009, 128620011, "RPN1", "protein_coding", "S_Shore", -0.183396, 0.041332, 0.832438, 0.767663, 0.90268, -4.437176, 9.114672e-06, 0.019109, 5.040259], ["cg15711288", "chr3", 150783114, 150783116, ".", ".", "S_Shelf", -0.214262, 0.048304, 0.807137, 0.734228, 0.887285, -4.43573, 9.176087e-06, 0.019118, 5.037342], ["cg03233656", "chr2", 64987490, 64987492, "SLC1A4", "protein_coding", "N_Shore", -0.182013, 0.04106, 0.83359, 0.769135, 0.903447, -4.432904, 9.297209e-06, 0.01925, 5.031647], ["cg14287724", "chr1", 1170137, 1170139, "RP11-465B22.8", "lincRNA", "N_Shelf", 0.14828, 0.033486, 1.159838, 1.086161, 1.238512, 4.428152, 9.504391e-06, 0.019455, 5.022076], ["cg08530484", "chr2", 219159883, 219159885, "NHEJ1;RP11-33O4.2", "protein_coding", "N_Shore", -0.132838, 0.03, 0.875607, 0.825607, 0.928635, -4.427956, 9.513029e-06, 0.019455, 5.021681], ["cg22760287", "chr11", 107705174, 107705176, ".", ".", ".", -0.170521, 0.038545, 0.843225, 0.781868, 0.909397, -4.423905, 9.693277e-06, 0.019593, 5.013529], ["cg22855993", "chr1", 16071749, 16071751, "FAM131C", "protein_coding", "N_Shore", 0.317077, 0.071675, 1.373108, 1.193149, 1.58021, 4.423796, 9.698182e-06, 0.019593, 5.01331], ["cg09941713", "chr8", 60792772, 60792774, "CHD7", "protein_coding", ".", 0.140702, 0.031848, 1.151081, 1.081426, 1.225224, 4.417872, 9.967746e-06, 0.019671, 5.001403], ["cg06956444", "chr1", 7941248, 7941250, "RP5-892F13.2;TNFRSF9", "protein_coding;processed_pseudogene", "N_Shore", -0.171499, 0.03882, 0.842401, 0.780683, 0.908997, -4.417788, 9.971623e-06, 0.019671, 5.001234]];

    // Define the dt_args
    let dt_args = {"order": []};
    dt_args["data"] = data;

    $(document).ready(function () {
        
        $('#9cc509ba-8ae9-47c2-aa29-147c140a9924').DataTable(dt_args);
    });
</script>
</div>
</div></div>
</div>
</section>
<section id="efs-suggested-significance-probes">
<h4><a class="toc-backref" href="#id23" role="doc-backlink">EFS suggested-significance probes:</a><a class="headerlink" href="#efs-suggested-significance-probes" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">show</span><span class="p">(</span><span class="n">top_efs</span><span class="p">,</span> <span class="n">classes</span><span class="o">=</span><span class="s2">&quot;display nowrap cell-border&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><style>.itables table td {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables table th {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables thead input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}

.itables tfoot input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}
</style>
<div class="itables">
<table id="7f31de53-f6ba-4fc5-ae10-588030157104" class="display nowrap cell-border"style="table-layout:auto;width:auto;margin:auto;caption-side:bottom"><thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>CGIPosition</th>
      <th>Coefficient</th>
      <th>StandardError</th>
      <th>HazardsRatio</th>
      <th>95%CI_lower</th>
      <th>95%CI_upper</th>
      <th>ZScore</th>
      <th>PValue</th>
      <th>FDR_QValue</th>
      <th>minuslog10value</th>
    </tr>
  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css">
<script type="module">
    // Import jquery and DataTable
    import 'https://code.jquery.com/jquery-3.6.0.min.js';
    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';
    dt($);

    // Define the table data
    const data = [["cg00192882", "chr17", 19387806, 19387808, "MFAP4", "protein_coding", ".", 0.245704, 0.041175, 1.278521, 1.179395, 1.385978, 5.967247, 2.412895e-09, 0.000596, 8.617462], ["cg14928764", "chr20", 23083970, 23083972, "CD93", "protein_coding", ".", -0.177767, 0.030116, 0.837137, 0.789155, 0.888038, -5.902802, 3.573791e-09, 0.000596, 8.446871], ["cg08900363", "chr7", 597816, 597818, "PRKAR1B", "protein_coding", ".", -0.257579, 0.044674, 0.77292, 0.708122, 0.843649, -5.765718, 8.131105e-09, 0.000888, 8.08985], ["cg15215077", "chr6", 119240673, 119240675, "MAN1A1", "protein_coding", ".", -0.300453, 0.052526, 0.740483, 0.668043, 0.820778, -5.720043, 1.064974e-08, 0.000888, 7.972661], ["cg06748884", "chr16", 11138900, 11138902, "CLEC16A", "protein_coding", ".", -0.245305, 0.044856, 0.782466, 0.716612, 0.854372, -5.468728, 4.532758e-08, 0.003022, 7.343637], ["cg17540192", "chr7", 98245946, 98245948, "TECPR1", "protein_coding", ".", -0.112744, 0.021809, 0.89338, 0.855997, 0.932395, -5.169578, 2.34623e-07, 0.013035, 6.629629], ["cg21637653", "chr17", 17213452, 17213454, "FLCN;MPRIP;RP11-45M22.4", "protein_coding", ".", -0.369615, 0.073694, 0.691, 0.598067, 0.798375, -5.015511, 5.289275e-07, 0.020443, 6.276604], ["cg02918093", "chr16", 10624083, 10624085, ".", ".", "N_Shelf", -0.237813, 0.047591, 0.78835, 0.718141, 0.865423, -4.997025, 5.822146e-07, 0.020443, 6.234917], ["cg14978242", "chr5", 80205311, 80205313, "SERINC5", "protein_coding", ".", 0.162636, 0.032577, 1.176608, 1.10383, 1.254185, 4.992338, 5.965268e-07, 0.020443, 6.22437], ["cg06845655", "chr11", 85651326, 85651328, "TMEM126A", "protein_coding", "S_Shelf", -0.157387, 0.031559, 0.854374, 0.803127, 0.90889, -4.986992, 6.132655e-07, 0.020443, 6.212351], ["cg04578362", "chr2", 144835191, 144835193, "TEX41", "lincRNA", ".", 0.134131, 0.027108, 1.143543, 1.084371, 1.205944, 4.94795, 7.499905e-07, 0.022728, 6.124944], ["cg13958583", "chr8", 100761836, 100761838, ".", ".", ".", -0.178707, 0.036589, 0.836351, 0.778474, 0.898532, -4.884137, 1.038827e-06, 0.028292, 5.983457], ["cg11823637", "chr12", 22729886, 22729888, "RP11-114G22.1", "lincRNA", ".", -0.290071, 0.059613, 0.74821, 0.665704, 0.840942, -4.865913, 1.139297e-06, 0.028292, 5.943363], ["cg11851174", "chr17", 17809294, 17809296, "RAI1", "protein_coding", "N_Shore", 0.195685, 0.040284, 1.216143, 1.123815, 1.316057, 4.857592, 1.18822e-06, 0.028292, 5.925103], ["cg13213755", "chr15", 73634112, 73634114, "NPTN", "protein_coding", "S_Shore", -0.249899, 0.051724, 0.778879, 0.70379, 0.86198, -4.831434, 1.355533e-06, 0.030125, 5.86789], ["cg11083457", "chr13", 51341139, 51341141, "SERPINE3", "protein_coding", ".", -0.154795, 0.032325, 0.856591, 0.804004, 0.912617, -4.788686, 1.678772e-06, 0.033037, 5.775008], ["cg14165772", "chr1", 2931066, 2931068, ".", ".", "S_Shore", 0.186236, 0.038897, 1.204706, 1.116278, 1.30014, 4.787966, 1.684798e-06, 0.033037, 5.773452], ["cg06985664", "chr7", 116727544, 116727546, "MET", "protein_coding", ".", -0.167872, 0.035161, 0.845462, 0.789159, 0.905782, -4.774305, 1.803286e-06, 0.033396, 5.743935], ["cg11975397", "chr1", 159470190, 159470192, "RP11-180D21.3;RP11-550P17.5", "antisense;lincRNA", ".", -0.130531, 0.027549, 0.877629, 0.831499, 0.926319, -4.738189, 2.156367e-06, 0.037833, 5.666277], ["cg16721321", "chr4", 1208907, 1208909, "CTBP1-AS;SPON2", "protein_coding;antisense", "N_Shore", 0.085431, 0.018095, 1.089186, 1.051235, 1.128507, 4.721253, 2.343957e-06, 0.039068, 5.63005], ["cg06024557", "chr5", 150052438, 150052440, "HMGXB3", "protein_coding", ".", -0.347852, 0.073843, 0.706203, 0.611046, 0.816179, -4.710678, 2.468937e-06, 0.039192, 5.60749], ["cg09890699", "chr1", 54356179, 54356181, "SSBP3", "protein_coding", "N_Shore", 0.171525, 0.036529, 1.187114, 1.105092, 1.275223, 4.695541, 2.659017e-06, 0.04029, 5.575279], ["cg11269166", "chr2", 171347336, 171347338, "METTL8", "protein_coding", ".", -0.154627, 0.033036, 0.856735, 0.80302, 0.914044, -4.680528, 2.861374e-06, 0.041471, 5.543425], ["cg01224891", "chr10", 8054404, 8054406, "GATA3-AS1;RP11-379F12.3;GATA3", "protein_coding;lincRNA", "Island", -0.133168, 0.028534, 0.875318, 0.82771, 0.925665, -4.667062, 3.055379e-06, 0.042438, 5.514935], ["cg08069883", "chr3", 72032668, 72032670, ".", ".", ".", -0.198408, 0.042865, 0.820035, 0.753955, 0.891907, -4.628612, 3.681244e-06, 0.046396, 5.434005], ["cg07471203", "chr11", 65113438, 65113440, "TM7SF2", "protein_coding", "S_Shore", 0.163372, 0.035416, 1.177475, 1.098513, 1.262113, 4.612899, 3.970915e-06, 0.046396, 5.401109], ["cg22162847", "chr1", 150562302, 150562304, "ADAMTSL4-AS1", "processed_transcript", "S_Shore", 0.229158, 0.049726, 1.257541, 1.140762, 1.386275, 4.608403, 4.057736e-06, 0.046396, 5.391716], ["cg03572680", "chr11", 124898657, 124898659, "ROBO4", "protein_coding", ".", -0.171514, 0.037255, 0.842389, 0.783071, 0.906199, -4.603813, 4.148244e-06, 0.046396, 5.382136], ["cg27018309", "chr16", 8849264, 8849266, "RP11-77H9.2;RP11-152P23.2;PMM2", "protein_coding;antisense", ".", -0.168882, 0.036754, 0.844608, 0.785905, 0.907696, -4.594928, 4.328979e-06, 0.046396, 5.363614], ["cg26966083", "chr11", 2164145, 2164147, "TH", "protein_coding", "N_Shelf", 0.205469, 0.044739, 1.228101, 1.124999, 1.340652, 4.592597, 4.377639e-06, 0.046396, 5.35876], ["cg11937508", "chr2", 206218878, 206218880, "GPR1;GPR1-AS", "protein_coding;processed_transcript", ".", -0.194668, 0.042397, 0.823108, 0.757475, 0.894429, -4.591497, 4.400786e-06, 0.046396, 5.35647], ["cg26124526", "chr5", 61323104, 61323106, ".", ".", ".", -0.174741, 0.038161, 0.839674, 0.779163, 0.904885, -4.57906, 4.670703e-06, 0.046396, 5.330618], ["cg09411730", "chr11", 65423474, 65423476, "NEAT1_2;NEAT1_3;NEAT1", "misc_RNA;lincRNA", "Island", 0.119564, 0.026117, 1.127006, 1.070768, 1.186197, 4.578084, 4.692551e-06, 0.046396, 5.328591], ["cg18866597", "chr20", 45344681, 45344683, "SDC4", "protein_coding", "N_Shelf", -0.385184, 0.084302, 0.680326, 0.576712, 0.802555, -4.569111, 4.897981e-06, 0.046396, 5.309983], ["cg12731773", "chr11", 846054, 846056, "TSPAN4", "protein_coding", "S_Shelf", 0.170859, 0.037422, 1.186323, 1.102426, 1.276606, 4.565718, 4.977873e-06, 0.046396, 5.302956], ["cg23361265", "chr17", 16014345, 16014347, "TTC19", "protein_coding", ".", -0.167669, 0.036756, 0.845634, 0.786856, 0.908802, -4.561617, 5.076129e-06, 0.046396, 5.294467], ["cg01048810", "chr1", 44677142, 44677144, "C1orf228", "protein_coding", "S_Shelf", -0.170217, 0.037391, 0.843481, 0.783877, 0.907618, -4.552343, 5.305165e-06, 0.046396, 5.275301], ["cg09432758", "chr19", 1649123, 1649125, "TCF3", "protein_coding", "S_Shelf", 0.244579, 0.053751, 1.277084, 1.149388, 1.418966, 4.550252, 5.358179e-06, 0.046396, 5.270983], ["cg25535982", "chr11", 74319354, 74319356, "RP11-632K5.3", "lincRNA", ".", -0.294269, 0.06471, 0.745076, 0.656326, 0.845827, -4.547524, 5.428068e-06, 0.046396, 5.265355], ["cg07262001", "chr22", 42338081, 42338083, "TCF20", "protein_coding", "N_Shore", 0.384966, 0.084873, 1.469564, 1.244357, 1.73553, 4.535811, 5.738254e-06, 0.047695, 5.24122], ["cg01476568", "chr10", 5692752, 5692754, "FAM208B", "protein_coding", ".", 0.081508, 0.017988, 1.084922, 1.047338, 1.123855, 4.531155, 5.866198e-06, 0.047695, 5.231643], ["cg17875657", "chr2", 209496087, 209496089, "MAP2", "protein_coding", ".", -0.164829, 0.036495, 0.848039, 0.789498, 0.910921, -4.516425, 6.289245e-06, 0.048439, 5.201401], ["cg21611262", "chr5", 145872879, 145872881, "GRXCR2", "protein_coding", ".", -0.278149, 0.061602, 0.757184, 0.671066, 0.854353, -4.51524, 6.324523e-06, 0.048439, 5.198972], ["cg17413194", "chr17", 65057877, 65057879, "GNA13", "protein_coding", "S_Shore", -0.267505, 0.05932, 0.765287, 0.681288, 0.859642, -4.509496, 6.498187e-06, 0.048439, 5.187208], ["cg12449049", "chr6", 25088770, 25088772, "CMAHP", "unitary_pseudogene", ".", -0.153109, 0.033965, 0.858036, 0.802777, 0.917099, -4.507899, 6.547279e-06, 0.048439, 5.183939], ["cg10520790", "chr4", 16029222, 16029224, "PROM1", "protein_coding", ".", -0.169794, 0.037751, 0.843838, 0.783657, 0.908642, -4.497774, 6.866863e-06, 0.048439, 5.163242], ["cg25292309", "chr2", 58762291, 58762293, "LINC01122", "lincRNA", ".", -0.18007, 0.04006, 0.835212, 0.772143, 0.903433, -4.494996, 6.957114e-06, 0.048439, 5.157571], ["cg13359161", "chr4", 123631741, 123631743, ".", ".", ".", -0.174813, 0.038895, 0.839614, 0.777986, 0.906123, -4.494454, 6.974859e-06, 0.048439, 5.156465], ["cg20823481", "chr4", 184815092, 184815094, "ACSL1;RP11-701P16.2", "protein_coding;antisense", ".", 0.15412, 0.03434, 1.166631, 1.090693, 1.247855, 4.48799, 7.189829e-06, 0.048913, 5.143281], ["cg25922105", "chr10", 43359427, 43359429, ".", ".", "N_Shelf", -0.291549, 0.065322, 0.747105, 0.657324, 0.849149, -4.463254, 8.072435e-06, 0.053345, 5.092995], ["cg14919082", "chr2", 54673917, 54673919, "AC093110.3", "antisense", ".", -0.151434, 0.033947, 0.859475, 0.804151, 0.918605, -4.460906, 8.161381e-06, 0.053345, 5.088236], ["cg02203833", "chr7", 151057195, 151057197, "CDK5;SLC4A2", "protein_coding", "N_Shore", -0.181545, 0.040746, 0.833981, 0.769968, 0.903314, -4.455544, 8.368091e-06, 0.053644, 5.077374], ["cg18437839", "chr1", 201476584, 201476586, ".", ".", ".", -0.22453, 0.050584, 0.798892, 0.723487, 0.882155, -4.438756, 9.048045e-06, 0.056909, 5.043445], ["cg02387554", "chr7", 1193648, 1193650, ".", ".", ".", 0.165487, 0.037363, 1.179968, 1.096646, 1.269621, 4.429121, 9.461786e-06, 0.058409, 5.024027], ["cg26888941", "chr6", 33430744, 33430746, "SYNGAP1", "protein_coding", "S_Shelf", 0.436667, 0.098779, 1.547541, 1.27515, 1.878118, 4.420653, 9.840318e-06, 0.059641, 5.006991]];

    // Define the dt_args
    let dt_args = {"order": []};
    dt_args["data"] = data;

    $(document).ready(function () {
        
        $('#7f31de53-f6ba-4fc5-ae10-588030157104').DataTable(dt_args);
    });
</script>
</div>
</div></div>
</div>
</section>
</section>
</section>
<section id="standardize-and-save-top-probes-risk-adjusted-os-at-1-10e-6">
<h2><a class="toc-backref" href="#id24" role="doc-backlink">Standardize and save top probes (risk-adjusted OS at 1*10e-6)</a><a class="headerlink" href="#standardize-and-save-top-probes-risk-adjusted-os-at-1-10e-6" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Select only significant probes</span>
<span class="n">x</span> <span class="o">=</span> <span class="n">df_batchcorrected</span><span class="p">[</span><span class="n">top_os</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_list</span><span class="p">()]</span>

<span class="c1"># Standardize the data</span>
<span class="n">x</span> <span class="o">=</span> <span class="p">(</span><span class="n">x</span> <span class="o">-</span> <span class="n">x</span><span class="o">.</span><span class="n">mean</span><span class="p">())</span> <span class="o">/</span> <span class="n">x</span><span class="o">.</span><span class="n">std</span><span class="p">()</span>

<span class="c1"># Save x to pickle in output_path</span>
<span class="n">x</span><span class="o">.</span><span class="n">to_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_x.pkl&#39;</span><span class="p">)</span>

<span class="c1"># Print a statement saying that you have saved the data, the shape and the path</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully saved the data in &#39;</span> <span class="o">+</span> <span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_x.pkl&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The shape of the data is &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">shape</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Succesfully saved the data in ../Data/Processed_Data/ewas_dmr/ewas_top_x.pkl
The shape of the data is (924, 167)
</pre></div>
</div>
</div>
</div>
</section>
<section id="watermark">
<h2><a class="toc-backref" href="#id25" role="doc-backlink">Watermark</a><a class="headerlink" href="#watermark" title="Permalink to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Python implementation: CPython
Python version       : 3.8.16
IPython version      : 8.12.2

methylize: 1.1.1
pandas   : 2.0.1

Compiler    : GCC 11.3.0
OS          : Linux
Release     : 5.15.90.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 20
Architecture: 64bit
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="1_Data_Download_Process.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Data Download and Processing</p>
      </div>
    </a>
    <a class="right-next"
       href="3_Multivariate_CoxPH-Lasso_v2.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">CoxPH-Lasso on top EWAS results</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-bone-marrow-samples-only">Select diagnostic bone marrow samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas-os">Run CoxPH-EWAS OS</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#unadjusted">Unadjusted</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#risk-group-adjusted">Risk group adjusted</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas-efs">Run CoxPH-EWAS EFS</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Unadjusted</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Risk group adjusted</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-statistically-significant-cpgs">Select statistically significant CpGs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#threshold-based-on-suggested-significance-p-values-1-10e-6">Threshold based on suggested-significance p-values (1*10e-6)</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-genes">Fetch genes</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#os-suggested-significance-probes">OS suggested-significance probes:</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#efs-suggested-significance-probes">EFS suggested-significance probes:</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#standardize-and-save-top-probes-risk-adjusted-os-at-1-10e-6">Standardize and save top probes (risk-adjusted OS at 1*10e-6)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Francisco Marchi @ Lamba Lab
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      Â© Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>