

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome &#8212; Lamba Lab at UF</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '2_Univariate_CoxPH-EWAS';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="CoxPH-Lasso on top EWAS results" href="3_Multivariate_CoxPH-Lasso.html" />
    <link rel="prev" title="Data Download and Processing" href="1_Data_Download_Process.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
  
    <p class="title logo__title">Lamba Lab at UF</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="index.html">
                    DNA methylation-based diagnosis and prognosis of pediatric AML
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Specific Aims</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="docs/Specific_Aims.html">Specific Aims</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="docs/Intro_to_Methylation.html">Understanding DNA Methylation</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Data Preparation</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="1_Data_Download_Process.html">Data Download and Processing</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_Multivariate_CoxPH-Lasso.html">CoxPH-Lasso on top EWAS results</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/2_Univariate_CoxPH-EWAS.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-process-clinical-data">Load and process clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-bone-marrow-samples-only">Select diagnostic bone marrow samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-9-perform-batch-correction">Step 9. Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-ewas-os-with-batch-correction">Run EWAS OS with batch correction</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-ewas-efs-with-batch-correction">Run EWAS EFS with batch correction</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-probes-by-bonferroni-corrected-p-values-1-5-10e-7">Select probes by Bonferroni-corrected p-values (1.5*10e-7)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-probes-by-standard-genome-wide-significance-threshold-5-10e-8">Select probes by standard genome-wide significance threshold (5*10e-8)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-gene-annotation">Fetch gene annotation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ewas-os">EWAS-OS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ewas-efs">EWAS-EFS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overlap-ewas-os-and-ewas-efs">Overlap EWAS-OS and EWAS-EFS</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#identify-differentially-methylated-regions-dmrs">Identify differentially methylated regions (DMRs)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">EWAS-OS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">EWAS-EFS</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#standardize-beta-values-and-save-top-probes">Standardize beta values and save top probes</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="coxph-ewas-unadjusted-in-cog-trials-with-os-and-efs-as-outcome">
<h1><a class="toc-backref" href="#id3" role="doc-backlink">CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</a><a class="headerlink" href="#coxph-ewas-unadjusted-in-cog-trials-with-os-and-efs-as-outcome" title="Permalink to this heading">#</a></h1>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#coxph-ewas-unadjusted-in-cog-trials-with-os-and-efs-as-outcome" id="id3">CoxPH-EWAS unadjusted in COG trials with OS and EFS as outcome</a></p>
<ul>
<li><p><a class="reference internal" href="#load-and-process-clinical-data" id="id4">Load and process clinical data</a></p></li>
<li><p><a class="reference internal" href="#remove-samples-based-on-certain-clinical-features" id="id5">Remove samples based on certain clinical features</a></p>
<ul>
<li><p><a class="reference internal" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" id="id6">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></p></li>
<li><p><a class="reference internal" href="#select-diagnostic-bone-marrow-samples-only" id="id7">Select diagnostic bone marrow samples only</a></p></li>
<li><p><a class="reference internal" href="#remove-duplicate-samples" id="id8">Remove duplicate samples</a></p></li>
<li><p><a class="reference internal" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" id="id9">Match samples in clinical data to samples in methylation data</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#transform-to-m-values" id="id10">Transform to M-values</a></p></li>
<li><p><a class="reference internal" href="#step-9-perform-batch-correction" id="id11">Step 9. Perform batch correction</a></p>
<ul>
<li><p><a class="reference internal" href="#evaluate-batch-effects" id="id12">Evaluate batch effects</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#run-ewas-os-with-batch-correction" id="id13">Run EWAS OS with batch correction</a></p></li>
<li><p><a class="reference internal" href="#run-ewas-efs-with-batch-correction" id="id14">Run EWAS EFS with batch correction</a></p></li>
<li><p><a class="reference internal" href="#select-probes-by-bonferroni-corrected-p-values-1-5-10e-7" id="id15">Select probes by Bonferroni-corrected p-values (1.5*10e-7)</a></p></li>
<li><p><a class="reference internal" href="#select-probes-by-standard-genome-wide-significance-threshold-5-10e-8" id="id16">Select probes by standard genome-wide significance threshold (5*10e-8)</a></p></li>
<li><p><a class="reference internal" href="#fetch-gene-annotation" id="id17">Fetch gene annotation</a></p>
<ul>
<li><p><a class="reference internal" href="#ewas-os" id="id18">EWAS-OS</a></p></li>
<li><p><a class="reference internal" href="#ewas-efs" id="id19">EWAS-EFS</a></p></li>
<li><p><a class="reference internal" href="#overlap-ewas-os-and-ewas-efs" id="id20">Overlap EWAS-OS and EWAS-EFS</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#identify-differentially-methylated-regions-dmrs" id="id21">Identify differentially methylated regions (DMRs)</a></p>
<ul>
<li><p><a class="reference internal" href="#id1" id="id22">EWAS-OS</a></p></li>
<li><p><a class="reference internal" href="#id2" id="id23">EWAS-EFS</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#standardize-beta-values-and-save-top-probes" id="id24">Standardize beta values and save top probes</a></p></li>
<li><p><a class="reference internal" href="#watermark" id="id25">Watermark</a></p></li>
</ul>
</li>
</ul>
</nav>
<section id="load-and-process-clinical-data">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">Load and process clinical data</a><a class="headerlink" href="#load-and-process-clinical-data" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">input_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Intermediate_Files/&#39;</span>
<span class="n">clinical_data_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Raw_Data/Clinical_Data/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Processed_Data/&#39;</span>


<span class="n">df_methyl</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;df_batch_uncorrected.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39; Dataset (df) contains </span><span class="si">{</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> columns (5mC nucleotides/probes) and </span><span class="si">{</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> rows (samples).&#39;</span><span class="p">)</span>

<span class="c1"># Import functions to clean up clinical data</span>
<span class="kn">from</span> <span class="nn">FM_Functions.Clinical_Data_CleanUp</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Call functions to merge, index and clean clinical data files</span>
<span class="n">labels_0531</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_0531</span><span class="p">())</span>
<span class="n">labels_1031</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_1031</span><span class="p">())</span>
<span class="n">labels_aml05</span>        <span class="o">=</span> <span class="n">clean_aml05</span>     <span class="p">(</span><span class="n">merge_index_aml05</span><span class="p">())</span>
<span class="n">labels_beataml</span>      <span class="o">=</span> <span class="n">clean_beataml</span>   <span class="p">(</span><span class="n">merge_index_beataml</span><span class="p">())</span>
<span class="n">labels_amltcga</span>      <span class="o">=</span> <span class="n">clean_amltcga</span>   <span class="p">(</span><span class="n">merge_index_amltcga</span><span class="p">())</span>
<span class="n">labels_nordic_all</span>   <span class="o">=</span> <span class="n">clean_nordic_all</span><span class="p">(</span><span class="n">merge_index_nordic_all</span><span class="p">())</span>
<span class="n">labels_mds_taml</span>     <span class="o">=</span> <span class="n">clean_mds_taml</span>  <span class="p">(</span><span class="n">merge_index_mds_taml</span><span class="p">())</span>
<span class="n">labels_all_graal</span>    <span class="o">=</span> <span class="n">clean_all_graal</span> <span class="p">(</span><span class="n">merge_index_all_graal</span><span class="p">())</span>
<span class="n">labels_target_all</span>   <span class="o">=</span> <span class="n">clean_target_all</span><span class="p">(</span><span class="n">merge_index_target_all</span><span class="p">())</span>

<span class="c1"># Combine all clinical data labels into one dataframe</span>
<span class="n">labels_combined</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">labels_aml05</span><span class="p">,</span> <span class="n">labels_beataml</span><span class="p">,</span>
                         <span class="n">labels_0531</span><span class="p">,</span> <span class="n">labels_amltcga</span><span class="p">,</span> <span class="n">labels_1031</span><span class="p">,</span>
                        <span class="n">labels_nordic_all</span><span class="p">,</span> <span class="n">labels_mds_taml</span><span class="p">,</span>
                        <span class="n">labels_all_graal</span><span class="p">,</span><span class="n">labels_target_all</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;outer&#39;</span><span class="p">)</span>

<span class="c1"># Remove samples that are not in the methyl dataset</span>
<span class="n">df_labels</span> <span class="o">=</span> <span class="n">labels_combined</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">labels_combined</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> Dataset (df) contains 333352 columns (5mC nucleotides/probes) and 3330 rows (samples).
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/home/fmarchi/projects/MethylScore/Code/.venv_py38/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)
/home/fmarchi/projects/MethylScore/Code/.venv_py38/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-samples-based-on-certain-clinical-features">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Remove samples based on certain clinical features</a><a class="headerlink" href="#remove-samples-based-on-certain-clinical-features" title="Permalink to this heading">#</a></h2>
<section id="select-samples-from-aaml1031-0531-and-03p1-clinical-trials">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Select samples from AAML1031, 0531, and 03P1 clinical trials</a><a class="headerlink" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df1</span> <span class="o">=</span> <span class="n">df_labels</span><span class="p">[</span><span class="n">df_labels</span><span class="p">[</span><span class="s1">&#39;Clinical Trial&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;AAML0531&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML1031&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML03P1&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df_labels</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2049 samples were removed. 1281 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="select-diagnostic-bone-marrow-samples-only">
<h3><a class="toc-backref" href="#id7" role="doc-backlink">Select diagnostic bone marrow samples only</a><a class="headerlink" href="#select-diagnostic-bone-marrow-samples-only" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df2</span> <span class="o">=</span> <span class="n">df1</span><span class="p">[</span><span class="n">df1</span><span class="p">[</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Diagnosis&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Bone Marrow&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>350 samples were removed. 931 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-duplicate-samples">
<h3><a class="toc-backref" href="#id8" role="doc-backlink">Remove duplicate samples</a><a class="headerlink" href="#remove-duplicate-samples" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df3</span> <span class="o">=</span> <span class="n">df2</span><span class="p">[</span><span class="o">~</span><span class="n">df2</span><span class="p">[</span><span class="s1">&#39;Patient_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="s1">&#39;last&#39;</span><span class="p">)]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>7 samples were removed. 924 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="match-samples-in-clinical-data-to-samples-in-methylation-data">
<h3><a class="toc-backref" href="#id9" role="doc-backlink">Match samples in clinical data to samples in methylation data</a><a class="headerlink" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Match samples in clinical data to samples in methylation data</span>
<span class="n">df_methyl_filtered</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">:]</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Samples in clinical data matched to samples in methylation data.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Samples in clinical data matched to samples in methylation data.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="transform-to-m-values">
<h2><a class="toc-backref" href="#id10" role="doc-backlink">Transform to M-values</a><a class="headerlink" href="#transform-to-m-values" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>


<span class="k">def</span> <span class="nf">beta2m</span><span class="p">(</span><span class="n">val</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Transfrom beta-values into m-values&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">math</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">val</span><span class="o">/</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">val</span><span class="p">))</span>

<span class="n">x_train_m</span> <span class="o">=</span> <span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>

<span class="c1"># Get correlation between Beta and M Values</span>
<span class="n">corr</span> <span class="o">=</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">corrwith</span><span class="p">(</span><span class="n">df_methyl_filtered</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Glue them into same dataframe for vizualization purposes</span>
<span class="n">beta_to_m_values</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">x_train_m</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">corr</span><span class="p">],</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                             <span class="n">keys</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="s1">&#39;Correlation&#39;</span><span class="p">],</span>
                             <span class="n">join</span><span class="o">=</span> <span class="s1">&#39;inner&#39;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="n">hue</span><span class="p">,</span><span class="n">s</span><span class="p">):</span>
    
    <span class="c1"># Define theme</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>
    <span class="n">f</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">despine</span><span class="p">(</span><span class="n">f</span><span class="p">,</span> <span class="n">left</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">bottom</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

    <span class="c1"># Define scatterplot</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
                    <span class="n">hue</span><span class="o">=</span><span class="n">hue</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;flare&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">s</span><span class="p">,</span>
                    <span class="n">sizes</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">8</span><span class="p">),</span> <span class="n">linewidth</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span>
                    <span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1"># Calculate Pearson’s correlation coefficient (r)</span>
    <span class="c1"># and its two-tailed p-value (p)</span>

    <span class="n">r</span><span class="p">,</span><span class="n">p</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">x</span><span class="p">],</span><span class="n">df</span><span class="p">[</span><span class="n">y</span><span class="p">])</span>

    <span class="c1"># Annotate r and p into the plot</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="sa">fr</span><span class="s1">&#39;$\rho$: </span><span class="si">{</span><span class="n">r</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">, p-value: </span><span class="si">{</span><span class="n">p</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span>
                <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mf">6.0</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">),</span> <span class="n">xycoords</span><span class="o">=</span><span class="s1">&#39;data&#39;</span><span class="p">,</span>
                <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span>
                <span class="n">bbox</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;boxstyle&#39;</span><span class="p">:</span> <span class="s1">&#39;round&#39;</span><span class="p">,</span> <span class="s1">&#39;facecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;none&#39;</span><span class="p">,</span>
                <span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;lightgray&#39;</span><span class="p">})</span>

    <span class="c1"># Define plot specs</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="n">y</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;Concordance by Pearson&#39;s correlation coefficient ($\rho$) in &quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">))</span> <span class="o">+</span> <span class="s1">&#39; CpG probes&#39;</span><span class="p">,</span>
               <span class="n">fontsize</span> <span class="o">=</span> <span class="mi">12</span><span class="p">)</span> 

    <span class="c1"># Save plot figure</span>
    
    <span class="c1"># plt.savefig(&#39;../Figures/Beta_to_M_Values_&#39; + str(len(df)) + &#39;_scatterplot&#39; + &#39;.png&#39;,</span>
    <span class="c1">#              bbox_inches=&#39;tight&#39;, dpi=300)</span>

    <span class="k">return</span><span class="p">(</span><span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">())</span>

<span class="n">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">beta_to_m_values</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="n">hue</span><span class="o">=</span><span class="s1">&#39;Correlation&#39;</span><span class="p">,</span><span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/d427fda8c2d9324d392e13b8b2e9e0a19df3d6ea893b3a323da06059c5ed09e2.png" src="_images/d427fda8c2d9324d392e13b8b2e9e0a19df3d6ea893b3a323da06059c5ed09e2.png" />
</div>
</div>
</section>
<section id="step-9-perform-batch-correction">
<h2><a class="toc-backref" href="#id11" role="doc-backlink">Step 9. Perform batch correction</a><a class="headerlink" href="#step-9-perform-batch-correction" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>pyCombat</strong> is a Python tool for batch effects correction in high-throughput molecular data using empirical Bayes methods</p></li>
<li><p><strong>Github</strong>: <a class="reference external" href="https://epigenelabs.github.io/pyComBat/">https://epigenelabs.github.io/pyComBat/</a></p></li>
<li><p><strong>Implementation Preprint</strong>: <a class="reference external" href="https://doi.org/10.1101/2020.03.17.995431">bioRxiv</a></p></li>
<li><p><strong>Original Paper</strong>: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/16632515/">Adjusting batch effects in microarray expression data using empirical Bayes methods</a></p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">combat.pycombat</span> <span class="kn">import</span> <span class="n">pycombat</span>

<span class="c1"># Correct batch effects in the training dataset</span>
<span class="n">df_batchcorrected</span> <span class="o">=</span> <span class="n">pycombat</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">batch</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully corrected batch effects in the training dataset.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Found 3 batches.
Adjusting for 0 covariate(s) or covariate level(s).
Standardizing Data across genes.
Fitting L/S model and finding priors.
Finding parametric adjustments.
Adjusting the Data
Succesfully corrected batch effects in the training dataset.
</pre></div>
</div>
</div>
</div>
<section id="evaluate-batch-effects">
<h3><a class="toc-backref" href="#id12" role="doc-backlink">Evaluate batch effects</a><a class="headerlink" href="#evaluate-batch-effects" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">sklearn.preprocessing</span> <span class="kn">import</span> <span class="n">StandardScaler</span>
<span class="kn">from</span> <span class="nn">sklearn.decomposition</span> <span class="kn">import</span> <span class="n">PCA</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="k">def</span> <span class="nf">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Standardizes the data.&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">StandardScaler</span><span class="p">()</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">perform_pca</span><span class="p">(</span><span class="n">std_data</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Performs PCA on standardized data.&#39;&#39;&#39;</span>
    <span class="n">pca</span> <span class="o">=</span> <span class="n">PCA</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="mi">42</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">pca</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">std_data</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">create_principal_df</span><span class="p">(</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Creates a DataFrame with the PCA data.&#39;&#39;&#39;</span>
    <span class="n">principal_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="s1">&#39;PC2&#39;</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span>
    <span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;hue&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">principal_df</span>

<span class="k">def</span> <span class="nf">filter_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">principal_df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Filters the DataFrame by removing outliers.&#39;&#39;&#39;</span>
    <span class="n">outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">threshold</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">outliers</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df_batchcorrected</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_batchcorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df_batchcorrected</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_uncorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Plots PCA data.&#39;&#39;&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">,</span> <span class="n">legend</span> <span class="ow">in</span> <span class="p">[(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA after batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="s1">&#39;full&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="n">non_outliers</span><span class="p">,</span> <span class="n">ax1</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA before batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">non_outliers</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="kc">False</span><span class="p">)]:</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PC2&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;hue&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="n">legend</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;PC 1&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;PC 2&#39;</span><span class="p">)</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.05</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">plot_data</span><span class="p">(</span><span class="n">pcs_batchcorrected</span><span class="p">,</span> <span class="n">pcs_uncorrected</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/c1d98325e380f09b01735341403a11342d2048df65771a3aa9a625290a974759.png" src="_images/c1d98325e380f09b01735341403a11342d2048df65771a3aa9a625290a974759.png" />
</div>
</div>
</section>
</section>
<section id="run-ewas-os-with-batch-correction">
<h2><a class="toc-backref" href="#id13" role="doc-backlink">Run EWAS OS with batch correction</a><a class="headerlink" href="#run-ewas-os-with-batch-correction" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">methylize</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>

<span class="c1"># ewas_cog_corrected = methylize.diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = df3[&#39;os.time&#39;],</span>
<span class="c1">#                                     event_observed = df3[&#39;os.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = df3[&#39;os.time&#39;], # not used</span>
<span class="c1">#                                     covariates = False,</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = True,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected&#39;)</span>

<span class="n">ewas_cog_corrected</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_batch_corrected.pkl&#39;</span><span class="p">)</span>

<span class="n">methylize</span><span class="o">.</span><span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_cog_corrected</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG OS batch corrected&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/home/fmarchi/projects/MethylScore/Code/.venv_py38/lib/python3.8/site-packages/methylize/diff_meth_pos.py:15: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from tqdm.autonotebook import tqdm
</pre></div>
</div>
<img alt="_images/1abe7d9e517f33fdd20b4872a11d448e059022e9a72412f7a44561fd98a2f0e1.png" src="_images/1abe7d9e517f33fdd20b4872a11d448e059022e9a72412f7a44561fd98a2f0e1.png" />
</div>
</div>
</section>
<section id="run-ewas-efs-with-batch-correction">
<h2><a class="toc-backref" href="#id14" role="doc-backlink">Run EWAS EFS with batch correction</a><a class="headerlink" href="#run-ewas-efs-with-batch-correction" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">methylize</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>

<span class="c1"># ewas_cog_corrected_efs = methylize.diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = df3[&#39;efs.time&#39;],</span>
<span class="c1">#                                     event_observed = df3[&#39;efs.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = df3[&#39;efs.time&#39;], # not used</span>
<span class="c1">#                                     covariates = False,</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = True,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_efs_batch_corrected&#39;)</span>

<span class="n">ewas_cog_corrected_efs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_efs_batch_corrected.pkl&#39;</span><span class="p">)</span>

<span class="n">methylize</span><span class="o">.</span><span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_cog_corrected_efs</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG EFS batch corrected&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_efs_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/8fc120c02e526d0445c168c47129d5b48734ff95d81061464896df23779b4513.png" src="_images/8fc120c02e526d0445c168c47129d5b48734ff95d81061464896df23779b4513.png" />
</div>
</div>
</section>
<section id="select-probes-by-bonferroni-corrected-p-values-1-5-10e-7">
<h2><a class="toc-backref" href="#id15" role="doc-backlink">Select probes by Bonferroni-corrected p-values (1.5*10e-7)</a><a class="headerlink" href="#select-probes-by-bonferroni-corrected-p-values-1-5-10e-7" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Select rows in which `PValue` is less than 0.05/number of rows/probes (bonferroni correction)</span>
<span class="n">ewas_os_top</span> <span class="o">=</span> <span class="n">ewas_cog_corrected</span><span class="p">[</span><span class="n">ewas_cog_corrected</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="p">(</span><span class="mf">0.05</span><span class="o">/</span><span class="n">ewas_cog_corrected</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])]</span>
<span class="n">ewas_efs_top</span> <span class="o">=</span> <span class="n">ewas_cog_corrected_efs</span><span class="p">[</span><span class="n">ewas_cog_corrected_efs</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="p">(</span><span class="mf">0.05</span><span class="o">/</span><span class="n">ewas_cog_corrected_efs</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])]</span>

<span class="c1"># Get overlapping probes</span>
<span class="n">ewas_top</span> <span class="o">=</span> <span class="n">ewas_os_top</span><span class="p">[</span><span class="n">ewas_os_top</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">ewas_efs_top</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># Load manifest</span>
<span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../Data/UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span><span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span> <span class="s1">&#39;TranscriptIDs&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># print number of probes selected</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Number of CpG probes selected in EWAS OS: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">ewas_os_top</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Number of CpG probes selected in EWAS EFS: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">ewas_efs_top</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Number of CpG probes selected in EWAS OS and EFS: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">ewas_top</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of CpG probes selected in EWAS OS: 1076
Number of CpG probes selected in EWAS EFS: 593
Number of CpG probes selected in EWAS OS and EFS: 418
</pre></div>
</div>
</div>
</div>
</section>
<section id="select-probes-by-standard-genome-wide-significance-threshold-5-10e-8">
<h2><a class="toc-backref" href="#id16" role="doc-backlink">Select probes by standard genome-wide significance threshold (5*10e-8)</a><a class="headerlink" href="#select-probes-by-standard-genome-wide-significance-threshold-5-10e-8" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">## Select probes by Bonferroni-corrected p-values</span>
<span class="c1"># Select rows in which `PValue` is less than 0.05/1,000,000 (standard genome-wide significance threshold)</span>
<span class="n">ewas_os_top</span> <span class="o">=</span> <span class="n">ewas_cog_corrected</span><span class="p">[</span><span class="n">ewas_cog_corrected</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="mi">10</span><span class="o">**-</span><span class="mi">8</span><span class="p">)]</span>
<span class="n">ewas_efs_top</span> <span class="o">=</span> <span class="n">ewas_cog_corrected_efs</span><span class="p">[</span><span class="n">ewas_cog_corrected_efs</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="mi">10</span><span class="o">**-</span><span class="mi">8</span><span class="p">)]</span>

<span class="c1"># Get overlapping probes</span>
<span class="n">ewas_top</span> <span class="o">=</span> <span class="n">ewas_os_top</span><span class="p">[</span><span class="n">ewas_os_top</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">ewas_efs_top</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># print number of probes selected</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Number of CpG probes selected in EWAS OS: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">ewas_os_top</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Number of CpG probes selected in EWAS EFS: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">ewas_efs_top</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Number of CpG probes selected in EWAS OS and EFS: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">ewas_top</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Number of CpG probes selected in EWAS OS: 791
Number of CpG probes selected in EWAS EFS: 431
Number of CpG probes selected in EWAS OS and EFS: 304
</pre></div>
</div>
</div>
</div>
</section>
<section id="fetch-gene-annotation">
<h2><a class="toc-backref" href="#id17" role="doc-backlink">Fetch gene annotation</a><a class="headerlink" href="#fetch-gene-annotation" title="Permalink to this heading">#</a></h2>
<section id="ewas-os">
<h3><a class="toc-backref" href="#id18" role="doc-backlink">EWAS-OS</a><a class="headerlink" href="#ewas-os" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load manifest</span>
<span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../Data/UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span><span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span> <span class="s1">&#39;TranscriptIDs&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">ewas_os_top</span> <span class="o">=</span> <span class="n">zhou2016_probes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">ewas_os_top</span><span class="p">[[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]],</span><span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span>
<span class="n">ewas_os_top</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>TranscriptIDs</th>
      <th>CGIPosition</th>
      <th>PValue</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>cg01543603</th>
      <td>chr16</td>
      <td>89403132</td>
      <td>89403134</td>
      <td>ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD1...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000301030.7;ENST00000378330.5;ENST000003...</td>
      <td>.</td>
      <td>1.354369e-17</td>
    </tr>
    <tr>
      <th>cg08404702</th>
      <td>chr16</td>
      <td>2083386</td>
      <td>2083388</td>
      <td>TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000219476.6;ENST00000350773.7;ENST000003...</td>
      <td>N_Shore</td>
      <td>7.837118e-17</td>
    </tr>
    <tr>
      <th>cg00920892</th>
      <td>chr8</td>
      <td>37874705</td>
      <td>37874707</td>
      <td>RAB11FIP1;RAB11FIP1;RAB11FIP1;RAB11FIP1;RAB11F...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000287263.7;ENST00000330843.7;ENST000005...</td>
      <td>.</td>
      <td>8.475752e-17</td>
    </tr>
    <tr>
      <th>cg08329113</th>
      <td>chr16</td>
      <td>70737238</td>
      <td>70737240</td>
      <td>VAC14;VAC14;VAC14;VAC14;VAC14;VAC14;VAC14</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000261776.8;ENST00000561879.1;ENST000005...</td>
      <td>.</td>
      <td>8.050759e-16</td>
    </tr>
    <tr>
      <th>cg26361535</th>
      <td>chr8</td>
      <td>143494433</td>
      <td>143494435</td>
      <td>ZC3H3;ZC3H3</td>
      <td>protein_coding;protein_coding</td>
      <td>ENST00000262577.5;ENST00000528401.1</td>
      <td>.</td>
      <td>4.356361e-15</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>cg19198993</th>
      <td>chr12</td>
      <td>131223754</td>
      <td>131223756</td>
      <td>.</td>
      <td>.</td>
      <td>.</td>
      <td>.</td>
      <td>4.931295e-08</td>
    </tr>
    <tr>
      <th>cg09269619</th>
      <td>chr4</td>
      <td>719389</td>
      <td>719391</td>
      <td>PCGF3;PCGF3;PCGF3;PCGF3;PCGF3;PCGF3;PCGF3;PCGF...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000362003.8;ENST00000400151.5;ENST000004...</td>
      <td>N_Shore</td>
      <td>4.951836e-08</td>
    </tr>
    <tr>
      <th>cg21175873</th>
      <td>chr9</td>
      <td>136840885</td>
      <td>136840887</td>
      <td>RABL6;RABL6;RABL6;RABL6;RABL6;RABL6</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000311502.10;ENST00000357466.5;ENST00000...</td>
      <td>S_Shore</td>
      <td>4.960181e-08</td>
    </tr>
    <tr>
      <th>cg10535320</th>
      <td>chr17</td>
      <td>63831508</td>
      <td>63831510</td>
      <td>FTSJ3;PSMC5;PSMC5;PSMC5;PSMC5;PSMC5;PSMC5;PSMC...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000580295.1;ENST00000310144.9;ENST000003...</td>
      <td>S_Shelf</td>
      <td>4.985238e-08</td>
    </tr>
    <tr>
      <th>cg16538343</th>
      <td>chr11</td>
      <td>69005492</td>
      <td>69005494</td>
      <td>MRGPRF;MRGPRF</td>
      <td>protein_coding;protein_coding</td>
      <td>ENST00000309099.6;ENST00000441623.1</td>
      <td>Island</td>
      <td>4.991807e-08</td>
    </tr>
  </tbody>
</table>
<p>791 rows × 8 columns</p>
</div></div></div>
</div>
</section>
<section id="ewas-efs">
<h3><a class="toc-backref" href="#id19" role="doc-backlink">EWAS-EFS</a><a class="headerlink" href="#ewas-efs" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">ewas_efs_top</span> <span class="o">=</span> <span class="n">zhou2016_probes</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">ewas_efs_top</span><span class="p">[[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]],</span><span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span>
<span class="n">ewas_efs_top</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>TranscriptIDs</th>
      <th>CGIPosition</th>
      <th>PValue</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>cg00192882</th>
      <td>chr17</td>
      <td>19387806</td>
      <td>19387808</td>
      <td>MFAP4;MFAP4;MFAP4;MFAP4</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000299610.4;ENST00000395592.5;ENST000004...</td>
      <td>.</td>
      <td>4.631221e-16</td>
    </tr>
    <tr>
      <th>cg06748884</th>
      <td>chr16</td>
      <td>11138900</td>
      <td>11138902</td>
      <td>CLEC16A;CLEC16A;CLEC16A;CLEC16A;CLEC16A</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000261657.5;ENST00000409790.4;ENST000004...</td>
      <td>.</td>
      <td>1.568135e-14</td>
    </tr>
    <tr>
      <th>cg01543603</th>
      <td>chr16</td>
      <td>89403132</td>
      <td>89403134</td>
      <td>ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD1...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000301030.7;ENST00000378330.5;ENST000003...</td>
      <td>.</td>
      <td>1.543976e-13</td>
    </tr>
    <tr>
      <th>cg16721321</th>
      <td>chr4</td>
      <td>1208907</td>
      <td>1208909</td>
      <td>CTBP1-AS;SPON2;SPON2;SPON2;SPON2;SPON2;SPON2;S...</td>
      <td>antisense;protein_coding;protein_coding;protei...</td>
      <td>ENST00000625256.1;ENST00000400762.6;ENST000005...</td>
      <td>N_Shore</td>
      <td>2.869478e-13</td>
    </tr>
    <tr>
      <th>cg00206490</th>
      <td>chr16</td>
      <td>89403320</td>
      <td>89403322</td>
      <td>ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD1...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000301030.7;ENST00000378330.5;ENST000003...</td>
      <td>.</td>
      <td>1.780858e-12</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>cg14165772</th>
      <td>chr1</td>
      <td>2931066</td>
      <td>2931068</td>
      <td>.</td>
      <td>.</td>
      <td>.</td>
      <td>S_Shore</td>
      <td>4.901819e-08</td>
    </tr>
    <tr>
      <th>cg16708012</th>
      <td>chr12</td>
      <td>108598337</td>
      <td>108598339</td>
      <td>TMEM119;TMEM119;TMEM119;TMEM119</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000392806.3;ENST00000547567.1;ENST000005...</td>
      <td>.</td>
      <td>4.910177e-08</td>
    </tr>
    <tr>
      <th>cg26657404</th>
      <td>chr16</td>
      <td>85651214</td>
      <td>85651216</td>
      <td>GSE1;GSE1;GSE1;GSE1</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000253458.10;ENST00000393243.4;ENST00000...</td>
      <td>S_Shore</td>
      <td>4.930162e-08</td>
    </tr>
    <tr>
      <th>cg04981940</th>
      <td>chr1</td>
      <td>1462967</td>
      <td>1462969</td>
      <td>ATAD3C;ATAD3C</td>
      <td>protein_coding;protein_coding</td>
      <td>ENST00000378785.5;ENST00000484537.2</td>
      <td>S_Shelf</td>
      <td>4.931577e-08</td>
    </tr>
    <tr>
      <th>cg09606564</th>
      <td>chr17</td>
      <td>19387039</td>
      <td>19387041</td>
      <td>MFAP4;MFAP4;MFAP4;MFAP4</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000299610.4;ENST00000395592.5;ENST000004...</td>
      <td>.</td>
      <td>4.952723e-08</td>
    </tr>
  </tbody>
</table>
<p>431 rows × 8 columns</p>
</div></div></div>
</div>
</section>
<section id="overlap-ewas-os-and-ewas-efs">
<h3><a class="toc-backref" href="#id20" role="doc-backlink">Overlap EWAS-OS and EWAS-EFS</a><a class="headerlink" href="#overlap-ewas-os-and-ewas-efs" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ewas_os_top2</span> <span class="o">=</span> <span class="n">ewas_os_top</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;level_0&#39;</span><span class="p">:</span><span class="s1">&#39;OS Rank&#39;</span><span class="p">})</span>\
    <span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;index&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;PValue&#39;</span><span class="p">:</span><span class="s1">&#39;OS PValue&#39;</span><span class="p">})</span>
<span class="n">ewas_efs_top2</span> <span class="o">=</span> <span class="n">ewas_efs_top</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;level_0&#39;</span><span class="p">:</span><span class="s1">&#39;EFS Rank&#39;</span><span class="p">})</span>\
    <span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;index&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;PValue&#39;</span><span class="p">:</span><span class="s1">&#39;EFS PValue&#39;</span><span class="p">})[[</span><span class="s1">&#39;EFS Rank&#39;</span><span class="p">,</span><span class="s1">&#39;EFS PValue&#39;</span><span class="p">]]</span>

<span class="c1"># concatenate the two dataframes</span>
<span class="n">ewas_top2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">ewas_os_top2</span><span class="p">,</span><span class="n">ewas_efs_top2</span><span class="p">],</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span><span class="n">join</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span>

<span class="c1"># Save to csv</span>
<span class="n">ewas_top2</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_overlapping_probes.csv&#39;</span><span class="p">)</span>

<span class="n">ewas_top2</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>OS Rank</th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>TranscriptIDs</th>
      <th>CGIPosition</th>
      <th>OS PValue</th>
      <th>EFS Rank</th>
      <th>EFS PValue</th>
    </tr>
    <tr>
      <th>index</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>cg01543603</th>
      <td>0</td>
      <td>chr16</td>
      <td>89403132</td>
      <td>89403134</td>
      <td>ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD11;ANKRD1...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000301030.7;ENST00000378330.5;ENST000003...</td>
      <td>.</td>
      <td>1.354369e-17</td>
      <td>2</td>
      <td>1.543976e-13</td>
    </tr>
    <tr>
      <th>cg08404702</th>
      <td>1</td>
      <td>chr16</td>
      <td>2083386</td>
      <td>2083388</td>
      <td>TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2;TSC2</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000219476.6;ENST00000350773.7;ENST000003...</td>
      <td>N_Shore</td>
      <td>7.837118e-17</td>
      <td>32</td>
      <td>1.185579e-10</td>
    </tr>
    <tr>
      <th>cg00920892</th>
      <td>2</td>
      <td>chr8</td>
      <td>37874705</td>
      <td>37874707</td>
      <td>RAB11FIP1;RAB11FIP1;RAB11FIP1;RAB11FIP1;RAB11F...</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000287263.7;ENST00000330843.7;ENST000005...</td>
      <td>.</td>
      <td>8.475752e-17</td>
      <td>54</td>
      <td>3.603416e-10</td>
    </tr>
    <tr>
      <th>cg08329113</th>
      <td>3</td>
      <td>chr16</td>
      <td>70737238</td>
      <td>70737240</td>
      <td>VAC14;VAC14;VAC14;VAC14;VAC14;VAC14;VAC14</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000261776.8;ENST00000561879.1;ENST000005...</td>
      <td>.</td>
      <td>8.050759e-16</td>
      <td>21</td>
      <td>5.440070e-11</td>
    </tr>
    <tr>
      <th>cg26361535</th>
      <td>4</td>
      <td>chr8</td>
      <td>143494433</td>
      <td>143494435</td>
      <td>ZC3H3;ZC3H3</td>
      <td>protein_coding;protein_coding</td>
      <td>ENST00000262577.5;ENST00000528401.1</td>
      <td>.</td>
      <td>4.356361e-15</td>
      <td>11</td>
      <td>2.125994e-11</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>cg13478045</th>
      <td>759</td>
      <td>chr9</td>
      <td>134585473</td>
      <td>134585475</td>
      <td>.</td>
      <td>.</td>
      <td>.</td>
      <td>.</td>
      <td>4.473791e-08</td>
      <td>108</td>
      <td>1.606878e-09</td>
    </tr>
    <tr>
      <th>cg08387463</th>
      <td>760</td>
      <td>chr8</td>
      <td>142613185</td>
      <td>142613187</td>
      <td>ARC;ARC</td>
      <td>protein_coding;protein_coding</td>
      <td>ENST00000356613.3;ENST00000581404.1</td>
      <td>Island</td>
      <td>4.475284e-08</td>
      <td>158</td>
      <td>3.832653e-09</td>
    </tr>
    <tr>
      <th>cg01476568</th>
      <td>761</td>
      <td>chr10</td>
      <td>5692752</td>
      <td>5692754</td>
      <td>FAM208B;FAM208B;FAM208B;FAM208B;FAM208B;FAM208B</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000328090.8;ENST00000473789.4;ENST000004...</td>
      <td>.</td>
      <td>4.502326e-08</td>
      <td>382</td>
      <td>3.588294e-08</td>
    </tr>
    <tr>
      <th>cg18728264</th>
      <td>775</td>
      <td>chr11</td>
      <td>118895781</td>
      <td>118895783</td>
      <td>BCL9L;CXCR5</td>
      <td>protein_coding;protein_coding</td>
      <td>ENST00000334801.6;ENST00000292174.4</td>
      <td>.</td>
      <td>4.624835e-08</td>
      <td>384</td>
      <td>3.653998e-08</td>
    </tr>
    <tr>
      <th>cg21175873</th>
      <td>788</td>
      <td>chr9</td>
      <td>136840885</td>
      <td>136840887</td>
      <td>RABL6;RABL6;RABL6;RABL6;RABL6;RABL6</td>
      <td>protein_coding;protein_coding;protein_coding;p...</td>
      <td>ENST00000311502.10;ENST00000357466.5;ENST00000...</td>
      <td>S_Shore</td>
      <td>4.960181e-08</td>
      <td>173</td>
      <td>4.751285e-09</td>
    </tr>
  </tbody>
</table>
<p>304 rows × 11 columns</p>
</div></div></div>
</div>
</section>
</section>
<section id="identify-differentially-methylated-regions-dmrs">
<h2><a class="toc-backref" href="#id21" role="doc-backlink">Identify differentially methylated regions (DMRs)</a><a class="headerlink" href="#identify-differentially-methylated-regions-dmrs" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>Comb-p</strong> is a command-line tool and a python library that manipulates BED files of possibly irregularly spaced P-values and (1) calculates auto-correlation, (2) combines adjacent P-values, (3) performs false discovery adjustment, (4) finds regions of enrichment (i.e. series of adjacent low P-values) and (5) assigns significance to those regions.</p></li>
<li><p><strong>Github</strong>: <a class="reference external" href="https://github.com/brentp/combined-pvalues">brentp/combined-pvalues</a></p></li>
<li><p><strong>Implementation Software</strong>: <a class="reference external" href="https://life-epigenetics-methylize.readthedocs-hosted.com/en/latest/docs/diff_meth_regions.html">Methylize</a></p></li>
<li><p><strong>Original Paper</strong>: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/22954632/">Comb-p: software for combining, analyzing, grouping and correcting spatially correlated P-values</a></p></li>
</ul>
<div class="cell tag_hide-input tag_hide-output docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># ewas_os_dmr = methylize.diff_meth_regions(stats = ewas_cog_corrected,</span>
<span class="c1">#                                             manifest_or_array_type = &#39;450k&#39;,</span>
<span class="c1">#                                             verbose = True,</span>
<span class="c1">#                                             prefix = output_path + &#39;ewas_dmr/ewas_cog_os_dmr&#39;,</span>
<span class="c1">#                                             plot=False,</span>
<span class="c1">#                                             annotate=True,</span>
<span class="c1">#                                             tissue=&#39;blood&#39;</span>
<span class="c1">#                                             )</span>

<span class="c1"># ewas_efs_dmr = methylize.diff_meth_regions(stats = ewas_cog_corrected_efs,</span>
<span class="c1">#                                             manifest_or_array_type = &#39;450k&#39;,</span>
<span class="c1">#                                             verbose = True,</span>
<span class="c1">#                                             prefix = output_path + &#39;ewas_dmr/ewas_cog_efs_dmr&#39;,</span>
<span class="c1">#                                             plot=False,</span>
<span class="c1">#                                             annotate=True,</span>
<span class="c1">#                                             tissue=&#39;blood&#39;</span>
<span class="c1">#                                             )</span>

<span class="n">ewas_os_dmr</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_dmr_regions_genes.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">ewas_efs_dmr</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_efs_dmr_regions_genes.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
<section id="id1">
<h3><a class="toc-backref" href="#id22" role="doc-backlink">EWAS-OS</a><a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ewas_os_dmr</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;n_probes&#39;</span><span class="p">,</span> <span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)[[</span><span class="s1">&#39;n_probes&#39;</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">,</span> <span class="s1">&#39;genes&#39;</span><span class="p">,</span> <span class="s1">&#39;descriptions&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">25</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>n_probes</th>
      <th>name</th>
      <th>genes</th>
      <th>descriptions</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2673</th>
      <td>29</td>
      <td>cg11626629</td>
      <td>FBXL2,STARD13-AS</td>
      <td>| Homo sapiens F-box and leucine rich repeat ...</td>
    </tr>
    <tr>
      <th>2408</th>
      <td>14</td>
      <td>cg09865015</td>
      <td>APBA2,LOC649352</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>621</th>
      <td>9</td>
      <td>cg04717045</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>516</th>
      <td>9</td>
      <td>cg08558873</td>
      <td>NLRP14</td>
      <td>Homo sapiens NLR family pyrin domain containin...</td>
    </tr>
    <tr>
      <th>1532</th>
      <td>8</td>
      <td>cg02667467</td>
      <td>OTOP2</td>
      <td>Homo sapiens otopetrin 2 (OTOP2), mRNA. (from ...</td>
    </tr>
    <tr>
      <th>72</th>
      <td>8</td>
      <td>cg01800926</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1092</th>
      <td>8</td>
      <td>cg25994931</td>
      <td>CYP26A1</td>
      <td>Homo sapiens cytochrome P450 family 26 subfami...</td>
    </tr>
    <tr>
      <th>2659</th>
      <td>8</td>
      <td>cg26978172</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>411</th>
      <td>8</td>
      <td>cg03161190</td>
      <td>WBP1L</td>
      <td>Homo sapiens WW domain binding protein 1 like ...</td>
    </tr>
    <tr>
      <th>2548</th>
      <td>8</td>
      <td>cg15145693</td>
      <td>CAST</td>
      <td>| Homo sapiens calpastatin (CAST), transcript...</td>
    </tr>
    <tr>
      <th>726</th>
      <td>7</td>
      <td>cg19035004</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>525</th>
      <td>7</td>
      <td>cg10387551</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2652</th>
      <td>7</td>
      <td>cg11565299</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>743</th>
      <td>7</td>
      <td>cg04685534</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>447</th>
      <td>7</td>
      <td>cg04652645</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>949</th>
      <td>7</td>
      <td>cg21101356</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>382</th>
      <td>7</td>
      <td>cg19880751</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>650</th>
      <td>7</td>
      <td>cg12391667</td>
      <td>CEP126</td>
      <td>| Homo sapiens centrosomal protein 126 (CEP12...</td>
    </tr>
    <tr>
      <th>2079</th>
      <td>7</td>
      <td>cg22227354</td>
      <td>RAD21L1,ABR</td>
      <td>| Homo sapiens RAD21 cohesin complex componen...</td>
    </tr>
    <tr>
      <th>786</th>
      <td>6</td>
      <td>cg12590695</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2654</th>
      <td>6</td>
      <td>cg02684104</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2661</th>
      <td>6</td>
      <td>cg15174834</td>
      <td>HSPA1A</td>
      <td>Homo sapiens heat shock protein family A (Hsp7...</td>
    </tr>
    <tr>
      <th>654</th>
      <td>6</td>
      <td>cg15603568</td>
      <td>GRIA4</td>
      <td>Homo sapiens glutamate ionotropic receptor AMP...</td>
    </tr>
    <tr>
      <th>2677</th>
      <td>6</td>
      <td>cg22904406</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2197</th>
      <td>6</td>
      <td>cg01373166</td>
      <td>ADORA2A</td>
      <td>| Homo sapiens adenosine A2a receptor (ADORA2...</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="id2">
<h3><a class="toc-backref" href="#id23" role="doc-backlink">EWAS-EFS</a><a class="headerlink" href="#id2" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ewas_efs_dmr</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;n_probes&#39;</span><span class="p">,</span> <span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)[[</span><span class="s1">&#39;n_probes&#39;</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">,</span> <span class="s1">&#39;genes&#39;</span><span class="p">,</span> <span class="s1">&#39;descriptions&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">25</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>n_probes</th>
      <th>name</th>
      <th>genes</th>
      <th>descriptions</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>949</th>
      <td>13</td>
      <td>cg09865015</td>
      <td>APBA2,LOC649352</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1048</th>
      <td>10</td>
      <td>cg22878489</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>275</th>
      <td>8</td>
      <td>cg04717045</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1009</th>
      <td>8</td>
      <td>cg15145693</td>
      <td>CAST</td>
      <td>| Homo sapiens calpastatin (CAST), transcript...</td>
    </tr>
    <tr>
      <th>387</th>
      <td>7</td>
      <td>cg21101356</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>545</th>
      <td>7</td>
      <td>cg13254898</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1037</th>
      <td>7</td>
      <td>cg11565299</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1049</th>
      <td>7</td>
      <td>cg16090881</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>216</th>
      <td>6</td>
      <td>cg27261733</td>
      <td>LSP1,PPP1R10,GNL1</td>
      <td>| Homo sapiens G protein nucleolar 1 (putativ...</td>
    </tr>
    <tr>
      <th>425</th>
      <td>6</td>
      <td>cg25994931</td>
      <td>CYP26A1</td>
      <td>Homo sapiens cytochrome P450 family 26 subfami...</td>
    </tr>
    <tr>
      <th>937</th>
      <td>6</td>
      <td>cg25670330</td>
      <td>FOXL2NB</td>
      <td>Homo sapiens FOXL2 neighbor (FOXL2NB), mRNA. (...</td>
    </tr>
    <tr>
      <th>1051</th>
      <td>6</td>
      <td>cg22904406</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>234</th>
      <td>6</td>
      <td>cg16820186</td>
      <td>NCR3LG1,PLVAP</td>
      <td>Homo sapiens plasmalemma vesicle associated pr...</td>
    </tr>
    <tr>
      <th>167</th>
      <td>6</td>
      <td>cg12336290</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>592</th>
      <td>6</td>
      <td>cg08014994</td>
      <td>ABCC3</td>
      <td>Homo sapiens ATP binding cassette subfamily C ...</td>
    </tr>
    <tr>
      <th>835</th>
      <td>6</td>
      <td>cg22227354</td>
      <td>RAD21L1,ABR</td>
      <td>| Homo sapiens RAD21 cohesin complex componen...</td>
    </tr>
    <tr>
      <th>287</th>
      <td>6</td>
      <td>cg16925090</td>
      <td>CEP126</td>
      <td>| Homo sapiens centrosomal protein 126 (CEP12...</td>
    </tr>
    <tr>
      <th>1011</th>
      <td>6</td>
      <td>cg14441976</td>
      <td>SLC27A6</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1034</th>
      <td>5</td>
      <td>cg11282676</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>978</th>
      <td>5</td>
      <td>cg01886556</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1041</th>
      <td>5</td>
      <td>cg09130091</td>
      <td>TTLL9</td>
      <td>| Homo sapiens tubulin tyrosine ligase like 9...</td>
    </tr>
    <tr>
      <th>282</th>
      <td>5</td>
      <td>cg23593537</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>318</th>
      <td>5</td>
      <td>cg12590695</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>857</th>
      <td>5</td>
      <td>cg12664464</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1159</th>
      <td>5</td>
      <td>cg01894631</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
</section>
<section id="standardize-beta-values-and-save-top-probes">
<h2><a class="toc-backref" href="#id24" role="doc-backlink">Standardize beta values and save top probes</a><a class="headerlink" href="#standardize-beta-values-and-save-top-probes" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Select only significant probes</span>
<span class="n">x</span> <span class="o">=</span> <span class="n">df_batchcorrected</span><span class="p">[</span><span class="n">ewas_top2</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_list</span><span class="p">()]</span>

<span class="c1"># Standardize the data</span>
<span class="n">x</span> <span class="o">=</span> <span class="p">(</span><span class="n">x</span> <span class="o">-</span> <span class="n">x</span><span class="o">.</span><span class="n">mean</span><span class="p">())</span> <span class="o">/</span> <span class="n">x</span><span class="o">.</span><span class="n">std</span><span class="p">()</span>

<span class="c1"># Save x to pickle in output_path</span>
<span class="n">x</span><span class="o">.</span><span class="n">to_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_x.pkl&#39;</span><span class="p">)</span>

<span class="c1"># Print a statement saying that you have saved the data, the shape and the path</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully saved the data in &#39;</span> <span class="o">+</span> <span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_x.pkl&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The shape of the data is &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">shape</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="watermark">
<h2><a class="toc-backref" href="#id25" role="doc-backlink">Watermark</a><a class="headerlink" href="#watermark" title="Permalink to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Python implementation: CPython
Python version       : 3.8.16
IPython version      : 8.12.2

methylize: 1.1.1
pandas   : 2.0.1

Compiler    : GCC 11.3.0
OS          : Linux
Release     : 5.15.90.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 20
Architecture: 64bit
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="1_Data_Download_Process.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Data Download and Processing</p>
      </div>
    </a>
    <a class="right-next"
       href="3_Multivariate_CoxPH-Lasso.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">CoxPH-Lasso on top EWAS results</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-process-clinical-data">Load and process clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-bone-marrow-samples-only">Select diagnostic bone marrow samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-9-perform-batch-correction">Step 9. Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-ewas-os-with-batch-correction">Run EWAS OS with batch correction</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-ewas-efs-with-batch-correction">Run EWAS EFS with batch correction</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-probes-by-bonferroni-corrected-p-values-1-5-10e-7">Select probes by Bonferroni-corrected p-values (1.5*10e-7)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-probes-by-standard-genome-wide-significance-threshold-5-10e-8">Select probes by standard genome-wide significance threshold (5*10e-8)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-gene-annotation">Fetch gene annotation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ewas-os">EWAS-OS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#ewas-efs">EWAS-EFS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overlap-ewas-os-and-ewas-efs">Overlap EWAS-OS and EWAS-EFS</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#identify-differentially-methylated-regions-dmrs">Identify differentially methylated regions (DMRs)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">EWAS-OS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">EWAS-EFS</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#standardize-beta-values-and-save-top-probes">Standardize beta values and save top probes</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Francisco Marchi
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>