

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>MethylScoreAML Px Development &#8212; MethylScoreAML</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '3.1_MethylScoreAML_Px_Development_v5';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="MethylScoreAML Px Validation" href="3.2_MethylScoreAML_Px_Validation_Peds_v2.html" />
    <link rel="prev" title="Epigenome Wide Association Study" href="2_Univariate_CoxPH-EWAS_v3.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
  
    <p class="title logo__title">MethylScoreAML</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="index.html">
                    The DNA methylome refines clinical diagnosis and prognosis of Acute Myeloid Leukemia
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Data Download and Processing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="1_Data_Download_Process.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="1_Test_Data_Process_v1.html">Validation Dataset</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">AML Prognostic Model</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="2_Univariate_CoxPH-EWAS_v3.html">Epigenome Wide Association Study</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">MethylScoreAML Px Development</a></li>
<li class="toctree-l1"><a class="reference internal" href="3.2_MethylScoreAML_Px_Validation_Peds_v2.html">MethylScoreAML Px Validation</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">AML Diagnostic Model</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="4_Methylome_Atlas_PaCMAP_v4.html">Methylome Atlas of Acute Leukemia</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_Multiclass_AMLsubtype_Classifier_v2.html">AML Diagnostic Subtype Classifier</a></li>
<li class="toctree-l1"><a class="reference internal" href="6_Sankey_Plots_v1.html">Evaluating Utility of Dx Model</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Nanopore Sequencing Testing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="8_Nanopore_Sample.html">Applying models to Nanopore data</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Other Benchmarks and Analyses</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="5_Multiclass_Julia_Gen_v1.html">Bayesian Inference with Julia - Gen</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_PyTorch_Multiclass_Classifier_v3.html">PyTorch Classifier Benchmark</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_Multiclass_PedsPacmap_Classifier_v2.html">AML Peds PaCMAP Classifier Benchmark</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/3.1_MethylScoreAML_Px_Development_v5.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>MethylScoreAML Px Development</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load Data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#standardize-data">Standardize Data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-multivariate-coxph-lasso">Perform multivariate CoxPH-Lasso</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#set-threshold">Set Threshold</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-table-of-35-cpgs-coefficients-with-ewas-results">Save table of 35 CpGs coefficients with EWAS results</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generate-model-results">Generate Model Results</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#discovery-dataset">Discovery Dataset</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#kaplan-meier-plots">Kaplan-Meier Plots</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-study-population">Overall study population</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#per-risk-group">Per risk group</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#forest-plots">Forest Plots</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#with-mrd-1">With MRD 1</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#without-mrd-1">Without MRD 1</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#roc-auc">ROC AUC</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#box-plots">Box Plots</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#stacked-bar-plots">Stacked Bar Plots</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#patient-characteristics-table">Patient Characteristics Table</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Overall study population</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#by-methylscore-category">By MethylScore category</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-gene-annotation">Fetch gene annotation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="methylscoreaml-px-development">
<h1><a class="toc-backref" href="#id2" role="doc-backlink">MethylScoreAML Px Development</a><a class="headerlink" href="#methylscoreaml-px-development" title="Permalink to this heading">#</a></h1>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#methylscoreaml-px-development" id="id2">MethylScoreAML Px Development</a></p>
<ul>
<li><p><a class="reference internal" href="#load-data" id="id3">Load Data</a></p></li>
<li><p><a class="reference internal" href="#standardize-data" id="id4">Standardize Data</a></p></li>
<li><p><a class="reference internal" href="#perform-multivariate-coxph-lasso" id="id5">Perform multivariate CoxPH-Lasso</a></p>
<ul>
<li><p><a class="reference internal" href="#set-threshold" id="id6">Set Threshold</a></p></li>
<li><p><a class="reference internal" href="#save-table-of-35-cpgs-coefficients-with-ewas-results" id="id7">Save table of 35 CpGs coefficients with EWAS results</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#generate-model-results" id="id8">Generate Model Results</a></p>
<ul>
<li><p><a class="reference internal" href="#discovery-dataset" id="id9">Discovery Dataset</a></p></li>
<li><p><a class="reference internal" href="#kaplan-meier-plots" id="id10">Kaplan-Meier Plots</a></p>
<ul>
<li><p><a class="reference internal" href="#overall-study-population" id="id11">Overall study population</a></p></li>
<li><p><a class="reference internal" href="#per-risk-group" id="id12">Per risk group</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#forest-plots" id="id13">Forest Plots</a></p>
<ul>
<li><p><a class="reference internal" href="#with-mrd-1" id="id14">With MRD 1</a></p></li>
<li><p><a class="reference internal" href="#without-mrd-1" id="id15">Without MRD 1</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#roc-auc" id="id16">ROC AUC</a></p></li>
<li><p><a class="reference internal" href="#box-plots" id="id17">Box Plots</a></p></li>
<li><p><a class="reference internal" href="#stacked-bar-plots" id="id18">Stacked Bar Plots</a></p></li>
<li><p><a class="reference internal" href="#patient-characteristics-table" id="id19">Patient Characteristics Table</a></p>
<ul>
<li><p><a class="reference internal" href="#id1" id="id20">Overall study population</a></p></li>
<li><p><a class="reference internal" href="#by-methylscore-category" id="id21">By MethylScore category</a></p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference internal" href="#fetch-gene-annotation" id="id22">Fetch gene annotation</a></p></li>
<li><p><a class="reference internal" href="#watermark" id="id23">Watermark</a></p></li>
</ul>
</li>
</ul>
</nav>
<section id="load-data">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Load Data</a><a class="headerlink" href="#load-data" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">input_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Intermediate_Files/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="s1">&#39;../Data/Processed_Data/&#39;</span>

<span class="c1"># read top CpGs selected from previous code file (univariate cox-ph EWAS)</span>
<span class="n">ewas_top_cpgs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">output_path</span><span class="o">+</span><span class="s1">&#39;ewas_dmr/ewas_top_cpgs.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Load clinical data</span>
<span class="n">discovery_clinical_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;discovery_clinical_data.csv&#39;</span><span class="p">,</span>
                                      <span class="n">low_memory</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">ewas_top_cpgs</span><span class="o">.</span><span class="n">index</span><span class="p">]</span>

<span class="c1"># save clinical data</span>
<span class="n">discovery_clinical_data</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;data/ewas_top_cpgs_clinical_data_discovery.xlsx&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39; Dataset (df) contains </span><span class="si">{</span><span class="n">ewas_top_cpgs</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> columns (5mC nucleotides/probes) and </span><span class="si">{</span><span class="n">ewas_top_cpgs</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> rows (samples).&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> Dataset (df) contains 167 columns (5mC nucleotides/probes) and 924 rows (samples).
</pre></div>
</div>
</div>
</div>
</section>
<section id="standardize-data">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">Standardize Data</a><a class="headerlink" href="#standardize-data" title="Permalink to this heading">#</a></h2>
<p>Standardize the data so that mean becomes 0 and standard deviation becomes 1.</p>
<p>This is done to avoid biasing the model towards features with large values.</p>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">reference_df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Standardize data using mean and standard deviation of reference dataset&quot;&quot;&quot;</span>

    <span class="c1"># Keep only columns that are in both datasets</span>
    <span class="n">reference_df</span> <span class="o">=</span> <span class="n">reference_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">]</span>

    <span class="c1"># Standardize data</span>
    <span class="n">df</span> <span class="o">=</span> <span class="p">(</span><span class="n">df</span> <span class="o">-</span> <span class="n">reference_df</span><span class="o">.</span><span class="n">mean</span><span class="p">())</span> <span class="o">/</span> <span class="n">reference_df</span><span class="o">.</span><span class="n">std</span><span class="p">()</span>

    <span class="k">return</span> <span class="n">df</span>

<span class="c1"># Standardize data</span>
<span class="n">x_test_m_z</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="o">=</span> <span class="n">ewas_top_cpgs</span><span class="p">,</span> <span class="n">reference_df</span><span class="o">=</span> <span class="n">ewas_top_cpgs</span><span class="p">)</span>

<span class="c1"># Save standardized data</span>
<span class="n">x_test_m_z</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;data/ewas_top_167_cpgs_M-Values_standardized_discovery_cohort.xlsx&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="perform-multivariate-coxph-lasso">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Perform multivariate CoxPH-Lasso</a><a class="headerlink" href="#perform-multivariate-coxph-lasso" title="Permalink to this heading">#</a></h2>
<p>Here we will run on 167 CpGs that were selected from EWAS risk-adjusted OS at 1*10e-6 significance:</p>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import functions to clean up clinical data</span>
<span class="kn">from</span> <span class="nn">source.cox_lasso_functions</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># raw_coefs = train_coxph_lasso(df=discovery_clinical_data,</span>
<span class="c1">#                                 event= &#39;os.evnt&#39;,</span>
<span class="c1">#                                 time= &#39;os.time&#39;,</span>
<span class="c1">#                                 train_x=x,</span>
<span class="c1">#                                 cv_n_split=10,</span>
<span class="c1">#                                 loops=1000)</span>
<span class="c1"># raw_coefs.to_csv(output_path + &#39;multivariate_cox_lasso/ewas_cog_os_raw_coefs.csv&#39;)</span>


<span class="n">raw_coefs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;multivariate_cox_lasso/ewas_cog_os_raw_coefs.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
<section id="set-threshold">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Set Threshold</a><a class="headerlink" href="#set-threshold" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mean_coefs</span> <span class="o">=</span> <span class="n">set_cutoff</span><span class="p">(</span><span class="n">coefs</span><span class="o">=</span><span class="n">raw_coefs</span><span class="p">,</span><span class="n">threshold</span><span class="o">=</span><span class="mf">0.99</span><span class="p">)</span>
                                
<span class="n">plot_nonzero_coef_freq</span><span class="p">(</span><span class="n">raw_coefs</span><span class="p">,</span> <span class="n">mean_coefs</span><span class="p">,</span> <span class="n">threshold</span><span class="o">=</span><span class="mf">0.99</span><span class="p">,</span> <span class="n">savefig</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">fig_path</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">4</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/bc7c98a8227111f3d373632aabfbf33c9e6dc4bb17c953098d0a70bfc2843873.png" src="_images/bc7c98a8227111f3d373632aabfbf33c9e6dc4bb17c953098d0a70bfc2843873.png" />
</div>
</div>
</section>
<section id="save-table-of-35-cpgs-coefficients-with-ewas-results">
<h3><a class="toc-backref" href="#id7" role="doc-backlink">Save table of 35 CpGs coefficients with EWAS results</a><a class="headerlink" href="#save-table-of-35-cpgs-coefficients-with-ewas-results" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load `ewas_cpgs_top_OS_with_EFSinfo.xlsx`</span>
<span class="n">ewas_cpgs_top_OS_with_EFSinfo</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="s1">&#39;data/ewas_cpgs_top_OS_with_EFSinfo.xlsx&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>\
    <span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">mean_coefs</span><span class="o">.</span><span class="n">to_frame</span><span class="p">(</span><span class="s1">&#39;MethylScoreAML Px Coefs&#39;</span><span class="p">))</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;MethylScoreAML Px Coefs&#39;</span><span class="p">,</span> <span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>\
        <span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;data/ewas_cpgs_top_OS_with_EFSinfo_with_MethylScoreAML_Px_Coefs.xlsx&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
</section>
<section id="generate-model-results">
<h2><a class="toc-backref" href="#id8" role="doc-backlink">Generate Model Results</a><a class="headerlink" href="#generate-model-results" title="Permalink to this heading">#</a></h2>
<section id="discovery-dataset">
<h3><a class="toc-backref" href="#id9" role="doc-backlink">Discovery Dataset</a><a class="headerlink" href="#discovery-dataset" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">score_name</span> <span class="o">=</span> <span class="s1">&#39;MethylScoreAML_Px&#39;</span>

<span class="n">df</span><span class="p">,</span> <span class="n">threshold</span> <span class="o">=</span> <span class="n">generate_coxph_score</span><span class="p">(</span><span class="n">coef_mean</span><span class="o">=</span><span class="n">mean_coefs</span><span class="p">,</span>
                                        <span class="n">x</span><span class="o">=</span><span class="n">x_test_m_z</span><span class="p">,</span>
                                        <span class="n">df</span><span class="o">=</span><span class="n">discovery_clinical_data</span><span class="p">,</span>
                                        <span class="n">score_name</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                                        <span class="n">train_test</span><span class="o">=</span><span class="s2">&quot;train&quot;</span><span class="p">,</span>
                                        <span class="n">rpart_outcome</span><span class="o">=</span><span class="s1">&#39;os.time&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Continuous score cut at the value of 0.2208 for [&#39;Other&#39;, &#39;Normal&#39;, &#39;MLL&#39;, &#39;inv(16)&#39;, &#39;t(8;21)&#39;, nan]
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">hue</span> <span class="o">=</span> <span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span> <span class="p">)</span>

<span class="c1"># add vertical line at threshold and third quartile</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">threshold</span> <span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">score_name</span><span class="p">]</span><span class="o">.</span><span class="n">quantile</span><span class="p">(</span><span class="mf">0.6</span><span class="p">),</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">)</span>


<span class="c1"># add legend</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">({</span><span class="s1">&#39;RPart&#39;</span><span class="p">:</span><span class="n">threshold</span><span class="p">,</span> <span class="s1">&#39;60</span><span class="si">%-40%</span><span class="s1">&#39;</span><span class="p">:</span><span class="n">df</span><span class="p">[</span><span class="n">score_name</span><span class="p">]</span><span class="o">.</span><span class="n">quantile</span><span class="p">(</span><span class="mf">0.6</span><span class="p">)})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Distribution of &#39;</span> <span class="o">+</span> <span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; in COG trials&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Text(0.5, 1.0, &#39;Distribution of MethylScoreAML_Px in COG trials&#39;)
</pre></div>
</div>
<img alt="_images/674cd4fc7c8a23abfddb8875e4816df7cd0e3eb97f6904da6fa4673fa14dce36.png" src="_images/674cd4fc7c8a23abfddb8875e4816df7cd0e3eb97f6904da6fa4673fa14dce36.png" />
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="c1"># create scatterplot </span>
<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;os.time&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>

<span class="c1"># use hexbin plot to show density</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">gca</span><span class="p">()</span>
<span class="n">ax</span><span class="o">.</span><span class="n">hexbin</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">score_name</span><span class="p">],</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;os.time&#39;</span><span class="p">],</span> <span class="n">gridsize</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;Blues&#39;</span><span class="p">)</span>

<span class="c1"># add title</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Hexbin Scatterplot of </span><span class="si">{}</span><span class="s1"> in COG trials&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">score_name</span><span class="p">))</span> 

<span class="c1"># add vertical line at threshold</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">threshold</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Threshold&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>

<span class="c1"># save plot as png</span>
<span class="c1"># plt.savefig(path + &#39;/Hexbin_Scatterplot_{}_efs.time_{}.png&#39;.format(score_name, len(df)), </span>
<span class="c1">#             bbox_inches=&#39;tight&#39;, dpi=300)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/e4cd75eba03b95a2ee025832ea688250f68f6a2bc5fd691f963cad9c9275c5f8.png" src="_images/e4cd75eba03b95a2ee025832ea688250f68f6a2bc5fd691f963cad9c9275c5f8.png" />
</div>
</div>
</section>
<section id="kaplan-meier-plots">
<h3><a class="toc-backref" href="#id10" role="doc-backlink">Kaplan-Meier Plots</a><a class="headerlink" href="#kaplan-meier-plots" title="Permalink to this heading">#</a></h3>
<section id="overall-study-population">
<h4><a class="toc-backref" href="#id11" role="doc-backlink">Overall study population</a><a class="headerlink" href="#overall-study-population" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import Plotting Functions</span>
<span class="kn">from</span> <span class="nn">source.data_visualization_functions</span> <span class="kn">import</span> <span class="o">*</span>

<span class="n">draw_kaplan_meier</span><span class="p">(</span><span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                        <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
                        <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">save_survival_table</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">add_risk_counts</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/6dd70621263f6de091ba69bbfe118338dcd521a2b0030a81e434150f53dfadfa.png" src="_images/6dd70621263f6de091ba69bbfe118338dcd521a2b0030a81e434150f53dfadfa.png" />
</div>
</div>
</section>
<section id="per-risk-group">
<h4><a class="toc-backref" href="#id12" role="doc-backlink">Per risk group</a><a class="headerlink" href="#per-risk-group" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">draw_kaplan_meier</span><span class="p">(</span><span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                        <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;High Risk&#39;</span><span class="p">],</span>
                        <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">save_survival_table</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">add_risk_counts</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials, High Risk Group&#39;</span><span class="p">)</span>

<span class="n">draw_kaplan_meier</span><span class="p">(</span><span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                        <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Low Risk&#39;</span><span class="p">],</span>
                        <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">save_survival_table</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">add_risk_counts</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials, Low Risk Group&#39;</span><span class="p">)</span>

<span class="n">draw_kaplan_meier</span><span class="p">(</span><span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                        <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">],</span>
                        <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">save_survival_table</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">add_risk_counts</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials, Standard Risk Group&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/791e9c9039ea3d1532eada887e06147de4748d35c5384e86459ec56caa1c4c68.png" src="_images/791e9c9039ea3d1532eada887e06147de4748d35c5384e86459ec56caa1c4c68.png" />
<img alt="_images/5faefdeb8e24ca0d43068c65e4a479c5243fc7b0a5af217fb4fc075cb2384900.png" src="_images/5faefdeb8e24ca0d43068c65e4a479c5243fc7b0a5af217fb4fc075cb2384900.png" />
<img alt="_images/e3be77e6f0ffb956c7ceb4fa237a4ed2735150d466b0dcf21c0bed72865333fe.png" src="_images/e3be77e6f0ffb956c7ceb4fa237a4ed2735150d466b0dcf21c0bed72865333fe.png" />
</div>
</div>
</section>
</section>
<section id="forest-plots">
<h3><a class="toc-backref" href="#id13" role="doc-backlink">Forest Plots</a><a class="headerlink" href="#forest-plots" title="Permalink to this heading">#</a></h3>
<section id="with-mrd-1">
<h4><a class="toc-backref" href="#id14" role="doc-backlink">With MRD 1</a><a class="headerlink" href="#with-mrd-1" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">draw_forest_plot</span><span class="p">(</span><span class="n">time</span><span class="o">=</span><span class="s1">&#39;os.time&#39;</span><span class="p">,</span>
                    <span class="n">event</span><span class="o">=</span><span class="s1">&#39;os.evnt&#39;</span><span class="p">,</span>
                    <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
                    <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials:&#39;</span><span class="p">,</span>
                    <span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                    <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">draw_forest_plot</span><span class="p">(</span><span class="n">time</span><span class="o">=</span><span class="s1">&#39;efs.time&#39;</span><span class="p">,</span>
                    <span class="n">event</span><span class="o">=</span><span class="s1">&#39;efs.evnt&#39;</span><span class="p">,</span>
                    <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
                    <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials:&#39;</span><span class="p">,</span>
                    <span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                    <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/a008f30c8f320b03fdcbe30bba28925e83944c631429359f7e59306406f07dd6.png" src="_images/a008f30c8f320b03fdcbe30bba28925e83944c631429359f7e59306406f07dd6.png" />
<img alt="_images/7f510f0f323c0bcbacc1a63b82633e982c88f18dbebcb424bf1725d7db5cc01b.png" src="_images/7f510f0f323c0bcbacc1a63b82633e982c88f18dbebcb424bf1725d7db5cc01b.png" />
</div>
</div>
</section>
<section id="without-mrd-1">
<h4><a class="toc-backref" href="#id15" role="doc-backlink">Without MRD 1</a><a class="headerlink" href="#without-mrd-1" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">draw_forest_plot_noMRD</span><span class="p">(</span><span class="n">time</span><span class="o">=</span><span class="s1">&#39;os.time&#39;</span><span class="p">,</span>
                    <span class="n">event</span><span class="o">=</span><span class="s1">&#39;os.evnt&#39;</span><span class="p">,</span>
                    <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
                    <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials:&#39;</span><span class="p">,</span>
                    <span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                    <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="n">draw_forest_plot_noMRD</span><span class="p">(</span><span class="n">time</span><span class="o">=</span><span class="s1">&#39;efs.time&#39;</span><span class="p">,</span>
                    <span class="n">event</span><span class="o">=</span><span class="s1">&#39;efs.evnt&#39;</span><span class="p">,</span>
                    <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
                    <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG trials:&#39;</span><span class="p">,</span>
                    <span class="n">scorename</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                    <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/9a10791b7701dac34f00ae8f5c288a0df3c90d5703ba545d07ff798a2f72183a.png" src="_images/9a10791b7701dac34f00ae8f5c288a0df3c90d5703ba545d07ff798a2f72183a.png" />
<img alt="_images/929c29930a04a282d2bca957a2894a531967c379966541e4494d6dfa747a74e0.png" src="_images/929c29930a04a282d2bca957a2894a531967c379966541e4494d6dfa747a74e0.png" />
</div>
</div>
</section>
</section>
<section id="roc-auc">
<h3><a class="toc-backref" href="#id16" role="doc-backlink">ROC AUC</a><a class="headerlink" href="#roc-auc" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Your current preprocessing</span>
<span class="n">df</span><span class="p">[</span><span class="s1">&#39;Risk Group bins&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">replace</span><span class="p">({</span><span class="s1">&#39;Low Risk&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">:</span><span class="mf">0.5</span><span class="p">,</span> <span class="s1">&#39;High Risk&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">})</span>
<span class="n">df</span><span class="p">[</span><span class="s1">&#39;MRD 1 bins&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;MRD 1 Status&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">replace</span><span class="p">({</span><span class="s1">&#39;Negative&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;Positive&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">})</span>
<span class="n">df2</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">&#39;os.evnt&#39;</span><span class="p">,</span> <span class="s1">&#39;MethylScoreAML_Px_cat_bin&#39;</span><span class="p">,</span> <span class="s1">&#39;Risk Group bins&#39;</span><span class="p">,</span> <span class="s1">&#39;MRD 1 bins&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span>

<span class="c1"># rename column `MethylScoreAML_Px_cat_bin` to `MethylScoreAML Px`</span>
<span class="n">df2</span> <span class="o">=</span> <span class="n">df2</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;MethylScoreAML_Px_cat_bin&#39;</span><span class="p">:</span><span class="s1">&#39;MethylScoreAML_Px&#39;</span><span class="p">})</span>

<span class="c1"># Add new columns based on standardized values</span>
<span class="n">df2</span><span class="p">[</span><span class="s1">&#39;MRD1 + Risk Group&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;MRD 1 bins&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Risk Group bins&#39;</span><span class="p">]</span>
<span class="n">df2</span><span class="p">[</span><span class="s1">&#39;MRD1 + Risk Group + MethylScore&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;MRD 1 bins&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Risk Group bins&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;MethylScoreAML_Px&#39;</span><span class="p">]</span>


<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">from</span> <span class="nn">sklearn.metrics</span> <span class="kn">import</span> <span class="n">roc_curve</span><span class="p">,</span> <span class="n">auc</span>
<span class="kn">from</span> <span class="nn">sklearn.preprocessing</span> <span class="kn">import</span> <span class="n">LabelBinarizer</span>

<span class="k">def</span> <span class="nf">plot_roc_auc</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">score_columns</span><span class="p">,</span> <span class="n">outcome_column</span><span class="p">,</span> <span class="n">trial_name</span><span class="o">=</span><span class="s1">&#39;discovery cohort&#39;</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Plots the ROC AUC curves for multiple models given a dataframe and multiple score columns.</span>
<span class="sd">    </span>
<span class="sd">    Parameters:</span>
<span class="sd">    - df (pd.DataFrame): Dataframe containing the score and outcome columns.</span>
<span class="sd">    - score_columns (list of str): List of names of columns that contain the scores.</span>
<span class="sd">    - outcome_column (str): The name of the column that contains the true outcomes.</span>
<span class="sd">    </span>
<span class="sd">    Returns:</span>
<span class="sd">    None</span>
<span class="sd">    &quot;&quot;&quot;</span>
    
    <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">()</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;ROC AUC in &#39;</span> <span class="o">+</span> <span class="n">trial_name</span> <span class="o">+</span> <span class="s1">&#39;, n=</span><span class="si">{}</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)))</span>
    
    <span class="c1"># plot random guessing line</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">([</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="s1">&#39;r--&#39;</span><span class="p">)</span>

    <span class="c1"># binarize the outcome variable</span>
    <span class="n">lb</span> <span class="o">=</span> <span class="n">LabelBinarizer</span><span class="p">()</span>
    <span class="n">lb</span><span class="o">.</span><span class="n">fit</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">outcome_column</span><span class="p">])</span>
    <span class="n">y</span> <span class="o">=</span> <span class="n">lb</span><span class="o">.</span><span class="n">transform</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">outcome_column</span><span class="p">])</span>
    
    <span class="c1"># Loop over score_columns to plot multiple ROC curves</span>
    <span class="k">for</span> <span class="n">score_column</span> <span class="ow">in</span> <span class="n">score_columns</span><span class="p">:</span>
        
        <span class="c1"># calculate the fpr and tpr for all thresholds of the classification</span>
        <span class="n">fpr</span><span class="p">,</span> <span class="n">tpr</span><span class="p">,</span> <span class="n">threshold</span> <span class="o">=</span> <span class="n">roc_curve</span><span class="p">(</span><span class="n">y</span><span class="p">,</span> <span class="n">df</span><span class="p">[</span><span class="n">score_column</span><span class="p">])</span>
        <span class="n">roc_auc</span> <span class="o">=</span> <span class="n">auc</span><span class="p">(</span><span class="n">fpr</span><span class="p">,</span> <span class="n">tpr</span><span class="p">)</span>
        
        <span class="c1"># plot ROC curve for this score_column</span>
        <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">fpr</span><span class="p">,</span> <span class="n">tpr</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">score_column</span><span class="si">}</span><span class="s1"> AUC = %0.2f&#39;</span> <span class="o">%</span> <span class="n">roc_auc</span><span class="p">)</span>

    <span class="c1"># set x and y limits</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlim</span><span class="p">([</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">])</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylim</span><span class="p">([</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">])</span>

    <span class="c1"># set x and y labels</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;True Positive Rate&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;False Positive Rate&#39;</span><span class="p">)</span>

    <span class="c1"># add legend</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="s1">&#39;lower right&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="c1"># Example Usage:</span>
<span class="n">score_columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;MethylScoreAML_Px&#39;</span><span class="p">,</span> <span class="s1">&#39;Risk Group bins&#39;</span><span class="p">,</span> <span class="s1">&#39;MRD 1 bins&#39;</span><span class="p">,</span> <span class="s1">&#39;MRD1 + Risk Group&#39;</span><span class="p">,</span> <span class="s1">&#39;MRD1 + Risk Group + MethylScore&#39;</span><span class="p">]</span>
<span class="n">outcome_column</span> <span class="o">=</span> <span class="s1">&#39;os.evnt&#39;</span>
<span class="n">plot_roc_auc</span><span class="p">(</span><span class="n">df2</span><span class="p">,</span> <span class="n">score_columns</span><span class="p">,</span> <span class="n">outcome_column</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/b2e5ea20ae14f4997217dc7a48d91341ba77526e56bf43395d59ced31503ec0e.png" src="_images/b2e5ea20ae14f4997217dc7a48d91341ba77526e56bf43395d59ced31503ec0e.png" />
</div>
</div>
</section>
<section id="box-plots">
<h3><a class="toc-backref" href="#id17" role="doc-backlink">Box Plots</a><a class="headerlink" href="#box-plots" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">draw_boxplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">discovery_clinical_data</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                <span class="n">order</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;High Risk&#39;</span><span class="p">,</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">,</span> <span class="s1">&#39;Low Risk&#39;</span><span class="p">],</span>
                <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">,</span>
                <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>

<span class="n">draw_boxplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">discovery_clinical_data</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;MRD 1 Status&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                <span class="n">order</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Positive&#39;</span><span class="p">,</span><span class="s1">&#39;Negative&#39;</span><span class="p">],</span>
                <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">,</span>
                <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>

<span class="n">draw_boxplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">discovery_clinical_data</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;Primary Cytogenetic Code&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                <span class="n">order</span><span class="o">=</span><span class="s1">&#39;auto&#39;</span><span class="p">,</span>
                <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">,</span>
                <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>p-value annotation legend:
      ns: p &lt;= 1.00e+00
       *: 1.00e-02 &lt; p &lt;= 5.00e-02
      **: 1.00e-03 &lt; p &lt;= 1.00e-02
     ***: 1.00e-04 &lt; p &lt;= 1.00e-03
    ****: p &lt;= 1.00e-04

High Risk vs. Standard Risk: Kruskal-Wallis independent samples (pairwise between groups) with Bonferroni correction, P_val:3.325e-07 Stat=2.817e+01
Standard Risk vs. Low Risk: Kruskal-Wallis independent samples (pairwise between groups) with Bonferroni correction, P_val:4.225e-79 Stat=3.568e+02
High Risk vs. Low Risk: Kruskal-Wallis independent samples (pairwise between groups) with Bonferroni correction, P_val:4.232e-27 Stat=1.184e+02
</pre></div>
</div>
<img alt="_images/e36135054c3dc169be170064f910f67cd9afb9c04268ec67fefe19883e344f10.png" src="_images/e36135054c3dc169be170064f910f67cd9afb9c04268ec67fefe19883e344f10.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>p-value annotation legend:
      ns: p &lt;= 1.00e+00
       *: 1.00e-02 &lt; p &lt;= 5.00e-02
      **: 1.00e-03 &lt; p &lt;= 1.00e-02
     ***: 1.00e-04 &lt; p &lt;= 1.00e-03
    ****: p &lt;= 1.00e-04

Positive vs. Negative: Kruskal-Wallis independent samples (pairwise between groups) with Bonferroni correction, P_val:2.989e-05 Stat=1.742e+01
</pre></div>
</div>
<img alt="_images/f32bee2937c9e454cab4c0908fa80c14fee4fe66ee91a3c554975c3fe46492bf.png" src="_images/f32bee2937c9e454cab4c0908fa80c14fee4fe66ee91a3c554975c3fe46492bf.png" />
<img alt="_images/56890b8b3ab5867baaeebfe74250ff205a6aaa1fdeb5cc4ab0d3f60f69e86b5e.png" src="_images/56890b8b3ab5867baaeebfe74250ff205a6aaa1fdeb5cc4ab0d3f60f69e86b5e.png" />
</div>
</div>
</section>
<section id="stacked-bar-plots">
<h3><a class="toc-backref" href="#id18" role="doc-backlink">Stacked Bar Plots</a><a class="headerlink" href="#stacked-bar-plots" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">draw_stacked_barplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">discovery_clinical_data</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;MRD 1 Status&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
             <span class="n">order</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Positive&#39;</span><span class="p">,</span><span class="s1">&#39;Negative&#39;</span><span class="p">],</span>
             <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">,</span>
             <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>

<span class="n">draw_stacked_barplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">discovery_clinical_data</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;Risk Group&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                <span class="n">order</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;High Risk&#39;</span><span class="p">,</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">,</span> <span class="s1">&#39;Low Risk&#39;</span><span class="p">],</span>
                <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">,</span>
                <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>

<span class="n">draw_stacked_barplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">discovery_clinical_data</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;Primary Cytogenetic Code&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                <span class="n">order</span><span class="o">=</span><span class="s1">&#39;auto&#39;</span><span class="p">,</span>
                <span class="n">trialname</span><span class="o">=</span><span class="s1">&#39;COG&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">,</span>
                <span class="n">save_plot</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">8</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="_images/1c845c57eb5fa7f6e7251026fe1712bf238006ebb0225644e45c9fcdd9aa5bc1.png" src="_images/1c845c57eb5fa7f6e7251026fe1712bf238006ebb0225644e45c9fcdd9aa5bc1.png" />
<img alt="_images/a0cf35213644865afe56bf26165acf3f71de3ba151db792ec036efeab44d80f3.png" src="_images/a0cf35213644865afe56bf26165acf3f71de3ba151db792ec036efeab44d80f3.png" />
<img alt="_images/dfcae5ea8e11a7f78b6b728d56dc3daa0be84d3e49cc282decd7b58db8753e35.png" src="_images/dfcae5ea8e11a7f78b6b728d56dc3daa0be84d3e49cc282decd7b58db8753e35.png" />
</div>
</div>
</section>
<section id="patient-characteristics-table">
<h3><a class="toc-backref" href="#id19" role="doc-backlink">Patient Characteristics Table</a><a class="headerlink" href="#patient-characteristics-table" title="Permalink to this heading">#</a></h3>
<section id="id1">
<h4><a class="toc-backref" href="#id20" role="doc-backlink">Overall study population</a><a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">tableone</span> <span class="kn">import</span> <span class="n">TableOne</span>

<span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">,</span><span class="s1">&#39;Age group (years)&#39;</span><span class="p">,</span><span class="s1">&#39;Sex&#39;</span><span class="p">,</span><span class="s1">&#39;Race or ethnic group&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Hispanic or Latino ethnic group&#39;</span><span class="p">,</span> <span class="s1">&#39;MRD 1 Status&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Leucocyte counts (10/L)&#39;</span><span class="p">,</span> <span class="s1">&#39;BM Leukemic blasts (%)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Risk Group&#39;</span><span class="p">,</span> <span class="s1">&#39;Clinical Trial&#39;</span><span class="p">,</span><span class="s1">&#39;FLT3 ITD&#39;</span><span class="p">]</span>

<span class="n">discovery_clinical_data</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">discovery_clinical_data</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">float</span><span class="p">)</span>

<span class="n">mytable_cog</span> <span class="o">=</span> <span class="n">TableOne</span><span class="p">(</span><span class="n">discovery_clinical_data</span><span class="p">,</span> <span class="n">columns</span><span class="p">,</span>
                        <span class="n">overall</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">missing</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">pval</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">pval_adjust</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">htest_name</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">dip_test</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">tukey_test</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">normal_test</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>

                        <span class="n">order</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;FLT3 ITD&#39;</span><span class="p">:[</span><span class="s1">&#39;Yes&#39;</span><span class="p">,</span><span class="s1">&#39;No&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Race or ethnic group&#39;</span><span class="p">:[</span><span class="s1">&#39;White&#39;</span><span class="p">,</span><span class="s1">&#39;Black or African American&#39;</span><span class="p">,</span><span class="s1">&#39;Asian&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;MRD 1 Status&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Positive&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Risk Group&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;High Risk&#39;</span><span class="p">,</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;FLT3 ITD&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Yes&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Leucocyte counts (10/L)&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;30&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Age group (years)&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;10&#39;</span><span class="p">]})</span>

<span class="c1"># mytable_cog.to_csv(output_path + &#39;multivariate_cox_lasso/tableone_discovery_cohort.csv&#39;)</span>

<span class="n">mytable_cog</span><span class="o">.</span><span class="n">tabulate</span><span class="p">(</span><span class="n">tablefmt</span><span class="o">=</span><span class="s2">&quot;html&quot;</span><span class="p">,</span> 
                        <span class="n">headers</span><span class="o">=</span><span class="p">[</span><span class="n">score_name</span><span class="p">,</span><span class="s2">&quot;&quot;</span><span class="p">,</span><span class="s1">&#39;Missing&#39;</span><span class="p">,</span><span class="s1">&#39;Discovery Cohort&#39;</span><span class="p">,</span><span class="s1">&#39;Low&#39;</span><span class="p">,</span><span class="s1">&#39;p-value&#39;</span><span class="p">,</span><span class="s1">&#39;Statistical Test&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<thead>
<tr><th>MethylScoreAML_Px                     </th><th>                                         </th><th>Missing  </th><th>Discovery Cohort  </th></tr>
</thead>
<tbody>
<tr><td>n                                     </td><td>                                         </td><td>         </td><td>924               </td></tr>
<tr><td>Age (years), mean (SD)                </td><td>                                         </td><td>0        </td><td>9.4 (6.3)         </td></tr>
<tr><td>Age group (years), n (%)              </td><td>10                                      </td><td>0        </td><td>452 (48.9)        </td></tr>
<tr><td>                                      </td><td>&lt;10                                      </td><td>         </td><td>472 (51.1)        </td></tr>
<tr><td>Sex, n (%)                            </td><td>Female                                   </td><td>0        </td><td>457 (49.5)        </td></tr>
<tr><td>                                      </td><td>Male                                     </td><td>         </td><td>467 (50.5)        </td></tr>
<tr><td>Race or ethnic group, n (%)           </td><td>White                                    </td><td>63       </td><td>680 (79.0)        </td></tr>
<tr><td>                                      </td><td>Black or African American                </td><td>         </td><td>100 (11.6)        </td></tr>
<tr><td>                                      </td><td>Asian                                    </td><td>         </td><td>42 (4.9)          </td></tr>
<tr><td>                                      </td><td>American Indian or Alaska Native         </td><td>         </td><td>5 (0.6)           </td></tr>
<tr><td>                                      </td><td>Native Hawaiian or other Pacific Islander</td><td>         </td><td>6 (0.7)           </td></tr>
<tr><td>                                      </td><td>Other                                    </td><td>         </td><td>28 (3.3)          </td></tr>
<tr><td>Hispanic or Latino ethnic group, n (%)</td><td>Hispanic or Latino                       </td><td>28       </td><td>179 (20.0)        </td></tr>
<tr><td>                                      </td><td>Not Hispanic or Latino                   </td><td>         </td><td>717 (80.0)        </td></tr>
<tr><td>MRD 1 Status, n (%)                   </td><td>Positive                                 </td><td>118      </td><td>251 (31.1)        </td></tr>
<tr><td>                                      </td><td>Negative                                 </td><td>         </td><td>555 (68.9)        </td></tr>
<tr><td>Leucocyte counts (10/L), n (%)       </td><td>30                                      </td><td>0        </td><td>453 (49.0)        </td></tr>
<tr><td>                                      </td><td>&lt;30                                      </td><td>         </td><td>471 (51.0)        </td></tr>
<tr><td>BM Leukemic blasts (%), mean (SD)     </td><td>                                         </td><td>65       </td><td>64.0 (24.4)       </td></tr>
<tr><td>Risk Group, n (%)                     </td><td>High Risk                                </td><td>14       </td><td>126 (13.8)        </td></tr>
<tr><td>                                      </td><td>Standard Risk                            </td><td>         </td><td>443 (48.7)        </td></tr>
<tr><td>                                      </td><td>Low Risk                                 </td><td>         </td><td>341 (37.5)        </td></tr>
<tr><td>Clinical Trial, n (%)                 </td><td>AAML03P1                                 </td><td>0        </td><td>36 (3.9)          </td></tr>
<tr><td>                                      </td><td>AAML0531                                 </td><td>         </td><td>491 (53.1)        </td></tr>
<tr><td>                                      </td><td>AAML1031                                 </td><td>         </td><td>397 (43.0)        </td></tr>
<tr><td>FLT3 ITD, n (%)                       </td><td>Yes                                      </td><td>2        </td><td>162 (17.6)        </td></tr>
<tr><td>                                      </td><td>No                                       </td><td>         </td><td>760 (82.4)        </td></tr>
</tbody>
</table></div></div>
</div>
</section>
<section id="by-methylscore-category">
<h4><a class="toc-backref" href="#id21" role="doc-backlink">By MethylScore category</a><a class="headerlink" href="#by-methylscore-category" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">tableone</span> <span class="kn">import</span> <span class="n">TableOne</span>

<span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">,</span><span class="s1">&#39;Age group (years)&#39;</span><span class="p">,</span><span class="s1">&#39;Sex&#39;</span><span class="p">,</span><span class="s1">&#39;Race or ethnic group&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Hispanic or Latino ethnic group&#39;</span><span class="p">,</span> <span class="s1">&#39;MRD 1 Status&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Leucocyte counts (10/L)&#39;</span><span class="p">,</span> <span class="s1">&#39;BM Leukemic blasts (%)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Risk Group&#39;</span><span class="p">,</span> <span class="s1">&#39;Clinical Trial&#39;</span><span class="p">,</span><span class="s1">&#39;FLT3 ITD&#39;</span><span class="p">]</span>

<span class="n">discovery_clinical_data</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">discovery_clinical_data</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">float</span><span class="p">)</span>

<span class="n">mytable_cog</span> <span class="o">=</span> <span class="n">TableOne</span><span class="p">(</span><span class="n">discovery_clinical_data</span><span class="p">,</span> <span class="n">columns</span><span class="p">,</span>
                        <span class="n">overall</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">missing</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">pval</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">pval_adjust</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">htest_name</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span><span class="n">dip_test</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">tukey_test</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">normal_test</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>

                        <span class="n">order</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;FLT3 ITD&#39;</span><span class="p">:[</span><span class="s1">&#39;Yes&#39;</span><span class="p">,</span><span class="s1">&#39;No&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Race or ethnic group&#39;</span><span class="p">:[</span><span class="s1">&#39;White&#39;</span><span class="p">,</span><span class="s1">&#39;Black or African American&#39;</span><span class="p">,</span><span class="s1">&#39;Asian&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;MRD 1 Status&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Positive&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Risk Group&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;High Risk&#39;</span><span class="p">,</span> <span class="s1">&#39;Standard Risk&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;FLT3 ITD&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;Yes&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Leucocyte counts (10/L)&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;30&#39;</span><span class="p">],</span>
                                <span class="s1">&#39;Age group (years)&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;10&#39;</span><span class="p">]},</span>
                        <span class="n">groupby</span><span class="o">=</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">)</span>

<span class="n">mytable_cog</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;multivariate_cox_lasso/tableone_discovery_methylscoreaml_px.csv&#39;</span><span class="p">)</span>
<span class="n">mytable_cog</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;data/tableone_discovery_methylscoreaml_px.xlsx&#39;</span><span class="p">)</span>

<span class="n">mytable_cog</span><span class="o">.</span><span class="n">tabulate</span><span class="p">(</span><span class="n">tablefmt</span><span class="o">=</span><span class="s2">&quot;html&quot;</span><span class="p">,</span> 
                        <span class="n">headers</span><span class="o">=</span><span class="p">[</span><span class="n">score_name</span><span class="p">,</span><span class="s2">&quot;&quot;</span><span class="p">,</span><span class="s1">&#39;Missing&#39;</span><span class="p">,</span><span class="s1">&#39;High&#39;</span><span class="p">,</span><span class="s1">&#39;Low&#39;</span><span class="p">,</span><span class="s1">&#39;p-value&#39;</span><span class="p">,</span><span class="s1">&#39;Statistical Test&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<thead>
<tr><th>MethylScoreAML_Px                     </th><th>                                         </th><th>Missing  </th><th>High       </th><th>Low        </th><th>p-value  </th><th>Statistical Test                         </th></tr>
</thead>
<tbody>
<tr><td>n                                     </td><td>                                         </td><td>         </td><td>563        </td><td>361        </td><td>         </td><td>                                         </td></tr>
<tr><td>Age (years), mean (SD)                </td><td>                                         </td><td>0        </td><td>10.1 (6.1) </td><td>8.4 (6.5)  </td><td>&lt;0.001   </td><td>Two Sample T-test                        </td></tr>
<tr><td>Age group (years), n (%)              </td><td>10                                      </td><td>0        </td><td>297 (52.8) </td><td>155 (42.9) </td><td>0.004    </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>&lt;10                                      </td><td>         </td><td>266 (47.2) </td><td>206 (57.1) </td><td>         </td><td>                                         </td></tr>
<tr><td>Sex, n (%)                            </td><td>Female                                   </td><td>0        </td><td>277 (49.2) </td><td>180 (49.9) </td><td>0.898    </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>Male                                     </td><td>         </td><td>286 (50.8) </td><td>181 (50.1) </td><td>         </td><td>                                         </td></tr>
<tr><td>Race or ethnic group, n (%)           </td><td>White                                    </td><td>63       </td><td>416 (79.2) </td><td>264 (78.6) </td><td>0.106    </td><td>Chi-squared (warning: expected count &lt; 5)</td></tr>
<tr><td>                                      </td><td>Black or African American                </td><td>         </td><td>52 (9.9)   </td><td>48 (14.3)  </td><td>         </td><td>                                         </td></tr>
<tr><td>                                      </td><td>Asian                                    </td><td>         </td><td>26 (5.0)   </td><td>16 (4.8)   </td><td>         </td><td>                                         </td></tr>
<tr><td>                                      </td><td>American Indian or Alaska Native         </td><td>         </td><td>4 (0.8)    </td><td>1 (0.3)    </td><td>         </td><td>                                         </td></tr>
<tr><td>                                      </td><td>Native Hawaiian or other Pacific Islander</td><td>         </td><td>5 (1.0)    </td><td>1 (0.3)    </td><td>         </td><td>                                         </td></tr>
<tr><td>                                      </td><td>Other                                    </td><td>         </td><td>22 (4.2)   </td><td>6 (1.8)    </td><td>         </td><td>                                         </td></tr>
<tr><td>Hispanic or Latino ethnic group, n (%)</td><td>Hispanic or Latino                       </td><td>28       </td><td>114 (20.9) </td><td>65 (18.6)  </td><td>0.449    </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>Not Hispanic or Latino                   </td><td>         </td><td>432 (79.1) </td><td>285 (81.4) </td><td>         </td><td>                                         </td></tr>
<tr><td>MRD 1 Status, n (%)                   </td><td>Positive                                 </td><td>118      </td><td>128 (26.1) </td><td>123 (38.9) </td><td>&lt;0.001   </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>Negative                                 </td><td>         </td><td>362 (73.9) </td><td>193 (61.1) </td><td>         </td><td>                                         </td></tr>
<tr><td>Leucocyte counts (10/L), n (%)       </td><td>30                                      </td><td>0        </td><td>292 (51.9) </td><td>161 (44.6) </td><td>0.037    </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>&lt;30                                      </td><td>         </td><td>271 (48.1) </td><td>200 (55.4) </td><td>         </td><td>                                         </td></tr>
<tr><td>BM Leukemic blasts (%), mean (SD)     </td><td>                                         </td><td>65       </td><td>62.0 (24.2)</td><td>67.1 (24.6)</td><td>0.003    </td><td>Two Sample T-test                        </td></tr>
<tr><td>Risk Group, n (%)                     </td><td>High Risk                                </td><td>14       </td><td>76 (13.7)  </td><td>50 (14.1)  </td><td>&lt;0.001   </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>Standard Risk                            </td><td>         </td><td>169 (30.4) </td><td>274 (77.4) </td><td>         </td><td>                                         </td></tr>
<tr><td>                                      </td><td>Low Risk                                 </td><td>         </td><td>311 (55.9) </td><td>30 (8.5)   </td><td>         </td><td>                                         </td></tr>
<tr><td>Clinical Trial, n (%)                 </td><td>AAML03P1                                 </td><td>0        </td><td>26 (4.6)   </td><td>10 (2.8)   </td><td>0.117    </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>AAML0531                                 </td><td>         </td><td>286 (50.8) </td><td>205 (56.8) </td><td>         </td><td>                                         </td></tr>
<tr><td>                                      </td><td>AAML1031                                 </td><td>         </td><td>251 (44.6) </td><td>146 (40.4) </td><td>         </td><td>                                         </td></tr>
<tr><td>FLT3 ITD, n (%)                       </td><td>Yes                                      </td><td>2        </td><td>114 (20.3) </td><td>48 (13.3)  </td><td>0.009    </td><td>Chi-squared                              </td></tr>
<tr><td>                                      </td><td>No                                       </td><td>         </td><td>448 (79.7) </td><td>312 (86.7) </td><td>         </td><td>                                         </td></tr>
</tbody>
</table></div></div>
</div>
</section>
</section>
</section>
<section id="fetch-gene-annotation">
<h2><a class="toc-backref" href="#id22" role="doc-backlink">Fetch gene annotation</a><a class="headerlink" href="#fetch-gene-annotation" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load manifest</span>
<span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../Data/UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span><span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">mean_coefs_genes</span> <span class="o">=</span> <span class="n">zhou2016_probes</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="n">ewas_top_cpgs</span><span class="o">.</span><span class="n">columns</span><span class="p">)]</span>\
                                    <span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">mean_coefs</span><span class="o">.</span><span class="n">to_frame</span><span class="p">(</span><span class="s1">&#39;MethylScore Coefficients&#39;</span><span class="p">),</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">)</span>\
                                    <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;MethylScore Coefficients&#39;</span><span class="p">,</span> <span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1"># `GeneNames`, `GeneClasses`, and `TranscriptIDs` columns contain multiple genes separated by `;` so we need to keep only the unique ones</span>
<span class="n">mean_coefs_genes</span><span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">mean_coefs_genes</span><span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">))))</span>
<span class="n">mean_coefs_genes</span><span class="p">[</span><span class="s1">&#39;GeneClasses&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">mean_coefs_genes</span><span class="p">[</span><span class="s1">&#39;GeneClasses&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">))))</span>

<span class="c1"># Transform `GeneNames` values separated by `;` into a list</span>
<span class="n">mean_coefs_genes</span><span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">mean_coefs_genes</span><span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">))</span>

<span class="c1"># If there are multiple genes in `GeneNames` or `GeneClasses`, we will add one row for each gene</span>
<span class="n">mean_coefs_genes</span> <span class="o">=</span> <span class="n">mean_coefs_genes</span><span class="o">.</span><span class="n">explode</span><span class="p">(</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">)</span>

<span class="c1"># Save the dataframe to a CSV file</span>
<span class="n">mean_coefs_genes</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;data/MethylScoreAML_Px_167_cpg_genes.xlsx&#39;</span><span class="p">)</span>
<span class="n">mean_coefs_genes</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;multivariate_cox_lasso/methylScore_coxph_167_cpg_signature.csv&#39;</span><span class="p">)</span>


<span class="kn">from</span> <span class="nn">itables</span> <span class="kn">import</span> <span class="n">show</span>

<span class="n">show</span><span class="p">(</span><span class="n">mean_coefs_genes</span><span class="p">,</span> <span class="n">classes</span><span class="o">=</span><span class="s2">&quot;display nowrap cell-border&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><style>.itables table td {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables table th {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables thead input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}

.itables tfoot input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}
</style>
<div class="itables">
<table id="cc8fa5a6-28af-458b-9db0-bdb1227f3214" class="display nowrap cell-border"style="table-layout:auto;width:auto;margin:auto;caption-side:bottom"><thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>CGIPosition</th>
      <th>MethylScore Coefficients</th>
    </tr>
    <tr>
      <th>probeID</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css">
<script type="module">
    // Import jquery and DataTable
    import 'https://code.jquery.com/jquery-3.6.0.min.js';
    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';
    dt($);

    // Define the table data
    const data = [["cg22855993", "chr1", 16071749, 16071751, "FAM131C", "protein_coding", "N_Shore", 0.164903], ["cg17099306", "chr13", 49549508, 49549510, "RCBTB1", "protein_coding", ".", 0.150585], ["cg10089193", "chr16", 1090938, 1090940, "C1QTNF8", "protein_coding", "S_Shore", 0.089058], ["cg22162847", "chr1", 150562302, 150562304, "ADAMTSL4-AS1", "processed_transcript", "S_Shore", 0.086697], ["cg09890699", "chr1", 54356179, 54356181, "SSBP3", "protein_coding", "N_Shore", 0.077382], ["cg08329113", "chr16", 70737238, 70737240, "VAC14", "protein_coding", ".", 0.07355], ["cg18072388", "chr19", 10323084, 10323086, "RAVER1", "protein_coding", "N_Shore", 0.073201], ["cg00920892", "chr8", 37874705, 37874707, "RAB11FIP1", "protein_coding", ".", 0.072263], ["cg19706516", "chr1", 151131392, 151131394, "RP11-68I18.10", "sense_overlapping", "Island", 0.062809], ["cg17669009", "chr19", 51417630, 51417632, "SIGLEC10", "protein_coding", ".", 0.056796], ["cg06142946", "chr11", 17351576, 17351578, "NCR3LG1", "protein_coding", "Island", 0.048525], ["cg07143043", "chr10", 131970793, 131970795, "RP11-45A17.2", "antisense;protein_coding", ".", 0.048089], ["cg07143043", "chr10", 131970793, 131970795, "BNIP3", "antisense;protein_coding", ".", 0.048089], ["cg14978242", "chr5", 80205311, 80205313, "SERINC5", "protein_coding", ".", 0.045868], ["cg09411730", "chr11", 65423474, 65423476, "NEAT1", "lincRNA;misc_RNA", "Island", 0.03028], ["cg09411730", "chr11", 65423474, 65423476, "NEAT1_3", "lincRNA;misc_RNA", "Island", 0.03028], ["cg09411730", "chr11", 65423474, 65423476, "NEAT1_2", "lincRNA;misc_RNA", "Island", 0.03028], ["cg05348324", "chr11", 119797455, 119797457, ".", ".", ".", 0.022253], ["cg13213755", "chr15", 73634112, 73634114, "NPTN", "protein_coding", "S_Shore", -0.002808], ["cg10280339", "chr15", 65519989, 65519991, ".", ".", "S_Shelf", -0.008283], ["cg13702536", "chr12", 122731005, 122731007, "HCAR1", "protein_coding", ".", -0.013434], ["cg23458843", "chr12", 57742661, 57742663, "AGAP2", "protein_coding", "S_Shelf", -0.022953], ["cg23458843", "chr12", 57742661, 57742663, "TSPAN31", "protein_coding", "S_Shelf", -0.022953], ["cg18964582", "chr1", 236393881, 236393883, "EDARADD", "protein_coding", "N_Shore", -0.023012], ["cg03601372", "chr19", 36776123, 36776125, "CTD-2162K18.4", "lincRNA", "Island", -0.025899], ["cg03601372", "chr19", 36776123, 36776125, "CTD-2162K18.3", "lincRNA", "Island", -0.025899], ["cg10817441", "chr6", 32855950, 32855952, "PSMB9", "protein_coding", "S_Shore", -0.026268], ["cg10805039", "chr17", 847000, 847002, "NXN", "protein_coding", ".", -0.029879], ["cg01771213", "chr1", 57270164, 57270166, "DAB1", "protein_coding", ".", -0.032153], ["cg14928764", "chr20", 23083970, 23083972, "CD93", "protein_coding", ".", -0.045346], ["cg21637653", "chr17", 17213452, 17213454, "RP11-45M22.4", "protein_coding", ".", -0.047542], ["cg21637653", "chr17", 17213452, 17213454, "MPRIP", "protein_coding", ".", -0.047542], ["cg21637653", "chr17", 17213452, 17213454, "FLCN", "protein_coding", ".", -0.047542], ["cg26223555", "chr19", 51824187, 51824189, "FPR3", "protein_coding", ".", -0.048636], ["cg17632028", "chr4", 652752, 652754, "PDE6B", "protein_coding", "N_Shore", -0.060051], ["cg06748884", "chr16", 11138900, 11138902, "CLEC16A", "protein_coding", ".", -0.063771], ["cg26846864", "chr16", 3257325, 3257327, "MEFV", "protein_coding", "S_Shelf", -0.069062], ["cg04839706", "chr16", 46886162, 46886164, "GPT2", "protein_coding", "S_Shore", -0.07115], ["cg23875404", "chr13", 36154731, 36154733, ".", ".", "N_Shore", -0.071194], ["cg16424082", "chr12", 50929956, 50929958, "METTL7A", "protein_coding", ".", -0.073138], ["cg06339275", "chr3", 62948996, 62948998, "LINC00698", "lincRNA", ".", -0.087628], ["cg02918093", "chr16", 10624083, 10624085, ".", ".", "N_Shelf", -0.108928], ["cg08900363", "chr7", 597816, 597818, "PRKAR1B", "protein_coding", ".", NaN], ["cg01543603", "chr16", 89403132, 89403134, "ANKRD11", "sense_intronic;protein_coding", ".", NaN], ["cg01543603", "chr16", 89403132, 89403134, "RP1-168P16.2", "sense_intronic;protein_coding", ".", NaN], ["cg00192882", "chr17", 19387806, 19387808, "MFAP4", "protein_coding", ".", NaN], ["cg08404702", "chr16", 2083386, 2083388, "TSC2", "protein_coding", "N_Shore", NaN], ["cg19432033", "chr17", 7613901, 7613903, "FXR2", "protein_coding", "N_Shore", NaN], ["cg19432033", "chr17", 7613901, 7613903, "SHBG", "protein_coding", "N_Shore", NaN], ["cg07660635", "chr16", 85902540, 85902542, "IRF8", "protein_coding", "S_Shelf", NaN], ["cg05480169", "chr10", 133253903, 133253905, ".", ".", ".", NaN], ["cg18886274", "chr7", 85160554, 85160556, "SEMA3D", "protein_coding", ".", NaN], ["cg10009497", "chr7", 1055779, 1055781, "GPR146", "antisense;protein_coding", "N_Shore", NaN], ["cg10009497", "chr7", 1055779, 1055781, "C7orf50", "antisense;protein_coding", "N_Shore", NaN], ["cg10009497", "chr7", 1055779, 1055781, "RP11-449P15.1", "antisense;protein_coding", "N_Shore", NaN], ["cg20703895", "chr1", 157438011, 157438013, ".", ".", ".", NaN], ["cg21750233", "chr11", 6483080, 6483082, "TIMM10B", "protein_coding", "S_Shore", NaN], ["cg20018344", "chr20", 3223491, 3223493, "ITPA", "protein_coding", ".", NaN], ["cg23128949", "chr12", 124501577, 124501579, "NCOR2", "protein_coding", ".", NaN], ["cg11823637", "chr12", 22729886, 22729888, "RP11-114G22.1", "lincRNA", ".", NaN], ["cg25292309", "chr2", 58762291, 58762293, "LINC01122", "lincRNA", ".", NaN], ["cg15787807", "chr11", 64741618, 64741620, "RASGRP2", "protein_coding", "N_Shore", NaN], ["cg13984533", "chr1", 37796839, 37796841, "MANEAL", "protein_coding", "S_Shelf", NaN], ["cg16721321", "chr4", 1208907, 1208909, "CTBP1-AS", "antisense;protein_coding", "N_Shore", NaN], ["cg16721321", "chr4", 1208907, 1208909, "SPON2", "antisense;protein_coding", "N_Shore", NaN], ["cg10098541", "chr3", 123066802, 123066804, "PDIA5", "protein_coding", "N_Shore", NaN], ["cg01614643", "chr6", 30156991, 30156993, "TRIM10", "protein_coding", ".", NaN], ["cg13360395", "chr2", 123907703, 123907705, ".", ".", ".", NaN], ["cg18586780", "chr2", 235044659, 235044661, "SH3BP4", "protein_coding", "S_Shore", NaN], ["cg18151487", "chr10", 102715582, 102715584, "SFXN2", "protein_coding", "S_Shore", NaN], ["cg18151487", "chr10", 102715582, 102715584, "ARL3", "protein_coding", "S_Shore", NaN], ["cg00059652", "chr17", 69609356, 69609358, "LINC01483", "lincRNA", ".", NaN], ["cg26310429", "chr17", 73028309, 73028311, "SLC39A11", "protein_coding", ".", NaN], ["cg06585893", "chr1", 230747805, 230747807, "CAPN9", "antisense;protein_coding", ".", NaN], ["cg06585893", "chr1", 230747805, 230747807, "RP11-99J16__A.2", "antisense;protein_coding", ".", NaN], ["cg04205653", "chr21", 44805974, 44805976, "SUMO3", "protein_coding", "S_Shelf", NaN], ["cg00550080", "chr6", 30157014, 30157016, "TRIM10", "protein_coding", ".", NaN], ["cg16449402", "chr3", 9733040, 9733042, "BRPF1", "protein_coding", "S_Shore", NaN], ["cg12251145", "chr6", 30156966, 30156968, "TRIM10", "protein_coding", ".", NaN], ["cg26361535", "chr8", 143494433, 143494435, "ZC3H3", "protein_coding", ".", NaN], ["cg21785847", "chr2", 216081555, 216081557, "TMEM169", "protein_coding", "N_Shore", NaN], ["cg21785847", "chr2", 216081555, 216081557, "PECR", "protein_coding", "N_Shore", NaN], ["cg15081720", "chr1", 119323994, 119323996, "RP11-418J17.3", "lincRNA", "N_Shelf", NaN], ["cg13857210", "chr9", 136107370, 136107372, ".", ".", ".", NaN], ["cg15022671", "chr2", 203637474, 203637476, "AC125238.2", "processed_pseudogene", ".", NaN], ["cg15119221", "chr17", 19387441, 19387443, "MFAP4", "protein_coding", ".", NaN], ["cg18437839", "chr1", 201476584, 201476586, ".", ".", ".", NaN], ["cg08480739", "chr11", 67997549, 67997551, "UNC93B1", "protein_coding", "Island", NaN], ["cg03769055", "chr6", 30157026, 30157028, "TRIM10", "protein_coding", ".", NaN], ["cg01312394", "chr2", 46882240, 46882242, ".", ".", ".", NaN], ["cg11818438", "chr8", 118904290, 118904292, ".", ".", ".", NaN], ["cg00223135", "chr15", 43507046, 43507048, "TP53BP1", "protein_coding", "N_Shelf", NaN], ["cg08467866", "chr16", 75208215, 75208217, "CTRB2", "protein_coding", ".", NaN], ["cg00206490", "chr16", 89403320, 89403322, "ANKRD11", "sense_intronic;protein_coding", ".", NaN], ["cg00206490", "chr16", 89403320, 89403322, "RP1-168P16.2", "sense_intronic;protein_coding", ".", NaN], ["cg00945409", "chr10", 78977907, 78977909, "ZMIZ1-AS1", "antisense", "S_Shelf", NaN], ["cg21739530", "chr4", 173264916, 173264918, "GALNT7", "protein_coding", ".", NaN], ["cg03596113", "chr1", 3213404, 3213406, "PRDM16", "protein_coding", ".", NaN], ["cg01591830", "chr16", 89599591, 89599593, ".", ".", "S_Shelf", NaN], ["cg10057841", "chr11", 130126348, 130126350, "APLP2", "protein_coding", "S_Shelf", NaN], ["cg01443071", "chr17", 41968231, 41968233, "CNP", "protein_coding", "Island", NaN], ["cg04092699", "chr2", 76748093, 76748095, "LRRTM4", "protein_coding", ".", NaN], ["cg09281220", "chr2", 219595362, 219595364, ".", ".", "N_Shelf", NaN], ["cg21477075", "chr6", 168032755, 168032757, "KIF25", "protein_coding", "N_Shelf", NaN], ["cg16822174", "chr12", 19483011, 19483013, "AEBP2", "protein_coding", ".", NaN], ["cg18581405", "chr19", 3180036, 3180038, "S1PR4", "protein_coding", "S_Shore", NaN], ["cg04407248", "chr16", 31136874, 31136876, "PRSS8", "protein_coding", ".", NaN], ["cg19479744", "chr6", 155701180, 155701182, ".", ".", ".", NaN], ["cg20964021", "chr7", 156641708, 156641710, "RNF32", "lincRNA;protein_coding", "S_Shore", NaN], ["cg20964021", "chr7", 156641708, 156641710, "LINC01006", "lincRNA;protein_coding", "S_Shore", NaN], ["cg25898577", "chr22", 21937868, 21937870, "PPM1F", "antisense;protein_coding", "N_Shore", NaN], ["cg25898577", "chr22", 21937868, 21937870, "LL22NC03-86G7.1", "antisense;protein_coding", "N_Shore", NaN], ["cg27018309", "chr16", 8849264, 8849266, "PMM2", "antisense;protein_coding", ".", NaN], ["cg27018309", "chr16", 8849264, 8849266, "RP11-152P23.2", "antisense;protein_coding", ".", NaN], ["cg27018309", "chr16", 8849264, 8849266, "RP11-77H9.2", "antisense;protein_coding", ".", NaN], ["cg12762600", "chr15", 89647922, 89647924, "KIF7", "protein_coding", "N_Shore", NaN], ["cg27658601", "chr17", 82836204, 82836206, "ZNF750", "protein_coding", "S_Shelf", NaN], ["cg27658601", "chr17", 82836204, 82836206, "TBCD", "protein_coding", "S_Shelf", NaN], ["cg07593405", "chr7", 118900, 118902, ".", ".", "S_Shore", NaN], ["cg19461880", "chr11", 6483135, 6483137, "TIMM10B", "protein_coding", "S_Shore", NaN], ["cg04400145", "chr20", 1893035, 1893037, "SIRPA", "antisense;protein_coding", "N_Shore", NaN], ["cg04400145", "chr20", 1893035, 1893037, "RP5-858L17.1", "antisense;protein_coding", "N_Shore", NaN], ["cg11661187", "chr4", 89835381, 89835383, "SNCA", "antisense;protein_coding", "N_Shore", NaN], ["cg11661187", "chr4", 89835381, 89835383, "SNCA-AS1", "antisense;protein_coding", "N_Shore", NaN], ["cg18019132", "chr11", 65424235, 65424237, "NEAT1", "lincRNA", "S_Shore", NaN], ["cg00539049", "chr2", 149077019, 149077021, "LYPD6B", "protein_coding", ".", NaN], ["cg16429055", "chr17", 17051538, 17051540, "MPRIP", "protein_coding", ".", NaN], ["cg09140274", "chr1", 220896850, 220896852, ".", ".", "S_Shore", NaN], ["cg07253321", "chr6", 146598717, 146598719, "ADGB", "lincRNA;protein_coding", "N_Shore", NaN], ["cg07253321", "chr6", 146598717, 146598719, "RP11-715G15.1", "lincRNA;protein_coding", "N_Shore", NaN], ["cg26130533", "chr4", 1208911, 1208913, "CTBP1-AS", "antisense;protein_coding", "N_Shore", NaN], ["cg26130533", "chr4", 1208911, 1208913, "SPON2", "antisense;protein_coding", "N_Shore", NaN], ["cg11600161", "chr11", 67404113, 67404115, "TBC1D10C", "protein_coding", "S_Shelf", NaN], ["cg11600161", "chr11", 67404113, 67404115, "PPP1CA", "protein_coding", "S_Shelf", NaN], ["cg11840849", "chr10", 93363960, 93363962, "MYOF", "protein_coding", ".", NaN], ["cg18062721", "chr3", 11601952, 11601954, "VGLL4", "protein_coding", ".", NaN], ["cg03445800", "chr3", 481328, 481330, ".", ".", ".", NaN], ["cg15219347", "chr19", 17378501, 17378503, "PLVAP", "protein_coding", ".", NaN], ["cg04516952", "chr19", 50968785, 50968787, "KLK6", "antisense;protein_coding", ".", NaN], ["cg04516952", "chr19", 50968785, 50968787, "CTB-147C22.9", "antisense;protein_coding", ".", NaN], ["cg11958234", "chr1", 117684205, 117684207, ".", ".", ".", NaN], ["cg10187365", "chr21", 33488648, 33488650, "DNAJC28", "protein_coding", "N_Shelf", NaN], ["cg15538767", "chr5", 150490832, 150490834, "NDST1", "protein_coding", ".", NaN], ["cg03272932", "chr16", 87555286, 87555288, ".", ".", ".", NaN], ["cg01140244", "chr10", 132685455, 132685457, "INPP5A", "protein_coding", ".", NaN], ["cg17078656", "chr1", 12475057, 12475059, "VPS13D", "protein_coding", ".", NaN], ["cg24800630", "chr6", 156426519, 156426521, ".", ".", ".", NaN], ["cg17518965", "chr19", 3178956, 3178958, "S1PR4", "protein_coding", "Island", NaN], ["cg04339462", "chr14", 99561858, 99561860, "CCDC85C", "protein_coding", ".", NaN], ["cg25392500", "chr6", 32127129, 32127131, "ATF6B", "protein_coding", ".", NaN], ["cg21955549", "chr15", 65026930, 65026932, "MTFMT", "protein_coding", "N_Shelf", NaN], ["cg20460366", "chr20", 46305392, 46305394, "CDH22", "protein_coding", "N_Shore", NaN], ["cg23716141", "chr17", 77364394, 77364396, "SEPT9", "protein_coding", ".", NaN], ["cg05482050", "chr5", 177982332, 177982334, "RP11-1252I4.2", "lincRNA", "N_Shelf", NaN], ["cg10695490", "chr19", 49156041, 49156043, "HRC", "protein_coding", "N_Shore", NaN], ["cg20378520", "chr12", 125219264, 125219266, "TMEM132B", "protein_coding", ".", NaN], ["cg12984636", "chr9", 104693221, 104693223, "OR13D1", "protein_coding", ".", NaN], ["cg09797463", "chr10", 89235539, 89235541, "LIPA", "protein_coding", ".", NaN], ["cg01052291", "chr6", 132715433, 132715435, "VNN1", "protein_coding", ".", NaN], ["cg13691108", "chr20", 38804603, 38804605, "PPP1R16B", "protein_coding", "N_Shore", NaN], ["cg09744397", "chr15", 101725089, 101725091, "TARSL2", "protein_coding", "S_Shore", NaN], ["cg22401679", "chr1", 94614490, 94614492, ".", ".", ".", NaN], ["cg18530324", "chr18", 48538620, 48538622, "CTIF", "protein_coding", "N_Shore", NaN], ["cg07892374", "chr3", 130849277, 130849279, "ATP2C1", "protein_coding", ".", NaN], ["cg01782668", "chr4", 148026284, 148026286, "ARHGAP10", "protein_coding", ".", NaN], ["cg21094613", "chr11", 79189641, 79189643, "TENM4", "protein_coding", ".", NaN], ["cg27125093", "chr19", 3179829, 3179831, "S1PR4", "protein_coding", "Island", NaN], ["cg13927938", "chr1", 204490455, 204490457, "PIK3C2B", "protein_coding", "N_Shelf", NaN], ["cg15586054", "chr6", 165658541, 165658543, "PDE10A", "protein_coding", "N_Shore", NaN], ["cg22571135", "chr7", 55309356, 55309358, ".", ".", ".", NaN], ["cg21853989", "chr1", 45339053, 45339055, "MUTYH", "protein_coding", "N_Shore", NaN], ["cg21853989", "chr1", 45339053, 45339055, "TOE1", "protein_coding", "N_Shore", NaN], ["cg08742262", "chr7", 158549590, 158549592, "PTPRN2", "protein_coding", "N_Shore", NaN], ["cg16280363", "chr12", 57489107, 57489109, "ARHGAP9", "protein_coding", "S_Shore", NaN], ["cg16280363", "chr12", 57489107, 57489109, "MARS", "protein_coding", "S_Shore", NaN], ["cg09218727", "chr10", 104350553, 104350555, ".", ".", ".", NaN], ["cg25074809", "chr2", 15945893, 15945895, "MYCN", "protein_coding", "S_Shelf", NaN], ["cg06715885", "chr12", 57201608, 57201610, "LRP1", "protein_coding", "N_Shore", NaN], ["cg03040821", "chr1", 212792866, 212792868, "NSL1", "protein_coding", "S_Shore", NaN], ["cg03040821", "chr1", 212792866, 212792868, "TATDN3", "protein_coding", "S_Shore", NaN], ["cg27064178", "chr22", 49784571, 49784573, "BRD1", "protein_coding", "Island", NaN], ["cg05974498", "chr19", 52096002, 52096004, "ZNF432", "protein_coding", "S_Shore", NaN], ["cg05974498", "chr19", 52096002, 52096004, "ZNF841", "protein_coding", "S_Shore", NaN], ["cg26925717", "chr14", 20999156, 20999158, "SLC39A2", "antisense;protein_coding", ".", NaN], ["cg26925717", "chr14", 20999156, 20999158, "RP11-84C10.4", "antisense;protein_coding", ".", NaN], ["cg27396056", "chr3", 188947203, 188947205, "TPRG1", "lincRNA;protein_coding", "N_Shore", NaN], ["cg27396056", "chr3", 188947203, 188947205, "TPRG1-AS1", "lincRNA;protein_coding", "N_Shore", NaN], ["cg04543865", "chr16", 81313945, 81313947, "GAN", "antisense;protein_coding", "N_Shore", NaN], ["cg04543865", "chr16", 81313945, 81313947, "RP11-55K13.1", "antisense;protein_coding", "N_Shore", NaN], ["cg11039729", "chr10", 35379396, 35379398, "CCNY", "protein_coding", ".", NaN], ["cg01476568", "chr10", 5692752, 5692754, "FAM208B", "protein_coding", ".", NaN], ["cg06605704", "chr16", 4501145, 4501147, "HMOX2", "protein_coding", ".", NaN], ["cg20435097", "chr10", 124632254, 124632256, "FAM53B", "protein_coding", ".", NaN], ["cg20435097", "chr10", 124632254, 124632256, "RP11-12J10.3", "protein_coding", ".", NaN], ["cg17082549", "chr3", 128620009, 128620011, "RPN1", "protein_coding", "S_Shore", NaN], ["cg15711288", "chr3", 150783114, 150783116, ".", ".", "S_Shelf", NaN], ["cg03233656", "chr2", 64987490, 64987492, "SLC1A4", "protein_coding", "N_Shore", NaN], ["cg14287724", "chr1", 1170137, 1170139, "RP11-465B22.8", "lincRNA", "N_Shelf", NaN], ["cg08530484", "chr2", 219159883, 219159885, "NHEJ1", "protein_coding", "N_Shore", NaN], ["cg08530484", "chr2", 219159883, 219159885, "RP11-33O4.2", "protein_coding", "N_Shore", NaN], ["cg22760287", "chr11", 107705174, 107705176, ".", ".", ".", NaN], ["cg09941713", "chr8", 60792772, 60792774, "CHD7", "protein_coding", ".", NaN], ["cg06956444", "chr1", 7941248, 7941250, "TNFRSF9", "processed_pseudogene;protein_coding", "N_Shore", NaN], ["cg06956444", "chr1", 7941248, 7941250, "RP5-892F13.2", "processed_pseudogene;protein_coding", "N_Shore", NaN]];

    // Define the dt_args
    let dt_args = {"order": []};
    dt_args["data"] = data;

    $(document).ready(function () {
        
        $('#cc8fa5a6-28af-458b-9db0-bdb1227f3214').DataTable(dt_args);
    });
</script>
</div>
</div></div>
</div>
</section>
<section id="watermark">
<h2><a class="toc-backref" href="#id23" role="doc-backlink">Watermark</a><a class="headerlink" href="#watermark" title="Permalink to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Python implementation: CPython
Python version       : 3.8.16
IPython version      : 8.12.2

pandas         : 2.0.2
seaborn        : 0.12.2
matplotlib     : 3.7.1
tableone       : 0.7.12
sklearn        : 1.2.2
lifelines      : 0.27.7
statannotations: 0.5.0

Compiler    : GCC 11.3.0
OS          : Linux
Release     : 5.15.90.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 20
Architecture: 64bit
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="2_Univariate_CoxPH-EWAS_v3.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Epigenome Wide Association Study</p>
      </div>
    </a>
    <a class="right-next"
       href="3.2_MethylScoreAML_Px_Validation_Peds_v2.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">MethylScoreAML Px Validation</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load Data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#standardize-data">Standardize Data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-multivariate-coxph-lasso">Perform multivariate CoxPH-Lasso</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#set-threshold">Set Threshold</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-table-of-35-cpgs-coefficients-with-ewas-results">Save table of 35 CpGs coefficients with EWAS results</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generate-model-results">Generate Model Results</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#discovery-dataset">Discovery Dataset</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#kaplan-meier-plots">Kaplan-Meier Plots</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-study-population">Overall study population</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#per-risk-group">Per risk group</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#forest-plots">Forest Plots</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#with-mrd-1">With MRD 1</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#without-mrd-1">Without MRD 1</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#roc-auc">ROC AUC</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#box-plots">Box Plots</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#stacked-bar-plots">Stacked Bar Plots</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#patient-characteristics-table">Patient Characteristics Table</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Overall study population</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#by-methylscore-category">By MethylScore category</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-gene-annotation">Fetch gene annotation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Marchi et al., 2023. Lamba Lab @ University of Florida. Please do not share this notebook publicly.
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
       Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>