{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reclassification to Current Clinical Guidelines\n",
    "\n",
    "```{contents}\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Current thinking:\n",
    "\n",
    "14 WHO subtypes of AML\n",
    "+1 otherwise-normal control\n",
    "+1 mds-like and secondary neoplasms\n",
    "\n",
    "Need to figure out how to deal with really rare subtypes (e.g. AML FUS-ERG, KAT6A)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Guidelines"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WHO 2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.itables table td {\n",
       "    text-overflow: ellipsis;\n",
       "    overflow: hidden;\n",
       "}\n",
       "\n",
       ".itables table th {\n",
       "    text-overflow: ellipsis;\n",
       "    overflow: hidden;\n",
       "}\n",
       "\n",
       ".itables thead input {\n",
       "    width: 100%;\n",
       "    padding: 3px;\n",
       "    box-sizing: border-box;\n",
       "}\n",
       "\n",
       ".itables tfoot input {\n",
       "    width: 100%;\n",
       "    padding: 3px;\n",
       "    box-sizing: border-box;\n",
       "}\n",
       "</style>\n",
       "<div class=\"itables\">\n",
       "<table id=\"6bd7bb40-20b1-4b5b-91db-214175f2bf51\" class=\"display nowrap\"style=\"table-layout:auto;width:auto;margin:auto;caption-side:bottom\"><thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      \n",
       "      <th>Subtype</th>\n",
       "      <th>Frequency in paediatric AML</th>\n",
       "      <th>Morphology</th>\n",
       "      <th>Immunophenotype</th>\n",
       "      <th>Common co-occurring genetic aberrations</th>\n",
       "      <th>Prognosis</th>\n",
       "    </tr>\n",
       "  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>\n",
       "<link rel=\"stylesheet\" type=\"text/css\" href=\"https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css\">\n",
       "<script type=\"module\">\n",
       "    // Import jquery and DataTable\n",
       "    import 'https://code.jquery.com/jquery-3.6.0.min.js';\n",
       "    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';\n",
       "    dt($);\n",
       "\n",
       "    // Define the table data\n",
       "    const data = [[\"AML with RUNX1::RUNXT1 fusion\", \"13\\u201319% of cases\", \"Neutrophilic maturation, Auer rods; often increased basophils and/or mast cells\", \"Positive: CD34, HLA-DR, KIT (CD117), CD99, myeloperoxidase\\\\nUsually positive but can be heterogeneous: CD13, CD33 and CD38, CD19, PAX5, cytoplasmic CD79a, CD56\\\\nNegative: CD7\", \"Mutations in KIT, FLT3, CSF3R, EZH2, ASXL1, ASXL2, CBL, WT1, cohesion genes (SMC1A, SMC3, RAD21), NRAS, KRAS\", \"Good\"], [\"AML with CBFB::MYH11 fusion\", \"7\\u201312% of cases\", \"Myelomonocytic; often increased eosinophils at all maturation stages\", \"Often multiple blast populations with signs of differentiation towards both granulocytic and monocytic lineages; may express CD2 and/or NG2 in a small subpopulation of cells\", \"Mutations in KIT, KRAS, NRAS, CBL, FLT3, WT1\", \"Good\"], [\"APL with PML::RARA fusion\", \"5\\u201310% of cases but wide geographical variation (2\\u201350%)\", \"Exclusively FAB M3; may be hypogranular\", \"Positive: CD33, often KIT (CD117), sometimes CD2\\\\nNegative: CD34 (except hypogranular variant), HLA-DR\\\\nOften high autofluorescence\", \"Mutations in FLT3\", \"Good (but treated differently)\"], [\"AML with KMT2A rearrangement\", \"20% of cases\", \"All FAB types except M3 and M6, most commonly M5\", \"Positive: CD33, CD65, CD4, HLA-DR (in monocytic cases), CD15, NG2; platelet antigens (in megakaryocytic cases)\\\\nNegative/low: CD13, CD34, CD14\", \"Mutations in the tyrosine kinase, PI3K, and RAS pathways; FLT3-TKD, NRAS, KRAS\", \"Depends on fusion partner\"], [\"AML with DEK::NUP214 fusion\", \"1\\u20132% of cases\", \"All FAB types except M3 and M7, most commonly M2 and M4; Auer rods; basophilia in blood or bone marrow; erythroid and granulocytic dysplasia are common\", \"Positive: myeloperoxidase, CD13, CD33, CD38, CD123, HLA-DR; often CD34, KIT (CD117), CD15; sometimes CD64, TdT\", \"FLT3 ITD in two thirds of cases; complex karyotype\", \"Poor\"], [\"AML with MECOM rearrangement\", \"< 1% of cases\", \"Commonly FAB M1, M4, and M7; multilineage dysplasia is common, especially in megakaryopoiesis\", \"Positive: CD34, CD33, CD13, KIT (CD117), HLA-DR, CD7\\\\nNegative: lymphoid markers other than CD7\", \"NaN\", \"Poor\"], [\"AML with ETV6 fusion\", \"1% of all cases\", \"All FAB types, most commonly M0 and M1\", \"For MNX1::ETV6:\\\\nPositive: CD34, HLA-DR, often KIT (CD117), T-cell markers CD7 and CD4\", \"Almost always +19, sometimes +8\", \"Poor, but may be protocol dependent\"], [\"AML with KAT6A::CREBBP fusion\", \"0.5% of cases\", \"FAB M5\", \"Positive: HLA-DR, CD15, myeloperoxidase, CD13, CD33, sometimes CD56, CD14\\\\nNegative: CD34, KIT (CD117)\", \"NaN\", \"OS similar to paediatric AML in general\"], [\"AML with RBM15::MRTFA fusion\", \"0.8% of all cases\", \"Exclusively FAB M7\", \"Positive: > 1 platelet antigen, some CD34 and/or HLA-DR\\\\nNegative: CD45\", \"del(1p), +4, +7, +8, +9, +10, +14, +15, +17, +18, +20, +21(21p), complex hyperdiploid karyotype\", \"Lower CR rate but similar EFS/OS vs other AMKLs\"], [\"AML with CBFA2T3::GLIS2 fusion\", \"3% of all cases\", \"All FAB types, most commonly M7\", \"> 50% show the RAM phenotype; overexpression of CD56 and underexpression of HLA-DR and CD38\", \"Mutations in FLT3, GATA1, JAK/STAT pathway\", \"Poor\"], [\"AML with NUP98 rearrangement\", \"5\\u20139% of cases\", \"NUP98::NSD1: often FAB M4 and M5\\\\nNUP98::KDM5A: mostly FAB M7 but also M5 and M6\", \"NaN\", \"For NUP98::NSD1: FLT3 ITD in > 80%, mutation in WT1 in 50%\", \"Poor\"], [\"AML with FUS::ERG fusion\", \"0.3\\u20130.5% of cases\", \"All FAB types except M6; Auer rods\", \"NaN\", \"+8, +10, complex karyotype\", \"Poor\"], [\"AML with NPM1 mutation\", \"6\\u20138% of cases\", \"FAB M1, M2, M4, and M5\", \"Positive: CD33, sometimes KIT (CD117), often CD36, CD64\\\\nNegative: CD34, sometimes HLA-DR\\\\nOften weak CD13\", \"FLT3 ITD, mutations in NRAS, WT1\", \"Good, also when FLT3 ITD is present\"], [\"AML with CEBPA mutation\", \"5\\u20136% of cases\", \"Most commonly FAB M1 and M2\", \"Positive: CD34, HLA-DR, often CD7\\\\nNegative: other lymphoid antigens\", \"Mutations in CEBPA-TAD, CSF3R, GATA2, RAS genes, FLT3\", \"Good, except for CSF3R-mutated cases\"]];\n",
       "\n",
       "    // Define the dt_args\n",
       "    let dt_args = {\"order\": []};\n",
       "    dt_args[\"data\"] = data;\n",
       "\n",
       "    $(document).ready(function () {\n",
       "        \n",
       "        $('#6bd7bb40-20b1-4b5b-91db-214175f2bf51').DataTable(dt_args);\n",
       "    });\n",
       "</script>\n",
       "</div>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import itables\n",
    "\n",
    "from itables import show\n",
    "\n",
    "# Read the CSV file\n",
    "df = pd.read_csv('../data/who2022_aml_classification.csv')\n",
    "\n",
    "# Display the dataframe using itables\n",
    "itables.show(df)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`````{admonition} Source\n",
    ":class: tip\n",
    "Hasle H, Meshinchi S, Fogelstrand L, Alaggio R, et al. Acute myeloid leukaemias (AMLs) with defining genetic abnormalities. In: WHO Classification of Tumours Editorial Board. Paediatric tumours [Internet]. Lyon (France): International Agency for Research on Cancer; 2022 [cited 2024 Jan 1]. (WHO classification of tumours series, 5th ed.; vol. 7). Available from: https://tumourclassification.iarc.who.int/chapters/44.\n",
    "`````"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load and process clinical data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The clinical data has been indexed and cleaned.\n",
      "Exclusion of samples may be applied depending on the analysis.\n"
     ]
    }
   ],
   "source": [
    "# Import functions to clean up clinical data\n",
    "import sys\n",
    "sys.path.insert(0, '..')\n",
    "from source.clinical_data_cleanup_functions import *\n",
    "\n",
    "# Call functions to merge, index and clean clinical data files\n",
    "labels_0531         = clean_cog       (merge_index_0531())\n",
    "labels_1031         = clean_cog       (merge_index_1031())\n",
    "labels_aml05        = clean_aml05     (merge_index_aml05())\n",
    "labels_beataml      = clean_beataml   (merge_index_beataml())\n",
    "labels_amltcga      = clean_amltcga   (merge_index_amltcga())\n",
    "labels_nordic_all   = clean_nordic_all(merge_index_nordic_all())\n",
    "labels_mds_taml     = clean_mds_taml  (merge_index_mds_taml())\n",
    "labels_all_graal    = clean_all_graal (merge_index_all_graal())\n",
    "labels_target_all   = clean_target_all(merge_index_target_all())\n",
    "\n",
    "# Combine all clinical data labels into one dataframe\n",
    "labels_combined = pd.concat([labels_aml05, labels_beataml,\n",
    "                        labels_0531, labels_amltcga, labels_1031,\n",
    "                        labels_nordic_all, labels_mds_taml,\n",
    "                        labels_all_graal,labels_target_all],\n",
    "                        axis=0, join='outer')\n",
    "\n",
    "# Redefine output path (for troubleshooting purposes in case only this cell is run)\n",
    "output_path = '../../Data/Intermediate_Files/'\n",
    "\n",
    "# Read df\n",
    "df = pd.read_pickle(output_path + '3330samples-333351cpgs-withbatchcorrection-bvalues.pkl')\n",
    "\n",
    "# Remove samples that are not in the methyl dataset\n",
    "df_labels = labels_combined.loc[labels_combined.index.isin(df.index)].sort_index()\n",
    "\n",
    "# Add age categorization and main disease classification to the clinical data\n",
    "# df_labels = process_df_labels(df_labels)\n",
    "\n",
    "# Save the clinical data labels\n",
    "# df_labels.to_csv(output_path + 'discovery_clinical_data.csv')\n",
    "\n",
    "print('The clinical data has been indexed and cleaned.\\n\\\n",
    "Exclusion of samples may be applied depending on the analysis.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "# save all label dataframes into excel files using `pandas` 'to_excel' function\n",
    "# labels_aml05.to_excel(output_path + 'aml05_clinical_data.xlsx')\n",
    "# labels_beataml.to_excel(output_path + 'beataml_clinical_data.xlsx')\n",
    "labels_0531.to_excel(output_path + '0531_clinical_data.xlsx')\n",
    "# labels_amltcga.to_excel(output_path + 'amltcga_clinical_data.xlsx')\n",
    "labels_1031.to_excel(output_path + '1031_clinical_data.xlsx')\n",
    "# labels_nordic_all.to_excel(output_path + 'nordic_all_clinical_data.xlsx')\n",
    "# labels_mds_taml.to_excel(output_path + 'mds_taml_clinical_data.xlsx')\n",
    "# labels_all_graal.to_excel(output_path + 'all_graal_clinical_data.xlsx')\n",
    "# labels_target_all.to_excel(output_path + 'target_all_clinical_data.xlsx')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>IlmnID</th>\n",
       "      <th>Batch</th>\n",
       "      <th>cg00000109</th>\n",
       "      <th>cg00000236</th>\n",
       "      <th>cg00000292</th>\n",
       "      <th>cg00000363</th>\n",
       "      <th>cg00000622</th>\n",
       "      <th>cg00000658</th>\n",
       "      <th>cg00000714</th>\n",
       "      <th>cg00000721</th>\n",
       "      <th>cg00000734</th>\n",
       "      <th>...</th>\n",
       "      <th>ch.9.83519450F</th>\n",
       "      <th>ch.9.837340R</th>\n",
       "      <th>ch.9.84051654F</th>\n",
       "      <th>ch.9.84078312F</th>\n",
       "      <th>ch.9.86947500F</th>\n",
       "      <th>ch.9.88862796F</th>\n",
       "      <th>ch.9.90287778F</th>\n",
       "      <th>ch.9.90621653R</th>\n",
       "      <th>ch.9.93402636R</th>\n",
       "      <th>ch.9.98463211R</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>202897270043_R01C01</th>\n",
       "      <td>GSE190931</td>\n",
       "      <td>0.934</td>\n",
       "      <td>0.891</td>\n",
       "      <td>0.879</td>\n",
       "      <td>0.363</td>\n",
       "      <td>0.013</td>\n",
       "      <td>0.905</td>\n",
       "      <td>0.364</td>\n",
       "      <td>0.957</td>\n",
       "      <td>0.079</td>\n",
       "      <td>...</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.039</td>\n",
       "      <td>0.061</td>\n",
       "      <td>0.039</td>\n",
       "      <td>0.045</td>\n",
       "      <td>0.040</td>\n",
       "      <td>0.050</td>\n",
       "      <td>0.033</td>\n",
       "      <td>0.029</td>\n",
       "      <td>0.032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202897270043_R03C01</th>\n",
       "      <td>GSE190931</td>\n",
       "      <td>0.974</td>\n",
       "      <td>0.880</td>\n",
       "      <td>0.884</td>\n",
       "      <td>0.512</td>\n",
       "      <td>0.014</td>\n",
       "      <td>0.869</td>\n",
       "      <td>0.389</td>\n",
       "      <td>0.967</td>\n",
       "      <td>0.071</td>\n",
       "      <td>...</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.057</td>\n",
       "      <td>0.021</td>\n",
       "      <td>0.047</td>\n",
       "      <td>0.041</td>\n",
       "      <td>0.033</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.026</td>\n",
       "      <td>0.024</td>\n",
       "      <td>0.026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202897270043_R04C01</th>\n",
       "      <td>GSE190931</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.936</td>\n",
       "      <td>0.753</td>\n",
       "      <td>0.639</td>\n",
       "      <td>0.016</td>\n",
       "      <td>0.815</td>\n",
       "      <td>0.439</td>\n",
       "      <td>0.921</td>\n",
       "      <td>0.077</td>\n",
       "      <td>...</td>\n",
       "      <td>0.032</td>\n",
       "      <td>0.057</td>\n",
       "      <td>0.029</td>\n",
       "      <td>0.097</td>\n",
       "      <td>0.041</td>\n",
       "      <td>0.058</td>\n",
       "      <td>0.040</td>\n",
       "      <td>0.036</td>\n",
       "      <td>0.032</td>\n",
       "      <td>0.033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202897270043_R05C01</th>\n",
       "      <td>GSE190931</td>\n",
       "      <td>0.861</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.337</td>\n",
       "      <td>0.134</td>\n",
       "      <td>0.017</td>\n",
       "      <td>0.872</td>\n",
       "      <td>0.107</td>\n",
       "      <td>0.758</td>\n",
       "      <td>0.098</td>\n",
       "      <td>...</td>\n",
       "      <td>0.050</td>\n",
       "      <td>0.327</td>\n",
       "      <td>0.034</td>\n",
       "      <td>0.135</td>\n",
       "      <td>0.048</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.048</td>\n",
       "      <td>0.035</td>\n",
       "      <td>0.039</td>\n",
       "      <td>0.066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202897270043_R06C01</th>\n",
       "      <td>GSE190931</td>\n",
       "      <td>0.924</td>\n",
       "      <td>0.778</td>\n",
       "      <td>0.710</td>\n",
       "      <td>0.167</td>\n",
       "      <td>0.014</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.247</td>\n",
       "      <td>0.958</td>\n",
       "      <td>0.081</td>\n",
       "      <td>...</td>\n",
       "      <td>0.026</td>\n",
       "      <td>0.064</td>\n",
       "      <td>0.051</td>\n",
       "      <td>0.030</td>\n",
       "      <td>0.041</td>\n",
       "      <td>0.051</td>\n",
       "      <td>0.030</td>\n",
       "      <td>0.030</td>\n",
       "      <td>0.028</td>\n",
       "      <td>0.037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203724130020_R02C01</th>\n",
       "      <td>GSE147667</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.848</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.269</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.848</td>\n",
       "      <td>0.423</td>\n",
       "      <td>0.896</td>\n",
       "      <td>0.099</td>\n",
       "      <td>...</td>\n",
       "      <td>0.041</td>\n",
       "      <td>0.062</td>\n",
       "      <td>0.046</td>\n",
       "      <td>0.037</td>\n",
       "      <td>0.058</td>\n",
       "      <td>0.051</td>\n",
       "      <td>0.063</td>\n",
       "      <td>0.044</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203724130020_R03C01</th>\n",
       "      <td>GSE147667</td>\n",
       "      <td>0.899</td>\n",
       "      <td>0.844</td>\n",
       "      <td>0.862</td>\n",
       "      <td>0.308</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.854</td>\n",
       "      <td>0.353</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.096</td>\n",
       "      <td>...</td>\n",
       "      <td>0.055</td>\n",
       "      <td>0.041</td>\n",
       "      <td>0.040</td>\n",
       "      <td>0.043</td>\n",
       "      <td>0.060</td>\n",
       "      <td>0.060</td>\n",
       "      <td>0.060</td>\n",
       "      <td>0.040</td>\n",
       "      <td>0.030</td>\n",
       "      <td>0.055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203724130020_R04C01</th>\n",
       "      <td>GSE147667</td>\n",
       "      <td>0.905</td>\n",
       "      <td>0.840</td>\n",
       "      <td>0.854</td>\n",
       "      <td>0.296</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.864</td>\n",
       "      <td>0.362</td>\n",
       "      <td>0.911</td>\n",
       "      <td>0.099</td>\n",
       "      <td>...</td>\n",
       "      <td>0.050</td>\n",
       "      <td>0.046</td>\n",
       "      <td>0.045</td>\n",
       "      <td>0.060</td>\n",
       "      <td>0.062</td>\n",
       "      <td>0.044</td>\n",
       "      <td>0.056</td>\n",
       "      <td>0.044</td>\n",
       "      <td>0.030</td>\n",
       "      <td>0.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203724130020_R05C01</th>\n",
       "      <td>GSE147667</td>\n",
       "      <td>0.887</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.881</td>\n",
       "      <td>0.281</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.861</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.884</td>\n",
       "      <td>0.098</td>\n",
       "      <td>...</td>\n",
       "      <td>0.066</td>\n",
       "      <td>0.066</td>\n",
       "      <td>0.056</td>\n",
       "      <td>0.052</td>\n",
       "      <td>0.055</td>\n",
       "      <td>0.044</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.064</td>\n",
       "      <td>0.041</td>\n",
       "      <td>0.055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203724130020_R06C01</th>\n",
       "      <td>GSE147667</td>\n",
       "      <td>0.918</td>\n",
       "      <td>0.846</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.269</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.866</td>\n",
       "      <td>0.392</td>\n",
       "      <td>0.884</td>\n",
       "      <td>0.115</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034</td>\n",
       "      <td>0.069</td>\n",
       "      <td>0.056</td>\n",
       "      <td>0.043</td>\n",
       "      <td>0.055</td>\n",
       "      <td>0.062</td>\n",
       "      <td>0.054</td>\n",
       "      <td>0.064</td>\n",
       "      <td>0.043</td>\n",
       "      <td>0.048</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3330 rows × 333352 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "IlmnID                   Batch  cg00000109  cg00000236  cg00000292  \\\n",
       "202897270043_R01C01  GSE190931       0.934       0.891       0.879   \n",
       "202897270043_R03C01  GSE190931       0.974       0.880       0.884   \n",
       "202897270043_R04C01  GSE190931       0.868       0.936       0.753   \n",
       "202897270043_R05C01  GSE190931       0.861       0.650       0.337   \n",
       "202897270043_R06C01  GSE190931       0.924       0.778       0.710   \n",
       "...                        ...         ...         ...         ...   \n",
       "203724130020_R02C01  GSE147667       0.913       0.848       0.844   \n",
       "203724130020_R03C01  GSE147667       0.899       0.844       0.862   \n",
       "203724130020_R04C01  GSE147667       0.905       0.840       0.854   \n",
       "203724130020_R05C01  GSE147667       0.887       0.846       0.881   \n",
       "203724130020_R06C01  GSE147667       0.918       0.846       0.868   \n",
       "\n",
       "IlmnID               cg00000363  cg00000622  cg00000658  cg00000714  \\\n",
       "202897270043_R01C01       0.363       0.013       0.905       0.364   \n",
       "202897270043_R03C01       0.512       0.014       0.869       0.389   \n",
       "202897270043_R04C01       0.639       0.016       0.815       0.439   \n",
       "202897270043_R05C01       0.134       0.017       0.872       0.107   \n",
       "202897270043_R06C01       0.167       0.014       0.868       0.247   \n",
       "...                         ...         ...         ...         ...   \n",
       "203724130020_R02C01       0.269       0.025       0.848       0.423   \n",
       "203724130020_R03C01       0.308       0.025       0.854       0.353   \n",
       "203724130020_R04C01       0.296       0.022       0.864       0.362   \n",
       "203724130020_R05C01       0.281       0.023       0.861       0.387   \n",
       "203724130020_R06C01       0.269       0.025       0.866       0.392   \n",
       "\n",
       "IlmnID               cg00000721  cg00000734  ...  ch.9.83519450F  \\\n",
       "202897270043_R01C01       0.957       0.079  ...           0.023   \n",
       "202897270043_R03C01       0.967       0.071  ...           0.022   \n",
       "202897270043_R04C01       0.921       0.077  ...           0.032   \n",
       "202897270043_R05C01       0.758       0.098  ...           0.050   \n",
       "202897270043_R06C01       0.958       0.081  ...           0.026   \n",
       "...                         ...         ...  ...             ...   \n",
       "203724130020_R02C01       0.896       0.099  ...           0.041   \n",
       "203724130020_R03C01       0.876       0.096  ...           0.055   \n",
       "203724130020_R04C01       0.911       0.099  ...           0.050   \n",
       "203724130020_R05C01       0.884       0.098  ...           0.066   \n",
       "203724130020_R06C01       0.884       0.115  ...           0.034   \n",
       "\n",
       "IlmnID               ch.9.837340R  ch.9.84051654F  ch.9.84078312F  \\\n",
       "202897270043_R01C01         0.039           0.061           0.039   \n",
       "202897270043_R03C01         0.057           0.021           0.047   \n",
       "202897270043_R04C01         0.057           0.029           0.097   \n",
       "202897270043_R05C01         0.327           0.034           0.135   \n",
       "202897270043_R06C01         0.064           0.051           0.030   \n",
       "...                           ...             ...             ...   \n",
       "203724130020_R02C01         0.062           0.046           0.037   \n",
       "203724130020_R03C01         0.041           0.040           0.043   \n",
       "203724130020_R04C01         0.046           0.045           0.060   \n",
       "203724130020_R05C01         0.066           0.056           0.052   \n",
       "203724130020_R06C01         0.069           0.056           0.043   \n",
       "\n",
       "IlmnID               ch.9.86947500F  ch.9.88862796F  ch.9.90287778F  \\\n",
       "202897270043_R01C01           0.045           0.040           0.050   \n",
       "202897270043_R03C01           0.041           0.033           0.022   \n",
       "202897270043_R04C01           0.041           0.058           0.040   \n",
       "202897270043_R05C01           0.048           0.049           0.048   \n",
       "202897270043_R06C01           0.041           0.051           0.030   \n",
       "...                             ...             ...             ...   \n",
       "203724130020_R02C01           0.058           0.051           0.063   \n",
       "203724130020_R03C01           0.060           0.060           0.060   \n",
       "203724130020_R04C01           0.062           0.044           0.056   \n",
       "203724130020_R05C01           0.055           0.044           0.049   \n",
       "203724130020_R06C01           0.055           0.062           0.054   \n",
       "\n",
       "IlmnID               ch.9.90621653R  ch.9.93402636R  ch.9.98463211R  \n",
       "202897270043_R01C01           0.033           0.029           0.032  \n",
       "202897270043_R03C01           0.026           0.024           0.026  \n",
       "202897270043_R04C01           0.036           0.032           0.033  \n",
       "202897270043_R05C01           0.035           0.039           0.066  \n",
       "202897270043_R06C01           0.030           0.028           0.037  \n",
       "...                             ...             ...             ...  \n",
       "203724130020_R02C01           0.044           0.022           0.081  \n",
       "203724130020_R03C01           0.040           0.030           0.055  \n",
       "203724130020_R04C01           0.044           0.030           0.050  \n",
       "203724130020_R05C01           0.064           0.041           0.055  \n",
       "203724130020_R06C01           0.064           0.043           0.048  \n",
       "\n",
       "[3330 rows x 333352 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sex</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Age (years)</th>\n",
       "      <th>Tissue Type</th>\n",
       "      <th>WBC Count (10⁹/L)</th>\n",
       "      <th>BM Leukemic blasts (%)</th>\n",
       "      <th>FLT3/ITD allelic ratio</th>\n",
       "      <th>Other genetic alterations</th>\n",
       "      <th>PRDM16 expression</th>\n",
       "      <th>Non-CR</th>\n",
       "      <th>...</th>\n",
       "      <th>Protocol</th>\n",
       "      <th>WBC at Diagnosis</th>\n",
       "      <th>Initial Therapy</th>\n",
       "      <th>ALAL Presentation</th>\n",
       "      <th>WHO ALAL Classification</th>\n",
       "      <th>WHO Dx</th>\n",
       "      <th>WHO RL1</th>\n",
       "      <th>WHO RL2</th>\n",
       "      <th>St. Jude Genome ID</th>\n",
       "      <th>Comments</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid</th>\n",
       "      <td>Male</td>\n",
       "      <td>PANVPB</td>\n",
       "      <td>14.624658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.1</td>\n",
       "      <td>87.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0031efba-f564-4fff-bd7b-2b97f37218c1_noid</th>\n",
       "      <td>Female</td>\n",
       "      <td>PASDKZ</td>\n",
       "      <td>5.136986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>53.5</td>\n",
       "      <td>73.2</td>\n",
       "      <td>0.61</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid</th>\n",
       "      <td>Female</td>\n",
       "      <td>PASDKZ</td>\n",
       "      <td>5.136986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>53.5</td>\n",
       "      <td>73.2</td>\n",
       "      <td>0.61</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>004c953f-d999-4d82-898d-c091e692df3c_noid</th>\n",
       "      <td>Female</td>\n",
       "      <td>PASAUT</td>\n",
       "      <td>1.542466</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.8</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0066d4af-8019-46a0-ba29-d2962c9537a7_noid</th>\n",
       "      <td>Male</td>\n",
       "      <td>PASYWA</td>\n",
       "      <td>14.194521</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>0.93</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid</th>\n",
       "      <td>Male</td>\n",
       "      <td>PASPLU</td>\n",
       "      <td>9.312329</td>\n",
       "      <td>NaN</td>\n",
       "      <td>446.0</td>\n",
       "      <td>92.6</td>\n",
       "      <td>0.52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fcfa89d8-f11e-4264-be00-e112c4c619f5_noid</th>\n",
       "      <td>Male</td>\n",
       "      <td>PASCRZ</td>\n",
       "      <td>14.673973</td>\n",
       "      <td>NaN</td>\n",
       "      <td>203.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid</th>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAPHGD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fd4e4be4-3306-4037-b4e9-7886283243ed_noid</th>\n",
       "      <td>Male</td>\n",
       "      <td>PASTZK</td>\n",
       "      <td>5.876712</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.9</td>\n",
       "      <td>55.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ffeac099-79b4-4fbf-a283-275d31baf531_noid</th>\n",
       "      <td>Male</td>\n",
       "      <td>PARFXF</td>\n",
       "      <td>21.054795</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.1</td>\n",
       "      <td>20.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1453 rows × 2278 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Sex        Patient_ID  \\\n",
       "Sample_ID                                                             \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid    Male            PANVPB   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid  Female            PASDKZ   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid  Female            PASDKZ   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid  Female            PASAUT   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid    Male            PASYWA   \n",
       "...                                           ...               ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid    Male            PASPLU   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid    Male            PASCRZ   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid     NaN  TARGET-10-PAPHGD   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid    Male            PASTZK   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid    Male            PARFXF   \n",
       "\n",
       "                                          Age (years) Tissue Type  \\\n",
       "Sample_ID                                                           \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid   14.624658         NaN   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid    5.136986         NaN   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid    5.136986         NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid    1.542466         NaN   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid   14.194521         NaN   \n",
       "...                                               ...         ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid    9.312329         NaN   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid   14.673973         NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid         NaN         NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid    5.876712         NaN   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid   21.054795         NaN   \n",
       "\n",
       "                                           WBC Count (10⁹/L)  \\\n",
       "Sample_ID                                                      \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid               34.1   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid               53.5   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid               53.5   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid               10.8   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid               19.1   \n",
       "...                                                      ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid              446.0   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid              203.3   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid                NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid               17.9   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid                4.1   \n",
       "\n",
       "                                           BM Leukemic blasts (%)  \\\n",
       "Sample_ID                                                           \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid                    87.3   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid                    73.2   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid                    73.2   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid                    34.0   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid                    95.0   \n",
       "...                                                           ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid                    92.6   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid                     NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid                     NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid                    55.0   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid                    20.0   \n",
       "\n",
       "                                           FLT3/ITD allelic ratio  \\\n",
       "Sample_ID                                                           \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid                     NaN   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid                    0.61   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid                    0.61   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid                     NaN   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid                    0.93   \n",
       "...                                                           ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid                    0.52   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid                     NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid                     NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid                     NaN   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid                     NaN   \n",
       "\n",
       "                                          Other genetic alterations  \\\n",
       "Sample_ID                                                             \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid                       NaN   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid                       NaN   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid                       NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid                       NaN   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid                       NaN   \n",
       "...                                                             ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid                       NaN   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid                       NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid                       NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid                       NaN   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid                       NaN   \n",
       "\n",
       "                                          PRDM16 expression Non-CR  ...  \\\n",
       "Sample_ID                                                           ...   \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid               NaN    NaN  ...   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid               NaN    NaN  ...   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid               NaN    NaN  ...   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid               NaN    NaN  ...   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid               NaN    NaN  ...   \n",
       "...                                                     ...    ...  ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid               NaN    NaN  ...   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid               NaN    NaN  ...   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid               NaN    NaN  ...   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid               NaN    NaN  ...   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid               NaN    NaN  ...   \n",
       "\n",
       "                                          Protocol WBC at Diagnosis  \\\n",
       "Sample_ID                                                             \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid      NaN              NaN   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid      NaN              NaN   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid      NaN              NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid      NaN              NaN   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid      NaN              NaN   \n",
       "...                                            ...              ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid      NaN              NaN   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid      NaN              NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid      NaN              NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid      NaN              NaN   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid      NaN              NaN   \n",
       "\n",
       "                                          Initial Therapy  ALAL Presentation  \\\n",
       "Sample_ID                                                                      \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid             NaN                NaN   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid             NaN                NaN   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid             NaN                NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid             NaN                NaN   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid             NaN                NaN   \n",
       "...                                                   ...                ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid             NaN                NaN   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid             NaN                NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid             NaN                NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid             NaN                NaN   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid             NaN                NaN   \n",
       "\n",
       "                                          WHO ALAL Classification WHO Dx  \\\n",
       "Sample_ID                                                                  \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid                     NaN    NaN   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid                     NaN    NaN   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid                     NaN    NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid                     NaN    NaN   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid                     NaN    NaN   \n",
       "...                                                           ...    ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid                     NaN    NaN   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid                     NaN    NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid                     NaN    NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid                     NaN    NaN   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid                     NaN    NaN   \n",
       "\n",
       "                                          WHO RL1 WHO RL2 St. Jude Genome ID  \\\n",
       "Sample_ID                                                                      \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid     NaN     NaN                NaN   \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid     NaN     NaN                NaN   \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid     NaN     NaN                NaN   \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid     NaN     NaN                NaN   \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid     NaN     NaN                NaN   \n",
       "...                                           ...     ...                ...   \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid     NaN     NaN                NaN   \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid     NaN     NaN                NaN   \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid     NaN     NaN                NaN   \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid     NaN     NaN                NaN   \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid     NaN     NaN                NaN   \n",
       "\n",
       "                                          Comments  \n",
       "Sample_ID                                           \n",
       "0003fe29-7b8f-4ef1-b9bc-40306205f1fd_noid      NaN  \n",
       "0031efba-f564-4fff-bd7b-2b97f37218c1_noid      NaN  \n",
       "0037ec75-bb9e-4dbb-a2d9-de1f9bfd2362_noid      NaN  \n",
       "004c953f-d999-4d82-898d-c091e692df3c_noid      NaN  \n",
       "0066d4af-8019-46a0-ba29-d2962c9537a7_noid      NaN  \n",
       "...                                            ...  \n",
       "fcdaabff-e76c-48dc-9acd-dc36beb0f4ac_noid      NaN  \n",
       "fcfa89d8-f11e-4264-be00-e112c4c619f5_noid      NaN  \n",
       "fd0d7fba-01de-4035-b3fa-41a84a4c562d_noid      NaN  \n",
       "fd4e4be4-3306-4037-b4e9-7886283243ed_noid      NaN  \n",
       "ffeac099-79b4-4fbf-a283-275d31baf531_noid      NaN  \n",
       "\n",
       "[1453 rows x 2278 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check if `Patient_ID` column from `df_labels` contains PAWDWA\n",
    "df_labels[df_labels['Patient_ID'].str.contains('P')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Classification Strategy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "    def first_classify_controls(normal_samples):\n",
    "        mapping = {\n",
    "            'Bone Marrow Normal': 'Otherwise-Normal Control',\n",
    "            'Blood Derived Normal': 'Otherwise-Normal Control'}\n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in normal_samples:\n",
    "                return value\n",
    "\n",
    "    def second_classify_annotated_diagnosis(diagnosis):\n",
    "        mapping = {\n",
    "            'mutated NPM1': 'AML with mutated NPM1',\n",
    "            'mutated CEBPA': 'AML with bZIP mutated CEBPA',\n",
    "            'myelodysplasia-related changes': 'MDS-related or secondary myeloid neoplasms'\n",
    "            }\n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in diagnosis:\n",
    "                return value\n",
    "\n",
    "    def third_classify_fusions(fusion_partner):\n",
    "        mapping = {\n",
    "        'RUNX1-RUNX1T1': 'AML with t(8;21)(q22;q22.1)/RUNX1::RUNX1T1',\n",
    "        'CBFB-MYH11':    'AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22)/CBFB::MYH11',\n",
    "        'KMT2A':         'AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement',\n",
    "        'add(11)(q23)':  'AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement',\n",
    "        'MLL':           'AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement',\n",
    "        'PML-RARA':      'APL with t(15;17)(q24.1;q21.2)/PML::RARA',\n",
    "        'DEK-NUP214':    'AML with t(6;9)(p23;q34.1)/DEK::NUP214',\n",
    "        'MECOM':         'AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement',\n",
    "        'ETV6':          'AML with ETV6 fusion',\n",
    "        'NPM1':          'AML with mutated NPM1',\n",
    "        'RBM15-MKL1':    'AML with t(1;22)(p13.3;q13.1); RBM15::MKL1',\n",
    "        'NUP98':         'AML with NUP98-fusion',\n",
    "        'KAT6A-CREBBP':  'AML with t(8;16)(p11.2;p13.3); KAT6A::CREBBP',\n",
    "        'FUS-ERG':       'AML with t(16;21)(p11;q22); FUS::ERG',\n",
    "        'CBFA2T3-GLIS2': 'AML with CBFA2T3::GLIS2 (inv(16)(p13q24))',\n",
    "        'BCR-ABL1':       'AML with t(9;22)(q34.1;q11.2)/BCR::ABL1',        \n",
    "        # 'RUNX1-CBFA2T3': 'AML with other rare recurring translocations',\n",
    "        # 'PRDM16-RPN1':   'AML with other rare recurring translocations',\n",
    "        # 'PICALM-MLLT10': 'AML with other rare recurring translocations',\n",
    "        # 'RBM15-MRTFA':   'AML with other rare recurring translocations',\n",
    "        'CBFA2T3-GLIS3': 'AML with CBFA2T3::GLIS2 (inv(16)(p13q24))',\n",
    "        'PSIP1-NUP214':  'AML with t(6;9)(p23;q34.1)/DEK::NUP214',\n",
    "        # 'XPO1-TNRC18':   'AML with other rare recurring translocations', \n",
    "        # 'HNRNPH1-ERG':   'AML with other rare recurring translocations',\n",
    "        # 'NIPBL-HOXB9':   'AML with other rare recurring translocations', \n",
    "        'SET-NUP214':    'AML with t(6;9)(p23;q34.1)/DEK::NUP214', \n",
    "        # 'FLI1-IFIT2':    'AML with other rare recurring translocations', \n",
    "        # 'TCF4-ZEB2':     'AML with other rare recurring translocations',\n",
    "        # 'MBTD1-ZMYND11': 'AML with other rare recurring translocations', \n",
    "        # 'FOSB-KLF6':     'AML with other rare recurring translocations', \n",
    "        # 'SFPQ-ZFP36L2':  'AML with other rare recurring translocations', \n",
    "        # 'RUNX1-LINC00478':'AML with other rare recurring translocations',\n",
    "        # 'RUNX1-EVX1':     'AML with other rare recurring translocations',  \n",
    "        # 'PSPC1-ZFP36L1':  'AML with other rare recurring translocations', \n",
    "        # 'EWSR1-FEV':      'AML with other rare recurring translocations',\n",
    "        # 'STAG2-AFF2':     'AML with other rare recurring translocations', \n",
    "        # 'MYB-GATA1':      'AML with other rare recurring translocations', \n",
    "        # 'RUNX1-ZFPM2':    'AML with other rare recurring translocations', \n",
    "        # 'RUNX1-CBFA2T2':  'AML with other rare recurring translocations',\n",
    "        # 'PIM3-BRD1':      'AML with other rare recurring translocations',\n",
    "        # 'KAT6A-EP300':    'AML with other rare recurring translocations',\n",
    "        # 'DOT1L-RPS15':    'AML with other rare recurring translocations',\n",
    "        # 'FUS-FEV':        'AML with other rare recurring translocations',\n",
    "        # 'KAT6A-NCOA2':    'AML with other rare recurring translocations',\n",
    "        # 'JARID2-PTP4A1':  'AML with other rare recurring translocations',\n",
    "        # 'FUS-FLI1':       'AML with other rare recurring translocations'\n",
    "        }    \n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in fusion_partner:\n",
    "                return value\n",
    "\n",
    "    # def fourth_classify_karyotype(structural_variation):  \n",
    "    #     mapping = {\n",
    "    #         't(8;21)(q22;q22.1)': 'AML with t(8;21)(q22;q22.1)/RUNX1::RUNX1T1',\n",
    "    #         'inv(16)(p13.1q22) or t(16;16)(p13.1;q22)': 'AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22)/CBFB::MYH11',\n",
    "    #         't(9;11)(p22;q23.3)': 'AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement',\n",
    "    #         't(15;17)(q24.1;q21.2)': 'APL with t(15;17)(q24.1;q21.2)/PML::RARA',\n",
    "    #         't(6;9)(p23;q34.1)': 'AML with t(6;9)(p23;q34.1)/DEK::NUP214',\n",
    "    #         'inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)': 'AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement',\n",
    "    #         't(1;22)(p13.3;q13.1)': 'AML with t(1;22)(p13.3;q13.1); RBM15::MKL1',\n",
    "    #         't(8;16)(p11.2;p13.3)': 'AML with t(8;16)(p11.2;p13.3); KAT6A::CREBBP',\n",
    "    #         't(16;21)(p11;q22)': 'AML with t(16;21)(p11;q22); FUS::ERG',\n",
    "    #         'inv(16)(p13q24)': 'AML with CBFA2T3::GLIS2 (inv(16)(p13q24))',\n",
    "    #         't(9;22)(q34.1;q11.2)': 'AML with t(9;22)(q34.1;q11.2)/BCR::ABL1'\n",
    "    #     }\n",
    "\n",
    "    #     for key, value in mapping.items():\n",
    "    #         if key in structural_variation:\n",
    "    #             return value      \n",
    "\n",
    "def process_labels_who22(df):\n",
    "    # Apply functions and concatenate results in one step\n",
    "    df['WHO 2022 Combined Diagnoses'] = df.apply(lambda x: ','.join(filter(None, [\n",
    "        classify_controls(str(x['Sample Type'])),\n",
    "        classify_fusion(str(x['Gene Fusion'])),\n",
    "        classify_cebpa(str(x['CEBPA mutation'])),\n",
    "        classify_npm(str(x['NPM mutation'])),\n",
    "        classify_annotated_diagnosis(str(x['Comment']))\n",
    "    ])), axis=1)\n",
    "\n",
    "    # Replace empty strings with NaN\n",
    "    df['WHO 2022 Combined Diagnoses'] = df['WHO 2022 Combined Diagnoses'].replace('', np.nan)\n",
    "\n",
    "    # Create `WHO AML 2022 Diagnosis` by extracting the first non-empty, non-NaN element\n",
    "    df['WHO AML 2022 Diagnosis'] = df['WHO 2022 Combined Diagnoses'].str.split(',').str[0]\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "    def classify_controls(normal_samples):\n",
    "        mapping = {\n",
    "            'Bone Marrow Normal': 'Otherwise-Normal Control',\n",
    "            'Blood Derived Normal': 'Otherwise-Normal Control'}\n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in normal_samples:\n",
    "                return value\n",
    "\n",
    "    def classify_fusion(gene_fusion):\n",
    "        mapping = {\n",
    "        'RUNX1-RUNX1T1': 'AML with t(8;21)(q22;q22.1)/RUNX1::RUNX1T1',\n",
    "        'CBFB-MYH11':    'AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22)/CBFB::MYH11',\n",
    "        'KMT2A':         'AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement',\n",
    "        'add(11)(q23)':  'AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement',\n",
    "        'MLL':           'AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement',\n",
    "        'PML-RARA':      'APL with t(15;17)(q24.1;q21.2)/PML::RARA',\n",
    "        'DEK-NUP214':    'AML with t(6;9)(p23;q34.1)/DEK::NUP214',\n",
    "        'MECOM':         'AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement',\n",
    "        'ETV6':          'AML with ETV6 fusion',\n",
    "        'NPM1':          'AML with mutated NPM1',\n",
    "        'RBM15-MKL1':    'AML with t(1;22)(p13.3;q13.1); RBM15::MKL1',\n",
    "        'NUP98':         'AML with NUP98-fusion',\n",
    "        'KAT6A-CREBBP':  'AML with t(8;16)(p11.2;p13.3); KAT6A::CREBBP',\n",
    "        'FUS-ERG':       'AML with t(16;21)(p11;q22); FUS::ERG',\n",
    "        'CBFA2T3-GLIS2': 'AML with CBFA2T3::GLIS2 (inv(16)(p13q24))',\n",
    "        'BCR-ABL1':       'AML with t(9;22)(q34.1;q11.2)/BCR::ABL1',\n",
    "\n",
    "        # Other uncharacterized abdnormalities present in the dataset but not in guidelines\n",
    "        \n",
    "        # 'RUNX1-CBFA2T3': 'AML with other rare recurring translocations',\n",
    "        # 'PRDM16-RPN1':   'AML with other rare recurring translocations',\n",
    "        # 'PICALM-MLLT10': 'AML with other rare recurring translocations',\n",
    "        # 'RBM15-MRTFA':   'AML with other rare recurring translocations',\n",
    "        # 'CBFA2T3-GLIS3': 'AML with CBFA2T3::GLIS2 (inv(16)(p13q24))',\n",
    "        # 'PSIP1-NUP214':  'AML with t(6;9)(p23;q34.1)/DEK::NUP214',\n",
    "        # 'XPO1-TNRC18':   'AML with other rare recurring translocations', \n",
    "        # 'HNRNPH1-ERG':   'AML with other rare recurring translocations',\n",
    "        # 'NIPBL-HOXB9':   'AML with other rare recurring translocations', \n",
    "        # 'SET-NUP214':    'AML with t(6;9)(p23;q34.1)/DEK::NUP214', \n",
    "        # 'FLI1-IFIT2':    'AML with other rare recurring translocations', \n",
    "        # 'TCF4-ZEB2':     'AML with other rare recurring translocations',\n",
    "        # 'MBTD1-ZMYND11': 'AML with other rare recurring translocations', \n",
    "        # 'FOSB-KLF6':     'AML with other rare recurring translocations', \n",
    "        # 'SFPQ-ZFP36L2':  'AML with other rare recurring translocations', \n",
    "        # 'RUNX1-LINC00478':'AML with other rare recurring translocations',\n",
    "        # 'RUNX1-EVX1':     'AML with other rare recurring translocations',  \n",
    "        # 'PSPC1-ZFP36L1':  'AML with other rare recurring translocations', \n",
    "        # 'EWSR1-FEV':      'AML with other rare recurring translocations',\n",
    "        # 'STAG2-AFF2':     'AML with other rare recurring translocations', \n",
    "        # 'MYB-GATA1':      'AML with other rare recurring translocations', \n",
    "        # 'RUNX1-ZFPM2':    'AML with other rare recurring translocations', \n",
    "        # 'RUNX1-CBFA2T2':  'AML with other rare recurring translocations',\n",
    "        # 'PIM3-BRD1':      'AML with other rare recurring translocations',\n",
    "        # 'KAT6A-EP300':    'AML with other rare recurring translocations',\n",
    "        # 'DOT1L-RPS15':    'AML with other rare recurring translocations',\n",
    "        # 'FUS-FEV':        'AML with other rare recurring translocations',\n",
    "        # 'KAT6A-NCOA2':    'AML with other rare recurring translocations',\n",
    "        # 'JARID2-PTP4A1':  'AML with other rare recurring translocations',\n",
    "        # 'FUS-FLI1':       'AML with other rare recurring translocations'\n",
    "        }    \n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in gene_fusion:\n",
    "                return value\n",
    "\n",
    "    def classify_cebpa(cebpa_mutation):\n",
    "        mapping = {\n",
    "            'Yes': 'AML with bZIP mutated CEBPA'}\n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in cebpa_mutation:\n",
    "                return value\n",
    "\n",
    "    def classify_npm(npm_mutation):\n",
    "        mapping = {\n",
    "            'Yes': 'AML with mutated NPM1',\n",
    "        }\n",
    "\n",
    "        for key, value in mapping.items():\n",
    "            if key in npm_mutation:\n",
    "                return value\n",
    "            \n",
    "    def classify_annotated_diagnosis(diagnosis):\n",
    "        mapping = {\n",
    "            'mutated NPM1': 'AML with mutated NPM1',\n",
    "            'mutated CEBPA': 'AML with bZIP mutated CEBPA',\n",
    "            # 'myelodysplasia-related changes': 'MDS-related or secondary myeloid neoplasms'\n",
    "            }\n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in diagnosis:\n",
    "                return value\n",
    "\n",
    "    def classify_karyotype(structural_variation):\n",
    "        mapping = {\n",
    "            't(8;16)': 'AML with t(8;16)(p11.2;p13.3); KAT6A::CREBBP',\n",
    "            't(16;21)': 'AML with t(16;21)(p11;q22); FUS::ERG',\n",
    "            't(6;9)': 'AML with t(6;9)(p23;q34.1)/DEK::NUP214',\n",
    "            't(1;22)': 'AML with t(1;22)(p13.3;q13.1); RBM15::MKL1',\n",
    "            'inv(3)': 'AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement',\n",
    "            't(3;3)': 'AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement',}\n",
    "        \n",
    "        for key, value in mapping.items():\n",
    "            if key in structural_variation:\n",
    "                return value\n",
    "\n",
    "    def process_labels_who22(df):\n",
    "        df['WHO 2022_Controls'] = df['Sample Type'].astype(str).apply(classify_controls)\n",
    "        df['WHO 2022_Gene Fusion'] = df['Gene Fusion'].astype(str).apply(classify_fusion)\n",
    "        df['WHO 2022_CEBPA'] = df['CEBPA mutation'].astype(str).apply(classify_cebpa)\n",
    "        df['WHO 2022_NPM1'] = df['NPM mutation'].astype(str).apply(classify_npm)\n",
    "        df['WHO 2022_Comment'] = df['Comment'].astype(str).apply(classify_annotated_diagnosis)\n",
    "        df['WHO 2022_Karyotype'] = df['Karyotype'].astype(str).apply(classify_karyotype)\n",
    "\n",
    "        df['WHO 2022 Combined Diagnoses'] = df[['WHO 2022_Controls','WHO 2022_Gene Fusion',\n",
    "        'WHO 2022_CEBPA', 'WHO 2022_NPM1', 'WHO 2022_Comment', 'WHO 2022_Karyotype']]\\\n",
    "            .apply(lambda x: ','.join(filter(lambda i: i is not None and i==i, x)), axis=1)\n",
    "\n",
    "        # Replace empty strings with NaN\n",
    "        df['WHO 2022 Combined Diagnoses'] = df['WHO 2022 Combined Diagnoses'].replace('', np.nan)\n",
    "\n",
    "        # Create `WHO 2022 Final Diagnosis` column by splitting `Combined Diagnosis` by comma and taking the first element\n",
    "        df['WHO AML 2022 Diagnosis'] = df['WHO 2022 Combined Diagnoses'].str.split(',').str[0]\n",
    "\n",
    "        # Drop columns created except for `WHO 2022 Final Diagnosis` and `Combined Diagnosis` columns\n",
    "        df = df.drop(['WHO 2022_Controls','WHO 2022_Gene Fusion',\n",
    "         'WHO 2022_CEBPA', 'WHO 2022_NPM1', 'WHO 2022_Comment', 'WHO 2022_Karyotype'], axis=1)\n",
    "            \n",
    "        return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Execution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels_cog = pd.concat([labels_0531, labels_1031],\n",
    "                        axis=0, join='outer')\n",
    "\n",
    "labels_cog_who = process_labels_who22(labels_cog)\n",
    "\n",
    "labels_cog_who.to_excel(output_path + 'cog_clinical_data.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "WHO AML 2022 Diagnosis\n",
       "AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement                           416\n",
       "AML with t(8;21)(q22;q22.1)/RUNX1::RUNX1T1                                245\n",
       "AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22)/CBFB::MYH11             199\n",
       "Otherwise-Normal Control                                                  181\n",
       "AML with mutated NPM1                                                     142\n",
       "AML with NUP98-fusion                                                     123\n",
       "AML with bZIP mutated CEBPA                                                85\n",
       "AML with CBFA2T3::GLIS2 (inv(16)(p13q24))                                  39\n",
       "AML with t(6;9)(p23;q34.1)/DEK::NUP214                                     33\n",
       "AML with ETV6 fusion                                                       25\n",
       "AML with t(1;22)(p13.3;q13.1); RBM15::MKL1                                 15\n",
       "AML with t(16;21)(p11;q22); FUS::ERG                                       12\n",
       "AML with t(8;16)(p11.2;p13.3); KAT6A::CREBBP                               10\n",
       "AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement     10\n",
       "APL with t(15;17)(q24.1;q21.2)/PML::RARA                                    1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels_cog_who['WHO AML 2022 Diagnosis'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "WHO AML 2022 Diagnosis\n",
       "AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement                           416\n",
       "AML with t(8;21)(q22;q22.1)/RUNX1::RUNX1T1                                245\n",
       "AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22)/CBFB::MYH11             199\n",
       "Otherwise-Normal Control                                                  181\n",
       "AML with mutated NPM1                                                     142\n",
       "AML with NUP98-fusion                                                     123\n",
       "AML with bZIP mutated CEBPA                                                85\n",
       "AML with CBFA2T3::GLIS2 (inv(16)(p13q24))                                  40\n",
       "AML with t(6;9)(p23;q34.1)/DEK::NUP214                                     35\n",
       "AML with ETV6 fusion                                                       25\n",
       "AML with t(1;22)(p13.3;q13.1); RBM15::MKL1                                 13\n",
       "AML with t(8;16)(p11.2;p13.3); KAT6A::CREBBP                                9\n",
       "AML with t(16;21)(p11;q22); FUS::ERG                                        7\n",
       "AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement      6\n",
       "APL with t(15;17)(q24.1;q21.2)/PML::RARA                                    1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels_cog_who['WHO AML 2022 Diagnosis'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Later"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_df_labels(df):\n",
    "    \"\"\"\n",
    "    Function to process a pandas dataframe, performing age categorization \n",
    "    and main disease classification.\n",
    "\n",
    "    Args:\n",
    "        df (pandas.DataFrame): Input dataframe.\n",
    "\n",
    "    Returns:\n",
    "        pandas.DataFrame: Processed dataframe with age categorized and main disease classified.\n",
    "    \"\"\"\n",
    "    import numpy as np\n",
    "    import pandas as pd\n",
    "\n",
    "    def categorize_age(age):\n",
    "        \"\"\"\n",
    "        Function to categorize age into a specific range.\n",
    "\n",
    "        Args:\n",
    "            age (int or float): The age to be categorized.\n",
    "\n",
    "        Returns:\n",
    "            str: A string representation of the age category.\n",
    "        \"\"\"\n",
    "        if pd.isnull(age):\n",
    "            return np.nan\n",
    "        elif age < 5:\n",
    "            return '0-5'\n",
    "        elif age < 13:\n",
    "            return '5-13'\n",
    "        elif age < 39:\n",
    "            return '13-39'\n",
    "        elif age < 60:\n",
    "            return '39-60'\n",
    "        else:\n",
    "            return '60+'\n",
    "\n",
    "    def classify_main_disease(subtype):\n",
    "        \"\"\"\n",
    "        Function to classify the main disease based on a given subtype.\n",
    "\n",
    "        Args:\n",
    "            subtype (str): The subtype of the disease.\n",
    "\n",
    "        Returns:\n",
    "            str: A string representation of the main disease.\n",
    "        \"\"\"\n",
    "        mapping = {\n",
    "            'AML': 'Acute myeloid leukemia (AML)',\n",
    "            'ALL': 'Acute lymphoblastic leukemia (ALL)',\n",
    "            'MDS': 'Myelodysplastic syndrome (MDS or MDS-like)',\n",
    "            'Mixed phenotype acute leukemia': 'Mixed phenotype acute leukemia (MPAL)',\n",
    "            'APL': 'Acute promyelocytic leukemia (APL)',\n",
    "            'Otherwise-Normal Control': 'Otherwise-Normal (Control)'\n",
    "        }\n",
    "\n",
    "        for key, value in mapping.items():\n",
    "            if key in subtype:\n",
    "                return value\n",
    "\n",
    "    def main_disease_class(df):\n",
    "        \"\"\"\n",
    "        Function to classify the main disease and create a pathology class.\n",
    "\n",
    "        Args:\n",
    "            df (pandas.DataFrame): The dataframe to be processed.\n",
    "\n",
    "        Returns:\n",
    "            pandas.DataFrame: The processed dataframe with new columns for main disease and pathology class.\n",
    "        \"\"\"\n",
    "\n",
    "        df['WHO_ALL'] = df['WHO ALL 2022 Diagnosis'].astype(str).apply(classify_main_disease)\n",
    "        df['ELN_AML'] = df['ELN AML 2022 Diagnosis'].astype(str).apply(classify_main_disease)\n",
    "\n",
    "        df['Hematopoietic Group'] = df[['ELN_AML', 'WHO_ALL']] \\\n",
    "            .apply(lambda x: ','.join(filter(lambda i: i is not None and i == i, x)), axis=1) \\\n",
    "            .replace('', np.nan)\n",
    "\n",
    "        # Drop columns created except for `WHO Final Diagnosis` and `Combined Diagnosis` columns\n",
    "        df = df.drop(['ELN_AML', 'WHO_ALL'], axis=1)\n",
    "\n",
    "        return df\n",
    "\n",
    "    # Convert 'Age (years)' to numeric, errors='coerce' will turn non-numeric data to NaN\n",
    "    df['Age (years)'] = pd.to_numeric(df['Age (years)'], errors='coerce')\n",
    "\n",
    "    # Then apply your function\n",
    "    df['Age (group years)'] = df['Age (years)'].apply(categorize_age)\n",
    "    \n",
    "    # Process labels\n",
    "    df = main_disease_class(df)\n",
    "    \n",
    "    # Create `WHO 2022 Diagnosis` column\n",
    "    df['WHO 2022 Diagnosis'] = df[['WHO AML 2022 Diagnosis', 'WHO ALL 2022 Diagnosis']] \\\n",
    "            .apply(lambda x: ','.join(filter(lambda i: i is not None and i == i, x)), axis=1) \\\n",
    "            .replace('', np.nan)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WHO AML 2022 Diagnosis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "WHO AML 2022 Diagnosis\n",
       "AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement                           313\n",
       "MDS-related or secondary myeloid neoplasms                                230\n",
       "AML with mutated NPM1                                                     179\n",
       "AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22)/CBFB::MYH11             178\n",
       "AML with t(8;21)(q22;q22.1)/RUNX1::RUNX1T1                                169\n",
       "Otherwise-Normal Control                                                  162\n",
       "AML with NUP98-fusion                                                      97\n",
       "AML with bZIP mutated CEBPA                                                69\n",
       "APL with t(15;17)(q24.1;q21.2)/PML::RARA                                   31\n",
       "AML with CBFA2T3::GLIS2 (inv(16)(p13q24))                                  30\n",
       "AML with t(6;9)(p23;q34.1)/DEK::NUP214                                     28\n",
       "AML with ETV6 fusion                                                       16\n",
       "AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement     10\n",
       "AML with t(1;22)(p13.3;q13.1); RBM15::MKL1                                  9\n",
       "AML with t(8;16)(p11.2;p13.3); KAT6A::CREBBP                                7\n",
       "AML with t(16;21)(p11;q22); FUS::ERG                                        4\n",
       "Myeloid leukemia associated with Down syndrome                              1\n",
       "Mixed phenotype acute leukemia T/myeloid                                    1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_labels['WHO AML 2022 Diagnosis'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ELN AML 2022 Diagnosis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ELN AML 2022 Diagnosis\n",
       "AML with t(9;11)(p22;q23.3)/KMT2A-rearrangement                           313\n",
       "MDS-related or secondary myeloid neoplasms                                228\n",
       "AML with other rare recurring translocations                              185\n",
       "AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22)/CBFB::MYH11             178\n",
       "AML with mutated NPM1                                                     172\n",
       "AML with t(8;21)(q22;q22.1)/RUNX1::RUNX1T1                                169\n",
       "Otherwise-Normal Control                                                  162\n",
       "AML with in-frame bZIP mutated CEBPA                                       69\n",
       "APL with t(15;17)(q24.1;q21.2)/PML::RARA                                   31\n",
       "AML with t(6;9)(p23;q34.1)/DEK::NUP214                                     28\n",
       "AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2)/MECOM-rearrangement     10\n",
       "AML with t(9;22)(q34.1;q11.2)/BCR::ABL1                                     3\n",
       "Myeloid leukaemia associated with Down syndrome                             1\n",
       "Mixed phenotype acute leukemia T/myeloid                                    1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_labels['ELN AML 2022 Diagnosis'].value_counts()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Evaluate final sample size by batch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Batch\n",
       "GSE49031          933\n",
       "GSE190931         581\n",
       "GSE124413         495\n",
       "GSE159907         316\n",
       "GDC_TARGET-AML    287\n",
       "GDC_TCGA-AML      194\n",
       "GSE152710         166\n",
       "GSE147667         153\n",
       "GDC_TARGET-ALL    141\n",
       "GSE133986          64\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Batch'].value_counts(dropna=False)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Tags",
  "kernelspec": {
   "display_name": ".venv_sklearn",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "vscode": {
   "interpreter": {
    "hash": "bea0b97f2aeab51a41f683aa60895cc0e821b0a2b6d8fed7be73bd28506d07e0"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
