
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Clinical Data &#8212; Acute Leukemia Methylome Atlas Study</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css?v=564be945" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'source/4_clinical_data';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="37-CpG Signature" href="5_signature.html" />
    <link rel="prev" title="Test Dataset" href="3_test.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="0_overview.html">
  
  
  
  
  
  
    <p class="title logo__title">Acute Leukemia Methylome Atlas Study</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="0_overview.html">
                    Overview
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Latest</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="1_discovery.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_validation.html">Validation Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_test.html">Test Dataset</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Clinical Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_signature.html">37-CpG Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_classifiers.html">Classifiers</a></li>
<li class="toctree-l1"><a class="reference internal" href="6_plots.html">Data Visualization</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Previous</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="6_plots_original.html">Data Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/1_discovery_dataset.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/2_validation_dataset_v2.html">Validation Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/3_reclassification_strategy.html">Dx and Px Reclassification</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/4_models.html">Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/5_plots.html">Data Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/6_test_dataset.html">Preprocess long reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/7_test_models.html">Run epigenomic models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/8_test_atlas_1852.html">UF HemBank 1852 case study</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/ewasCox_vs_pacmapLGBM.html">Px Models Benchmarks</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/5_plots_6-17-24.html">AML-only map with new interactive prognostic and race histograms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/2_Univariate_CoxPH-EWAS_v4.html">EWAS Runs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v5.html">924pts-OS-35CpGs-ASH23</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v6.html">940pts-OS-48CpGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v7.html">940pts-EFS-55CpGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v2.html">OS-35CpGs-Validation-ASH23</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v3.html">OS-48CpGs-Validation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v4.html">EFS-55CpGs-Validation</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/source/4_clinical_data.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Clinical Data</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-index-merge-and-clean-discovery-clinical-data">Load, index, merge, and clean discovery clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#reclassification-strategy-dx-who-2022">Reclassification strategy: Dx WHO 2022</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#reclassification-strategy-risk-group-cog-aaml1831">Reclassification strategy: Risk Group COG AAML1831</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#main-reclassification-functions-applied-to-all-samples">Main reclassification functions (applied to all samples)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#execution">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#categorize-eln-2022-hematopoietic-entity-age-trisomy-8-and-5y-os-cutoff">Categorize ELN 2022, hematopoietic entity, age, trisomy 8, and 5y OS cutoff</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#functions">Functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-classification-to-validation-dataset">Apply classification to validation dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-classification-to-test-dataset">Apply classification to test dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#save-clinical-datasets">Save clinical datasets</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="clinical-data">
<h1>Clinical Data<a class="headerlink" href="#clinical-data" title="Link to this heading">#</a></h1>
<section id="load-index-merge-and-clean-discovery-clinical-data">
<h2>Load, index, merge, and clean discovery clinical data<a class="headerlink" href="#load-index-merge-and-clean-discovery-clinical-data" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import functions to clean up clinical data</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;../&#39;</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">utils.clinical_data_cleanup</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Call functions to merge, index and clean clinical data files</span>
<span class="n">labels_0531</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_0531</span><span class="p">())</span>
<span class="n">labels_1031</span>         <span class="o">=</span> <span class="n">clean_cog</span>       <span class="p">(</span><span class="n">merge_index_1031</span><span class="p">())</span>
<span class="n">labels_aml05</span>        <span class="o">=</span> <span class="n">clean_aml05</span>     <span class="p">(</span><span class="n">merge_index_aml05</span><span class="p">())</span>
<span class="n">labels_beataml</span>      <span class="o">=</span> <span class="n">clean_beataml</span>   <span class="p">(</span><span class="n">merge_index_beataml</span><span class="p">())</span>
<span class="n">labels_amltcga</span>      <span class="o">=</span> <span class="n">clean_amltcga</span>   <span class="p">(</span><span class="n">merge_index_amltcga</span><span class="p">())</span>
<span class="n">labels_nordic_all</span>   <span class="o">=</span> <span class="n">clean_nordic_all</span><span class="p">(</span><span class="n">merge_index_nordic_all</span><span class="p">())</span>
<span class="n">labels_mds_taml</span>     <span class="o">=</span> <span class="n">clean_mds_taml</span>  <span class="p">(</span><span class="n">merge_index_mds_taml</span><span class="p">())</span>
<span class="n">labels_all_graal</span>    <span class="o">=</span> <span class="n">clean_all_graal</span> <span class="p">(</span><span class="n">merge_index_all_graal</span><span class="p">())</span>
<span class="n">labels_target_all</span>   <span class="o">=</span> <span class="n">clean_target_all</span><span class="p">(</span><span class="n">merge_index_target_all</span><span class="p">())</span>

<span class="c1"># Combine all clinical data labels into one dataframe</span>
<span class="n">labels_combined</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">labels_aml05</span><span class="p">,</span> <span class="n">labels_beataml</span><span class="p">,</span>
                        <span class="n">labels_0531</span><span class="p">,</span> <span class="n">labels_amltcga</span><span class="p">,</span> <span class="n">labels_1031</span><span class="p">,</span>
                        <span class="n">labels_nordic_all</span><span class="p">,</span> <span class="n">labels_mds_taml</span><span class="p">,</span>
                        <span class="n">labels_all_graal</span><span class="p">,</span><span class="n">labels_target_all</span><span class="p">],</span>
                        <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;outer&#39;</span><span class="p">)</span>

<span class="c1"># Redefine output path (for troubleshooting purposes in case only this cell is run)</span>
<span class="n">MOUNT</span> <span class="o">=</span> <span class="s1">&#39;/mnt/e/ALMA/&#39;</span>
<span class="n">PATH</span> <span class="o">=</span> <span class="n">MOUNT</span> <span class="o">+</span> <span class="s1">&#39;/Intermediate_Files/&#39;</span>

<span class="c1"># Read df</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">PATH</span> <span class="o">+</span> <span class="s1">&#39;3314samples_331556cpgs_withbatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span>

<span class="c1"># Add columns for later analyses</span>
<span class="n">labels_combined</span><span class="p">[</span><span class="s1">&#39;Train-Test&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;Train Sample&#39;</span>
<span class="n">labels_combined</span><span class="p">[</span><span class="s1">&#39;Batch&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Batch&#39;</span><span class="p">]</span>

<span class="c1"># Remove samples that are not in the methyl dataset</span>
<span class="n">df_labels</span> <span class="o">=</span> <span class="n">labels_combined</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">labels_combined</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The clinical data has been indexed and cleaned.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The clinical data has been indexed and cleaned.
</pre></div>
</div>
</div>
</div>
</section>
<section id="reclassification-strategy-dx-who-2022">
<h2>Reclassification strategy: Dx WHO 2022<a class="headerlink" href="#reclassification-strategy-dx-who-2022" title="Link to this heading">#</a></h2>
<div class="tip admonition">
<p class="admonition-title">Source</p>
<p>Hasle H, Meshinchi S, Fogelstrand L, Alaggio R, et al. Acute myeloid leukaemias (AMLs) with defining genetic abnormalities. In: WHO Classification of Tumours Editorial Board. Paediatric tumours [Internet]. Lyon (France): International Agency for Research on Cancer; 2022 [cited 2024 Jan 1]. (WHO classification of tumours series, 5th ed.; vol. 7). Available from: <a class="reference external" href="https://tumourclassification.iarc.who.int/chapters/44">https://tumourclassification.iarc.who.int/chapters/44</a>.</p>
<p>Pfister, Stefan M et al. “A Summary of the Inaugural WHO Classification of Pediatric Tumors: Transitioning from the Optical into the Molecular Era.” Cancer discovery vol. 12,2 (2022): 331-355. doi:10.1158/2159-8290.CD-21-1094</p>
</div>
<div class="cell tag_hide-input tag_full-width docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">itables</span>
<span class="kn">from</span> <span class="nn">itables</span> <span class="kn">import</span> <span class="n">show</span>

<span class="c1"># Read the CSV file</span>
<span class="n">dx_criteria</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../data/who2022_aml_classification.csv&#39;</span><span class="p">)</span>

<span class="c1"># Display the dataframe using itables</span>
<span class="n">itables</span><span class="o">.</span><span class="n">show</span><span class="p">(</span><span class="n">dx_criteria</span><span class="p">,</span> <span class="n">columnDefs</span><span class="o">=</span><span class="p">[{</span><span class="s2">&quot;className&quot;</span><span class="p">:</span> <span class="s2">&quot;dt-left&quot;</span><span class="p">,</span> <span class="s2">&quot;targets&quot;</span><span class="p">:</span> <span class="s2">&quot;_all&quot;</span><span class="p">}],</span> <span class="n">paging</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><table id="itables_b73eee2a_a3b8_47ab_bc1f_65e08459661d" class="display nowrap" data-quarto-disable-processing="true" style="table-layout:auto;width:auto;margin:auto;caption-side:bottom">
<thead>
    <tr style="text-align: right;">
      
      <th>Subtype</th>
      <th>Frequency in paediatric AML</th>
      <th>Morphology</th>
      <th>Immunophenotype</th>
      <th>Common co-occurring genetic aberrations</th>
      <th>Prognosis</th>
    </tr>
  </thead><tbody><tr>
<td style="vertical-align:middle; text-align:left">
<div style="float:left; margin-right: 10px;">
<a href=https://mwouts.github.io/itables/><svg class="main-svg" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
width="64" viewBox="0 0 500 400" style="font-family: 'Droid Sans', sans-serif;">
    <g style="fill:#d9d7fc">
        <path d="M100,400H500V357H100Z" />
        <path d="M100,300H400V257H100Z" />
        <path d="M0,200H400V157H0Z" />
        <path d="M100,100H500V57H100Z" />
        <path d="M100,350H500V307H100Z" />
        <path d="M100,250H400V207H100Z" />
        <path d="M0,150H400V107H0Z" />
        <path d="M100,50H500V7H100Z" />
    </g>
    <g style="fill:#1a1366;stroke:#1a1366;">
   <rect x="100" y="7" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="5s"
      repeatCount="indefinite" />
      <animate
      attributeName="x"
      values="100;100;500"
      dur="5s"
      repeatCount="indefinite" />
  </rect>
        <rect x="0" y="107" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="3.5s"
      repeatCount="indefinite" />
    <animate
      attributeName="x"
      values="0;0;400"
      dur="3.5s"
      repeatCount="indefinite" />
  </rect>
        <rect x="100" y="207" width="300" height="43">
    <animate
      attributeName="width"
      values="0;300;0"
      dur="3s"
      repeatCount="indefinite" />
    <animate
      attributeName="x"
      values="100;100;400"
      dur="3s"
      repeatCount="indefinite" />
  </rect>
        <rect x="100" y="307" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="4s"
      repeatCount="indefinite" />
      <animate
      attributeName="x"
      values="100;100;500"
      dur="4s"
      repeatCount="indefinite" />
  </rect>
        <g style="fill:transparent;stroke-width:8; stroke-linejoin:round" rx="5">
            <g transform="translate(45 50) rotate(-45)">
                <circle r="33" cx="0" cy="0" />
                <rect x="-8" y="32" width="16" height="30" />
            </g>

            <g transform="translate(450 152)">
                <polyline points="-15,-20 -35,-20 -35,40 25,40 25,20" />
                <rect x="-15" y="-40" width="60" height="60" />
            </g>

            <g transform="translate(50 352)">
                <polygon points="-35,-5 0,-40 35,-5" />
                <polygon points="-35,10 0,45 35,10" />
            </g>

            <g transform="translate(75 250)">
                <polyline points="-30,30 -60,0 -30,-30" />
                <polyline points="0,30 -30,0 0,-30" />
            </g>

            <g transform="translate(425 250) rotate(180)">
                <polyline points="-30,30 -60,0 -30,-30" />
                <polyline points="0,30 -30,0 0,-30" />
            </g>
        </g>
    </g>
</svg>
</a>
</div>
<div>
Loading ITables v2.1.5 from the internet...
(need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td>
</div>
</tr></tbody>

</table>
<link href="https://www.unpkg.com/dt_for_itables@2.0.12/dt_bundle.css" rel="stylesheet">
<script type="module">
    import {DataTable, jQuery as $} from 'https://www.unpkg.com/dt_for_itables@2.0.12/dt_bundle.js';

    document.querySelectorAll("#itables_b73eee2a_a3b8_47ab_bc1f_65e08459661d:not(.dataTable)").forEach(table => {
        // Define the table data
        const data = [["AML with RUNX1::RUNXT1 fusion", "13\u201319% of cases", "Neutrophilic maturation, Auer rods; often increased basophils and/or mast cells", "Positive: CD34, HLA-DR, KIT (CD117), CD99, myeloperoxidase\\nUsually positive but can be heterogeneous: CD13, CD33 and CD38, CD19, PAX5, cytoplasmic CD79a, CD56\\nNegative: CD7", "Mutations in KIT, FLT3, CSF3R, EZH2, ASXL1, ASXL2, CBL, WT1, cohesion genes (SMC1A, SMC3, RAD21), NRAS, KRAS", "Good"], ["AML with CBFB::MYH11 fusion", "7\u201312% of cases", "Myelomonocytic; often increased eosinophils at all maturation stages", "Often multiple blast populations with signs of differentiation towards both granulocytic and monocytic lineages; may express CD2 and/or NG2 in a small subpopulation of cells", "Mutations in KIT, KRAS, NRAS, CBL, FLT3, WT1", "Good"], ["APL with PML::RARA fusion", "5\u201310% of cases but wide geographical variation (2\u201350%)", "Exclusively FAB M3; may be hypogranular", "Positive: CD33, often KIT (CD117), sometimes CD2\\nNegative: CD34 (except hypogranular variant), HLA-DR\\nOften high autofluorescence", "Mutations in FLT3", "Good (but treated differently)"], ["AML with KMT2A rearrangement", "20% of cases", "All FAB types except M3 and M6, most commonly M5", "Positive: CD33, CD65, CD4, HLA-DR (in monocytic cases), CD15, NG2; platelet antigens (in megakaryocytic cases)\\nNegative/low: CD13, CD34, CD14", "Mutations in the tyrosine kinase, PI3K, and RAS pathways; FLT3-TKD, NRAS, KRAS", "Depends on fusion partner"], ["AML with DEK::NUP214 fusion", "1\u20132% of cases", "All FAB types except M3 and M7, most commonly M2 and M4; Auer rods; basophilia in blood or bone marrow; erythroid and granulocytic dysplasia are common", "Positive: myeloperoxidase, CD13, CD33, CD38, CD123, HLA-DR; often CD34, KIT (CD117), CD15; sometimes CD64, TdT", "FLT3 ITD in two thirds of cases; complex karyotype", "Poor"], ["AML with MECOM rearrangement", "< 1% of cases", "Commonly FAB M1, M4, and M7; multilineage dysplasia is common, especially in megakaryopoiesis", "Positive: CD34, CD33, CD13, KIT (CD117), HLA-DR, CD7\\nNegative: lymphoid markers other than CD7", "NaN", "Poor"], ["AML with ETV6 fusion", "1% of all cases", "All FAB types, most commonly M0 and M1", "For MNX1::ETV6:\\nPositive: CD34, HLA-DR, often KIT (CD117), T-cell markers CD7 and CD4", "Almost always +19, sometimes +8", "Poor, but may be protocol dependent"], ["AML with KAT6A::CREBBP fusion", "0.5% of cases", "FAB M5", "Positive: HLA-DR, CD15, myeloperoxidase, CD13, CD33, sometimes CD56, CD14\\nNegative: CD34, KIT (CD117)", "NaN", "OS similar to paediatric AML in general"], ["AML with RBM15::MRTFA fusion", "0.8% of all cases", "Exclusively FAB M7", "Positive: > 1 platelet antigen, some CD34 and/or HLA-DR\\nNegative: CD45", "del(1p), +4, +7, +8, +9, +10, +14, +15, +17, +18, +20, +21(21p), complex hyperdiploid karyotype", "Lower CR rate but similar EFS/OS vs other AMKLs"], ["AML with CBFA2T3::GLIS2 fusion", "3% of all cases", "All FAB types, most commonly M7", "> 50% show the RAM phenotype; overexpression of CD56 and underexpression of HLA-DR and CD38", "Mutations in FLT3, GATA1, JAK/STAT pathway", "Poor"], ["AML with NUP98 rearrangement", "5\u20139% of cases", "NUP98::NSD1: often FAB M4 and M5\\nNUP98::KDM5A: mostly FAB M7 but also M5 and M6", "NaN", "For NUP98::NSD1: FLT3 ITD in > 80%, mutation in WT1 in 50%", "Poor"], ["AML with FUS::ERG fusion", "0.3\u20130.5% of cases", "All FAB types except M6; Auer rods", "NaN", "+8, +10, complex karyotype", "Poor"], ["AML with NPM1 mutation", "6\u20138% of cases", "FAB M1, M2, M4, and M5", "Positive: CD33, sometimes KIT (CD117), often CD36, CD64\\nNegative: CD34, sometimes HLA-DR\\nOften weak CD13", "FLT3 ITD, mutations in NRAS, WT1", "Good, also when FLT3 ITD is present"], ["AML with CEBPA mutation", "5\u20136% of cases", "Most commonly FAB M1 and M2", "Positive: CD34, HLA-DR, often CD7\\nNegative: other lymphoid antigens", "Mutations in CEBPA-TAD, CSF3R, GATA2, RAS genes, FLT3", "Good, except for CSF3R-mutated cases"]];

        // Define the dt_args
        let dt_args = {"columnDefs": [{"className": "dt-left", "targets": "_all"}], "paging": false, "layout": {"topStart": "pageLength", "topEnd": "search", "bottomStart": "info", "bottomEnd": "paging"}, "order": []};
        dt_args["data"] = data;

        
        new DataTable(table, dt_args);
    });
</script>
</div></div>
</div>
</section>
<section id="reclassification-strategy-risk-group-cog-aaml1831">
<h2>Reclassification strategy: Risk Group COG AAML1831<a class="headerlink" href="#reclassification-strategy-risk-group-cog-aaml1831" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input tag_full-width docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Raw_Data/Clinical_Data/updated_risk_group_COG_RR_1-10-2024/&#39;</span>

<span class="n">COG_RR_dataset_Jan2024</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">path</span> <span class="o">+</span> <span class="s1">&#39;COG_risk_added_v2_cleanedFM.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,)[</span>
                                    <span class="p">[</span><span class="s1">&#39;Primary Fusion&#39;</span><span class="p">,</span> <span class="s1">&#39;Cyto/Molecular Risk&#39;</span><span class="p">,</span> <span class="s1">&#39;Reason&#39;</span><span class="p">,</span> <span class="s1">&#39;MRD at EOI I&#39;</span><span class="p">]</span>
                                    <span class="p">]</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;Cyto/Molecular Risk&#39;</span><span class="p">:</span> <span class="s1">&#39;Risk Group AAML1831&#39;</span><span class="p">})</span>

<span class="c1"># Join the two dataframes based on sample index</span>
<span class="n">df_labels2</span> <span class="o">=</span> <span class="n">df_labels</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">COG_RR_dataset_Jan2024</span><span class="p">)</span>

<span class="c1"># Create binary version for column &#39;Risk Group AAML1831&#39;</span>
<span class="n">df_labels2</span><span class="p">[</span><span class="s1">&#39;Risk_Group_AAML1831_int&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_labels2</span><span class="p">[</span><span class="s1">&#39;Risk Group AAML1831&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">({</span><span class="s1">&#39;Low&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;Standard&#39;</span><span class="p">:</span><span class="mf">0.5</span><span class="p">,</span><span class="s1">&#39;High&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">})</span>

<span class="c1"># Combine `Gene Fusion` and `Primary Fusion` columns with comma while ignoring NaNs/empty strings</span>
<span class="n">df_labels2</span><span class="p">[</span><span class="s1">&#39;Gene Fusion&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_labels2</span><span class="p">[[</span><span class="s1">&#39;Primary Fusion&#39;</span><span class="p">,</span> <span class="s1">&#39;Gene Fusion&#39;</span><span class="p">,]]</span>\
<span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">filter</span><span class="p">(</span><span class="k">lambda</span> <span class="n">i</span><span class="p">:</span> <span class="n">i</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="ow">and</span> <span class="n">i</span><span class="o">==</span><span class="n">i</span><span class="p">,</span> <span class="n">x</span><span class="p">)),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="c1"># Replace &#39;&#39; with np.nan</span>
<span class="n">df_labels2</span><span class="p">[</span><span class="s1">&#39;Gene Fusion&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_labels2</span><span class="p">[</span><span class="s1">&#39;Gene Fusion&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">replace</span><span class="p">({</span><span class="s1">&#39;&#39;</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">})</span>

<span class="c1"># Read criteria</span>
<span class="n">px_criteria</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">path</span> <span class="o">+</span> <span class="s1">&#39;AAML1831_risk_stratification_and_proposed_additions_FMcleaned.xlsx&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">({</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">})</span>

<span class="c1"># Display the dataframe using itables</span>
<span class="kn">import</span> <span class="nn">itables</span>

<span class="n">itables</span><span class="o">.</span><span class="n">show</span><span class="p">(</span><span class="n">px_criteria</span><span class="p">,</span> <span class="n">columnDefs</span><span class="o">=</span><span class="p">[{</span><span class="s2">&quot;className&quot;</span><span class="p">:</span> <span class="s2">&quot;dt-left&quot;</span><span class="p">,</span> <span class="s2">&quot;targets&quot;</span><span class="p">:</span> <span class="s2">&quot;_all&quot;</span><span class="p">}])</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><table id="itables_167f931a_54dc_4b24_a6e0_79b55b64906d" class="display nowrap" data-quarto-disable-processing="true" style="table-layout:auto;width:auto;margin:auto;caption-side:bottom">
<thead>
    <tr style="text-align: right;">
      
      <th>Low Risk</th>
      <th>Standard Risk</th>
      <th>High Risk</th>
    </tr>
  </thead><tbody><tr>
<td style="vertical-align:middle; text-align:left">
<div style="float:left; margin-right: 10px;">
<a href=https://mwouts.github.io/itables/><svg class="main-svg" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
width="64" viewBox="0 0 500 400" style="font-family: 'Droid Sans', sans-serif;">
    <g style="fill:#d9d7fc">
        <path d="M100,400H500V357H100Z" />
        <path d="M100,300H400V257H100Z" />
        <path d="M0,200H400V157H0Z" />
        <path d="M100,100H500V57H100Z" />
        <path d="M100,350H500V307H100Z" />
        <path d="M100,250H400V207H100Z" />
        <path d="M0,150H400V107H0Z" />
        <path d="M100,50H500V7H100Z" />
    </g>
    <g style="fill:#1a1366;stroke:#1a1366;">
   <rect x="100" y="7" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="5s"
      repeatCount="indefinite" />
      <animate
      attributeName="x"
      values="100;100;500"
      dur="5s"
      repeatCount="indefinite" />
  </rect>
        <rect x="0" y="107" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="3.5s"
      repeatCount="indefinite" />
    <animate
      attributeName="x"
      values="0;0;400"
      dur="3.5s"
      repeatCount="indefinite" />
  </rect>
        <rect x="100" y="207" width="300" height="43">
    <animate
      attributeName="width"
      values="0;300;0"
      dur="3s"
      repeatCount="indefinite" />
    <animate
      attributeName="x"
      values="100;100;400"
      dur="3s"
      repeatCount="indefinite" />
  </rect>
        <rect x="100" y="307" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="4s"
      repeatCount="indefinite" />
      <animate
      attributeName="x"
      values="100;100;500"
      dur="4s"
      repeatCount="indefinite" />
  </rect>
        <g style="fill:transparent;stroke-width:8; stroke-linejoin:round" rx="5">
            <g transform="translate(45 50) rotate(-45)">
                <circle r="33" cx="0" cy="0" />
                <rect x="-8" y="32" width="16" height="30" />
            </g>

            <g transform="translate(450 152)">
                <polyline points="-15,-20 -35,-20 -35,40 25,40 25,20" />
                <rect x="-15" y="-40" width="60" height="60" />
            </g>

            <g transform="translate(50 352)">
                <polygon points="-35,-5 0,-40 35,-5" />
                <polygon points="-35,10 0,45 35,10" />
            </g>

            <g transform="translate(75 250)">
                <polyline points="-30,30 -60,0 -30,-30" />
                <polyline points="0,30 -30,0 0,-30" />
            </g>

            <g transform="translate(425 250) rotate(180)">
                <polyline points="-30,30 -60,0 -30,-30" />
                <polyline points="0,30 -30,0 0,-30" />
            </g>
        </g>
    </g>
</svg>
</a>
</div>
<div>
Loading ITables v2.1.5 from the internet...
(need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td>
</div>
</tr></tbody>

</table>
<link href="https://www.unpkg.com/dt_for_itables@2.0.12/dt_bundle.css" rel="stylesheet">
<script type="module">
    import {DataTable, jQuery as $} from 'https://www.unpkg.com/dt_for_itables@2.0.12/dt_bundle.js';

    document.querySelectorAll("#itables_167f931a_54dc_4b24_a6e0_79b55b64906d:not(.dataTable)").forEach(table => {
        // Define the table data
        const data = [["t(8;21)(q21.2;q22) [RUNX1-RUNX1T1]", "patients without HR or LR features ", "Monosomy 7"], ["inv(16)/t(16;16)(p13.1q22.1) [CBFB-MYH11]", "", "Monosomy 5/del5q-"], ["NPM1 positive", "", "FLT3/ITD+ with allelic ratio > 0.1%"], ["CEBPA positive\u00a0(Bzip mutation)", "", "t(6;9)(p22.3;q34.1) [DEK-NUP214]"], ["", "", "inv(3)(q21.3q26.2) [RPN1-MECOM]"], ["", "", "t(3;21)(26.2;q22) [RUNX1-MECOM]"], ["", "", "t(3;5)(q25q34) [NPM1-MLF1]"], ["", "", "11p15 (NUP98) rearrangements"], ["", "", "inv(16)(p13.3q24.3) [CBFA2T3-GLIS2]"], ["", "", "10p12.3 (MLLT10) rearrangements"], ["", "", "KMT2A rearrangements"], ["", "", "t(4;11)(q21;q23) [KMT2A-AFF1]"], ["", "", "t(6;11)(q27;q23) [KMT2A-MLLT4] (aka AF6, AFDN)"], ["", "", "t(10;11)(p11.2;q23) [KMT2A-ABI1]"], ["", "", "t(10;11)(p12;q23) [KMT2A-MLLT10]"], ["", "", "t(11;19)(q23;p13.3) [KMT2A-MLLT1]"], ["", "", "KMT2A-PTD"], ["", "", "ETS family rearrangements"], ["", "", "12p13.2 (ETV6) rearrangements"], ["", "", "12p13.2 (ETV6) deletion"], ["", "", "21q22.2 (ERG) rearrangements"], ["", "", "FEV rearrangements"], ["", "", "FLI1 rearrangements"], ["", "", "CREBBP"], ["", "", "t(8;16)(p11.2;p13.3) [KAT6A-CREBBP]"], ["", "", "CREBBP single nucleotide variants"], ["", "", "CREBBP insertion/deletions"], ["", "", "CEBPA with CSF3R"], ["", "", "IDH alterations"], ["", "", "IDH without NPM1"], ["", "", "IDH with NPM1 and DNMT3A"], ["", "", "IDH with NPM1 and FLT3-ITD"], ["", "", "Exon 5 \u2013 Exon 33 inv(16)/t16;16)(p13.1q22.1) [CBFB-MYH11]  with exon 17 c-KIT"], ["", "", "t(8;21)(q21.2;q22) [RUNX1-RUNX1T1] with ASXL1"], ["", "", "TP53"], ["", "", "UBTF/ITD"], ["", "", "NPM1 + DNMT3A"], ["", "", "RUNX1-RUNX1T1 + MGA alteration"], ["", "", "MYC SNV/indels"]];

        // Define the dt_args
        let dt_args = {"columnDefs": [{"className": "dt-left", "targets": "_all"}], "layout": {"topStart": "pageLength", "topEnd": "search", "bottomStart": "info", "bottomEnd": "paging"}, "order": []};
        dt_args["data"] = data;

        
        new DataTable(table, dt_args);
    });
</script>
</div></div>
</div>
<section id="main-reclassification-functions-applied-to-all-samples">
<h3>Main reclassification functions (applied to all samples)<a class="headerlink" href="#main-reclassification-functions-applied-to-all-samples" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">classify_controls</span><span class="p">(</span><span class="n">normal_samples</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Bone Marrow Normal&#39;</span>  <span class="p">:</span> <span class="s1">&#39;Otherwise-Normal Control&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Blood Derived Normal&#39;</span><span class="p">:</span> <span class="s1">&#39;Otherwise-Normal Control&#39;</span><span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">normal_samples</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_fusion</span><span class="p">(</span><span class="n">gene_fusion</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;RUNX1-RUNX1T1&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span><span class="p">,</span>
    <span class="s1">&#39;CBFB-MYH11&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span><span class="p">,</span>
    <span class="s1">&#39;KMT2A&#39;</span><span class="p">:</span>         <span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>
    <span class="s1">&#39;MLL&#39;</span><span class="p">:</span>           <span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>
    <span class="s1">&#39;PML-RARA&#39;</span><span class="p">:</span>      <span class="s1">&#39;APL with t(15;17); PML::RARA&#39;</span><span class="p">,</span>
    <span class="s1">&#39;DEK-NUP214&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span><span class="p">,</span>
    <span class="s1">&#39;MECOM&#39;</span><span class="p">:</span>         <span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span><span class="p">,</span>
    <span class="s1">&#39;ETV6&#39;</span><span class="p">:</span>          <span class="s1">&#39;AML with ETV6 fusion&#39;</span><span class="p">,</span>
    <span class="s1">&#39;NPM1&#39;</span><span class="p">:</span>          <span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
    <span class="s1">&#39;RBM15-MKL1&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(1;22); RBM15::MKL1&#39;</span><span class="p">,</span>
    <span class="s1">&#39;NUP98&#39;</span><span class="p">:</span>         <span class="s1">&#39;AML with NUP98-fusion&#39;</span><span class="p">,</span>
    <span class="s1">&#39;KAT6A-CREBBP&#39;</span><span class="p">:</span>  <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span><span class="p">,</span>
    <span class="s1">&#39;FUS-ERG&#39;</span><span class="p">:</span>       <span class="s1">&#39;AML with t(16;21); FUS::ERG&#39;</span><span class="p">,</span>
    <span class="s1">&#39;CBFA2T3-GLIS2&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with CBFA2T3::GLIS2&#39;</span><span class="p">,</span>
    <span class="s1">&#39;BCR-ABL1&#39;</span><span class="p">:</span>       <span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>

    <span class="c1"># Other uncharacterized abdnormalities present in the dataset but not specifically in guidelines</span>

    <span class="c1"># &#39;RUNX1-CBFA2T3&#39;: &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;PRDM16-RPN1&#39;:   &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;RBM15-MRTFA&#39;:   &#39;AML with t(1;22); RBM15::MKL1&#39;,</span>
    <span class="s1">&#39;CBFA2T3-GLIS3&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with CBFA2T3::GLIS2&#39;</span><span class="p">,</span> <span class="c1"># including here based on PMID: 35927023</span>
    <span class="c1"># &#39;PSIP1-NUP214&#39;:  &#39;AML with t(6;9); DEK::NUP214&#39;,</span>
    <span class="c1"># &#39;XPO1-TNRC18&#39;:   &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;HNRNPH1-ERG&#39;:   &#39;AML with t(16;21); FUS::ERG&#39;,</span>
    <span class="c1"># &#39;NIPBL-HOXB9&#39;:   &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;SET-NUP214&#39;:    &#39;AML with t(6;9); DEK::NUP214&#39;, </span>
    <span class="c1"># &#39;FLI1-IFIT2&#39;:    &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;TCF4-ZEB2&#39;:     &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;MBTD1-ZMYND11&#39;: &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;FOSB-KLF6&#39;:     &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;SFPQ-ZFP36L2&#39;:  &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;RUNX1-LINC00478&#39;:&#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;RUNX1-EVX1&#39;:     &#39;AML NOS&#39;,  </span>
    <span class="c1"># &#39;PSPC1-ZFP36L1&#39;:  &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;EWSR1-FEV&#39;:      &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;STAG2-AFF2&#39;:     &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;MYB-GATA1&#39;:      &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;RUNX1-ZFPM2&#39;:    &#39;AML NOS&#39;, </span>
    <span class="c1"># &#39;RUNX1-CBFA2T2&#39;:  &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;PIM3-BRD1&#39;:      &#39;AML NOS&#39;,</span>
    <span class="s1">&#39;KAT6A-EP300&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span><span class="p">,</span>
    <span class="c1"># &#39;DOT1L-RPS15&#39;:    &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;FUS-FEV&#39;:        &#39;AML with t(16;21); FUS::ERG&#39;,</span>
    <span class="s1">&#39;KAT6A-NCOA2&#39;</span><span class="p">:</span>    <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span><span class="p">,</span>
    <span class="c1"># &#39;JARID2-PTP4A1&#39;:  &#39;AML NOS&#39;,</span>
    <span class="c1"># &#39;FUS-FLI1&#39;:       &#39;AML with t(16;21); FUS::ERG&#39;,</span>
    <span class="p">}</span>    
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">gene_fusion</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_cebpa</span><span class="p">(</span><span class="n">cebpa_mutation</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Yes&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Positive&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">cebpa_mutation</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_npm</span><span class="p">(</span><span class="n">npm_mutation</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Yes&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Positive&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with mutated NPM1&#39;</span>
    <span class="p">}</span>

    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">npm_mutation</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">classify_karyotype</span><span class="p">(</span><span class="n">structural_variation</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;t(8;16)&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span><span class="p">,</span>
        <span class="c1"># &#39;t(16;21)&#39;: &#39;AML with t(16;21); FUS::ERG&#39;,</span>
        <span class="c1"># &#39;t(6;9)&#39;: &#39;AML with t(6;9); DEK::NUP214&#39;,</span>
        <span class="c1"># &#39;t(1;22)&#39;: &#39;AML with t(1;22); RBM15::MKL1&#39;,</span>
        <span class="s1">&#39;inv(3)&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span><span class="p">,</span>
        <span class="s1">&#39;t(3;3)&#39;</span><span class="p">:</span> <span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span><span class="p">,</span>
        <span class="c1"># &#39;t(6;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;,</span>
        <span class="c1"># &#39;t(1;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;,</span>
        <span class="c1"># &#39;t(4;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;,</span>
        <span class="c1"># &#39;ins(6;11)&#39;: &#39;AML with t(v;11q23); KMT2A-r&#39;, # Interestingly, karyotype did not prove reliable for subtyping. Who would have thought!</span>

        <span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">structural_variation</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>
            
<span class="k">def</span> <span class="nf">classify_annotated_diagnosis</span><span class="p">(</span><span class="n">diagnosis</span><span class="p">):</span>
    <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;mutated NPM1&#39;</span>                                                  <span class="p">:</span> <span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;mutated CEBPA&#39;</span>                                                 <span class="p">:</span> <span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">,</span>
        <span class="s1">&#39;myelodysplasia-related changes&#39;</span>                                <span class="p">:</span> <span class="s1">&#39;MDS-related; secondary myeloid&#39;</span><span class="p">,</span>
        <span class="s2">&quot;AML with mutated NPM1&quot;</span>                                         <span class="p">:</span> <span class="s2">&quot;AML with mutated NPM1&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with myelodysplasia-related changes&quot;</span>                       <span class="p">:</span> <span class="s2">&quot;MDS-related; secondary myeloid&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22); CBFB-MYH11&quot;</span> <span class="p">:</span> <span class="s2">&quot;AML with inv(16); t(16;16); CBFB::MYH11&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with mutated CEBPA&quot;</span>                                        <span class="p">:</span> <span class="s2">&quot;AML with mutated CEBPA&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Therapy-related myeloid neoplasms&quot;</span>                             <span class="p">:</span> <span class="s2">&quot;MDS-related; secondary myeloid&quot;</span><span class="p">,</span>
        <span class="s2">&quot;PML-RARA&quot;</span>                                                      <span class="p">:</span> <span class="s2">&quot;APL with t(15;17); PML::RARA&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with t(9;11)(p22;q23); MLLT3-MLL&quot;</span>                          <span class="p">:</span> <span class="s2">&quot;AML with t(v;11q23); KMT2A-r&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with t(8;21)(q22;q22.1); RUNX1-RUNX1T1&quot;</span>                    <span class="p">:</span> <span class="s2">&quot;AML with t(8;21); RUNX1::RUNX1T1&quot;</span><span class="p">,</span>
        <span class="s2">&quot;AML with inv(3)(q21q26.2) or t(3;3)(q21;q26.2); RPN1-EVI1&quot;</span>     <span class="p">:</span> <span class="s2">&quot;AML with inv(3); t(3;3); MECOM-r&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Mixed phenotype acute leukaemia, T/myeloid&quot;</span>                    <span class="p">:</span> <span class="s2">&quot;MPAL T-Lymphoblastic/Myeloid&quot;</span><span class="p">,</span>
        <span class="s2">&quot;Myeloid leukaemia associated with Down syndrome&quot;</span>               <span class="p">:</span> <span class="s2">&quot;AML with Down syndrome&quot;</span><span class="p">,</span>
        <span class="s1">&#39;AML with CBFB-MYH11&#39;</span>                                           <span class="p">:</span> <span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span><span class="p">,</span>
        <span class="s1">&#39;AML with RUNX1-RUNX1T1&#39;</span>                                        <span class="p">:</span> <span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;AML with BCR-ABL1&#39;</span>                                             <span class="p">:</span> <span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>
        <span class="s1">&#39;MDS-related or secondary myeloid neoplasms&#39;</span>                    <span class="p">:</span> <span class="s1">&#39;MDS-related; secondary myeloid&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with t(8;21) (q22;q22) [AML1/ETO]&#39;</span>            <span class="p">:</span> <span class="s2">&quot;AML with t(8;21); RUNX1::RUNX1T1&quot;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with 11q23/MLL Abnormalities&#39;</span>                 <span class="p">:</span> <span class="s2">&quot;AML with t(v;11q23); KMT2A-r&quot;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with t(6;9)(p23;q34) DEK-NUP214&#39;</span>              <span class="p">:</span> <span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span><span class="p">,</span>
        <span class="s1">&#39;Myeloid Leukemia with inv(16) or t(16;16) [CBFbeta/MYH11]&#39;</span>     <span class="p">:</span> <span class="s2">&quot;AML with inv(16); t(16;16); CBFB::MYH11&quot;</span><span class="p">,</span>
        <span class="s1">&#39;MLL+, FLT3 TKD + monocytic AML&#39;</span>                                <span class="p">:</span> <span class="s2">&quot;AML with t(v;11q23); KMT2A-r&quot;</span><span class="p">,</span>
        
        <span class="c1"># ALL</span>
        <span class="s1">&#39;T-ALL&#39;</span>                                                         <span class="p">:</span><span class="s1">&#39;T-ALL NOS&#39;</span><span class="p">,</span>
        <span class="s1">&#39;T-cell acute lymphoblastic leukemia&#39;</span>                           <span class="p">:</span><span class="s1">&#39;T-ALL NOS&#39;</span><span class="p">,</span>
        <span class="s1">&#39;BCP-ALL HeH&#39;</span>                                                   <span class="p">:</span><span class="s1">&#39;B-ALL with hyperdiploidy&#39;</span><span class="p">,</span>
        <span class="s1">&#39;BCP-ALL t(12;21)&#39;</span>                                              <span class="p">:</span><span class="s1">&#39;B-ALL with t(12;21); ETV6::RUNX1&#39;</span><span class="p">,</span>         
        <span class="s1">&#39;BCP-ALL 11q23/MLL&#39;</span>                                             <span class="p">:</span><span class="s1">&#39;B-ALL with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL t(1;19)&#39;</span>                                               <span class="p">:</span><span class="s1">&#39;B-ALL with t(1;19); TCF3::PBX1&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL dic(9;20)&#39;</span>                                             <span class="p">:</span><span class="s1">&#39;B-ALL dic(9;20)&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL t(9;22)&#39;</span>                                               <span class="p">:</span><span class="s1">&#39;B-ALL with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>       
        <span class="s1">&#39;BCP-ALL iAMP21&#39;</span>                                                <span class="p">:</span><span class="s1">&#39;B-ALL with iAMP21&#39;</span><span class="p">,</span>        
        <span class="s1">&#39;BCP-ALL &lt;45chr&#39;</span>                                                <span class="p">:</span><span class="s1">&#39;B-ALL with hypodiploidy&#39;</span><span class="p">,</span>        
        <span class="s1">&#39;BCP-ALL &gt;67chr&#39;</span>                                                <span class="p">:</span><span class="s1">&#39;B-ALL with hyperdiploidy&#39;</span><span class="p">,</span>

        <span class="c1"># MPAL</span>
        <span class="s1">&#39;MPAL with MLL&#39;</span>                                                 <span class="p">:</span><span class="s1">&#39;MPAL with t(v;11q23.3)/KMT2A-r&#39;</span><span class="p">,</span>
        <span class="s1">&#39;MPAL with T/M&#39;</span>                                                 <span class="p">:</span><span class="s1">&#39;MPAL T-Lymphoblastic/Myeloid&#39;</span><span class="p">,</span>
        <span class="s1">&#39;MPAL with B/M&#39;</span>                                                 <span class="p">:</span><span class="s1">&#39;MPAL B-Lymphoblastic/Myeloid&#39;</span><span class="p">,</span>

        <span class="p">}</span>
    
    <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">diagnosis</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">value</span>

<span class="k">def</span> <span class="nf">process_labels_who22</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
    <span class="c1"># Merge processing functions with source columns into a single dictionary</span>
    <span class="n">column_mappings</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Annotated Controls&#39;</span>            <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">,</span> <span class="n">classify_controls</span><span class="p">),</span>
        <span class="s1">&#39;Annotated Gene Fusion&#39;</span>         <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Gene Fusion&#39;</span><span class="p">,</span> <span class="n">classify_fusion</span><span class="p">),</span>
        <span class="s1">&#39;Annotated CEBPA&#39;</span>               <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;CEBPA mutation&#39;</span><span class="p">,</span> <span class="n">classify_cebpa</span><span class="p">),</span>
        <span class="s1">&#39;Annotated NPM1&#39;</span>                <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;NPM mutation&#39;</span><span class="p">,</span> <span class="n">classify_npm</span><span class="p">),</span>
        <span class="s1">&#39;Annotated Karyotype&#39;</span>           <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Karyotype&#39;</span><span class="p">,</span> <span class="n">classify_karyotype</span><span class="p">),</span>
        <span class="s1">&#39;Annotated Dx at Acquisition&#39;</span>   <span class="p">:</span> <span class="p">(</span><span class="s1">&#39;Dx at Acquisition&#39;</span><span class="p">,</span> <span class="n">classify_annotated_diagnosis</span><span class="p">),</span>
    <span class="p">}</span>

    <span class="c1"># Loop through the mapping to process and create new columns</span>
    <span class="k">for</span> <span class="n">new_col</span><span class="p">,</span> <span class="p">(</span><span class="n">old_col</span><span class="p">,</span> <span class="n">processing_func</span><span class="p">)</span> <span class="ow">in</span> <span class="n">column_mappings</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="c1"># Apply the corresponding classification function to each column</span>
        <span class="n">df</span><span class="p">[</span><span class="n">new_col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">old_col</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">processing_func</span><span class="p">)</span>

    <span class="c1"># Combine processed columns into a single column, separating values with commas</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Combined Diagnoses&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="nb">list</span><span class="p">(</span><span class="n">column_mappings</span><span class="o">.</span><span class="n">keys</span><span class="p">())]</span>\
        <span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">dropna</span><span class="p">()),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">)</span>

    <span class="c1"># Extract the first diagnosis from the combined diagnoses as the final diagnosis</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Combined Diagnoses&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">str</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>

    <span class="c1"># Drop intermediate columns used for processing</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">column_mappings</span><span class="o">.</span><span class="n">keys</span><span class="p">()),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">df</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="execution">
<h3>Execution<a class="headerlink" href="#execution" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Reclassify according to WHO 2022 guidelines</span>
<span class="n">df_labels3</span> <span class="o">=</span> <span class="n">process_labels_who22</span><span class="p">(</span><span class="n">df_labels2</span><span class="p">)</span>

<span class="c1"># Display the value counts of the new diagnosis column</span>
<span class="n">df_labels3</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>WHO 2022 Diagnosis
NaN                                        840
AML with t(v;11q23); KMT2A-r               324
Otherwise-Normal Control                   251
T-ALL NOS                                  242
AML with mutated NPM1                      232
MDS-related; secondary myeloid             223
B-ALL with hyperdiploidy                   190
AML with inv(16); t(16;16); CBFB::MYH11    178
AML with t(8;21); RUNX1::RUNX1T1           178
B-ALL with t(12;21); ETV6::RUNX1           163
AML with NUP98-fusion                      100
AML with mutated CEBPA                      83
APL with t(15;17); PML::RARA                31
AML with CBFA2T3::GLIS2                     29
B-ALL with t(v;11q23); KMT2A-r              28
AML with t(6;9); DEK::NUP214                27
MPAL T-Lymphoblastic/Myeloid                25
B-ALL with t(1;19); TCF3::PBX1              23
AML with ETV6 fusion                        20
B-ALL dic(9;20)                             20
AML with inv(3); t(3;3); MECOM-r            20
B-ALL with t(9;22); BCR::ABL1               19
MPAL B-Lymphoblastic/Myeloid                17
AML with t(8;16); KAT6A::CREBBP             12
AML with t(1;22); RBM15::MKL1               10
B-ALL with iAMP21                           10
MPAL with t(v;11q23.3)/KMT2A-r               6
AML with t(16;21); FUS::ERG                  5
B-ALL with hypodiploidy                      5
AML with t(9;22); BCR::ABL1                  3
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="categorize-eln-2022-hematopoietic-entity-age-trisomy-8-and-5y-os-cutoff">
<h2>Categorize ELN 2022, hematopoietic entity, age, trisomy 8, and 5y OS cutoff<a class="headerlink" href="#categorize-eln-2022-hematopoietic-entity-age-trisomy-8-and-5y-os-cutoff" title="Link to this heading">#</a></h2>
<section id="functions">
<h3>Functions<a class="headerlink" href="#functions" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">process_df_labels</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Function to process a pandas dataframe, performing age categorization </span>
<span class="sd">    and main disease classification.</span>

<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">def</span> <span class="nf">categorize_age</span><span class="p">(</span><span class="n">age</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to categorize age into a specific range.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isnull</span><span class="p">(</span><span class="n">age</span><span class="p">):</span>
            <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">5</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;0-5&#39;</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">13</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;5-13&#39;</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">39</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;13-39&#39;</span>
        <span class="k">elif</span> <span class="n">age</span> <span class="o">&lt;</span> <span class="mi">60</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;39-60&#39;</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="s1">&#39;60+&#39;</span>

    <span class="c1"># Convert &#39;Age (years)&#39; to numeric, errors=&#39;coerce&#39; will turn non-numeric data to NaN</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">to_numeric</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">],</span> <span class="n">errors</span><span class="o">=</span><span class="s1">&#39;coerce&#39;</span><span class="p">)</span>

    <span class="c1"># Then apply your function</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (group years)&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Age (years)&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_age</span><span class="p">)</span>
    
    <span class="k">def</span> <span class="nf">categorize_subtypes</span><span class="p">(</span><span class="n">subtype</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to classify the main disease based on a given subtype.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s1">&#39;AML&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Acute myeloid leukemia (AML)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;ALL&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Acute lymphoblastic leukemia (ALL)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;MDS&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Myelodysplastic syndrome (MDS or MDS-like)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;MPAL&#39;</span>                          <span class="p">:</span><span class="s1">&#39;Mixed phenotype acute leukemia (MPAL)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;APL&#39;</span>                           <span class="p">:</span><span class="s1">&#39;Acute promyelocytic leukemia (APL)&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Otherwise-Normal Control&#39;</span>      <span class="p">:</span><span class="s1">&#39;Otherwise-Normal (Control)&#39;</span><span class="p">,</span>
        <span class="p">}</span>

        <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">subtype</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">value</span>
    
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Hematopoietic Entity&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_subtypes</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">categorize_trisomy8</span><span class="p">(</span><span class="n">karyotype</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to classify trisomy 8 by karyotype as a separate column.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s1">&#39;+8&#39;</span><span class="p">:</span> <span class="s1">&#39;Positive&#39;</span><span class="p">,</span>
            <span class="s1">&#39;trisomy 8&#39;</span><span class="p">:</span> <span class="s1">&#39;Positive&#39;</span><span class="p">,</span>
        <span class="p">}</span>

        <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">karyotype</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">value</span>
    
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Trisomy 8 Status&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;Karyotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_trisomy8</span><span class="p">)</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s1">&#39;Negative or Unknown&#39;</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">categorize_ELN2022</span><span class="p">(</span><span class="n">subtype</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Function to reclassify samples according to ELN AML 2022.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">mapping</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span>                  <span class="p">:</span><span class="s1">&#39;AML with t(v;11q23); KMT2A-r&#39;</span><span class="p">,</span>
            <span class="s1">&#39;Otherwise-Normal Control&#39;</span>                      <span class="p">:</span><span class="s1">&#39;Otherwise-Normal Control&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with mutated CEBPA&#39;</span>                        <span class="p">:</span><span class="s1">&#39;AML with mutated CEBPA&#39;</span><span class="p">,</span>
            <span class="s1">&#39;MDS-related; secondary myeloid&#39;</span>                <span class="p">:</span><span class="s1">&#39;MDS-related; secondary myeloid&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span>              <span class="p">:</span><span class="s1">&#39;AML with inv(3); t(3;3); MECOM-r&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with mutated NPM1&#39;</span>                         <span class="p">:</span><span class="s1">&#39;AML with mutated NPM1&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span>       <span class="p">:</span><span class="s1">&#39;AML with inv(16); t(16;16); CBFB::MYH11&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span>              <span class="p">:</span><span class="s1">&#39;AML with t(8;21); RUNX1::RUNX1T1&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span>                   <span class="p">:</span><span class="s1">&#39;AML with t(9;22); BCR::ABL1&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span>                  <span class="p">:</span><span class="s1">&#39;AML with t(6;9); DEK::NUP214&#39;</span><span class="p">,</span>
            <span class="s1">&#39;APL with t(15;17); PML::RARA&#39;</span>                  <span class="p">:</span><span class="s1">&#39;APL with t(15;17); PML::RARA&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(1;22); RBM15::MKL1&#39;</span>                 <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(16;21); FUS::ERG&#39;</span>                   <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with t(8;16); KAT6A::CREBBP&#39;</span>               <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with CBFA2T3::GLIS2&#39;</span>                       <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with NUP98-fusion&#39;</span>                         <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>
            <span class="s1">&#39;AML with ETV6 fusion&#39;</span>                          <span class="p">:</span><span class="s1">&#39;AML with other recurring translocations&#39;</span><span class="p">,</span>   
        <span class="p">}</span>

        <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">mapping</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">subtype</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">value</span>
    
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;ELN 2022 Diagnosis&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">categorize_ELN2022</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">add_clinical_outcome_cutoff</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">e</span><span class="p">,</span> <span class="n">t</span><span class="p">,</span> <span class="n">cutoff_years</span><span class="o">=</span><span class="mi">5</span><span class="p">):</span>

        <span class="n">df_</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
        
        <span class="c1"># Create &#39;Vital Status at {cutoff_years}y&#39; column</span>
        <span class="n">conditions</span> <span class="o">=</span> <span class="p">[</span>
            <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">e</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">t</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">cutoff_years</span><span class="p">),</span>
            <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">e</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">t</span><span class="p">]</span> <span class="o">&gt;</span> <span class="n">cutoff_years</span><span class="p">),</span>
            <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">e</span><span class="p">]</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">t</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">cutoff_years</span><span class="p">),</span>
            <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">e</span><span class="p">]</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df_</span><span class="p">[</span><span class="n">t</span><span class="p">]</span> <span class="o">&gt;</span> <span class="n">cutoff_years</span><span class="p">)</span>
        <span class="p">]</span>
        <span class="n">choices</span> <span class="o">=</span> <span class="p">[</span><span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">]</span>
        
        <span class="n">df_</span><span class="p">[</span><span class="n">e</span> <span class="o">+</span> <span class="sa">f</span><span class="s1">&#39; at </span><span class="si">{</span><span class="n">cutoff_years</span><span class="si">}</span><span class="s1">y&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">select</span><span class="p">(</span><span class="n">conditions</span><span class="p">,</span> <span class="n">choices</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s1">&#39;&#39;</span><span class="p">)</span>
        
        <span class="c1"># Create &#39;os.time at {cutoff_years}y&#39; column</span>
        <span class="n">df_</span><span class="p">[</span><span class="n">t</span> <span class="o">+</span> <span class="sa">f</span><span class="s1">&#39; at </span><span class="si">{</span><span class="n">cutoff_years</span><span class="si">}</span><span class="s1">y&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">where</span><span class="p">(</span>
            <span class="n">df_</span><span class="p">[</span><span class="n">t</span><span class="p">]</span> <span class="o">&gt;</span> <span class="n">cutoff_years</span><span class="p">,</span> <span class="n">cutoff_years</span><span class="p">,</span> <span class="n">df_</span><span class="p">[</span><span class="n">t</span><span class="p">])</span>
        

        <span class="k">return</span> <span class="n">df_</span>
    
    <span class="c1"># Apply the function to the DataFrame</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">add_clinical_outcome_cutoff</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">e</span><span class="o">=</span><span class="s1">&#39;os.evnt&#39;</span><span class="p">,</span> <span class="n">t</span><span class="o">=</span><span class="s1">&#39;os.time&#39;</span><span class="p">,</span> <span class="n">cutoff_years</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">add_clinical_outcome_cutoff</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">e</span><span class="o">=</span><span class="s1">&#39;efs.evnt&#39;</span><span class="p">,</span> <span class="n">t</span><span class="o">=</span><span class="s1">&#39;efs.time&#39;</span><span class="p">,</span> <span class="n">cutoff_years</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>

    <span class="n">df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;Vital Status at 5y&#39;</span><span class="p">]</span> <span class="o">=</span>  <span class="n">df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;os.evnt at 5y&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">({</span><span class="s1">&#39;0&#39;</span><span class="p">:</span><span class="s1">&#39;Alive&#39;</span><span class="p">,</span> <span class="s1">&#39;1&#39;</span> <span class="p">:</span><span class="s1">&#39;Dead&#39;</span><span class="p">})</span>
    <span class="n">df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;Event-Free Survival at 5y&#39;</span><span class="p">]</span> <span class="o">=</span>  <span class="n">df</span><span class="p">[</span><span class="sa">f</span><span class="s1">&#39;efs.evnt at 5y&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">({</span><span class="s1">&#39;0&#39;</span><span class="p">:</span><span class="s1">&#39;Yes&#39;</span><span class="p">,</span> <span class="s1">&#39;1&#39;</span> <span class="p">:</span><span class="s1">&#39;No&#39;</span><span class="p">})</span>

    <span class="k">return</span> <span class="n">df</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="id1">
<h3>Execution<a class="headerlink" href="#id1" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_labels4</span> <span class="o">=</span> <span class="n">process_df_labels</span><span class="p">(</span><span class="n">df_labels3</span><span class="p">)</span>

<span class="n">df_labels4</span><span class="p">[</span><span class="s1">&#39;Hematopoietic Entity&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Hematopoietic Entity
Acute myeloid leukemia (AML)                  1221
Acute lymphoblastic leukemia (ALL)             700
Otherwise-Normal (Control)                     251
Myelodysplastic syndrome (MDS or MDS-like)     223
Mixed phenotype acute leukemia (MPAL)           48
Acute promyelocytic leukemia (APL)              31
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="apply-classification-to-validation-dataset">
<h2>Apply classification to validation dataset<a class="headerlink" href="#apply-classification-to-validation-dataset" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import functions to clean up clinical data</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;../&#39;</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">utils.clinical_data_cleanup</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Call functions to merge, index and clean clinical data files</span>
<span class="n">labels_aml02</span>         <span class="o">=</span> <span class="n">clean_aml02</span>       <span class="p">(</span><span class="n">merge_index_aml02</span><span class="p">())</span>
<span class="n">labels_aml08</span>         <span class="o">=</span> <span class="n">clean_aml08</span>       <span class="p">(</span><span class="n">merge_index_aml08</span><span class="p">())</span>

<span class="c1"># Combine all clinical data labels into one dataframe</span>
<span class="n">labels_validation</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">labels_aml02</span><span class="p">,</span> <span class="n">labels_aml08</span><span class="p">]</span> <span class="p">,</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;outer&#39;</span><span class="p">)</span>

<span class="c1"># Redefine output path (for troubleshooting purposes in case only this cell is run)</span>
<span class="n">mount</span> <span class="o">=</span> <span class="s1">&#39;/mnt/e/ALMA/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Intermediate_Files/&#39;</span>

<span class="c1"># Read df_test</span>
<span class="n">df_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;201samples_331556cpgs_nobatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span>

<span class="c1"># Add columns for later analyses</span>
<span class="n">labels_validation</span><span class="p">[</span><span class="s1">&#39;Train-Test&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;Test Sample&#39;</span>
<span class="n">labels_validation</span><span class="p">[</span><span class="s1">&#39;Batch&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;Multicenter AML02,08 Trials&quot;</span>
<span class="n">labels_validation</span><span class="p">[</span><span class="s1">&#39;Risk Group AAML1831&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>
<span class="n">labels_validation</span><span class="p">[</span><span class="s1">&#39;Risk_Group_AAML1831_int&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>

<span class="c1"># Remove samples that are not in the methyl dataset</span>
<span class="n">df_labels_validation</span> <span class="o">=</span> <span class="n">labels_validation</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">labels_validation</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_test</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="c1"># Reclassify samples according to WHO 2022</span>
<span class="n">df_labels_validation2</span> <span class="o">=</span> <span class="n">process_labels_who22</span><span class="p">(</span><span class="n">df_labels_validation</span><span class="p">)</span>

<span class="c1"># Add age categorization and main disease classification to the clinical data</span>
<span class="n">df_labels_validation3</span> <span class="o">=</span> <span class="n">process_df_labels</span><span class="p">(</span><span class="n">df_labels_validation2</span><span class="p">)</span>

<span class="c1"># Display the value counts of the WHO 2022 Diagnosis</span>
<span class="n">df_labels_validation3</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>WHO 2022 Diagnosis
NaN                                        97
AML with t(v;11q23); KMT2A-r               47
AML with t(8;21); RUNX1::RUNX1T1           29
AML with inv(16); t(16;16); CBFB::MYH11    23
AML with t(8;16); KAT6A::CREBBP             2
AML with mutated CEBPA                      2
AML with t(6;9); DEK::NUP214                1
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
<section id="apply-classification-to-test-dataset">
<h2>Apply classification to test dataset<a class="headerlink" href="#apply-classification-to-test-dataset" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Read test data</span>
<span class="n">labels_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">PATH</span> <span class="o">+</span> <span class="s1">&#39;UF_HemBank/uf_hembank_FM.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Read df_test</span>
<span class="n">df_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;17samples_331556cpgs_nanoporeseq_bvalues.pkl&#39;</span><span class="p">)</span>

<span class="c1"># Remove samples that are not in the methyl dataset</span>
<span class="n">df_labels_test</span> <span class="o">=</span> <span class="n">labels_test</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">labels_test</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_test</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="c1"># Reclassify samples according to WHO 2022</span>
<span class="n">df_labels_test2</span> <span class="o">=</span> <span class="n">process_labels_who22</span><span class="p">(</span><span class="n">df_labels_test</span><span class="p">)</span>

<span class="c1"># Add age categorization and main disease classification to the clinical data</span>
<span class="n">df_labels_test3</span> <span class="o">=</span> <span class="n">process_df_labels</span><span class="p">(</span><span class="n">df_labels_test2</span><span class="p">)</span>

<span class="n">df_labels_test3</span><span class="p">[</span><span class="s1">&#39;WHO 2022 Diagnosis&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>WHO 2022 Diagnosis
NaN                             14
AML with t(v;11q23); KMT2A-r     2
T-ALL NOS                        1
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
<section id="save-clinical-datasets">
<h2>Save clinical datasets<a class="headerlink" href="#save-clinical-datasets" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Concatenate clinical data files and keep overlapping columns</span>
<span class="n">common_columns</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df_labels4</span><span class="o">.</span><span class="n">columns</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df_labels_validation3</span><span class="o">.</span><span class="n">columns</span><span class="p">]</span>

<span class="c1"># Apply `common_columns` to all clinical datasets</span>
<span class="n">df_labels5</span> <span class="o">=</span> <span class="n">df_labels4</span><span class="p">[</span><span class="n">common_columns</span><span class="p">]</span>
<span class="n">df_labels_validation4</span> <span class="o">=</span> <span class="n">df_labels_validation3</span><span class="p">[</span><span class="n">common_columns</span><span class="p">]</span>
<span class="n">df_labels_test4</span> <span class="o">=</span> <span class="n">df_labels_test3</span><span class="p">[</span><span class="n">common_columns</span><span class="p">]</span>

<span class="c1"># Concatenate clinical data files and keep overlapping columns</span>
<span class="n">clinical_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">df_labels5</span><span class="p">,</span> <span class="n">df_labels_validation4</span><span class="p">,</span> <span class="n">df_labels_test4</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Save the merged clinical data</span>
<span class="n">clinical_data</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;clinical_data.xlsx&#39;</span><span class="p">)</span>

<span class="c1"># Save individual datasets (for troubleshooting purposes)</span>

<span class="c1"># labels_aml05.to_excel(output_path + &#39;aml05_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_beataml.to_excel(output_path + &#39;beataml_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_0531.to_excel(output_path + &#39;0531_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_amltcga.to_excel(output_path + &#39;amltcga_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_1031.to_excel(output_path + &#39;1031_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_nordic_all.to_excel(output_path + &#39;nordic_all_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_mds_taml.to_excel(output_path + &#39;mds_taml_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_all_graal.to_excel(output_path + &#39;all_graal_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_target_all.to_excel(output_path + &#39;target_all_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_aml02.to_excel(output_path + &#39;aml02_clinical_data.xlsx&#39;)</span>
<span class="c1"># labels_aml08.to_excel(output_path + &#39;aml08_clinical_data.xlsx&#39;)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="watermark">
<h2>Watermark<a class="headerlink" href="#watermark" title="Link to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Last updated: 2024-09-14

Python implementation: CPython
Python version       : 3.8.18
IPython version      : 8.12.3

numpy   : 1.24.4
pandas  : 2.0.3
pacmap  : 0.7.0
sklearn : 1.2.2
lightgbm: 3.3.5

Compiler    : GCC 11.4.0
OS          : Linux
Release     : 5.15.133.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 32
Architecture: 64bit

Git repo: git@github.com:f-marchi/ALMA.git
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./source"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="3_test.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Test Dataset</p>
      </div>
    </a>
    <a class="right-next"
       href="5_signature.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">37-CpG Signature</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-index-merge-and-clean-discovery-clinical-data">Load, index, merge, and clean discovery clinical data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#reclassification-strategy-dx-who-2022">Reclassification strategy: Dx WHO 2022</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#reclassification-strategy-risk-group-cog-aaml1831">Reclassification strategy: Risk Group COG AAML1831</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#main-reclassification-functions-applied-to-all-samples">Main reclassification functions (applied to all samples)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#execution">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#categorize-eln-2022-hematopoietic-entity-age-trisomy-8-and-5y-os-cutoff">Categorize ELN 2022, hematopoietic entity, age, trisomy 8, and 5y OS cutoff</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#functions">Functions</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Execution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-classification-to-validation-dataset">Apply classification to validation dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-classification-to-test-dataset">Apply classification to test dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#save-clinical-datasets">Save clinical datasets</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Marchi et al. @ Lamba Lab @ University of Florida.
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>