
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>38-CpG AML Signature &#8212; Acute Leukemia Methylome Atlas Study</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css?v=564be945" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=36754332"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'source/5_signature';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Classifiers" href="5_classifiers.html" />
    <link rel="prev" title="Clinical Data" href="4_clinical_data.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../overview.html">
  
  
  
  
  
  
    <p class="title logo__title">Acute Leukemia Methylome Atlas Study</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../overview.html">
                    Overview
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Datasets</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="1_discovery.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_test.html">Test Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_nanopore_test.html">Nanopore Test Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="4_clinical_data.html">Clinical Data</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Models</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">38-CpG AML Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_classifiers.html">Classifiers</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Figures</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="6_plots.html">Plots</a></li>
<li class="toctree-l1"><a class="reference external" href="https://f-marchi.github.io/UF-LambaLab-ALMA-app/">Run Interactive App</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/source/5_signature.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>38-CpG AML Signature</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-samples-only">Select diagnostic samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas">Run CoxPH-EWAS</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-cpgs-based-on-suggested-significance-p-values-1-10e-5">Select CpGs based on suggested-significance p-values (1*10e-5)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#train-coxph-lasso-model-with-the-selected-cpgs">Train CoxPH+Lasso model with the selected CpGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-discovery-dataset-and-set-high-low-cutoff">Apply signature to discovery dataset and set high/low cutoff</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-test-dataset">Apply signature to test dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-nanopore-test-dataset">Apply signature to nanopore test dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#merge-and-save-files">Merge and save files</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="cpg-aml-signature">
<h1>38-CpG AML Signature<a class="headerlink" href="#cpg-aml-signature" title="Link to this heading">#</a></h1>
<section id="where-the-data-at">
<h2>Where the data at?<a class="headerlink" href="#where-the-data-at" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;../&#39;</span><span class="p">)</span>

<span class="n">mount</span> <span class="o">=</span> <span class="s1">&#39;/mnt/e/ALMA/&#39;</span>
<span class="n">input_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Intermediate_Files/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Processed_Files/&#39;</span>
<span class="n">zhou2016_probes_path</span> <span class="o">=</span>  <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="load-data">
<h2>Load data<a class="headerlink" href="#load-data" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read df_discovery and df_test</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;3314samples_331556cpgs_nobatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="c1"># Read clinical data</span>
<span class="n">clinical_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;clinical_data.xlsx&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="remove-samples-based-on-certain-clinical-features">
<h2>Remove samples based on certain clinical features<a class="headerlink" href="#remove-samples-based-on-certain-clinical-features" title="Link to this heading">#</a></h2>
<section id="select-samples-from-aaml1031-0531-and-03p1-clinical-trials">
<h3>Select samples from AAML1031, 0531, and 03P1 clinical trials<a class="headerlink" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">clinical_data2</span> <span class="o">=</span> <span class="n">clinical_data</span><span class="p">[</span><span class="n">clinical_data</span><span class="p">[</span><span class="s1">&#39;Clinical Trial&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;AAML0531&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML1031&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML03P1&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">clinical_data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">clinical_data2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">clinical_data2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2240 samples were removed. 1287 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="select-diagnostic-samples-only">
<h3>Select diagnostic samples only<a class="headerlink" href="#select-diagnostic-samples-only" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">clinical_data3</span> <span class="o">=</span> <span class="n">clinical_data2</span><span class="p">[</span><span class="n">clinical_data2</span><span class="p">[</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Diagnosis&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Bone Marrow&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Peripheral Blood&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">clinical_data2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">clinical_data3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">clinical_data3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>332 samples were removed. 955 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-duplicate-samples">
<h3>Remove duplicate samples<a class="headerlink" href="#remove-duplicate-samples" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">clinical_data4</span> <span class="o">=</span> <span class="n">clinical_data3</span><span class="p">[</span><span class="o">~</span><span class="n">clinical_data3</span><span class="p">[</span><span class="s1">&#39;Patient_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="s1">&#39;last&#39;</span><span class="p">)]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">clinical_data3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">clinical_data4</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">clinical_data4</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>9 samples were removed. 946 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="match-samples-in-clinical-data-to-samples-in-methylation-data">
<h3>Match samples in clinical data to samples in methylation data<a class="headerlink" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Match samples in clinical data to samples in methylation data</span>
<span class="n">df2</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">clinical_data4</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">:]</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Samples in clinical data matched to samples in methylation data.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Samples in clinical data matched to samples in methylation data.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="transform-to-m-values">
<h2>Transform to M-values<a class="headerlink" href="#transform-to-m-values" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="k">def</span> <span class="nf">beta2m</span><span class="p">(</span><span class="n">val</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Transfrom beta-values into m-values&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">math</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">val</span><span class="o">/</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">val</span><span class="p">))</span>

<span class="n">df3</span> <span class="o">=</span> <span class="n">df2</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="perform-batch-correction">
<h2>Perform batch correction<a class="headerlink" href="#perform-batch-correction" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>pyCombat</strong> is a Python tool for batch effects correction in high-throughput molecular data using empirical Bayes methods</p></li>
<li><p><strong>Github</strong>: <a class="reference external" href="https://epigenelabs.github.io/pyComBat/">https://epigenelabs.github.io/pyComBat/</a></p></li>
<li><p><strong>Implementation Preprint</strong>: <a class="reference external" href="https://doi.org/10.1101/2020.03.17.995431">bioRxiv</a></p></li>
<li><p><strong>Original Paper</strong>: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/16632515/">Adjusting batch effects in microarray expression data using empirical Bayes methods</a></p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">combat.pycombat</span> <span class="kn">import</span> <span class="n">pycombat</span>

<span class="c1"># Correct batch effects in the training dataset</span>
<span class="n">df4</span> <span class="o">=</span> <span class="n">pycombat</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">df3</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">batch</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully corrected batch effects in the training dataset.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Found 3 batches.
Adjusting for 0 covariate(s) or covariate level(s).
Standardizing Data across genes.
Fitting L/S model and finding priors.
Finding parametric adjustments.
Adjusting the Data
Succesfully corrected batch effects in the training dataset.
</pre></div>
</div>
</div>
</div>
<section id="evaluate-batch-effects">
<h3>Evaluate batch effects<a class="headerlink" href="#evaluate-batch-effects" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">sklearn.preprocessing</span> <span class="kn">import</span> <span class="n">StandardScaler</span>
<span class="kn">from</span> <span class="nn">sklearn.decomposition</span> <span class="kn">import</span> <span class="n">PCA</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;white&#39;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Standardizes the data.&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">StandardScaler</span><span class="p">()</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">perform_pca</span><span class="p">(</span><span class="n">std_data</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Performs PCA on standardized data.&#39;&#39;&#39;</span>
    <span class="n">pca</span> <span class="o">=</span> <span class="n">PCA</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="mi">42</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">pca</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">std_data</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">create_principal_df</span><span class="p">(</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">df4</span> <span class="p">,</span> <span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Creates a DataFrame with the PCA data.&#39;&#39;&#39;</span>
    <span class="n">principal_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="s1">&#39;PC2&#39;</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span><span class="n">df4</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span>
    <span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;hue&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">principal_df</span>

<span class="k">def</span> <span class="nf">filter_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">principal_df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Filters the DataFrame by removing outliers.&#39;&#39;&#39;</span>
    <span class="n">outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">threshold</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">outliers</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df4</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_batchcorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df4</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">df</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_uncorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df3</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">df</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Plots PCA data.&#39;&#39;&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">,</span> <span class="n">legend</span> <span class="ow">in</span> <span class="p">[(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA after batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="s1">&#39;full&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="n">non_outliers</span><span class="p">,</span> <span class="n">ax1</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA before batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">non_outliers</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="kc">False</span><span class="p">)]:</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PC2&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;hue&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="n">legend</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;PC 1&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;PC 2&#39;</span><span class="p">)</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.05</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">plot_data</span><span class="p">(</span><span class="n">pcs_batchcorrected</span><span class="p">,</span> <span class="n">pcs_uncorrected</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/de68ee61b22f776c1453d2399d80db129afb2d0c68fe0fbbdf749668a02ab76d.png" src="../_images/de68ee61b22f776c1453d2399d80db129afb2d0c68fe0fbbdf749668a02ab76d.png" />
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ol class="arabic simple">
<li><p>GDC_TARGET-AML contains AAML0531 and 03P1 samples from methylation array 450k.</p></li>
<li><p>GSE124413 contains AAML0531 and 03P1 samples from methylation array EPIC.</p></li>
<li><p>GSE190931 contains AAML1031 samples from methylation array EPIC.</p></li>
</ol>
</div>
</section>
</section>
<section id="run-coxph-ewas">
<h2>Run CoxPH-EWAS<a class="headerlink" href="#run-coxph-ewas" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">utils.methylize_custom_code.diff_meth_pos_custom</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># def impute_by_mean(df):</span>
<span class="c1">#     return df.fillna(df.mean())</span>

<span class="c1"># covar_os = round(impute_by_mean(clinical_data4[[&#39;os.evnt at 5y&#39;,&#39;os.time at 5y&#39;,&#39;Risk_Group_AAML1831_int&#39;]]),3)</span>

<span class="c1"># ewas_os_cog_adjusted = diff_meth_pos(meth_data = df4,</span>
<span class="c1">#                                     duration = covar_os[&#39;os.time at 5y&#39;],</span>
<span class="c1">#                                     event_observed = covar_os[&#39;os.evnt at 5y&#39;],</span>
<span class="c1">#                                     pheno_data = covar_os[&#39;os.time at 5y&#39;], # not used</span>
<span class="c1">#                                     covariates = covar_os[&#39;Risk_Group_AAML1831_int&#39;],</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = False,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted&#39;)</span>

<span class="n">ewas_os_cog_adjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_os_cog_adjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; discovery cohort - OS at 5y adjusted for AAML1831 risk group&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/7b77f777606c2831910503fffb1864c8d94b228c8afd39c0cb4dd890b5a11755.png" src="../_images/7b77f777606c2831910503fffb1864c8d94b228c8afd39c0cb4dd890b5a11755.png" />
</div>
</div>
</section>
<section id="select-cpgs-based-on-suggested-significance-p-values-1-10e-5">
<h2>Select CpGs based on suggested-significance p-values (1*10e-5)<a class="headerlink" href="#select-cpgs-based-on-suggested-significance-p-values-1-10e-5" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Select only rows where `PValue`&lt; 1*10e-5</span>
<span class="n">ewas_os_cog_adjusted_sig</span> <span class="o">=</span> <span class="n">ewas_os_cog_adjusted</span><span class="p">[</span><span class="n">ewas_os_cog_adjusted</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="o">*</span><span class="mf">10e-5</span><span class="p">]</span>

<span class="c1"># Define a function to handle repetitive tasks</span>
<span class="k">def</span> <span class="nf">process_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">other_df</span><span class="p">,</span> <span class="n">column_drop</span><span class="p">):</span>
    <span class="c1"># Join and sort values</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">other_df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">column_drop</span><span class="p">),</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span>

    <span class="c1"># Handle columns with multiple genes separated by `;`</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneClasses&#39;</span><span class="p">]:</span>
        <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">))))</span>

    <span class="k">return</span> <span class="n">df</span>

<span class="c1"># Load manifest</span>
<span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">zhou2016_probes_path</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span>
                                                     <span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">top_os</span> <span class="o">=</span> <span class="n">process_df</span><span class="p">(</span><span class="n">zhou2016_probes</span><span class="p">,</span> <span class="n">ewas_os_cog_adjusted_sig</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;chromosome&#39;</span><span class="p">,</span> <span class="s1">&#39;MAPINFO&#39;</span><span class="p">])</span>

<span class="c1"># Now include the date in the filename and save to excel</span>
<span class="c1"># import datetime</span>
<span class="c1"># top_os.to_excel(output_path + &#39;ewas_dmr/ewas_stats_&#39; + str(datetime.date.today()) + &#39;.xlsx&#39;)</span>

<span class="c1"># Select only significant probes</span>
<span class="n">ewas_top_cpgs_os</span> <span class="o">=</span> <span class="n">df4</span><span class="p">[</span><span class="n">top_os</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_list</span><span class="p">()]</span>

<span class="c1"># Save x to pickle in output_path</span>
<span class="n">ewas_top_cpgs_os</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_cpgs_os_&#39;</span><span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">date</span><span class="o">.</span><span class="n">today</span><span class="p">())</span><span class="o">+</span> <span class="s1">&#39;.csv&#39;</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">itables</span> <span class="kn">import</span> <span class="n">show</span>

<span class="n">show</span><span class="p">(</span><span class="n">top_os</span><span class="p">,</span> <span class="n">classes</span><span class="o">=</span><span class="s2">&quot;display nowrap compact&quot;</span><span class="p">,</span>  <span class="n">style</span><span class="o">=</span><span class="s2">&quot;width:1500px&quot;</span><span class="p">,</span> <span class="n">autoWidth</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><table id="itables_eee8f2aa_f73a_4a19_9424_f0417921939a" class="display nowrap compact" data-quarto-disable-processing="true" style="width:1500px">
<thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>CGIPosition</th>
      <th>Coefficient</th>
      <th>StandardError</th>
      <th>HazardsRatio</th>
      <th>95%CI_lower</th>
      <th>95%CI_upper</th>
      <th>ZScore</th>
      <th>PValue</th>
      <th>FDR_QValue</th>
      <th>minuslog10value</th>
    </tr>
  </thead><tbody><tr>
<td style="vertical-align:middle; text-align:left">
<div style="float:left; margin-right: 10px;">
<a href=https://mwouts.github.io/itables/><svg class="main-svg" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
width="64" viewBox="0 0 500 400" style="font-family: 'Droid Sans', sans-serif;">
    <g style="fill:#d9d7fc">
        <path d="M100,400H500V357H100Z" />
        <path d="M100,300H400V257H100Z" />
        <path d="M0,200H400V157H0Z" />
        <path d="M100,100H500V57H100Z" />
        <path d="M100,350H500V307H100Z" />
        <path d="M100,250H400V207H100Z" />
        <path d="M0,150H400V107H0Z" />
        <path d="M100,50H500V7H100Z" />
    </g>
    <g style="fill:#1a1366;stroke:#1a1366;">
   <rect x="100" y="7" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="5s"
      repeatCount="indefinite" />
      <animate
      attributeName="x"
      values="100;100;500"
      dur="5s"
      repeatCount="indefinite" />
  </rect>
        <rect x="0" y="107" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="3.5s"
      repeatCount="indefinite" />
    <animate
      attributeName="x"
      values="0;0;400"
      dur="3.5s"
      repeatCount="indefinite" />
  </rect>
        <rect x="100" y="207" width="300" height="43">
    <animate
      attributeName="width"
      values="0;300;0"
      dur="3s"
      repeatCount="indefinite" />
    <animate
      attributeName="x"
      values="100;100;400"
      dur="3s"
      repeatCount="indefinite" />
  </rect>
        <rect x="100" y="307" width="400" height="43">
    <animate
      attributeName="width"
      values="0;400;0"
      dur="4s"
      repeatCount="indefinite" />
      <animate
      attributeName="x"
      values="100;100;500"
      dur="4s"
      repeatCount="indefinite" />
  </rect>
        <g style="fill:transparent;stroke-width:8; stroke-linejoin:round" rx="5">
            <g transform="translate(45 50) rotate(-45)">
                <circle r="33" cx="0" cy="0" />
                <rect x="-8" y="32" width="16" height="30" />
            </g>

            <g transform="translate(450 152)">
                <polyline points="-15,-20 -35,-20 -35,40 25,40 25,20" />
                <rect x="-15" y="-40" width="60" height="60" />
            </g>

            <g transform="translate(50 352)">
                <polygon points="-35,-5 0,-40 35,-5" />
                <polygon points="-35,10 0,45 35,10" />
            </g>

            <g transform="translate(75 250)">
                <polyline points="-30,30 -60,0 -30,-30" />
                <polyline points="0,30 -30,0 0,-30" />
            </g>

            <g transform="translate(425 250) rotate(180)">
                <polyline points="-30,30 -60,0 -30,-30" />
                <polyline points="0,30 -30,0 0,-30" />
            </g>
        </g>
    </g>
</svg>
</a>
</div>
<div>
Loading ITables v2.1.5 from the internet...
(need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td>
</div>
</tr></tbody>

</table>
<link href="https://www.unpkg.com/dt_for_itables@2.0.12/dt_bundle.css" rel="stylesheet">
<script type="module">
    import {DataTable, jQuery as $} from 'https://www.unpkg.com/dt_for_itables@2.0.12/dt_bundle.js';

    document.querySelectorAll("#itables_eee8f2aa_f73a_4a19_9424_f0417921939a:not(.dataTable)").forEach(table => {
        // Define the table data
        const data = [["cg00920892", "chr8", 37874705, 37874707, "RAB11FIP1", "protein_coding", ".", 0.217791, 0.039091, 1.243327, 1.151626, 1.34233, 5.571441, 2.526415e-08, 0.008376, 7.597495], ["cg08900363", "chr7", 597816, 597818, "PRKAR1B", "protein_coding", ".", -0.291705, 0.054727, 0.746989, 0.671013, 0.831568, -5.330214, 9.809699e-08, 0.010419, 7.008344], ["cg01543603", "chr16", 89403132, 89403134, "RP1-168P16.2;ANKRD11", "protein_coding;sense_intronic", ".", 0.176991, 0.033708, 1.19362, 1.117311, 1.275141, 5.250731, 1.514971e-07, 0.010419, 6.819596], ["cg08404702", "chr16", 2083386, 2083388, "TSC2", "protein_coding", "N_Shore", 0.126939, 0.024193, 1.135347, 1.082768, 1.19048, 5.246847, 1.547245e-07, 0.010419, 6.810441], ["cg01052291", "chr6", 132715433, 132715435, "VNN1", "protein_coding", ".", -0.271835, 0.051837, 0.76198, 0.688366, 0.843466, -5.244005, 1.57128e-07, 0.010419, 6.803746], ["cg08329113", "chr16", 70737238, 70737240, "VAC14", "protein_coding", ".", 0.189572, 0.036594, 1.208732, 1.125074, 1.298611, 5.180385, 2.214278e-07, 0.012236, 6.654768], ["cg20018344", "chr20", 3223491, 3223493, "ITPA", "protein_coding", ".", 0.204926, 0.040286, 1.227434, 1.134244, 1.328281, 5.086743, 3.64264e-07, 0.016829, 6.438584], ["cg06339275", "chr3", 62948996, 62948998, "LINC00698", "lincRNA", ".", -0.175624, 0.034666, 0.838933, 0.783825, 0.897916, -5.066099, 4.060507e-07, 0.016829, 6.39142], ["cg00192882", "chr17", 19387806, 19387808, "MFAP4", "protein_coding", ".", 0.244635, 0.049449, 1.277155, 1.159185, 1.407131, 4.947237, 7.527415e-07, 0.024442, 6.123354], ["cg00945409", "chr10", 78977907, 78977909, "ZMIZ1-AS1", "antisense", "S_Shelf", -0.29687, 0.060008, 0.743141, 0.660681, 0.835891, -4.947178, 7.529715e-07, 0.024442, 6.123221], ["cg06748884", "chr16", 11138900, 11138902, "CLEC16A", "protein_coding", ".", -0.254821, 0.051659, 0.775055, 0.700423, 0.85764, -4.93272, 8.109238e-07, 0.024442, 6.09102], ["cg07660635", "chr16", 85902540, 85902542, "IRF8", "protein_coding", "S_Shelf", -0.156227, 0.032264, 0.855365, 0.802949, 0.911203, -4.84208, 1.284873e-06, 0.031612, 5.89114], ["cg16424082", "chr12", 50929956, 50929958, "METTL7A", "protein_coding", ".", -0.362921, 0.074988, 0.695641, 0.600559, 0.805778, -4.839732, 1.300143e-06, 0.031612, 5.886009], ["cg22760287", "chr11", 107705174, 107705176, ".", ".", ".", -0.182793, 0.03786, 0.83294, 0.773369, 0.8971, -4.82809, 1.378486e-06, 0.031612, 5.860598], ["cg04839706", "chr16", 46886162, 46886164, "GPT2", "protein_coding", "S_Shore", -0.187613, 0.038958, 0.828935, 0.767998, 0.894708, -4.815819, 1.465972e-06, 0.031612, 5.833874], ["cg18062721", "chr3", 11601952, 11601954, "VGLL4", "protein_coding", ".", 0.114374, 0.023789, 1.121171, 1.070096, 1.174684, 4.807864, 1.525517e-06, 0.031612, 5.816583], ["cg05430997", "chr13", 109744841, 109744843, ".", ".", ".", 0.137319, 0.028692, 1.147194, 1.084463, 1.213555, 4.786035, 1.701084e-06, 0.033177, 5.769274], ["cg17099306", "chr13", 49549508, 49549510, "RCBTB1", "protein_coding", ".", 0.205127, 0.043099, 1.227681, 1.128234, 1.335894, 4.759396, 1.941732e-06, 0.035766, 5.711811], ["cg10089193", "chr16", 1090938, 1090940, "C1QTNF8", "protein_coding", "S_Shore", 0.168562, 0.035591, 1.183601, 1.103852, 1.269113, 4.736133, 2.178346e-06, 0.037855, 5.661873], ["cg26361535", "chr8", 143494433, 143494435, "ZC3H3", "protein_coding", ".", 0.164587, 0.034889, 1.178907, 1.100986, 1.262342, 4.717434, 2.388375e-06, 0.037855, 5.621897], ["cg13213755", "chr15", 73634112, 73634114, "NPTN", "protein_coding", "S_Shore", -0.290423, 0.061643, 0.747947, 0.662827, 0.843997, -4.711399, 2.460224e-06, 0.037855, 5.609025], ["cg14928764", "chr20", 23083970, 23083972, "CD93", "protein_coding", ".", -0.170578, 0.036298, 0.843178, 0.785275, 0.90535, -4.69931, 2.61042e-06, 0.037855, 5.58329], ["cg27049094", "chr2", 74840588, 74840590, "HK2", "protein_coding", ".", 0.130594, 0.027825, 1.139505, 1.079026, 1.203374, 4.693483, 2.685919e-06, 0.037855, 5.570907], ["cg19432033", "chr17", 7613901, 7613903, "FXR2;SHBG", "protein_coding", "N_Shore", -0.199624, 0.042569, 0.819039, 0.753476, 0.890306, -4.689393, 2.740167e-06, 0.037855, 5.562223], ["cg03801691", "chr3", 11558341, 11558343, "VGLL4", "protein_coding", ".", 0.208867, 0.044919, 1.232281, 1.128431, 1.345689, 4.649915, 3.320715e-06, 0.04404, 5.478768], ["cg02312559", "chr3", 134254413, 134254415, ".", ".", "S_Shelf", -0.241455, 0.052079, 0.785484, 0.709264, 0.869895, -4.636351, 3.546135e-06, 0.044406, 5.450245], ["cg20310759", "chr1", 87152629, 87152631, "LINC01140;RP5-1052I5.2", "lincRNA;protein_coding", "S_Shore", -0.185792, 0.040108, 0.830446, 0.767665, 0.898362, -4.632303, 3.6162e-06, 0.044406, 5.441748], ["cg04220636", "chr16", 88967679, 88967681, "CBFA2T3", "protein_coding", "Island", 0.158674, 0.034357, 1.171956, 1.095637, 1.253592, 4.618405, 3.86701e-06, 0.04579, 5.412625], ["cg22410826", "chr13", 109744594, 109744596, ".", ".", ".", 0.15431, 0.033486, 1.166853, 1.09273, 1.246004, 4.608221, 4.061281e-06, 0.046432, 5.391337], ["cg13984533", "chr1", 37796839, 37796841, "MANEAL", "protein_coding", "S_Shelf", -0.181608, 0.039799, 0.833928, 0.77135, 0.901584, -4.563078, 5.040907e-06, 0.055711, 5.297491], ["cg11818438", "chr8", 118904290, 118904292, ".", ".", ".", -0.232889, 0.051122, 0.792242, 0.716708, 0.875735, -4.555545, 5.224983e-06, 0.055883, 5.281915], ["cg00206490", "chr16", 89403320, 89403322, "RP1-168P16.2;ANKRD11", "protein_coding;sense_intronic", ".", 0.134541, 0.029635, 1.144011, 1.079455, 1.212428, 4.539864, 5.629043e-06, 0.058036, 5.249565], ["cg04663203", "chr17", 47851211, 47851213, "SP6", "protein_coding", "Island", 0.219234, 0.048464, 1.245123, 1.132296, 1.369193, 4.523681, 6.077318e-06, 0.058036, 5.216288], ["cg18072388", "chr19", 10323084, 10323086, "RAVER1", "protein_coding", "N_Shore", 0.182522, 0.040359, 1.20024, 1.108956, 1.299039, 4.522411, 6.11392e-06, 0.058036, 5.21368], ["cg06585893", "chr1", 230747805, 230747807, "RP11-99J16__A.2;CAPN9", "antisense;protein_coding", ".", 0.253343, 0.056025, 1.288325, 1.154349, 1.437851, 4.521979, 6.12642e-06, 0.058036, 5.212793], ["cg20703895", "chr1", 157438011, 157438013, ".", ".", ".", -0.376435, 0.083454, 0.686304, 0.582747, 0.808263, -4.510695, 6.461557e-06, 0.05912, 5.189663], ["cg21637653", "chr17", 17213452, 17213454, "FLCN;MPRIP;RP11-45M22.4", "protein_coding", ".", -0.400461, 0.088954, 0.670011, 0.562813, 0.797627, -4.501884, 6.735368e-06, 0.05912, 5.171639], ["cg10009497", "chr7", 1055779, 1055781, "RP11-449P15.1;C7orf50;GPR146", "antisense;protein_coding", "N_Shore", 0.146034, 0.032448, 1.157236, 1.085931, 1.233222, 4.500612, 6.775811e-06, 0.05912, 5.169039], ["cg04205653", "chr21", 44805974, 44805976, "SUMO3", "protein_coding", "S_Shelf", -0.430236, 0.095956, 0.650355, 0.538857, 0.784925, -4.483705, 7.335804e-06, 0.060648, 5.134552], ["cg21170304", "chr9", 135987756, 135987758, ".", ".", "N_Shelf", -0.30148, 0.067243, 0.739723, 0.648383, 0.84393, -4.483423, 7.345512e-06, 0.060648, 5.133978], ["cg12885484", "chr7", 6077236, 6077238, ".", ".", "N_Shelf", -0.25898, 0.057821, 0.771839, 0.689143, 0.864458, -4.478992, 7.49965e-06, 0.060648, 5.124959], ["cg05480169", "chr10", 133253903, 133253905, ".", ".", ".", -0.148088, 0.033127, 0.862356, 0.808144, 0.920204, -4.470319, 7.810299e-06, 0.061656, 5.107332], ["cg21739530", "chr4", 173264916, 173264918, "GALNT7", "protein_coding", ".", 0.163044, 0.036534, 1.177088, 1.09575, 1.264464, 4.462842, 8.087976e-06, 0.062363, 5.09216], ["cg16179125", "chr20", 59007065, 59007067, "CTSZ", "protein_coding", "Island", 0.177477, 0.039873, 1.194201, 1.104427, 1.291271, 4.451037, 8.545642e-06, 0.064395, 5.068255], ["cg27064178", "chr22", 49784571, 49784573, "BRD1", "protein_coding", "Island", 0.156693, 0.035346, 1.169637, 1.091352, 1.253538, 4.433159, 9.286213e-06, 0.06842, 5.032161], ["cg10098541", "chr3", 123066802, 123066804, "PDIA5", "protein_coding", "N_Shore", -0.192805, 0.04366, 0.824643, 0.757011, 0.898317, -4.416015, 1.005371e-05, 0.071937, 4.997674], ["cg11668573", "chr4", 158788032, 158788034, "FNIP2", "protein_coding", ".", -0.466311, 0.105669, 0.627312, 0.509962, 0.771666, -4.412944, 1.019743e-05, 0.071937, 4.991509], ["cg19706516", "chr1", 151131392, 151131394, "RP11-68I18.10", "sense_overlapping", "Island", 0.190502, 0.043231, 1.209856, 1.111566, 1.316837, 4.406591, 1.050102e-05, 0.072535, 4.978768], ["cg18156551", "chr11", 64224444, 64224446, "TRPT1;NUDT22", "protein_coding", "N_Shore", -0.404332, 0.092016, 0.667422, 0.557284, 0.799328, -4.39415, 1.11207e-05, 0.072656, 4.953868], ["cg11064395", "chr6", 31407526, 31407528, "MICA;HCP5", "protein_coding;sense_overlapping", "S_Shelf", -0.51249, 0.116843, 0.599002, 0.4764, 0.753156, -4.386157, 1.153708e-05, 0.072656, 4.937904], ["cg10187365", "chr21", 33488648, 33488650, "DNAJC28", "protein_coding", "N_Shelf", -0.416829, 0.095047, 0.659134, 0.547103, 0.794105, -4.385496, 1.15722e-05, 0.072656, 4.936584], ["cg16429055", "chr17", 17051538, 17051540, "MPRIP", "protein_coding", ".", -0.286015, 0.065219, 0.751252, 0.661106, 0.853689, -4.385484, 1.15728e-05, 0.072656, 4.936562], ["cg10057841", "chr11", 130126348, 130126350, "APLP2", "protein_coding", "S_Shelf", -0.334702, 0.076334, 0.715551, 0.61612, 0.831029, -4.384706, 1.161427e-05, 0.072656, 4.935008], ["cg18437839", "chr1", 201476584, 201476586, ".", ".", ".", -0.250406, 0.057222, 0.778485, 0.695893, 0.870879, -4.376022, 1.208646e-05, 0.073745, 4.917701], ["cg04857395", "chr4", 3514909, 3514911, "LRPAP1", "protein_coding", "Island", 0.099386, 0.022735, 1.104492, 1.056358, 1.154821, 4.371549, 1.233682e-05, 0.073745, 4.908797], ["cg05102794", "chr16", 89101072, 89101074, "ACSF3", "protein_coding", "Island", 0.22047, 0.05046, 1.246662, 1.129267, 1.376261, 4.369158, 1.247267e-05, 0.073745, 4.904041], ["cg07143043", "chr10", 131970793, 131970795, "RP11-45A17.2;BNIP3", "antisense;protein_coding", ".", 0.188959, 0.043284, 1.207991, 1.109739, 1.314943, 4.365591, 1.267798e-05, 0.073745, 4.89695], ["cg20695297", "chr19", 3178845, 3178847, "S1PR4", "protein_coding", "Island", 0.194545, 0.044613, 1.214758, 1.113052, 1.325758, 4.360772, 1.296045e-05, 0.073963, 4.88738], ["cg17511968", "chr2", 218856441, 218856443, ".", ".", "N_Shelf", 0.147108, 0.033777, 1.158479, 1.084269, 1.237768, 4.355259, 1.329095e-05, 0.073963, 4.876444], ["cg11801959", "chr13", 49549476, 49549478, "RCBTB1", "protein_coding", ".", 0.179333, 0.041191, 1.196419, 1.103625, 1.297014, 4.353719, 1.338473e-05, 0.073963, 4.873391], ["cg21229337", "chr6", 36963766, 36963768, "PI16", "protein_coding", "S_Shore", 0.178894, 0.041205, 1.195894, 1.103109, 1.296482, 4.341519, 1.415013e-05, 0.07567, 4.84924], ["cg03596113", "chr1", 3213404, 3213406, "PRDM16", "protein_coding", ".", -0.241307, 0.055598, 0.7856, 0.704492, 0.876046, -4.340181, 1.423651e-05, 0.07567, 4.846597], ["cg12762600", "chr15", 89647922, 89647924, "KIF7", "protein_coding", "N_Shore", -0.252784, 0.058279, 0.776636, 0.692803, 0.870612, -4.33746, 1.441389e-05, 0.07567, 4.841219], ["cg00532502", "chr1", 246789970, 246789972, "LINC01341;KIF28P", "unitary_pseudogene;processed_transcript", "S_Shore", 0.257569, 0.059422, 1.293781, 1.151543, 1.453587, 4.334541, 1.460648e-05, 0.07567, 4.835454], ["cg09140274", "chr1", 220896850, 220896852, ".", ".", "S_Shore", -0.201704, 0.046899, 0.817337, 0.745557, 0.896027, -4.300861, 1.701358e-05, 0.083499, 4.769204], ["cg16958939", "chr6", 7224989, 7224991, "RREB1", "protein_coding", ".", 0.120737, 0.028083, 1.128328, 1.067901, 1.192173, 4.299324, 1.713196e-05, 0.083499, 4.766193], ["cg25292309", "chr2", 58762291, 58762293, "LINC01122", "lincRNA", ".", -0.230758, 0.053727, 0.793931, 0.71458, 0.882094, -4.295051, 1.746531e-05, 0.083499, 4.757824], ["cg01312394", "chr2", 46882240, 46882242, ".", ".", ".", -0.194051, 0.045181, 0.823616, 0.753819, 0.899876, -4.295011, 1.746845e-05, 0.083499, 4.757746], ["cg07536737", "chr2", 98013627, 98013629, ".", ".", ".", -0.125559, 0.029243, 0.882004, 0.832873, 0.934032, -4.293687, 1.757305e-05, 0.083499, 4.755153], ["cg00059652", "chr17", 69609356, 69609358, "LINC01483", "lincRNA", ".", -0.146816, 0.034211, 0.863453, 0.807455, 0.923335, -4.291481, 1.774853e-05, 0.083499, 4.750838], ["cg04400145", "chr20", 1893035, 1893037, "SIRPA;RP5-858L17.1", "antisense;protein_coding", "N_Shore", -0.191633, 0.044672, 0.82561, 0.756398, 0.901154, -4.289834, 1.788066e-05, 0.083499, 4.747617], ["cg18898209", "chr11", 86889207, 86889209, "PRSS23", "protein_coding", ".", -0.170329, 0.039949, 0.843387, 0.779871, 0.912077, -4.263696, 2.010726e-05, 0.092056, 4.696647], ["cg08480739", "chr11", 67997549, 67997551, "UNC93B1", "protein_coding", "Island", -0.100655, 0.023623, 0.904245, 0.863333, 0.947096, -4.260876, 2.036268e-05, 0.092056, 4.691165], ["cg04713531", "chr7", 158404595, 158404597, "PTPRN2", "protein_coding", "N_Shore", -0.194317, 0.045626, 0.823397, 0.75296, 0.900423, -4.258874, 2.054592e-05, 0.092056, 4.687275], ["cg02347859", "chr20", 3781872, 3781874, "SPEF1", "protein_coding", "S_Shelf", -0.353778, 0.083235, 0.702031, 0.596358, 0.826429, -4.250365, 2.134229e-05, 0.094349, 4.670759], ["cg14037250", "chr9", 134053771, 134053773, "RP11-374P20.4;BRD3", "antisense;protein_coding", "S_Shore", 0.168446, 0.039673, 1.183465, 1.094928, 1.27916, 4.245894, 2.177236e-05, 0.094984, 4.662094], ["cg00598235", "chr4", 17579056, 17579058, "LAP3", "protein_coding", "S_Shore", -0.29544, 0.069702, 0.744204, 0.649174, 0.853145, -4.238586, 2.249317e-05, 0.096854, 4.647949], ["cg07080653", "chr4", 53508081, 53508083, "FIP1L1;LNX1;LNX1-AS1", "antisense;protein_coding", ".", -0.122043, 0.02883, 0.88511, 0.836482, 0.936565, -4.233143, 2.304483e-05, 0.097957, 4.637426], ["cg26223555", "chr19", 51824187, 51824189, "FPR3", "protein_coding", ".", -0.265607, 0.062837, 0.766741, 0.677894, 0.867232, -4.226922, 2.369095e-05, 0.099429, 4.625417], ["cg15586054", "chr6", 165658541, 165658543, "PDE10A", "protein_coding", "N_Shore", -0.138573, 0.032814, 0.870599, 0.816371, 0.92843, -4.223049, 2.410192e-05, 0.099889, 4.617948], ["cg16449402", "chr3", 9733040, 9733042, "BRPF1", "protein_coding", "S_Shore", -0.207972, 0.04932, 0.81223, 0.737391, 0.894665, -4.216789, 2.478057e-05, 0.101434, 4.605889], ["cg19479744", "chr6", 155701180, 155701182, ".", ".", ".", -0.251659, 0.059823, 0.77751, 0.691488, 0.874233, -4.206742, 2.59079e-05, 0.103116, 4.586568], ["cg02988730", "chr1", 109624886, 109624888, "AMPD2", "protein_coding", ".", 0.26675, 0.063435, 1.305713, 1.153061, 1.478576, 4.205105, 2.609612e-05, 0.103116, 4.583424], ["cg23458843", "chr12", 57742661, 57742663, "AGAP2;TSPAN31", "protein_coding", "S_Shelf", -0.343971, 0.081803, 0.70895, 0.603927, 0.832236, -4.204858, 2.612465e-05, 0.103116, 4.582949], ["cg27018309", "chr16", 8849264, 8849266, "RP11-77H9.2;PMM2;RP11-152P23.2", "antisense;protein_coding", ".", -0.184988, 0.044073, 0.831114, 0.762335, 0.906099, -4.197312, 2.701019e-05, 0.104724, 4.568472], ["cg13720395", "chr7", 76328584, 76328586, "YWHAG", "protein_coding", "N_Shore", 0.123125, 0.029379, 1.131025, 1.067738, 1.198064, 4.190851, 2.779103e-05, 0.104724, 4.556095], ["cg26310429", "chr17", 73028309, 73028311, "SLC39A11", "protein_coding", ".", -0.1126, 0.026876, 0.893508, 0.84766, 0.941837, -4.189564, 2.794903e-05, 0.104724, 4.553633], ["cg00223135", "chr15", 43507046, 43507048, "TP53BP1", "protein_coding", "N_Shelf", -0.326563, 0.07795, 0.721399, 0.61919, 0.840479, -4.189372, 2.797272e-05, 0.104724, 4.553265], ["cg13857210", "chr9", 136107370, 136107372, ".", ".", ".", 0.13571, 0.032456, 1.14535, 1.07476, 1.220577, 4.181309, 2.898357e-05, 0.104724, 4.537848], ["cg21617903", "chr3", 11558238, 11558240, "VGLL4", "protein_coding", ".", 0.17338, 0.041487, 1.189318, 1.096439, 1.290065, 4.179158, 2.925908e-05, 0.104724, 4.533739], ["cg00460983", "chr10", 89374183, 89374185, "LIPA", "protein_coding", ".", -0.163649, 0.039161, 0.84904, 0.786311, 0.916772, -4.178934, 2.92879e-05, 0.104724, 4.533312], ["cg25376409", "chr17", 80999487, 80999489, "CHMP6", "protein_coding", "Island", 0.145841, 0.034918, 1.157013, 1.080479, 1.238968, 4.176738, 2.957191e-05, 0.104724, 4.529121], ["cg17669009", "chr19", 51417630, 51417632, "SIGLEC10", "protein_coding", ".", 0.294133, 0.070428, 1.341962, 1.168939, 1.540596, 4.176357, 2.962145e-05, 0.104724, 4.528394], ["cg20429981", "chr1", 1678532, 1678534, "SLC35E2B;MMP23B;CDK11B", "protein_coding", "S_Shore", 0.147103, 0.03525, 1.158473, 1.081139, 1.241339, 4.173166, 3.003956e-05, 0.104724, 4.522306], ["cg07914629", "chr2", 44362460, 44362462, "CAMKMT;PREPL", "protein_coding", "S_Shore", -0.826113, 0.198044, 0.437747, 0.296926, 0.645355, -4.17136, 3.027865e-05, 0.104724, 4.518863], ["cg03092447", "chr16", 89275990, 89275992, "ANKRD11", "protein_coding", "N_Shelf", 0.24376, 0.058476, 1.276037, 1.13786, 1.430995, 4.168554, 3.065376e-05, 0.104724, 4.513516], ["cg18151487", "chr10", 102715582, 102715584, "ARL3;SFXN2", "protein_coding", "S_Shore", -0.167568, 0.040206, 0.845719, 0.781633, 0.91506, -4.167742, 3.07632e-05, 0.104724, 4.511969], ["cg06605704", "chr16", 4501145, 4501147, "HMOX2", "protein_coding", ".", -0.368191, 0.088373, 0.691985, 0.581933, 0.822848, -4.166333, 3.095384e-05, 0.104724, 4.509285], ["cg18886274", "chr7", 85160554, 85160556, "SEMA3D", "protein_coding", ".", -0.148442, 0.03567, 0.86205, 0.803842, 0.924473, -4.161595, 3.160327e-05, 0.105841, 4.500268], ["cg23613253", "chr11", 70594241, 70594243, "SHANK2", "protein_coding", ".", -0.10314, 0.024849, 0.902, 0.859123, 0.947018, -4.150686, 3.314804e-05, 0.109753, 4.479542], ["cg02348449", "chr19", 58119061, 58119063, "ZSCAN18", "protein_coding", "S_Shore", -0.171289, 0.041287, 0.842578, 0.777081, 0.913596, -4.148684, 3.343917e-05, 0.109753, 4.475744], ["cg17078656", "chr1", 12475057, 12475059, "VPS13D", "protein_coding", ".", -0.306075, 0.073816, 0.736331, 0.637149, 0.850953, -4.146466, 3.376456e-05, 0.109753, 4.471539], ["cg14978242", "chr5", 80205311, 80205313, "SERINC5", "protein_coding", ".", 0.168524, 0.040688, 1.183557, 1.092837, 1.281808, 4.141852, 3.445122e-05, 0.110898, 4.462795], ["cg09890699", "chr1", 54356179, 54356181, "SSBP3", "protein_coding", "N_Shore", 0.184999, 0.044692, 1.203218, 1.102307, 1.313367, 4.139438, 3.481582e-05, 0.110994, 4.458223], ["cg02957270", "chr17", 48430734, 48430736, "SKAP1", "protein_coding", "S_Shore", -0.13357, 0.032308, 0.874966, 0.821279, 0.932162, -4.13433, 3.559912e-05, 0.11241, 4.448561], ["cg02905663", "chr8", 118111467, 118111469, "EXT1", "protein_coding", "N_Shore", -0.310791, 0.075236, 0.732867, 0.632389, 0.84931, -4.130885, 3.613686e-05, 0.113032, 4.44205], ["cg18586780", "chr2", 235044659, 235044661, "SH3BP4", "protein_coding", "S_Shore", -0.204286, 0.049509, 0.815229, 0.73984, 0.8983, -4.126283, 3.686736e-05, 0.114239, 4.433358], ["cg14882966", "chr2", 3651762, 3651764, ".", ".", "Island", 0.111938, 0.027158, 1.118444, 1.060468, 1.17959, 4.121799, 3.759254e-05, 0.115408, 4.424898], ["cg22624391", "chr11", 1916226, 1916228, ".", ".", ".", -0.229846, 0.055829, 0.794656, 0.712291, 0.886546, -4.11695, 3.839199e-05, 0.116657, 4.415759], ["cg10280339", "chr15", 65519989, 65519991, ".", ".", "S_Shelf", -0.18074, 0.043927, 0.834653, 0.7658, 0.909696, -4.11456, 3.879179e-05, 0.116657, 4.41126], ["cg20219911", "chr3", 52779635, 52779637, "ITIH1", "protein_coding", ".", 0.216693, 0.05271, 1.241962, 1.12006, 1.377132, 4.111017, 3.939206e-05, 0.116657, 4.404591], ["cg03601372", "chr19", 36776123, 36776125, "CTD-2162K18.4;CTD-2162K18.3", "lincRNA", "Island", -0.181463, 0.044199, 0.834049, 0.764838, 0.909524, -4.105585, 4.032929e-05, 0.116657, 4.394379], ["cg03762237", "chr2", 98085052, 98085054, ".", ".", "N_Shore", -0.160132, 0.039004, 0.852031, 0.789323, 0.919721, -4.10548, 4.034769e-05, 0.116657, 4.394181], ["cg25828093", "chr8", 26183486, 26183488, ".", ".", ".", -0.310269, 0.075607, 0.73325, 0.632258, 0.850372, -4.103688, 4.066165e-05, 0.116657, 4.390815], ["cg19827511", "chr14", 93114298, 93114300, "ITPK1", "protein_coding", "N_Shore", -0.216962, 0.052901, 0.804961, 0.72568, 0.892903, -4.101264, 4.108989e-05, 0.116657, 4.386265], ["cg10759293", "chr20", 20018581, 20018583, "NAA20", "protein_coding", "S_Shore", -0.179981, 0.043919, 0.835286, 0.766393, 0.910372, -4.098045, 4.166548e-05, 0.116657, 4.380224], ["cg10591771", "chr19", 3671119, 3671121, "AC004637.1;PIP5K1C", "antisense;protein_coding", "N_Shore", 0.124325, 0.030338, 1.132384, 1.067014, 1.201759, 4.098026, 4.166892e-05, 0.116657, 4.380188], ["cg11823637", "chr12", 22729886, 22729888, "RP11-114G22.1", "lincRNA", ".", -0.285082, 0.069584, 0.751952, 0.656085, 0.861828, -4.09695, 4.186289e-05, 0.116657, 4.378171], ["cg10817441", "chr6", 32855950, 32855952, "PSMB9", "protein_coding", "S_Shore", -0.22986, 0.056106, 0.794644, 0.711894, 0.887013, -4.096912, 4.186984e-05, 0.116657, 4.378099], ["cg27658601", "chr17", 82836204, 82836206, "TBCD;ZNF750", "protein_coding", "S_Shelf", -0.138516, 0.033844, 0.870649, 0.81477, 0.93036, -4.092804, 4.261888e-05, 0.117726, 4.370398], ["cg12623045", "chr10", 133253944, 133253946, ".", ".", ".", -0.167063, 0.040837, 0.846146, 0.781061, 0.916656, -4.090935, 4.296371e-05, 0.117726, 4.366898], ["cg13846675", "chr8", 52168344, 52168346, "ST18", "protein_coding", ".", 0.109542, 0.026794, 1.115767, 1.058684, 1.175929, 4.088263, 4.346158e-05, 0.118114, 4.361895], ["cg15576082", "chr2", 241813194, 241813196, "AC114730.3;NEU4", "antisense;protein_coding", "Island", 0.128719, 0.031504, 1.137371, 1.069267, 1.209813, 4.085849, 4.391597e-05, 0.118379, 4.357378], ["cg05348324", "chr11", 119797455, 119797457, ".", ".", ".", 0.15632, 0.038337, 1.169201, 1.084566, 1.260439, 4.077478, 4.552691e-05, 0.120786, 4.341732], ["cg23152216", "chr12", 25252150, 25252152, "KRAS", "protein_coding", "S_Shore", -0.167865, 0.041169, 0.845468, 0.779927, 0.916517, -4.077424, 4.553745e-05, 0.120786, 4.341631], ["cg13689831", "chr1", 226987375, 226987377, "ADCK3", "protein_coding", ".", 0.173215, 0.042523, 1.189122, 1.094034, 1.292475, 4.073435, 4.632487e-05, 0.121899, 4.334186], ["cg16721321", "chr4", 1208907, 1208909, "SPON2;CTBP1-AS", "antisense;protein_coding", "N_Shore", 0.093237, 0.022922, 1.097721, 1.049496, 1.148163, 4.067516, 4.751705e-05, 0.123197, 4.323151], ["cg02678414", "chr16", 2159838, 2159840, "TRAF7", "protein_coding", "S_Shelf", 0.189144, 0.046503, 1.208214, 1.102961, 1.323512, 4.067298, 4.756136e-05, 0.123197, 4.322746], ["cg16822174", "chr12", 19483011, 19483013, "AEBP2", "protein_coding", ".", 0.140085, 0.034529, 1.150371, 1.075095, 1.230918, 4.057026, 4.970157e-05, 0.127743, 4.30363], ["cg23601905", "chr17", 17757745, 17757747, "RAI1", "protein_coding", ".", 0.329083, 0.081166, 1.389693, 1.185306, 1.629323, 4.054459, 5.025048e-05, 0.12816, 4.29886], ["cg13691108", "chr20", 38804603, 38804605, "PPP1R16B", "protein_coding", "N_Shore", 0.182112, 0.044987, 1.199749, 1.098493, 1.310337, 4.048125, 5.162964e-05, 0.130673, 4.287101], ["cg23128949", "chr12", 124501577, 124501579, "NCOR2", "protein_coding", ".", 0.107634, 0.026606, 1.11364, 1.057056, 1.173253, 4.045543, 5.22019e-05, 0.13112, 4.282314], ["cg00549798", "chr22", 40418426, 40418428, "MKL1", "protein_coding", "N_Shore", 0.171746, 0.042489, 1.187376, 1.092499, 1.290492, 4.042079, 5.297943e-05, 0.132073, 4.275893], ["cg19345219", "chr12", 6638238, 6638240, "ACRBP", "protein_coding", ".", 0.195181, 0.048361, 1.215531, 1.105609, 1.336383, 4.035952, 5.438125e-05, 0.134555, 4.264551], ["cg25898577", "chr22", 21937868, 21937870, "LL22NC03-86G7.1;PPM1F", "antisense;protein_coding", "N_Shore", -0.133991, 0.033237, 0.874598, 0.81944, 0.933469, -4.031371, 5.545243e-05, 0.13619, 4.256079], ["cg03272932", "chr16", 87555286, 87555288, ".", ".", ".", -0.136212, 0.033824, 0.872658, 0.816682, 0.932471, -4.027041, 5.648313e-05, 0.137701, 4.248081], ["cg15219347", "chr19", 17378501, 17378503, "PLVAP", "protein_coding", ".", 0.18515, 0.046007, 1.203399, 1.099634, 1.316955, 4.024379, 5.712586e-05, 0.138251, 4.243167], ["cg00151914", "chr17", 35731530, 35731532, "TAF15;RASL10B", "protein_coding", "Island", 0.254981, 0.063417, 1.290437, 1.139609, 1.461227, 4.020686, 5.802886e-05, 0.139419, 4.236356], ["cg02584610", "chr11", 69089414, 69089416, "TPCN2", "protein_coding", ".", -0.154713, 0.038504, 0.856661, 0.794391, 0.923813, -4.018084, 5.867337e-05, 0.139953, 4.231559], ["cg08467866", "chr16", 75208215, 75208217, "CTRB2", "protein_coding", ".", 0.161232, 0.040171, 1.174957, 1.085998, 1.271204, 4.013684, 5.977831e-05, 0.14157, 4.223456], ["cg00550080", "chr6", 30157014, 30157016, "TRIM10", "protein_coding", ".", -0.179706, 0.044804, 0.835516, 0.765275, 0.912204, -4.010942, 6.0477e-05, 0.142209, 4.21841], ["cg00057663", "chr2", 105329171, 105329173, "TGFBRAP1;AC012360.4", "lincRNA;protein_coding", "N_Shore", 0.310434, 0.077464, 1.364016, 1.171878, 1.587657, 4.007479, 6.137034e-05, 0.143112, 4.212041], ["cg19708903", "chr10", 131970803, 131970805, "RP11-45A17.2;BNIP3", "antisense;protein_coding", ".", 0.159585, 0.039835, 1.173024, 1.084923, 1.268279, 4.006121, 6.172413e-05, 0.143112, 4.209545], ["cg03071245", "chr13", 36888896, 36888898, "SMAD9", "protein_coding", ".", -0.277627, 0.069367, 0.757579, 0.661276, 0.867908, -4.002271, 6.273735e-05, 0.143682, 4.202474], ["cg11817631", "chr11", 85811565, 85811567, "SYTL2", "protein_coding", "S_Shore", 0.179474, 0.044863, 1.196588, 1.095864, 1.306569, 4.000464, 6.32183e-05, 0.143682, 4.199157], ["cg11887376", "chr5", 1118554, 1118556, ".", ".", ".", 0.13035, 0.032596, 1.139227, 1.068721, 1.214384, 3.998929, 6.362981e-05, 0.143682, 4.196339], ["cg06721473", "chr3", 50317984, 50317986, "HYAL2", "protein_coding", "N_Shelf", 0.22622, 0.056574, 1.253851, 1.122251, 1.400883, 3.998656, 6.370326e-05, 0.143682, 4.195838], ["cg24343287", "chr19", 58058617, 58058619, "ZNF135", "protein_coding", "N_Shore", -0.20366, 0.050985, 0.81574, 0.738164, 0.901469, -3.99446, 6.4842e-05, 0.14524, 4.188144], ["cg02642565", "chr16", 4529400, 4529402, "CDIP1", "protein_coding", ".", 0.159742, 0.040007, 1.173208, 1.08473, 1.268904, 3.992898, 6.527051e-05, 0.14524, 4.185283], ["cg12466324", "chr6", 154085528, 154085530, "OPRM1", "protein_coding", ".", -0.286494, 0.071789, 0.750891, 0.652335, 0.864339, -3.99079, 6.585365e-05, 0.145259, 4.18142], ["cg21878918", "chr12", 57240978, 57240980, "NDUFA4L2", "protein_coding", "S_Shore", 0.316859, 0.079448, 1.372809, 1.174855, 1.604117, 3.988281, 6.655387e-05, 0.145259, 4.176827], ["cg21093005", "chr17", 82584038, 82584040, "FOXK2", "protein_coding", "Island", 0.133453, 0.033465, 1.142768, 1.070219, 1.220234, 3.987883, 6.666556e-05, 0.145259, 4.176098], ["cg26373351", "chr12", 11878369, 11878371, "ETV6", "protein_coding", ".", 0.261005, 0.065471, 1.298234, 1.14189, 1.475984, 3.986584, 6.703147e-05, 0.145259, 4.173721], ["cg01591830", "chr16", 89599591, 89599593, ".", ".", "S_Shelf", 0.158078, 0.039695, 1.171258, 1.083589, 1.26602, 3.982371, 6.823129e-05, 0.14559, 4.166016], ["cg21262032", "chr1", 154465216, 154465218, "IL6R", "protein_coding", ".", 0.141762, 0.035607, 1.152302, 1.074627, 1.235592, 3.981312, 6.853594e-05, 0.14559, 4.164082], ["cg05305099", "chr16", 57623601, 57623603, "RP11-405F3.5;GPR56", "antisense;protein_coding", ".", -0.245067, 0.061592, 0.782652, 0.693653, 0.883071, -3.9789, 6.923475e-05, 0.14559, 4.159676], ["cg06417460", "chr11", 61102287, 61102289, "CD5", "protein_coding", ".", 0.176499, 0.044367, 1.193033, 1.093671, 1.301421, 3.978129, 6.945965e-05, 0.14559, 4.158267], ["cg01914278", "chr8", 143764139, 143764141, "RP11-429J17.7", "lincRNA", "S_Shelf", 0.144114, 0.036234, 1.155016, 1.075836, 1.240024, 3.977361, 6.968419e-05, 0.14559, 4.156866], ["cg19461880", "chr11", 6483135, 6483137, "TIMM10B", "protein_coding", "S_Shore", -0.170387, 0.042851, 0.843338, 0.775402, 0.917227, -3.97627, 7.000451e-05, 0.14559, 4.154874], ["cg21026460", "chr3", 13497558, 13497560, "HDAC11", "protein_coding", ".", -0.136997, 0.034461, 0.871973, 0.815023, 0.932903, -3.975411, 7.025789e-05, 0.14559, 4.153305], ["cg04516952", "chr19", 50968785, 50968787, "KLK6;CTB-147C22.9", "antisense;protein_coding", ".", 0.227491, 0.057256, 1.255446, 1.122177, 1.404542, 3.97321, 7.091039e-05, 0.145977, 4.14929], ["cg14935078", "chr19", 17303589, 17303591, "ABHD8;MRPL34", "protein_coding", "Island", 0.245192, 0.061733, 1.277867, 1.13224, 1.442224, 3.971822, 7.132516e-05, 0.145977, 4.146757], ["cg06849931", "chr6", 32198115, 32198117, "NOTCH4", "protein_coding", "S_Shore", -0.310363, 0.078324, 0.73318, 0.628842, 0.854831, -3.962553, 7.415261e-05, 0.150321, 4.129874], ["cg19357999", "chr12", 121223958, 121223960, "P2RX4", "protein_coding", ".", 0.161934, 0.040873, 1.175782, 1.085265, 1.273848, 3.961904, 7.435439e-05, 0.150321, 4.128693], ["cg01881444", "chr3", 69012423, 69012425, "EOGT", "protein_coding", "N_Shore", -0.375504, 0.094829, 0.686943, 0.57043, 0.827254, -3.959815, 7.500789e-05, 0.150723, 4.124893], ["cg05413061", "chr3", 16183799, 16183801, "GALNT15", "protein_coding", ".", -0.319341, 0.080733, 0.726627, 0.620286, 0.8512, -3.955525, 7.636688e-05, 0.15253, 4.117095], ["cg09041251", "chr2", 10244393, 10244395, ".", ".", ".", -0.195036, 0.049414, 0.822805, 0.746854, 0.90648, -3.946969, 7.914678e-05, 0.156628, 4.101567], ["cg02205013", "chr16", 1708796, 1708798, "MAPK8IP3", "protein_coding", "S_Shelf", 0.117808, 0.029871, 1.125028, 1.061053, 1.192861, 3.94388, 8.017395e-05, 0.156628, 4.095967], ["cg04010915", "chr12", 110109247, 110109249, ".", ".", ".", -0.136154, 0.03453, 0.872708, 0.815599, 0.933816, -3.943033, 8.045747e-05, 0.156628, 4.094434], ["cg21865657", "chr19", 3687752, 3687754, "PIP5K1C", "protein_coding", "N_Shore", 0.114646, 0.029107, 1.121476, 1.059289, 1.187315, 3.938789, 8.189386e-05, 0.156628, 4.086749], ["cg16451525", "chr16", 89100873, 89100875, "ACSF3", "protein_coding", "Island", 0.12, 0.030468, 1.127496, 1.062137, 1.196878, 3.938487, 8.199717e-05, 0.156628, 4.086201], ["cg06212760", "chr1", 46199264, 46199266, "POMGNT1", "protein_coding", "N_Shelf", -0.158659, 0.040287, 0.853287, 0.788503, 0.923395, -3.938248, 8.207857e-05, 0.156628, 4.08577], ["cg24457649", "chr10", 71712586, 71712588, "CDH23;C10orf105", "protein_coding", ".", 0.149226, 0.037892, 1.160936, 1.077841, 1.250436, 3.938246, 8.207941e-05, 0.156628, 4.085766], ["cg06806210", "chr1", 228215425, 228215427, "OBSCN", "protein_coding", "N_Shore", -0.25113, 0.063772, 0.777922, 0.686519, 0.881493, -3.9379, 8.219791e-05, 0.156628, 4.085139], ["cg24743301", "chr2", 112187928, 112187930, "FBLN7;RP11-399B17.1", "TEC;protein_coding", "S_Shore", 0.116198, 0.029532, 1.123219, 1.060052, 1.19015, 3.934726, 8.329178e-05, 0.156959, 4.079398], ["cg13440692", "chr1", 1250976, 1250978, "RP5-902P8.12", "lincRNA", "N_Shelf", 0.151905, 0.038618, 1.16405, 1.079195, 1.255577, 3.933528, 8.370808e-05, 0.156959, 4.077233], ["cg25737891", "chr2", 11921150, 11921152, "AC096559.1", "lincRNA", ".", -0.159647, 0.040589, 0.852445, 0.787258, 0.923029, -3.933287, 8.379194e-05, 0.156959, 4.076798], ["cg27282206", "chr7", 158386620, 158386622, "PTPRN2", "protein_coding", ".", -0.202705, 0.051572, 0.816519, 0.73802, 0.903368, -3.930521, 8.476198e-05, 0.157884, 4.071799], ["cg19137417", "chr17", 81389134, 81389136, "RP11-1055B8.4", "lincRNA", "S_Shore", -0.136622, 0.034773, 0.8723, 0.814831, 0.933823, -3.929004, 8.529857e-05, 0.157996, 4.069058], ["cg10172675", "chr8", 103120454, 103120456, "LINC01181", "antisense", ".", 0.14315, 0.036458, 1.153903, 1.074327, 1.239373, 3.926465, 8.620353e-05, 0.158537, 4.064475], ["cg21468889", "chr6", 143512113, 143512115, "FUCA2", "protein_coding", "S_Shore", -0.297346, 0.075748, 0.742787, 0.640307, 0.86167, -3.925485, 8.65552e-05, 0.158537, 4.062707], ["cg00521620", "chr6", 29466638, 29466640, "UBDP1", "unprocessed_pseudogene", ".", -0.200677, 0.051139, 0.818176, 0.740147, 0.904433, -3.924181, 8.702538e-05, 0.158537, 4.060354], ["cg00539049", "chr2", 149077019, 149077021, "LYPD6B", "protein_coding", ".", -0.161139, 0.041124, 0.851174, 0.78526, 0.92262, -3.918376, 8.914742e-05, 0.161435, 4.049891], ["cg14543152", "chr6", 43886795, 43886797, ".", ".", ".", -0.172611, 0.044065, 0.841465, 0.771841, 0.917369, -3.917183, 8.958957e-05, 0.161435, 4.047743], ["cg13360395", "chr2", 123907703, 123907705, ".", ".", ".", -0.152189, 0.038889, 0.858826, 0.795798, 0.926845, -3.913465, 9.098108e-05, 0.16206, 4.041049], ["cg01397507", "chr11", 70594289, 70594291, "SHANK2", "protein_coding", ".", -0.124181, 0.031737, 0.88322, 0.829954, 0.939904, -3.912799, 9.123237e-05, 0.16206, 4.039851], ["cg05800336", "chr10", 131453805, 131453807, ".", ".", ".", 0.16867, 0.043128, 1.18373, 1.087782, 1.288141, 3.910901, 9.195252e-05, 0.16206, 4.036436], ["cg18964582", "chr1", 236393881, 236393883, "EDARADD", "protein_coding", "N_Shore", -0.137613, 0.035191, 0.871436, 0.813356, 0.933663, -3.910448, 9.212493e-05, 0.16206, 4.035623], ["cg01782668", "chr4", 148026284, 148026286, "ARHGAP10", "protein_coding", ".", -0.213384, 0.054577, 0.807846, 0.725892, 0.899051, -3.909778, 9.238083e-05, 0.16206, 4.034418], ["cg11039729", "chr10", 35379396, 35379398, "CCNY", "protein_coding", ".", 0.171791, 0.04398, 1.187429, 1.089361, 1.294326, 3.906098, 9.379853e-05, 0.162145, 4.027804], ["cg21932513", "chr1", 17425477, 17425479, "RCC2", "protein_coding", ".", 0.250786, 0.064206, 1.285035, 1.133086, 1.457361, 3.905973, 9.384697e-05, 0.162145, 4.02758], ["cg13707005", "chr10", 11204664, 11204666, "CELF2", "protein_coding", ".", 0.111557, 0.028567, 1.118018, 1.05714, 1.182402, 3.905098, 9.418736e-05, 0.162145, 4.026007], ["cg24355048", "chr14", 24575821, 24575823, "CTSG", "protein_coding", ".", -0.124383, 0.031856, 0.883041, 0.829594, 0.939932, -3.90459, 9.438514e-05, 0.162145, 4.025096], ["cg17389427", "chr3", 129612142, 129612144, ".", ".", ".", 0.31175, 0.079887, 1.365813, 1.167862, 1.597316, 3.902399, 9.524401e-05, 0.162777, 4.021162], ["cg06730285", "chr5", 140565521, 140565523, "APBB3;SLC35A4", "protein_coding", "S_Shore", -0.145189, 0.03724, 0.864859, 0.803982, 0.930345, -3.898714, 9.670486e-05, 0.163684, 4.014552], ["cg13696545", "chr3", 55750825, 55750827, "ERC2", "protein_coding", ".", -0.121025, 0.03105, 0.886011, 0.833699, 0.941607, -3.897699, 9.711101e-05, 0.163684, 4.012732], ["cg14458815", "chr22", 23141813, 23141815, "RSPH14", "protein_coding", "Island", 0.143925, 0.036929, 1.154797, 1.074167, 1.24148, 3.897338, 9.72558e-05, 0.163684, 4.012085], ["cg04371567", "chr4", 8247121, 8247123, ".", ".", ".", -0.307822, 0.079063, 0.735046, 0.62953, 0.858248, -3.893401, 9.884848e-05, 0.165275, 4.00503], ["cg03684062", "chr19", 3179365, 3179367, "S1PR4", "protein_coding", "Island", 0.16934, 0.043519, 1.184523, 1.087677, 1.289993, 3.891158, 9.976673e-05, 0.165275, 4.001014], ["cg17632028", "chr4", 652752, 652754, "PDE6B", "protein_coding", "N_Shore", -0.160261, 0.04119, 0.851921, 0.785847, 0.923551, -3.89073, 9.994326e-05, 0.165275, 4.000246]];

        // Define the dt_args
        let dt_args = {"autoWidth": false, "layout": {"topStart": "pageLength", "topEnd": "search", "bottomStart": "info", "bottomEnd": "paging"}, "order": []};
        dt_args["data"] = data;

        
        new DataTable(table, dt_args);
    });
</script>
</div></div>
</div>
</section>
<section id="train-coxph-lasso-model-with-the-selected-cpgs">
<h2>Train CoxPH+Lasso model with the selected CpGs<a class="headerlink" href="#train-coxph-lasso-model-with-the-selected-cpgs" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">utils.cox_lasso</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># raw_coefs = train_coxph_lasso(df=clinical_data4.copy(),</span>
<span class="c1">#                                 event= &#39;os.evnt at 5y&#39;,</span>
<span class="c1">#                                 time= &#39;os.time at 5y&#39;,</span>
<span class="c1">#                                 train_x=ewas_top_cpgs_os,</span>
<span class="c1">#                                 cv_n_split=10,</span>
<span class="c1">#                                 loops=1000)</span>
<span class="c1"># raw_coefs.to_csv(output_path + &#39;ewas_dmr/ewas_cog_os_raw_coefs_newrisk.csv&#39;)</span>


<span class="n">raw_coefs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_raw_coefs_newrisk.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">mean_coefs</span> <span class="o">=</span> <span class="n">set_cutoff</span><span class="p">(</span><span class="n">coefs</span><span class="o">=</span><span class="n">raw_coefs</span><span class="p">,</span><span class="n">threshold</span><span class="o">=</span><span class="mf">0.95</span><span class="p">)</span>
                                
<span class="n">plot_nonzero_coef_freq</span><span class="p">(</span><span class="n">raw_coefs</span><span class="p">,</span> <span class="n">mean_coefs</span><span class="p">,</span> <span class="n">threshold</span><span class="o">=</span><span class="mf">0.95</span><span class="p">,</span> <span class="n">savefig</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">fig_path</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mf">3.2</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/dd23eebe7f436d3fbb0270431689d6bcfe534726b3afd7c401d0be7e16b59ea3.png" src="../_images/dd23eebe7f436d3fbb0270431689d6bcfe534726b3afd7c401d0be7e16b59ea3.png" />
</div>
</div>
</section>
<section id="apply-signature-to-discovery-dataset-and-set-high-low-cutoff">
<h2>Apply signature to discovery dataset and set high/low cutoff<a class="headerlink" href="#apply-signature-to-discovery-dataset-and-set-high-low-cutoff" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">continuous_model</span> <span class="o">=</span> <span class="s1">&#39;38CpG-HazardScore&#39;</span>
<span class="n">categorical_model</span> <span class="o">=</span> <span class="s1">&#39;38CpG-AMLsignature&#39;</span>
<span class="n">CUTOFF</span> <span class="o">=</span> <span class="mf">0.5</span>


<span class="n">df5</span><span class="p">,</span> <span class="n">threshold</span> <span class="o">=</span> <span class="n">generate_coxph_score</span><span class="p">(</span><span class="n">coef_mean</span><span class="o">=</span><span class="n">mean_coefs</span><span class="p">,</span>
                                        <span class="n">x</span><span class="o">=</span><span class="n">ewas_top_cpgs_os</span><span class="p">,</span>
                                        <span class="n">df</span><span class="o">=</span><span class="n">clinical_data4</span><span class="p">,</span>
                                        <span class="n">cont_model_name</span> <span class="o">=</span> <span class="n">continuous_model</span><span class="p">,</span>
                                        <span class="n">cat_model_name</span> <span class="o">=</span> <span class="n">categorical_model</span><span class="p">,</span>
                                        <span class="n">cutoff</span><span class="o">=</span><span class="s1">&#39;quantile&#39;</span><span class="p">,</span>
                                        <span class="n">cutoff_value</span><span class="o">=</span><span class="n">CUTOFF</span><span class="p">)</span>

<span class="c1"># Plot results</span>

<span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df5</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">continuous_model</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">hue</span> <span class="o">=</span> <span class="n">categorical_model</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">threshold</span> <span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">({</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">CUTOFF</span><span class="o">*</span><span class="mi">100</span><span class="si">}</span><span class="s1">% Cutoff: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">threshold</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)):</span><span class="n">threshold</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="n">categorical_model</span> <span class="o">+</span> <span class="s1">&#39; in discovery cohort&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Continuous score cut at the value of -2.0431
</pre></div>
</div>
<img alt="../_images/23b9d0fb82fe2630b02f56bc66dc96960515194fd01027bb58f476e74c7cc368.png" src="../_images/23b9d0fb82fe2630b02f56bc66dc96960515194fd01027bb58f476e74c7cc368.png" />
</div>
</div>
</section>
<section id="apply-signature-to-test-dataset">
<h2>Apply signature to test dataset<a class="headerlink" href="#apply-signature-to-test-dataset" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load test data and pick only columns that are in mean_coefs</span>
<span class="n">df_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;201samples_331556cpgs_nobatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()[</span><span class="n">mean_coefs</span><span class="o">.</span><span class="n">index</span><span class="p">]</span>

<span class="c1"># Transform from beta-values to m-values</span>
<span class="n">df_test2</span> <span class="o">=</span> <span class="n">df_test</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>

<span class="c1"># Select samples from `clincal_data` that are in `df_test`</span>
<span class="n">clinical_data_test</span> <span class="o">=</span> <span class="n">clinical_data</span><span class="p">[</span><span class="n">clinical_data</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_test</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># apply signature to test data using `threshold` from training data</span>
<span class="n">clinical_data_test2</span> <span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">generate_coxph_score</span><span class="p">(</span><span class="n">coef_mean</span><span class="o">=</span><span class="n">mean_coefs</span><span class="p">,</span>
                                        <span class="n">x</span><span class="o">=</span><span class="n">df_test2</span><span class="p">,</span>
                                        <span class="n">df</span><span class="o">=</span><span class="n">clinical_data_test</span><span class="p">,</span>
                                        <span class="n">cont_model_name</span> <span class="o">=</span> <span class="n">continuous_model</span><span class="p">,</span>
                                        <span class="n">cat_model_name</span> <span class="o">=</span> <span class="n">categorical_model</span><span class="p">,</span>
                                        <span class="n">cutoff</span><span class="o">=</span><span class="n">threshold</span><span class="p">)</span>

<span class="c1"># Plot the results</span>
<span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">clinical_data_test2</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">continuous_model</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">hue</span> <span class="o">=</span> <span class="n">categorical_model</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">threshold</span> <span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">({</span><span class="s1">&#39;Discovery Cutoff: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">threshold</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)):</span><span class="n">threshold</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="n">categorical_model</span> <span class="o">+</span> <span class="s1">&#39; in test cohort&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Continuous score cut at the value of -2.0431
</pre></div>
</div>
<img alt="../_images/9cc34415f9848e592e54c51528537b53d5838be3a75047fc3c0d79038d656aff.png" src="../_images/9cc34415f9848e592e54c51528537b53d5838be3a75047fc3c0d79038d656aff.png" />
</div>
</div>
</section>
<section id="apply-signature-to-nanopore-test-dataset">
<h2>Apply signature to nanopore test dataset<a class="headerlink" href="#apply-signature-to-nanopore-test-dataset" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load nanopore_test data and pick only columns that are in mean_coefs</span>
<span class="n">df_nanopore_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;12samples_331556cpgs_nanoporeseq_bvalues.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()[</span><span class="n">mean_coefs</span><span class="o">.</span><span class="n">index</span><span class="p">]</span>

<span class="n">df_nanopore_test</span> <span class="o">=</span> <span class="n">df_nanopore_test</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">0.999</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">0.001</span><span class="p">)</span>

<span class="c1"># Transform from beta-values to m-values</span>
<span class="n">df_nanopore_test2</span> <span class="o">=</span> <span class="n">df_nanopore_test</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>

<span class="c1"># Select samples from `clincal_data` that are in `df_nanopore_test`</span>
<span class="n">clinical_data_nanopore_test</span> <span class="o">=</span> <span class="n">clinical_data</span><span class="p">[</span><span class="n">clinical_data</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_nanopore_test</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># apply signature to nanopore_test data using `threshold` from training data</span>
<span class="n">clinical_data_nanopore_test2</span> <span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">generate_coxph_score</span><span class="p">(</span><span class="n">coef_mean</span><span class="o">=</span><span class="n">mean_coefs</span><span class="p">,</span>
                                        <span class="n">x</span><span class="o">=</span><span class="n">df_nanopore_test2</span><span class="p">,</span>
                                        <span class="n">df</span><span class="o">=</span><span class="n">clinical_data_nanopore_test</span><span class="p">,</span>
                                        <span class="n">cont_model_name</span> <span class="o">=</span> <span class="n">continuous_model</span><span class="p">,</span>
                                        <span class="n">cat_model_name</span> <span class="o">=</span> <span class="n">categorical_model</span><span class="p">,</span>
                                        <span class="n">cutoff</span><span class="o">=</span><span class="n">threshold</span><span class="p">)</span>

<span class="c1"># Plot the results</span>
<span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">clinical_data_nanopore_test2</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">continuous_model</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">hue</span> <span class="o">=</span> <span class="n">categorical_model</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">threshold</span> <span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">({</span><span class="s1">&#39;Discovery Cutoff: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">threshold</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)):</span><span class="n">threshold</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="n">categorical_model</span> <span class="o">+</span> <span class="s1">&#39; in nanopore cohort&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Continuous score cut at the value of -2.0431
</pre></div>
</div>
<img alt="../_images/db11ff081ba9fac76d527b54cbdeb31b8b95258b892ad93b1e0199bdef735f98.png" src="../_images/db11ff081ba9fac76d527b54cbdeb31b8b95258b892ad93b1e0199bdef735f98.png" />
</div>
</div>
</section>
<section id="merge-and-save-files">
<h2>Merge and save files<a class="headerlink" href="#merge-and-save-files" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">final_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">clinical_data_test2</span><span class="p">,</span> <span class="n">clinical_data_nanopore_test2</span><span class="p">,</span> <span class="n">clinical_data4</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">final_df</span> <span class="o">=</span> <span class="n">final_df</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="o">-</span><span class="mi">3</span><span class="p">:]</span>

<span class="n">final_df</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/signature_results.xlsx&#39;</span><span class="p">)</span>

<span class="c1"># Save ewas results to data folder</span>
<span class="n">mean_coefs</span><span class="o">.</span><span class="n">to_frame</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s1">&#39;38CpG-AMLsignature coefficients&#39;</span><span class="p">)</span>\
    <span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">top_os</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>\
        <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;38CpG-AMLsignature coefficients&#39;</span><span class="p">,</span> <span class="n">ascending</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>\
            <span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/ewas_top_cpgs_os_aml_signature.xlsx&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Saved signature results to ../data/signature_results.xlsx&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Saved EWAS top CpGs + signature coefficients to ../data/ewas_top_cpgs_os_aml_signature.xlsx&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Saved signature results to ../data/signature_results.xlsx
Saved EWAS top CpGs + signature coefficients to ../data/ewas_top_cpgs_os_aml_signature.xlsx
</pre></div>
</div>
</div>
</div>
</section>
<section id="watermark">
<h2>Watermark<a class="headerlink" href="#watermark" title="Link to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Last updated: 2024-09-23

Python implementation: CPython
Python version       : 3.8.18
IPython version      : 8.12.3

methylize: not installed
pandas   : 2.0.3

Compiler    : GCC 11.4.0
OS          : Linux
Release     : 5.15.133.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 32
Architecture: 64bit

Git repo: git@github.com:f-marchi/ALMA.git
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./source"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="4_clinical_data.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Clinical Data</p>
      </div>
    </a>
    <a class="right-next"
       href="5_classifiers.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Classifiers</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-samples-only">Select diagnostic samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas">Run CoxPH-EWAS</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-cpgs-based-on-suggested-significance-p-values-1-10e-5">Select CpGs based on suggested-significance p-values (1*10e-5)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#train-coxph-lasso-model-with-the-selected-cpgs">Train CoxPH+Lasso model with the selected CpGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-discovery-dataset-and-set-high-low-cutoff">Apply signature to discovery dataset and set high/low cutoff</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-test-dataset">Apply signature to test dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-nanopore-test-dataset">Apply signature to nanopore test dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#merge-and-save-files">Merge and save files</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Marchi et al. | Lamba Lab | University of Florida.
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>