
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>37-CpG Signature &#8212; Acute Leukemia Methylome Atlas Study</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css?v=564be945" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'source/5_signature';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Classifiers" href="5_classifiers.html" />
    <link rel="prev" title="Clinical Data" href="4_clinical_data.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../index.html">
  
  
  
  
  
  
    <p class="title logo__title">Acute Leukemia Methylome Atlas Study</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../index.html">
                    Overview
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Latest</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="1_discovery.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_validation.html">Validation Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_test.html">Test Dataset (Work in Progress)</a></li>
<li class="toctree-l1"><a class="reference internal" href="4_clinical_data.html">Clinical Data</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">37-CpG Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_classifiers.html">Classifiers</a></li>
<li class="toctree-l1"><a class="reference internal" href="6_plots.html">Data Visualization</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Previous 2022-2024</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../v1/1_discovery_dataset.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/2_validation_dataset_v2.html">Validation Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/3_reclassification_strategy.html">Dx and Px Reclassification</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/4_models.html">Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/5_plots.html">Data Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/6_test_dataset.html">Preprocess long reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/7_test_models.html">Run epigenomic models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../v1/8_test_atlas_1852.html">UF HemBank 1852 case study</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/ewasCox_vs_pacmapLGBM.html">Px Models Benchmarks</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/5_plots_6-17-24.html">AML-only map with new interactive prognostic and race histograms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/2_Univariate_CoxPH-EWAS_v4.html">EWAS Runs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v5.html">924pts-OS-35CpGs-ASH23</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v6.html">940pts-OS-48CpGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v7.html">940pts-EFS-55CpGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v2.html">OS-35CpGs-Validation-ASH23</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v3.html">OS-48CpGs-Validation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v4.html">EFS-55CpGs-Validation</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/source/5_signature.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>37-CpG Signature</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-samples-only">Select diagnostic samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas">Run CoxPH-EWAS</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-cpgs-based-on-suggested-significance-p-values-1-10e-5">Select CpGs based on suggested-significance p-values (1*10e-5)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#train-coxph-lasso-model-with-the-selected-cpgs">Train CoxPH+Lasso model with the selected CpGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-discovery-dataset-and-set-high-low-cutoff">Apply signature to discovery dataset and set high/low cutoff</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-validation-dataset">Apply signature to validation dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#merge-and-save-files">Merge and save files</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="cpg-signature">
<h1>37-CpG Signature<a class="headerlink" href="#cpg-signature" title="Link to this heading">#</a></h1>
<section id="where-the-data-at">
<h2>Where the data at?<a class="headerlink" href="#where-the-data-at" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;../&#39;</span><span class="p">)</span>

<span class="n">mount</span> <span class="o">=</span> <span class="s1">&#39;/mnt/e/ALMA/&#39;</span>
<span class="n">input_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Intermediate_Files/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Processed_Files/&#39;</span>
<span class="n">zhou2016_probes_path</span> <span class="o">=</span>  <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="load-data">
<h2>Load data<a class="headerlink" href="#load-data" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read df_discovery and df_test</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;3314samples_331557cpgs_nobatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="c1"># Read clinical data</span>
<span class="n">clinical_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;clinical_data.xlsx&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="remove-samples-based-on-certain-clinical-features">
<h2>Remove samples based on certain clinical features<a class="headerlink" href="#remove-samples-based-on-certain-clinical-features" title="Link to this heading">#</a></h2>
<section id="select-samples-from-aaml1031-0531-and-03p1-clinical-trials">
<h3>Select samples from AAML1031, 0531, and 03P1 clinical trials<a class="headerlink" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">clinical_data2</span> <span class="o">=</span> <span class="n">clinical_data</span><span class="p">[</span><span class="n">clinical_data</span><span class="p">[</span><span class="s1">&#39;Clinical Trial&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;AAML0531&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML1031&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML03P1&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">clinical_data</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">clinical_data2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">clinical_data2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2228 samples were removed. 1287 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="select-diagnostic-samples-only">
<h3>Select diagnostic samples only<a class="headerlink" href="#select-diagnostic-samples-only" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">clinical_data3</span> <span class="o">=</span> <span class="n">clinical_data2</span><span class="p">[</span><span class="n">clinical_data2</span><span class="p">[</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Diagnosis&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Bone Marrow&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Peripheral Blood&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">clinical_data2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">clinical_data3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">clinical_data3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>332 samples were removed. 955 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-duplicate-samples">
<h3>Remove duplicate samples<a class="headerlink" href="#remove-duplicate-samples" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">clinical_data4</span> <span class="o">=</span> <span class="n">clinical_data3</span><span class="p">[</span><span class="o">~</span><span class="n">clinical_data3</span><span class="p">[</span><span class="s1">&#39;Patient_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="s1">&#39;last&#39;</span><span class="p">)]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">clinical_data3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">clinical_data4</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">clinical_data4</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>9 samples were removed. 946 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="match-samples-in-clinical-data-to-samples-in-methylation-data">
<h3>Match samples in clinical data to samples in methylation data<a class="headerlink" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Match samples in clinical data to samples in methylation data</span>
<span class="n">df2</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">clinical_data4</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">:]</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Samples in clinical data matched to samples in methylation data.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Samples in clinical data matched to samples in methylation data.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="transform-to-m-values">
<h2>Transform to M-values<a class="headerlink" href="#transform-to-m-values" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="k">def</span> <span class="nf">beta2m</span><span class="p">(</span><span class="n">val</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Transfrom beta-values into m-values&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">math</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">val</span><span class="o">/</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">val</span><span class="p">))</span>

<span class="n">df3</span> <span class="o">=</span> <span class="n">df2</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="perform-batch-correction">
<h2>Perform batch correction<a class="headerlink" href="#perform-batch-correction" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>pyCombat</strong> is a Python tool for batch effects correction in high-throughput molecular data using empirical Bayes methods</p></li>
<li><p><strong>Github</strong>: <a class="reference external" href="https://epigenelabs.github.io/pyComBat/">https://epigenelabs.github.io/pyComBat/</a></p></li>
<li><p><strong>Implementation Preprint</strong>: <a class="reference external" href="https://doi.org/10.1101/2020.03.17.995431">bioRxiv</a></p></li>
<li><p><strong>Original Paper</strong>: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/16632515/">Adjusting batch effects in microarray expression data using empirical Bayes methods</a></p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">combat.pycombat</span> <span class="kn">import</span> <span class="n">pycombat</span>

<span class="c1"># Correct batch effects in the training dataset</span>
<span class="n">df4</span> <span class="o">=</span> <span class="n">pycombat</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">df3</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">batch</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully corrected batch effects in the training dataset.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Found 3 batches.
Adjusting for 0 covariate(s) or covariate level(s).
Standardizing Data across genes.
Fitting L/S model and finding priors.
Finding parametric adjustments.
Adjusting the Data
Succesfully corrected batch effects in the training dataset.
</pre></div>
</div>
</div>
</div>
<section id="evaluate-batch-effects">
<h3>Evaluate batch effects<a class="headerlink" href="#evaluate-batch-effects" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">sklearn.preprocessing</span> <span class="kn">import</span> <span class="n">StandardScaler</span>
<span class="kn">from</span> <span class="nn">sklearn.decomposition</span> <span class="kn">import</span> <span class="n">PCA</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;white&#39;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Standardizes the data.&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">StandardScaler</span><span class="p">()</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">perform_pca</span><span class="p">(</span><span class="n">std_data</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Performs PCA on standardized data.&#39;&#39;&#39;</span>
    <span class="n">pca</span> <span class="o">=</span> <span class="n">PCA</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="mi">42</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">pca</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">std_data</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">create_principal_df</span><span class="p">(</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">df4</span> <span class="p">,</span> <span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Creates a DataFrame with the PCA data.&#39;&#39;&#39;</span>
    <span class="n">principal_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="s1">&#39;PC2&#39;</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span><span class="n">df4</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span>
    <span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;hue&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">principal_df</span>

<span class="k">def</span> <span class="nf">filter_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">principal_df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Filters the DataFrame by removing outliers.&#39;&#39;&#39;</span>
    <span class="n">outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">threshold</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">outliers</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df4</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_batchcorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df4</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">df</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_uncorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df3</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">df</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Plots PCA data.&#39;&#39;&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">,</span> <span class="n">legend</span> <span class="ow">in</span> <span class="p">[(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA after batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="s1">&#39;full&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="n">non_outliers</span><span class="p">,</span> <span class="n">ax1</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA before batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">non_outliers</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="kc">False</span><span class="p">)]:</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PC2&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;hue&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="n">legend</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;PC 1&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;PC 2&#39;</span><span class="p">)</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.05</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">plot_data</span><span class="p">(</span><span class="n">pcs_batchcorrected</span><span class="p">,</span> <span class="n">pcs_uncorrected</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/079c8965ff781306e701fab52b898142c01801027c9bf54bed6045b86edd52e8.png" src="../_images/079c8965ff781306e701fab52b898142c01801027c9bf54bed6045b86edd52e8.png" />
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ol class="arabic simple">
<li><p>GDC_TARGET-AML contains AAML0531 and 03P1 samples from methylation array 450k.</p></li>
<li><p>GSE124413 contains AAML0531 and 03P1 samples from methylation array EPIC.</p></li>
<li><p>GSE190931 contains AAML1031 samples from methylation array EPIC.</p></li>
</ol>
</div>
</section>
</section>
<section id="run-coxph-ewas">
<h2>Run CoxPH-EWAS<a class="headerlink" href="#run-coxph-ewas" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">utils.methylize_custom_code.diff_meth_pos_custom</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># def impute_by_mean(df):</span>
<span class="c1">#     return df.fillna(df.mean())</span>

<span class="c1"># covar_os = round(impute_by_mean(clinical_data4[[&#39;os.evnt at 5y&#39;,&#39;os.time at 5y&#39;,&#39;Risk_Group_AAML1831_int&#39;]]),3)</span>

<span class="c1"># ewas_os_cog_adjusted = diff_meth_pos(meth_data = df4,</span>
<span class="c1">#                                     duration = covar_os[&#39;os.time at 5y&#39;],</span>
<span class="c1">#                                     event_observed = covar_os[&#39;os.evnt at 5y&#39;],</span>
<span class="c1">#                                     pheno_data = covar_os[&#39;os.time at 5y&#39;], # not used</span>
<span class="c1">#                                     covariates = covar_os[&#39;Risk_Group_AAML1831_int&#39;],</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = False,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted&#39;)</span>

<span class="n">ewas_os_cog_adjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_os_cog_adjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG OS risk adjusted&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/d32c5fdc51a2c361e318176be0e8f280490df384bd8a212e6cb0906076b6c786.png" src="../_images/d32c5fdc51a2c361e318176be0e8f280490df384bd8a212e6cb0906076b6c786.png" />
</div>
</div>
</section>
<section id="select-cpgs-based-on-suggested-significance-p-values-1-10e-5">
<h2>Select CpGs based on suggested-significance p-values (1*10e-5)<a class="headerlink" href="#select-cpgs-based-on-suggested-significance-p-values-1-10e-5" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Select only rows where `PValue`&lt; 1*10e-5</span>
<span class="n">ewas_os_cog_adjusted_sig</span> <span class="o">=</span> <span class="n">ewas_os_cog_adjusted</span><span class="p">[</span><span class="n">ewas_os_cog_adjusted</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="o">*</span><span class="mf">10e-5</span><span class="p">]</span>

<span class="c1"># Define a function to handle repetitive tasks</span>
<span class="k">def</span> <span class="nf">process_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">other_df</span><span class="p">,</span> <span class="n">column_drop</span><span class="p">):</span>
    <span class="c1"># Join and sort values</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">other_df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">column_drop</span><span class="p">),</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span>

    <span class="c1"># Handle columns with multiple genes separated by `;`</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneClasses&#39;</span><span class="p">]:</span>
        <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">))))</span>

    <span class="k">return</span> <span class="n">df</span>

<span class="c1"># Load manifest</span>
<span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">zhou2016_probes_path</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span>
                                                     <span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">top_os</span> <span class="o">=</span> <span class="n">process_df</span><span class="p">(</span><span class="n">zhou2016_probes</span><span class="p">,</span> <span class="n">ewas_os_cog_adjusted_sig</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;chromosome&#39;</span><span class="p">,</span> <span class="s1">&#39;MAPINFO&#39;</span><span class="p">])</span>

<span class="c1"># Now include the date in the filename and save to excel</span>
<span class="c1"># import datetime</span>
<span class="c1"># top_os.to_excel(output_path + &#39;ewas_dmr/ewas_stats_&#39; + str(datetime.date.today()) + &#39;.xlsx&#39;)</span>

<span class="c1"># Select only significant probes</span>
<span class="n">ewas_top_cpgs_os</span> <span class="o">=</span> <span class="n">df4</span><span class="p">[</span><span class="n">top_os</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_list</span><span class="p">()]</span>

<span class="c1"># Save x to pickle in output_path</span>
<span class="n">ewas_top_cpgs_os</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_cpgs_os&#39;</span><span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">date</span><span class="o">.</span><span class="n">today</span><span class="p">())</span><span class="o">+</span> <span class="s1">&#39;.csv&#39;</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">itables</span> <span class="kn">import</span> <span class="n">show</span>

<span class="n">show</span><span class="p">(</span><span class="n">top_os</span><span class="p">,</span> <span class="n">classes</span><span class="o">=</span><span class="s2">&quot;display nowrap compact&quot;</span><span class="p">,</span>  <span class="n">style</span><span class="o">=</span><span class="s2">&quot;width:1500px&quot;</span><span class="p">,</span> <span class="n">autoWidth</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><style>.itables table td {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables table th {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables thead input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}

.itables tfoot input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}
</style>
<div class="itables">
<table id="84760c52-d468-4435-a071-95752f12288a" class="display nowrap compact"style="width:1500px"><thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>CGIPosition</th>
      <th>Coefficient</th>
      <th>StandardError</th>
      <th>HazardsRatio</th>
      <th>95%CI_lower</th>
      <th>95%CI_upper</th>
      <th>ZScore</th>
      <th>PValue</th>
      <th>FDR_QValue</th>
      <th>minuslog10value</th>
    </tr>
  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css">
<script type="module">
    // Import jquery and DataTable
    import 'https://code.jquery.com/jquery-3.6.0.min.js';
    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';
    dt($);

    // Define the table data
    const data = [["cg00920892", "chr8", 37874705, 37874707, "RAB11FIP1", "protein_coding", ".", 0.217552, 0.039032, 1.24303, 1.151483, 1.341856, 5.573679, 2.494145e-08, 0.00827, 7.603078], ["cg08900363", "chr7", 597816, 597818, "PRKAR1B", "protein_coding", ".", -0.270291, 0.050184, 0.763158, 0.691668, 0.842037, -5.385973, 7.205385e-08, 0.011019, 7.142343], ["cg01543603", "chr16", 89403132, 89403134, "ANKRD11;RP1-168P16.2", "protein_coding;sense_intronic", ".", 0.192383, 0.036474, 1.212135, 1.128506, 1.301961, 5.274475, 1.331365e-07, 0.011019, 6.875703], ["cg08404702", "chr16", 2083386, 2083388, "TSC2", "protein_coding", "N_Shore", 0.125705, 0.024039, 1.133948, 1.081761, 1.188652, 5.229314, 1.701404e-07, 0.011019, 6.769192], ["cg06748884", "chr16", 11138900, 11138902, "CLEC16A", "protein_coding", ".", -0.257325, 0.049282, 0.773117, 0.701934, 0.851518, -5.221497, 1.774824e-07, 0.011019, 6.750845], ["cg08329113", "chr16", 70737238, 70737240, "VAC14", "protein_coding", ".", 0.189879, 0.036516, 1.209103, 1.125592, 1.29881, 5.199897, 1.993987e-07, 0.011019, 6.700278], ["cg06339275", "chr3", 62948996, 62948998, "LINC00698", "lincRNA", ".", -0.173536, 0.034135, 0.840687, 0.786282, 0.898857, -5.083766, 3.70024e-07, 0.017526, 6.43177], ["cg20018344", "chr20", 3223491, 3223493, "ITPA", "protein_coding", ".", 0.201828, 0.040116, 1.223638, 1.131112, 1.323732, 5.031072, 4.877456e-07, 0.020214, 6.311807], ["cg00192882", "chr17", 19387806, 19387808, "MFAP4", "protein_coding", ".", 0.244104, 0.04929, 1.276477, 1.158929, 1.405947, 4.952367, 7.331633e-07, 0.027009, 6.134799], ["cg01052291", "chr6", 132715433, 132715435, "VNN1", "protein_coding", ".", -0.224744, 0.046278, 0.798721, 0.729462, 0.874555, -4.856342, 1.195741e-06, 0.033743, 5.922363], ["cg05430997", "chr13", 109744841, 109744843, ".", ".", ".", 0.138752, 0.028643, 1.148839, 1.086122, 1.215178, 4.844248, 1.27092e-06, 0.033743, 5.895882], ["cg07660635", "chr16", 85902540, 85902542, "IRF8", "protein_coding", "S_Shelf", -0.155406, 0.032114, 0.856068, 0.803845, 0.911683, -4.839133, 1.304069e-06, 0.033743, 5.884699], ["cg16424082", "chr12", 50929956, 50929958, "METTL7A", "protein_coding", ".", -0.359358, 0.074576, 0.698125, 0.603189, 0.808002, -4.81865, 1.445327e-06, 0.033743, 5.840034], ["cg10089193", "chr16", 1090938, 1090940, "C1QTNF8", "protein_coding", "S_Shore", 0.170649, 0.035466, 1.186074, 1.106427, 1.271455, 4.811564, 1.497539e-06, 0.033743, 5.824622], ["cg04839706", "chr16", 46886162, 46886164, "GPT2", "protein_coding", "S_Shore", -0.187157, 0.038928, 0.829314, 0.768392, 0.895065, -4.807727, 1.526557e-06, 0.033743, 5.816287], ["cg22760287", "chr11", 107705174, 107705176, ".", ".", ".", -0.180325, 0.03765, 0.834999, 0.775601, 0.898946, -4.789497, 1.672002e-06, 0.034648, 5.776763], ["cg17099306", "chr13", 49549508, 49549510, "RCBTB1", "protein_coding", ".", 0.205942, 0.043237, 1.228682, 1.128849, 1.337343, 4.763063, 1.906762e-06, 0.037188, 5.719703], ["cg19432033", "chr17", 7613901, 7613903, "FXR2;SHBG", "protein_coding", "N_Shore", -0.198699, 0.041944, 0.819797, 0.755099, 0.890039, -4.737266, 2.166203e-06, 0.039901, 5.664301], ["cg26361535", "chr8", 143494433, 143494435, "ZC3H3", "protein_coding", ".", 0.163702, 0.034713, 1.177863, 1.100392, 1.260789, 4.715929, 2.406102e-06, 0.039913, 5.618686], ["cg00945409", "chr10", 78977907, 78977909, "ZMIZ1-AS1", "antisense", "S_Shelf", -0.273171, 0.057927, 0.760962, 0.679291, 0.852453, -4.715803, 2.407591e-06, 0.039913, 5.618417], ["cg27049094", "chr2", 74840588, 74840590, "HK2", "protein_coding", ".", 0.130897, 0.027901, 1.13985, 1.079192, 1.203918, 4.691526, 2.711745e-06, 0.040193, 5.566751], ["cg18062721", "chr3", 11601952, 11601954, "VGLL4", "protein_coding", ".", 0.110221, 0.023507, 1.116525, 1.06625, 1.16917, 4.688837, 2.747622e-06, 0.040193, 5.561043], ["cg14928764", "chr20", 23083970, 23083972, "CD93", "protein_coding", ".", -0.170034, 0.036287, 0.843636, 0.78572, 0.905821, -4.685839, 2.788157e-06, 0.040193, 5.554683], ["cg04220636", "chr16", 88967679, 88967681, "CBFA2T3", "protein_coding", "Island", 0.158563, 0.03423, 1.171826, 1.095789, 1.253139, 4.632343, 3.615504e-06, 0.047448, 5.441831], ["cg22410826", "chr13", 109744594, 109744596, ".", ".", ".", 0.154879, 0.033453, 1.167516, 1.093421, 1.246633, 4.629669, 3.6625e-06, 0.047448, 5.436222], ["cg20310759", "chr1", 87152629, 87152631, "RP5-1052I5.2;LINC01140", "lincRNA;protein_coding", "S_Shore", -0.185051, 0.039999, 0.831062, 0.768398, 0.898836, -4.626398, 3.720792e-06, 0.047448, 5.429365], ["cg13213755", "chr15", 73634112, 73634114, "NPTN", "protein_coding", "S_Shore", -0.281483, 0.060998, 0.754663, 0.669625, 0.850501, -4.614634, 3.937893e-06, 0.048357, 5.404736], ["cg25292309", "chr2", 58762291, 58762293, "LINC01122", "lincRNA", ".", -0.218543, 0.047674, 0.803689, 0.731994, 0.882405, -4.584132, 4.558772e-06, 0.053982, 5.341152], ["cg06585893", "chr1", 230747805, 230747807, "CAPN9;RP11-99J16__A.2", "protein_coding;antisense", ".", 0.254647, 0.055855, 1.290006, 1.156239, 1.43925, 4.559041, 5.138765e-06, 0.058752, 5.289141], ["cg11818438", "chr8", 118904290, 118904292, ".", ".", ".", -0.231781, 0.051021, 0.79312, 0.717644, 0.876532, -4.54284, 5.550141e-06, 0.059828, 5.255696], ["cg18072388", "chr19", 10323084, 10323086, "RAVER1", "protein_coding", "N_Shore", 0.183326, 0.040372, 1.201205, 1.109819, 1.300116, 4.540872, 5.602199e-06, 0.059828, 5.251642], ["cg07143043", "chr10", 131970793, 131970795, "BNIP3;RP11-45A17.2", "protein_coding;antisense", ".", 0.195374, 0.043086, 1.215766, 1.117313, 1.322894, 4.53449, 5.774288e-06, 0.059828, 5.238502], ["cg04663203", "chr17", 47851211, 47851213, "SP6", "protein_coding", "Island", 0.219539, 0.04849, 1.245503, 1.132583, 1.36968, 4.52754, 5.967433e-06, 0.059956, 5.224212], ["cg10009497", "chr7", 1055779, 1055781, "GPR146;RP11-449P15.1;C7orf50", "protein_coding;antisense", "N_Shore", 0.145012, 0.03223, 1.156053, 1.085283, 1.231438, 4.499206, 6.820783e-06, 0.066514, 5.166166], ["cg04205653", "chr21", 44805974, 44805976, "SUMO3", "protein_coding", "S_Shelf", -0.431301, 0.096295, 0.649663, 0.537925, 0.784611, -4.478968, 7.500467e-06, 0.070309, 5.124912], ["cg20703895", "chr1", 157438011, 157438013, ".", ".", ".", -0.372221, 0.083554, 0.689202, 0.585093, 0.811835, -4.454861, 8.394762e-06, 0.070309, 5.075992], ["cg03801691", "chr3", 11558341, 11558343, "VGLL4", "protein_coding", ".", 0.18009, 0.040449, 1.197325, 1.106069, 1.296111, 4.452302, 8.49547e-06, 0.070309, 5.070813], ["cg27064178", "chr22", 49784571, 49784573, "BRD1", "protein_coding", "Island", 0.152976, 0.034366, 1.165297, 1.089392, 1.246491, 4.451365, 8.5326e-06, 0.070309, 5.068919], ["cg13984533", "chr1", 37796839, 37796841, "MANEAL", "protein_coding", "S_Shelf", -0.174472, 0.039216, 0.839901, 0.777762, 0.907003, -4.448955, 8.628901e-06, 0.070309, 5.064045], ["cg04857395", "chr4", 3514909, 3514911, "LRPAP1", "protein_coding", "Island", 0.099698, 0.022431, 1.104837, 1.057316, 1.154495, 4.444574, 8.806608e-06, 0.070309, 5.055191], ["cg05480169", "chr10", 133253903, 133253905, ".", ".", ".", -0.143363, 0.032284, 0.866439, 0.813313, 0.923035, -4.440681, 8.967454e-06, 0.070309, 5.047331], ["cg15787807", "chr11", 64741618, 64741620, "RASGRP2", "protein_coding", "N_Shore", -0.180039, 0.040553, 0.835238, 0.771421, 0.904334, -4.439601, 9.012561e-06, 0.070309, 5.045152], ["cg21229337", "chr6", 36963766, 36963768, "PI16", "protein_coding", "S_Shore", 0.182904, 0.041222, 1.200699, 1.107506, 1.301733, 4.437088, 9.118406e-06, 0.070309, 5.040081], ["cg18156551", "chr11", 64224444, 64224446, "TRPT1;NUDT22", "protein_coding", "N_Shore", -0.405008, 0.091795, 0.666971, 0.557149, 0.798441, -4.412115, 1.023658e-05, 0.077137, 4.989845], ["cg00460983", "chr10", 89374183, 89374185, "LIPA", "protein_coding", ".", -0.16734, 0.038023, 0.845912, 0.785163, 0.911361, -4.400999, 1.077539e-05, 0.077958, 4.967567], ["cg19706516", "chr1", 151131392, 151131394, "RP11-68I18.10", "sense_overlapping", "Island", 0.189426, 0.04305, 1.208556, 1.110767, 1.314954, 4.400187, 1.081578e-05, 0.077958, 4.965942], ["cg10098541", "chr3", 123066802, 123066804, "PDIA5", "protein_coding", "N_Shore", -0.19123, 0.043573, 0.825943, 0.758334, 0.899579, -4.388749, 1.140043e-05, 0.079382, 4.943079], ["cg05102794", "chr16", 89101072, 89101074, "ACSF3", "protein_coding", "Island", 0.221294, 0.050495, 1.24769, 1.130121, 1.377489, 4.382473, 1.173395e-05, 0.079382, 4.930556], ["cg16429055", "chr17", 17051538, 17051540, "MPRIP", "protein_coding", ".", -0.285725, 0.065223, 0.751469, 0.661292, 0.853944, -4.380713, 1.182918e-05, 0.079382, 4.927045], ["cg00059652", "chr17", 69609356, 69609358, "LINC01483", "lincRNA", ".", -0.148976, 0.034027, 0.86159, 0.806002, 0.921011, -4.378113, 1.197111e-05, 0.079382, 4.921866], ["cg11801959", "chr13", 49549476, 49549478, "RCBTB1", "protein_coding", ".", 0.179183, 0.041079, 1.19624, 1.103701, 1.296538, 4.361868, 1.289569e-05, 0.08146, 4.889555], ["cg18437839", "chr1", 201476584, 201476586, ".", ".", ".", -0.24931, 0.057203, 0.779338, 0.696681, 0.871801, -4.358308, 1.310717e-05, 0.08146, 4.882491], ["cg05413061", "chr3", 16183799, 16183801, "GALNT15", "protein_coding", ".", -0.30629, 0.070325, 0.736173, 0.641386, 0.84497, -4.35533, 1.328664e-05, 0.08146, 4.876585], ["cg10187365", "chr21", 33488648, 33488650, "DNAJC28", "protein_coding", "N_Shelf", -0.41295, 0.094888, 0.661696, 0.549401, 0.796943, -4.35195, 1.349322e-05, 0.08146, 4.869884], ["cg21094613", "chr11", 79189641, 79189643, "TENM4", "protein_coding", ".", -0.201398, 0.046281, 0.817587, 0.746688, 0.895217, -4.351629, 1.351298e-05, 0.08146, 4.869249], ["cg16179125", "chr20", 59007065, 59007067, "CTSZ", "protein_coding", "Island", 0.170324, 0.039214, 1.185688, 1.097974, 1.28041, 4.343487, 1.402385e-05, 0.082684, 4.853133], ["cg00532502", "chr1", 246789970, 246789972, "LINC01341;KIF28P", "unitary_pseudogene;processed_transcript", "S_Shore", 0.257653, 0.05936, 1.29389, 1.151781, 1.453532, 4.340518, 1.421473e-05, 0.082684, 4.847261], ["cg00206490", "chr16", 89403320, 89403322, "ANKRD11;RP1-168P16.2", "protein_coding;sense_intronic", ".", 0.126648, 0.029221, 1.135018, 1.07184, 1.20192, 4.33417, 1.463113e-05, 0.083639, 4.834722], ["cg10057841", "chr11", 130126348, 130126350, "APLP2", "protein_coding", "S_Shelf", -0.330927, 0.076436, 0.718258, 0.618326, 0.83434, -4.329442, 1.494876e-05, 0.084006, 4.825395], ["cg18886274", "chr7", 85160554, 85160556, "SEMA3D", "protein_coding", ".", -0.151105, 0.03506, 0.859757, 0.802662, 0.920914, -4.30992, 1.633134e-05, 0.090246, 4.786978], ["cg16958939", "chr6", 7224989, 7224991, "RREB1", "protein_coding", ".", 0.120307, 0.027974, 1.127844, 1.067672, 1.191407, 4.300759, 1.70214e-05, 0.092517, 4.769005], ["cg07536737", "chr2", 98013627, 98013629, ".", ".", ".", -0.123527, 0.028752, 0.883798, 0.83537, 0.935032, -4.296295, 1.736761e-05, 0.092877, 4.76026], ["cg04092699", "chr2", 76748093, 76748095, "LRRTM4", "protein_coding", ".", -0.202902, 0.047417, 0.816359, 0.743909, 0.895864, -4.279116, 1.876372e-05, 0.09875, 4.726681], ["cg20964021", "chr7", 156641708, 156641710, "RNF32;LINC01006", "lincRNA;protein_coding", "S_Shore", -0.312158, 0.073092, 0.731866, 0.634184, 0.844593, -4.270754, 1.948132e-05, 0.099989, 4.710382], ["cg01312394", "chr2", 46882240, 46882242, ".", ".", ".", -0.185239, 0.043388, 0.830906, 0.763168, 0.904656, -4.269372, 1.96024e-05, 0.099989, 4.707691], ["cg20429981", "chr1", 1678532, 1678534, "MMP23B;CDK11B;SLC35E2B", "protein_coding", "S_Shore", 0.15522, 0.036431, 1.167915, 1.087429, 1.254358, 4.260623, 2.038579e-05, 0.10241, 4.690672], ["cg07080653", "chr4", 53508081, 53508083, "FIP1L1;LNX1-AS1;LNX1", "protein_coding;antisense", ".", -0.121593, 0.028601, 0.885508, 0.837236, 0.936564, -4.251435, 2.124054e-05, 0.105111, 4.672834], ["cg02988730", "chr1", 109624886, 109624888, "AMPD2", "protein_coding", ".", 0.259603, 0.06123, 1.296416, 1.149807, 1.461718, 4.239786, 2.237326e-05, 0.108028, 4.650271], ["cg14037250", "chr9", 134053771, 134053773, "BRD3;RP11-374P20.4", "protein_coding;antisense", "S_Shore", 0.165632, 0.039097, 1.180138, 1.093083, 1.274127, 4.236388, 2.271439e-05, 0.108028, 4.643699], ["cg00598235", "chr4", 17579056, 17579058, "LAP3", "protein_coding", "S_Shore", -0.295602, 0.069856, 0.744084, 0.648873, 0.853264, -4.231565, 2.32071e-05, 0.108028, 4.634379], ["cg21739530", "chr4", 173264916, 173264918, "GALNT7", "protein_coding", ".", 0.15037, 0.035542, 1.162264, 1.084055, 1.246115, 4.230768, 2.328946e-05, 0.108028, 4.632841], ["cg18964582", "chr1", 236393881, 236393883, "EDARADD", "protein_coding", "N_Shore", -0.140142, 0.033165, 0.869235, 0.814529, 0.927614, -4.225559, 2.383487e-05, 0.108028, 4.622787], ["cg15586054", "chr6", 165658541, 165658543, "PDE10A", "protein_coding", "N_Shore", -0.138388, 0.032751, 0.87076, 0.816622, 0.928487, -4.225525, 2.383844e-05, 0.108028, 4.622722], ["cg11823637", "chr12", 22729886, 22729888, "RP11-114G22.1", "lincRNA", ".", -0.286756, 0.067924, 0.750695, 0.657122, 0.857591, -4.221711, 2.424547e-05, 0.108028, 4.615369], ["cg13720395", "chr7", 76328584, 76328586, "YWHAG", "protein_coding", "N_Shore", 0.123802, 0.029337, 1.131792, 1.068549, 1.198777, 4.219941, 2.443658e-05, 0.108028, 4.61196], ["cg13857210", "chr9", 136107370, 136107372, ".", ".", ".", 0.132202, 0.031453, 1.141338, 1.073104, 1.213911, 4.203194, 2.631744e-05, 0.114629, 4.579756], ["cg22624391", "chr11", 1916226, 1916228, ".", ".", ".", -0.23309, 0.055525, 0.792082, 0.710407, 0.883148, -4.19793, 2.693661e-05, 0.114629, 4.569657], ["cg26223555", "chr19", 51824187, 51824189, "FPR3", "protein_coding", ".", -0.263593, 0.062801, 0.768286, 0.679308, 0.868919, -4.197266, 2.70157e-05, 0.114629, 4.568384], ["cg21617903", "chr3", 11558238, 11558240, "VGLL4", "protein_coding", ".", 0.173796, 0.041431, 1.189813, 1.097014, 1.290462, 4.194788, 2.731268e-05, 0.114629, 4.563636], ["cg12762600", "chr15", 89647922, 89647924, "KIF7", "protein_coding", "N_Shore", -0.240656, 0.05748, 0.786112, 0.702356, 0.879857, -4.18675, 2.829773e-05, 0.114898, 4.548248], ["cg16449402", "chr3", 9733040, 9733042, "BRPF1", "protein_coding", "S_Shore", -0.205645, 0.049124, 0.814122, 0.739393, 0.896403, -4.186291, 2.835496e-05, 0.114898, 4.547371], ["cg18898209", "chr11", 86889207, 86889209, "PRSS23", "protein_coding", ".", -0.16666, 0.039816, 0.846487, 0.782941, 0.91519, -4.1858, 2.841635e-05, 0.114898, 4.546432], ["cg25376409", "chr17", 80999487, 80999489, "CHMP6", "protein_coding", "Island", 0.145611, 0.034823, 1.156746, 1.080431, 1.238452, 4.181482, 2.896149e-05, 0.115359, 4.538179], ["cg23458843", "chr12", 57742661, 57742663, "AGAP2;TSPAN31", "protein_coding", "S_Shelf", -0.341694, 0.081809, 0.710566, 0.605297, 0.834142, -4.176741, 2.957152e-05, 0.115359, 4.529126], ["cg26310429", "chr17", 73028309, 73028311, "SLC39A11", "protein_coding", ".", -0.111691, 0.026746, 0.894321, 0.848647, 0.942452, -4.176008, 2.9667e-05, 0.115359, 4.527726], ["cg03092447", "chr16", 89275990, 89275992, "ANKRD11", "protein_coding", "N_Shelf", 0.244219, 0.058545, 1.276623, 1.138227, 1.431847, 4.171445, 3.026741e-05, 0.115359, 4.519025], ["cg14978242", "chr5", 80205311, 80205313, "SERINC5", "protein_coding", ".", 0.169469, 0.040626, 1.184675, 1.094002, 1.282863, 4.171426, 3.026998e-05, 0.115359, 4.518988], ["cg09890699", "chr1", 54356179, 54356181, "SSBP3", "protein_coding", "N_Shore", 0.18524, 0.044481, 1.203507, 1.103027, 1.31314, 4.164456, 3.120962e-05, 0.116469, 4.505711], ["cg17511968", "chr2", 218856441, 218856443, ".", ".", "N_Shelf", 0.130723, 0.031399, 1.139652, 1.071631, 1.211989, 4.163308, 3.136701e-05, 0.116469, 4.503527], ["cg17669009", "chr19", 51417630, 51417632, "SIGLEC10", "protein_coding", ".", 0.292968, 0.070472, 1.340399, 1.167477, 1.538934, 4.157221, 3.221429e-05, 0.116469, 4.491951], ["cg23613253", "chr11", 70594241, 70594243, "SHANK2", "protein_coding", ".", -0.102879, 0.024749, 0.902236, 0.859516, 0.94708, -4.156911, 3.225793e-05, 0.116469, 4.491363], ["cg00223135", "chr15", 43507046, 43507048, "TP53BP1", "protein_coding", "N_Shelf", -0.322306, 0.077614, 0.724476, 0.622242, 0.843508, -4.152692, 3.285873e-05, 0.116469, 4.483349], ["cg19479744", "chr6", 155701180, 155701182, ".", ".", ".", -0.247722, 0.059677, 0.780577, 0.694415, 0.87743, -4.151067, 3.30929e-05, 0.116469, 4.480265], ["cg18151487", "chr10", 102715582, 102715584, "ARL3;SFXN2", "protein_coding", "S_Shore", -0.166462, 0.040113, 0.846655, 0.78264, 0.915905, -4.149857, 3.32683e-05, 0.116469, 4.477969], ["cg04400145", "chr20", 1893035, 1893037, "SIRPA;RP5-858L17.1", "antisense;protein_coding", "N_Shore", -0.184128, 0.044377, 0.831829, 0.762535, 0.90742, -4.149147, 3.337163e-05, 0.116469, 4.476623], ["cg21170304", "chr9", 135987756, 135987758, ".", ".", "N_Shelf", -0.248373, 0.05992, 0.780069, 0.693632, 0.877277, -4.145099, 3.396673e-05, 0.116817, 4.468946], ["cg10817441", "chr6", 32855950, 32855952, "PSMB9", "protein_coding", "S_Shore", -0.229617, 0.055414, 0.794838, 0.713034, 0.886027, -4.143692, 3.41759e-05, 0.116817, 4.46628], ["cg05348324", "chr11", 119797455, 119797457, ".", ".", ".", 0.148717, 0.035946, 1.160345, 1.081409, 1.245043, 4.137238, 3.515118e-05, 0.11699, 4.45406], ["cg02348449", "chr19", 58119061, 58119063, "ZSCAN18", "protein_coding", "S_Shore", -0.170086, 0.04114, 0.843592, 0.778241, 0.91443, -4.13436, 3.559445e-05, 0.11699, 4.448618], ["cg20219911", "chr3", 52779635, 52779637, "ITIH1", "protein_coding", ".", 0.21775, 0.052675, 1.243276, 1.121322, 1.378493, 4.133833, 3.567631e-05, 0.11699, 4.44762], ["cg11064395", "chr6", 31407526, 31407528, "MICA;HCP5", "sense_overlapping;protein_coding", "S_Shelf", -0.328401, 0.079497, 0.720074, 0.616182, 0.841483, -4.130965, 3.61244e-05, 0.11699, 4.442199], ["cg17078656", "chr1", 12475057, 12475059, "VPS13D", "protein_coding", ".", -0.305353, 0.073955, 0.736863, 0.637436, 0.851799, -4.128922, 3.644686e-05, 0.11699, 4.43834], ["cg14882966", "chr2", 3651762, 3651764, ".", ".", "Island", 0.112157, 0.027173, 1.118689, 1.060668, 1.179883, 4.127519, 3.666983e-05, 0.11699, 4.435691], ["cg21750233", "chr11", 6483080, 6483082, "TIMM10B", "protein_coding", "S_Shore", -0.183778, 0.044532, 0.832121, 0.762572, 0.908012, -4.126902, 3.676829e-05, 0.11699, 4.434527], ["cg17632028", "chr4", 652752, 652754, "PDE6B", "protein_coding", "N_Shore", -0.165817, 0.040197, 0.847201, 0.783017, 0.916647, -4.12515, 3.704936e-05, 0.11699, 4.431219], ["cg06605704", "chr16", 4501145, 4501147, "HMOX2", "protein_coding", ".", -0.363037, 0.088201, 0.69556, 0.585138, 0.826821, -4.11603, 3.854542e-05, 0.119901, 4.414027], ["cg02957270", "chr17", 48430734, 48430736, "SKAP1", "protein_coding", "S_Shore", -0.132555, 0.032211, 0.875855, 0.822269, 0.932933, -4.115141, 3.869427e-05, 0.119901, 4.412353], ["cg01771213", "chr1", 57270164, 57270166, "DAB1", "protein_coding", ".", -0.197116, 0.047941, 0.821096, 0.747457, 0.901989, -4.111641, 3.928562e-05, 0.120606, 4.405766], ["cg10591771", "chr19", 3671119, 3671121, "PIP5K1C;AC004637.1", "antisense;protein_coding", "N_Shore", 0.123999, 0.030256, 1.132015, 1.066836, 1.201175, 4.098264, 4.162596e-05, 0.1251, 4.380636], ["cg08467866", "chr16", 75208215, 75208217, "CTRB2", "protein_coding", ".", 0.162189, 0.039593, 1.176082, 1.088268, 1.270982, 4.096373, 4.196727e-05, 0.1251, 4.377089], ["cg03762237", "chr2", 98085052, 98085054, ".", ".", "N_Shore", -0.159399, 0.038966, 0.852656, 0.789962, 0.920326, -4.090722, 4.30032e-05, 0.1251, 4.366499], ["cg10280339", "chr15", 65519989, 65519991, ".", ".", "S_Shelf", -0.179141, 0.043827, 0.835988, 0.767175, 0.910974, -4.087433, 4.361729e-05, 0.1251, 4.360341], ["cg19708903", "chr10", 131970803, 131970805, "BNIP3;RP11-45A17.2", "protein_coding;antisense", ".", 0.159533, 0.039033, 1.172962, 1.086574, 1.266219, 4.08714, 4.367237e-05, 0.1251, 4.359793], ["cg18586780", "chr2", 235044659, 235044661, "SH3BP4", "protein_coding", "S_Shore", -0.201007, 0.04922, 0.817907, 0.742691, 0.900741, -4.083862, 4.429328e-05, 0.1251, 4.353662], ["cg13689831", "chr1", 226987375, 226987377, "ADCK3", "protein_coding", ".", 0.173684, 0.042541, 1.189679, 1.094507, 1.293127, 4.082709, 4.451373e-05, 0.1251, 4.351506], ["cg02642565", "chr16", 4529400, 4529402, "CDIP1", "protein_coding", ".", 0.162805, 0.039879, 1.176807, 1.08833, 1.272477, 4.082512, 4.45516e-05, 0.1251, 4.351137], ["cg16822174", "chr12", 19483011, 19483013, "AEBP2", "protein_coding", ".", 0.140494, 0.034417, 1.150842, 1.07577, 1.231153, 4.082059, 4.463843e-05, 0.1251, 4.350291], ["cg08480739", "chr11", 67997549, 67997551, "UNC93B1", "protein_coding", "Island", -0.09589, 0.023496, 0.908564, 0.867673, 0.951382, -4.081186, 4.480643e-05, 0.1251, 4.34866], ["cg13846675", "chr8", 52168344, 52168346, "ST18", "protein_coding", ".", 0.10461, 0.025638, 1.110278, 1.055865, 1.167495, 4.08023, 4.499108e-05, 0.1251, 4.346874], ["cg00549798", "chr22", 40418426, 40418428, "MKL1", "protein_coding", "N_Shore", 0.17306, 0.042436, 1.188937, 1.094051, 1.292052, 4.078175, 4.539059e-05, 0.1251, 4.343034], ["cg19827511", "chr14", 93114298, 93114300, "ITPK1", "protein_coding", "N_Shore", -0.215504, 0.052861, 0.806135, 0.726796, 0.894135, -4.076826, 4.565465e-05, 0.1251, 4.340515], ["cg03601372", "chr19", 36776123, 36776125, "CTD-2162K18.3;CTD-2162K18.4", "lincRNA", "Island", -0.169699, 0.041667, 0.843919, 0.777738, 0.91573, -4.07274, 4.646324e-05, 0.126272, 4.332891], ["cg17104202", "chr1", 3245839, 3245841, "PRDM16", "protein_coding", "N_Shore", -0.138098, 0.033927, 0.871013, 0.814978, 0.930902, -4.070385, 4.693555e-05, 0.126519, 4.328498], ["cg02678414", "chr16", 2159838, 2159840, "TRAF7", "protein_coding", "S_Shelf", 0.190337, 0.046807, 1.209658, 1.103622, 1.325881, 4.066445, 4.773579e-05, 0.126549, 4.321156], ["cg23128949", "chr12", 124501577, 124501579, "NCOR2", "protein_coding", ".", 0.109372, 0.026899, 1.115578, 1.058286, 1.175971, 4.066, 4.782697e-05, 0.126549, 4.320327], ["cg13691108", "chr20", 38804603, 38804605, "PPP1R16B", "protein_coding", "N_Shore", 0.182458, 0.044888, 1.200164, 1.099086, 1.310538, 4.064712, 4.80919e-05, 0.126549, 4.317928], ["cg09140274", "chr1", 220896850, 220896852, ".", ".", "S_Shore", -0.188668, 0.046468, 0.828061, 0.755978, 0.907018, -4.060212, 4.902827e-05, 0.127304, 4.309553], ["cg10759293", "chr20", 20018581, 20018583, "NAA20", "protein_coding", "S_Shore", -0.17743, 0.043706, 0.83742, 0.768672, 0.912317, -4.059648, 4.914668e-05, 0.127304, 4.308506], ["cg19345219", "chr12", 6638238, 6638240, "ACRBP", "protein_coding", ".", 0.195853, 0.048273, 1.216349, 1.106543, 1.337051, 4.057217, 4.966085e-05, 0.127639, 4.303986], ["cg22266824", "chr4", 158996903, 158996905, "C4orf45", "protein_coding", ".", -0.177377, 0.043835, 0.837464, 0.768518, 0.912596, -4.046493, 5.199077e-05, 0.132599, 4.284074], ["cg01983248", "chr5", 43482701, 43482703, "C5orf28;RP11-159F24.3", "processed_transcript;protein_coding", "N_Shore", -0.144147, 0.035662, 0.865761, 0.807314, 0.928439, -4.042012, 5.299448e-05, 0.133242, 4.275769], ["cg23152216", "chr12", 25252150, 25252152, "KRAS", "protein_coding", "S_Shore", -0.166037, 0.04108, 0.847015, 0.78149, 0.918033, -4.041781, 5.304666e-05, 0.133242, 4.275342], ["cg15576082", "chr2", 241813194, 241813196, "NEU4;AC114730.3", "antisense;protein_coding", "Island", 0.125862, 0.031189, 1.134125, 1.066874, 1.205615, 4.035498, 5.44867e-05, 0.13583, 4.26371], ["cg00151914", "chr17", 35731530, 35731532, "TAF15;RASL10B", "protein_coding", "Island", 0.254, 0.063005, 1.289172, 1.139414, 1.458615, 4.03146, 5.543137e-05, 0.136715, 4.256244], ["cg27658601", "chr17", 82836204, 82836206, "TBCD;ZNF750", "protein_coding", "S_Shelf", -0.135916, 0.033722, 0.872916, 0.817086, 0.93256, -4.030467, 5.566604e-05, 0.136715, 4.25441], ["cg13360395", "chr2", 123907703, 123907705, ".", ".", ".", -0.150679, 0.037404, 0.860123, 0.799323, 0.925549, -4.028386, 5.616101e-05, 0.136916, 4.250565], ["cg26373351", "chr12", 11878369, 11878371, "ETV6", "protein_coding", ".", 0.262772, 0.065328, 1.30053, 1.14423, 1.478181, 4.022367, 5.761617e-05, 0.139438, 4.239456], ["cg11887376", "chr5", 1118554, 1118556, ".", ".", ".", 0.130733, 0.032527, 1.139663, 1.069275, 1.214685, 4.019212, 5.839305e-05, 0.140294, 4.233639], ["cg03596113", "chr1", 3213404, 3213406, "PRDM16", "protein_coding", ".", -0.197555, 0.049192, 0.820735, 0.7453, 0.903805, -4.016042, 5.918359e-05, 0.141065, 4.227799], ["cg27282206", "chr7", 158386620, 158386622, "PTPRN2", "protein_coding", ".", -0.205892, 0.051287, 0.813921, 0.736083, 0.89999, -4.014528, 5.956481e-05, 0.141065, 4.22501], ["cg19357999", "chr12", 121223958, 121223960, "P2RX4", "protein_coding", ".", 0.163436, 0.04074, 1.17755, 1.087181, 1.275431, 4.011735, 6.027416e-05, 0.141733, 4.219869], ["cg14458815", "chr22", 23141813, 23141815, "RSPH14", "protein_coding", "Island", 0.147112, 0.036719, 1.158484, 1.078039, 1.244931, 4.006403, 6.16503e-05, 0.142976, 4.210065], ["cg02905663", "chr8", 118111467, 118111469, "EXT1", "protein_coding", "N_Shore", -0.293109, 0.073161, 0.745941, 0.646293, 0.860953, -4.006346, 6.166536e-05, 0.142976, 4.209959], ["cg21262032", "chr1", 154465216, 154465218, "IL6R", "protein_coding", ".", 0.142108, 0.035524, 1.152702, 1.075175, 1.235819, 4.000378, 6.324132e-05, 0.145466, 4.198999], ["cg20378520", "chr12", 125219264, 125219266, "TMEM132B", "protein_coding", ".", -0.185845, 0.046493, 0.830403, 0.758078, 0.909627, -3.997281, 6.407436e-05, 0.145466, 4.193316], ["cg21093005", "chr17", 82584038, 82584040, "FOXK2", "protein_coding", "Island", 0.133647, 0.033437, 1.142989, 1.070486, 1.220404, 3.997031, 6.41418e-05, 0.145466, 4.192859], ["cg24343287", "chr19", 58058617, 58058619, "ZNF135", "protein_coding", "N_Shore", -0.203005, 0.050829, 0.816274, 0.738873, 0.901783, -3.993855, 6.500765e-05, 0.145466, 4.187036], ["cg03272932", "chr16", 87555286, 87555288, ".", ".", ".", -0.134639, 0.033719, 0.874032, 0.818136, 0.933746, -3.99298, 6.52481e-05, 0.145466, 4.185432], ["cg00481639", "chr2", 16332838, 16332840, ".", ".", ".", -0.302499, 0.075766, 0.738969, 0.636992, 0.857272, -3.992532, 6.537164e-05, 0.145466, 4.184611], ["cg03922381", "chr10", 69217453, 69217455, "RP11-227H15.4", "antisense", ".", -0.180505, 0.045244, 0.834848, 0.764004, 0.912262, -3.989577, 6.619125e-05, 0.146308, 4.179199], ["cg00057663", "chr2", 105329171, 105329173, "TGFBRAP1;AC012360.4", "lincRNA;protein_coding", "N_Shore", 0.307309, 0.077065, 1.359761, 1.169136, 1.581468, 3.987674, 6.672425e-05, 0.146509, 4.175716], ["cg06417460", "chr11", 61102287, 61102289, "CD5", "protein_coding", ".", 0.176376, 0.044256, 1.192886, 1.093775, 1.300978, 3.985333, 6.738561e-05, 0.146555, 4.171433], ["cg10172675", "chr8", 103120454, 103120456, "LINC01181", "antisense", ".", 0.153032, 0.038408, 1.165362, 1.080856, 1.256475, 3.984363, 6.766127e-05, 0.146555, 4.16966], ["cg03071245", "chr13", 36888896, 36888898, "SMAD9", "protein_coding", ".", -0.275677, 0.069237, 0.759058, 0.662736, 0.86938, -3.981642, 6.844094e-05, 0.146555, 4.164684], ["cg16721321", "chr4", 1208907, 1208909, "SPON2;CTBP1-AS", "antisense;protein_coding", "N_Shore", 0.091089, 0.022879, 1.095367, 1.047334, 1.145602, 3.981391, 6.851305e-05, 0.146555, 4.164227], ["cg04415132", "chr19", 48718241, 48718243, "MAMSTR", "protein_coding", "N_Shore", -0.286318, 0.071976, 0.751024, 0.65221, 0.864808, -3.977966, 6.950728e-05, 0.147728, 4.15797], ["cg04713531", "chr7", 158404595, 158404597, "PTPRN2", "protein_coding", "N_Shore", -0.173031, 0.043539, 0.841111, 0.772313, 0.916038, -3.974207, 7.06142e-05, 0.148432, 4.151108], ["cg21865657", "chr19", 3687752, 3687754, "PIP5K1C", "protein_coding", "N_Shore", 0.115618, 0.029095, 1.122567, 1.060344, 1.188443, 3.973804, 7.073365e-05, 0.148432, 4.150374], ["cg09796602", "chr16", 23144881, 23144883, "USP31", "protein_coding", "N_Shelf", -0.168246, 0.042365, 0.845146, 0.777805, 0.918316, -3.971398, 7.145219e-05, 0.148997, 4.145984], ["cg13440692", "chr1", 1250976, 1250978, "RP5-902P8.12", "lincRNA", "N_Shelf", 0.152534, 0.038536, 1.164782, 1.080046, 1.256166, 3.958172, 7.552544e-05, 0.156506, 4.121907], ["cg13707005", "chr10", 11204664, 11204666, "CELF2", "protein_coding", ".", 0.112464, 0.028481, 1.119033, 1.058278, 1.183275, 3.948759, 7.855738e-05, 0.161778, 4.104813], ["cg05305099", "chr16", 57623601, 57623603, "RP11-405F3.5;GPR56", "protein_coding;antisense", ".", -0.242716, 0.061617, 0.784494, 0.69525, 0.885194, -3.939085, 8.179299e-05, 0.164313, 4.087284], ["cg24743301", "chr2", 112187928, 112187930, "RP11-399B17.1;FBLN7", "TEC;protein_coding", "S_Shore", 0.120844, 0.030686, 1.128449, 1.062581, 1.198399, 3.938113, 8.212503e-05, 0.164313, 4.085524], ["cg24457649", "chr10", 71712586, 71712588, "CDH23;C10orf105", "protein_coding", ".", 0.148701, 0.037802, 1.160325, 1.077464, 1.24956, 3.933664, 8.366069e-05, 0.164313, 4.077479], ["cg16451525", "chr16", 89100873, 89100875, "ACSF3", "protein_coding", "Island", 0.119891, 0.030481, 1.127375, 1.061996, 1.196778, 3.933352, 8.376938e-05, 0.164313, 4.076915], ["cg00550080", "chr6", 30157014, 30157016, "TRIM10", "protein_coding", ".", -0.171149, 0.043522, 0.842696, 0.773793, 0.917735, -3.93245, 8.408444e-05, 0.164313, 4.075284], ["cg24018814", "chr12", 132771529, 132771531, "GOLGA3", "protein_coding", "N_Shore", 0.127079, 0.032323, 1.135507, 1.065802, 1.209771, 3.931536, 8.440474e-05, 0.164313, 4.073633], ["cg19461880", "chr11", 6483135, 6483137, "TIMM10B", "protein_coding", "S_Shore", -0.167934, 0.042717, 0.84541, 0.777511, 0.919238, -3.931331, 8.4477e-05, 0.164313, 4.073262], ["cg13844237", "chr9", 125522544, 125522546, "MAPKAP1", "protein_coding", ".", -0.185201, 0.047113, 0.830937, 0.757644, 0.91132, -3.930992, 8.459616e-05, 0.164313, 4.072649], ["cg15346842", "chr2", 59249930, 59249932, "AC007131.2;AC007131.1;RP11-444A22.1", "lincRNA", ".", 0.119418, 0.030403, 1.126841, 1.061655, 1.196028, 3.927861, 8.570455e-05, 0.164313, 4.066996], ["cg01881444", "chr3", 69012423, 69012425, "EOGT", "protein_coding", "N_Shore", -0.371237, 0.094521, 0.68988, 0.573214, 0.830291, -3.927553, 8.581442e-05, 0.164313, 4.06644], ["cg21932513", "chr1", 17425477, 17425479, "RCC2", "protein_coding", ".", 0.251858, 0.064129, 1.286413, 1.134472, 1.458705, 3.927348, 8.588775e-05, 0.164313, 4.066069], ["cg15028995", "chr19", 46124861, 46124863, "IGFL3;AC007193.6", "lincRNA;protein_coding", ".", -0.139848, 0.035615, 0.869491, 0.810866, 0.932354, -3.926604, 8.615366e-05, 0.164313, 4.064726], ["cg06806210", "chr1", 228215425, 228215427, "OBSCN", "protein_coding", "N_Shore", -0.240675, 0.061297, 0.786097, 0.697109, 0.886446, -3.926388, 8.62309e-05, 0.164313, 4.064337], ["cg04516952", "chr19", 50968785, 50968787, "KLK6;CTB-147C22.9", "antisense;protein_coding", ".", 0.22097, 0.056325, 1.247286, 1.116919, 1.392869, 3.9231, 8.741685e-05, 0.165029, 4.058405], ["cg05197036", "chr5", 141444469, 141444471, "PCDHGA7;PCDHGB1;PCDHGB4;PCDHGA12;PCDHGA8;PCDHGB5;PCDHGA9;PCDHGB2;PCDHGA10;PCDHGB7;PCDHGB3;PCDHGA2;PCDHGA11;PCDHGA5;PCDHGA3;PCDHGA4;PCDHGB6;PCDHGA6;PCDHGA1", "protein_coding", "S_Shelf", -0.314059, 0.080064, 0.730476, 0.624389, 0.854588, -3.92259, 8.760221e-05, 0.165029, 4.057485], ["cg02205013", "chr16", 1708796, 1708798, "MAPK8IP3", "protein_coding", "S_Shelf", 0.116309, 0.029718, 1.123343, 1.059782, 1.190716, 3.913799, 9.085509e-05, 0.169796, 4.041651], ["cg15081720", "chr1", 119323994, 119323996, "RP11-418J17.3", "lincRNA", "N_Shelf", -0.148882, 0.038062, 0.86167, 0.799728, 0.92841, -3.911547, 9.170696e-05, 0.169796, 4.037598], ["cg04010915", "chr12", 110109247, 110109249, ".", ".", ".", -0.134552, 0.034407, 0.874107, 0.817104, 0.935087, -3.910593, 9.206989e-05, 0.169796, 4.035882], ["cg25737891", "chr2", 11921150, 11921152, "AC096559.1", "lincRNA", ".", -0.157998, 0.040413, 0.853852, 0.78883, 0.924234, -3.909611, 9.244484e-05, 0.169796, 4.034117], ["cg09041251", "chr2", 10244393, 10244395, ".", ".", ".", -0.192953, 0.049368, 0.82452, 0.748479, 0.908287, -3.908497, 9.287223e-05, 0.169796, 4.032114], ["cg15304012", "chr6", 166463001, 166463003, "RPS6KA2-IT1;RPS6KA2", "protein_coding;sense_intronic", ".", 0.155938, 0.039917, 1.168754, 1.080801, 1.263864, 3.906579, 9.361208e-05, 0.169796, 4.028668], ["cg17389427", "chr3", 129612142, 129612144, ".", ".", ".", 0.312048, 0.079883, 1.366221, 1.168219, 1.597782, 3.906307, 9.371758e-05, 0.169796, 4.028179], ["cg22560020", "chr1", 246789611, 246789613, "LINC01341;KIF28P", "unitary_pseudogene;processed_transcript", "S_Shore", 0.132386, 0.033933, 1.141549, 1.068097, 1.220052, 3.901406, 9.563552e-05, 0.170281, 4.019381], ["cg24355048", "chr14", 24575821, 24575823, "CTSG", "protein_coding", ".", -0.123901, 0.031761, 0.883468, 0.830148, 0.940212, -3.900987, 9.580136e-05, 0.170281, 4.018628], ["cg06650260", "chr1", 26913986, 26913988, "NUDC;NR0B2", "protein_coding", ".", -0.251502, 0.064477, 0.777632, 0.685316, 0.882383, -3.900649, 9.593515e-05, 0.170281, 4.018022], ["cg19597170", "chr12", 130134001, 130134003, ".", ".", ".", -0.141263, 0.036224, 0.868261, 0.808753, 0.932147, -3.899698, 9.631283e-05, 0.170281, 4.016316], ["cg05800336", "chr10", 131453805, 131453807, ".", ".", ".", 0.166671, 0.042756, 1.181365, 1.086402, 1.284629, 3.898215, 9.690453e-05, 0.170281, 4.013656], ["cg13696545", "chr3", 55750825, 55750827, "ERC2", "protein_coding", ".", -0.120629, 0.030948, 0.886363, 0.834197, 0.941791, -3.897809, 9.706687e-05, 0.170281, 4.012929], ["cg25794538", "chr15", 99562783, 99562785, ".", ".", "N_Shelf", 0.206989, 0.053185, 1.229969, 1.108213, 1.365103, 3.891881, 9.947018e-05, 0.173579, 4.002307]];

    // Define the dt_args
    let dt_args = {"autoWidth": false, "order": []};
    dt_args["data"] = data;

    $(document).ready(function () {
        
        $('#84760c52-d468-4435-a071-95752f12288a').DataTable(dt_args);
    });
</script>
</div>
</div></div>
</div>
</section>
<section id="train-coxph-lasso-model-with-the-selected-cpgs">
<h2>Train CoxPH+Lasso model with the selected CpGs<a class="headerlink" href="#train-coxph-lasso-model-with-the-selected-cpgs" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">utils.cox_lasso</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># raw_coefs = train_coxph_lasso(df=clinical_data4.copy(),</span>
<span class="c1">#                                 event= &#39;os.evnt at 5y&#39;,</span>
<span class="c1">#                                 time= &#39;os.time at 5y&#39;,</span>
<span class="c1">#                                 train_x=ewas_top_cpgs_os,</span>
<span class="c1">#                                 cv_n_split=10,</span>
<span class="c1">#                                 loops=1000)</span>
<span class="c1"># raw_coefs.to_csv(output_path + &#39;ewas_dmr/ewas_cog_os_raw_coefs_newrisk.csv&#39;)</span>


<span class="n">raw_coefs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_raw_coefs_newrisk.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">mean_coefs</span> <span class="o">=</span> <span class="n">set_cutoff</span><span class="p">(</span><span class="n">coefs</span><span class="o">=</span><span class="n">raw_coefs</span><span class="p">,</span><span class="n">threshold</span><span class="o">=</span><span class="mf">0.99</span><span class="p">)</span>
                                
<span class="n">plot_nonzero_coef_freq</span><span class="p">(</span><span class="n">raw_coefs</span><span class="p">,</span> <span class="n">mean_coefs</span><span class="p">,</span> <span class="n">threshold</span><span class="o">=</span><span class="mf">0.99</span><span class="p">,</span> <span class="n">savefig</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">fig_path</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span><span class="mi">3</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/59abcb063eb0e2b4814d12e251ae0d9e71e055564f082baa991354c39668d3e9.png" src="../_images/59abcb063eb0e2b4814d12e251ae0d9e71e055564f082baa991354c39668d3e9.png" />
</div>
</div>
</section>
<section id="apply-signature-to-discovery-dataset-and-set-high-low-cutoff">
<h2>Apply signature to discovery dataset and set high/low cutoff<a class="headerlink" href="#apply-signature-to-discovery-dataset-and-set-high-low-cutoff" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">score_name</span> <span class="o">=</span> <span class="s1">&#39;MethylScoreAML&#39;</span>

<span class="n">df5</span><span class="p">,</span> <span class="n">threshold</span> <span class="o">=</span> <span class="n">generate_coxph_score</span><span class="p">(</span><span class="n">coef_mean</span><span class="o">=</span><span class="n">mean_coefs</span><span class="p">,</span>
                                        <span class="n">x</span><span class="o">=</span><span class="n">ewas_top_cpgs_os</span><span class="p">,</span>
                                        <span class="n">df</span><span class="o">=</span><span class="n">clinical_data4</span><span class="p">,</span>
                                        <span class="n">score_name</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                                        <span class="n">train_test</span><span class="o">=</span><span class="s2">&quot;train&quot;</span><span class="p">,</span>
                                        <span class="n">rpart_outcome</span><span class="o">=</span><span class="s1">&#39;os.time at 5y&#39;</span><span class="p">)</span>

<span class="c1"># df.to_excel(output_path + &#39;multivariate_cox_lasso/ewas_cog_os_MethylScoreAML_Px.xlsx&#39;)</span>

<span class="c1"># Plot results</span>

<span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df5</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">hue</span> <span class="o">=</span> <span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span> <span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">threshold</span> <span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">({</span><span class="s1">&#39;Cutoff: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">threshold</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)):</span><span class="n">threshold</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; in discovery cohort&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Continuous score cut at the value of -1.5341
</pre></div>
</div>
<img alt="../_images/6d11d6bee5367df4aa906ddae721086a65c931ea52e6eae64173c212c73baa00.png" src="../_images/6d11d6bee5367df4aa906ddae721086a65c931ea52e6eae64173c212c73baa00.png" />
</div>
</div>
</section>
<section id="apply-signature-to-validation-dataset">
<h2>Apply signature to validation dataset<a class="headerlink" href="#apply-signature-to-validation-dataset" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load test data and pick only columns that are in mean_coefs</span>
<span class="n">df_test</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;201samples_427375cpgs_withbatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()[</span><span class="n">mean_coefs</span><span class="o">.</span><span class="n">index</span><span class="p">]</span>

<span class="c1"># Transform from beta-values to m-values</span>
<span class="n">df_test2</span> <span class="o">=</span> <span class="n">df_test</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>

<span class="c1"># Select samples from `clincal_data` that are in `df_test`</span>
<span class="n">clinical_data_test</span> <span class="o">=</span> <span class="n">clinical_data</span><span class="p">[</span><span class="n">clinical_data</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df_test</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># apply signature to test data using `threshold` from training data</span>
<span class="n">clinical_data_test2</span> <span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">generate_coxph_score</span><span class="p">(</span><span class="n">coef_mean</span><span class="o">=</span><span class="n">mean_coefs</span><span class="p">,</span>
                                        <span class="n">x</span><span class="o">=</span><span class="n">df_test2</span><span class="p">,</span>
                                        <span class="n">df</span><span class="o">=</span><span class="n">clinical_data_test</span><span class="p">,</span>
                                        <span class="n">score_name</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span>
                                        <span class="n">train_test</span><span class="o">=</span><span class="n">threshold</span><span class="p">)</span>

<span class="c1"># Plot the results</span>
<span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">sns</span><span class="o">.</span><span class="n">histplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">clinical_data_test2</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">score_name</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">hue</span> <span class="o">=</span> <span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; Categorical&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">threshold</span> <span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">({</span><span class="s1">&#39;Cutoff: &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">threshold</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)):</span><span class="n">threshold</span><span class="p">})</span>
<span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="n">score_name</span> <span class="o">+</span> <span class="s1">&#39; in validation cohort&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Continuous score cut at the value of -1.5341
</pre></div>
</div>
<img alt="../_images/e0fe7321d6819f1d9939b5ec4704e85301ae89695b80fc3c66abc4017afd58a4.png" src="../_images/e0fe7321d6819f1d9939b5ec4704e85301ae89695b80fc3c66abc4017afd58a4.png" />
</div>
</div>
</section>
<section id="merge-and-save-files">
<h2>Merge and save files<a class="headerlink" href="#merge-and-save-files" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">final_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">clinical_data_test2</span><span class="p">,</span> <span class="n">clinical_data4</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">final_df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="o">-</span><span class="mi">3</span><span class="p">:]</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/signature_results.xlsx&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="watermark">
<h2>Watermark<a class="headerlink" href="#watermark" title="Link to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Python implementation: CPython
Python version       : 3.8.18
IPython version      : 8.12.2

methylize: 1.1.1
pandas   : 2.0.2

Compiler    : GCC 11.4.0
OS          : Linux
Release     : 5.15.133.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 32
Architecture: 64bit
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./source"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="4_clinical_data.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Clinical Data</p>
      </div>
    </a>
    <a class="right-next"
       href="5_classifiers.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Classifiers</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-samples-only">Select diagnostic samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas">Run CoxPH-EWAS</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-cpgs-based-on-suggested-significance-p-values-1-10e-5">Select CpGs based on suggested-significance p-values (1*10e-5)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#train-coxph-lasso-model-with-the-selected-cpgs">Train CoxPH+Lasso model with the selected CpGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-discovery-dataset-and-set-high-low-cutoff">Apply signature to discovery dataset and set high/low cutoff</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#apply-signature-to-validation-dataset">Apply signature to validation dataset</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#merge-and-save-files">Merge and save files</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Marchi et al. @ Lamba Lab @ University of Florida.
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>