

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Epigenome Wide Association Study &#8212; MethylScoreAML</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'models/2_Univariate_CoxPH-EWAS_v4';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="MethylScoreAML Px Development" href="3.1_MethylScoreAML_Px_Development_v5.html" />
    <link rel="prev" title="Reclassification to Current Clinical Guidelines" href="../data-processing/reclassification_strategy.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="../index.html">
  
  
  
  
  
    <p class="title logo__title">MethylScoreAML</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../index.html">
                    The DNA methylome refines clinical diagnosis and prognosis of Acute Myeloid Leukemia
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Data Processing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../data-processing/discovery_dataset.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data-processing/test_dataset.html">Validation Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data-processing/reclassification_strategy.html">Reclassification to Current Clinical Guidelines</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">AML Prognostic Model</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Epigenome Wide Association Study</a></li>
<li class="toctree-l1"><a class="reference internal" href="3.1_MethylScoreAML_Px_Development_v5.html">MethylScoreAML Px Development</a></li>
<li class="toctree-l1"><a class="reference internal" href="3.1_MethylScoreAML_Px_Development_v6.html">Px Score Discovery - Updated Risk</a></li>
<li class="toctree-l1"><a class="reference internal" href="3.2_MethylScoreAML_Px_Validation_Peds_v2.html">MethylScoreAML Px Validation</a></li>
<li class="toctree-l1"><a class="reference internal" href="3.2_MethylScoreAML_Px_Validation_Peds_v3.html">Px Score Validation - Updated Risk</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">AML Diagnostic Model</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="dx-score_new.html">Diagnostic Classifier</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Nanopore Sequencing Testing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../tests/8_Nanopore_Sample.html">Applying models to Nanopore data</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Benchmarks</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-n_components-pacmap.html">Dx score - various components</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-pedsAML-only.html">Dx score - pedsAML only</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-with-looser-filtering.html">Dx score - less strict pre-processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-without-ALL-relapse.html">Dx score - without relapse and ALL</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/dx-score-without-relapse.html">Dx score - without relapse</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/models/2_Univariate_CoxPH-EWAS_v4.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Epigenome Wide Association Study</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-samples-only">Select diagnostic samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-dataframe-for-other-analyses">Save dataframe for other analyses</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas">Run CoxPH-EWAS</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#import-custom-functions-adapted-from-methylize">Import custom functions adapted from <code class="docutils literal notranslate"><span class="pre">methylize</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-survival-previous-risk-group">Overall survival - previous risk group</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-survival-updated-rr-risk-group">Overall survival - updated RR risk group</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#event-free-survival-previous-risk-group">Event-free survival - previous risk group</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#event-free-survival-updated-risk-group">Event-free survival - updated risk group</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-statistically-significant-cpgs">Select statistically significant CpGs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#threshold-based-on-suggested-significance-p-values-1-10e-6">Threshold based on suggested-significance p-values (1*10e-6)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#threshold-based-on-suggested-significance-p-values-1-10e-5">Threshold based on suggested-significance p-values (1*10e-5)</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-genes">Fetch genes</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#os-suggested-significance-probes">OS suggested-significance probes:</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#efs-suggested-significance-probes">EFS suggested-significance probes:</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#save-top-probes">Save top probes</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="epigenome-wide-association-study">
<h1><a class="toc-backref" href="#id1" role="doc-backlink">Epigenome Wide Association Study</a><a class="headerlink" href="#epigenome-wide-association-study" title="Permalink to this heading">#</a></h1>
<p>Epigenome-wide association study (EWAS) using Cox proportional hazards regression (CoxPH) model with event time as the outcome and DNA methylation as the predictor.</p>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#epigenome-wide-association-study" id="id1">Epigenome Wide Association Study</a></p>
<ul>
<li><p><a class="reference internal" href="#where-the-data-at" id="id2">Where the data at?</a></p></li>
<li><p><a class="reference internal" href="#load-data" id="id3">Load data</a></p></li>
<li><p><a class="reference internal" href="#remove-samples-based-on-certain-clinical-features" id="id4">Remove samples based on certain clinical features</a></p>
<ul>
<li><p><a class="reference internal" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" id="id5">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></p></li>
<li><p><a class="reference internal" href="#select-diagnostic-samples-only" id="id6">Select diagnostic samples only</a></p></li>
<li><p><a class="reference internal" href="#remove-duplicate-samples" id="id7">Remove duplicate samples</a></p></li>
<li><p><a class="reference internal" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" id="id8">Match samples in clinical data to samples in methylation data</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#transform-to-m-values" id="id9">Transform to M-values</a></p></li>
<li><p><a class="reference internal" href="#perform-batch-correction" id="id10">Perform batch correction</a></p>
<ul>
<li><p><a class="reference internal" href="#evaluate-batch-effects" id="id11">Evaluate batch effects</a></p></li>
<li><p><a class="reference internal" href="#save-dataframe-for-other-analyses" id="id12">Save dataframe for other analyses</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#run-coxph-ewas" id="id13">Run CoxPH-EWAS</a></p>
<ul>
<li><p><a class="reference internal" href="#import-custom-functions-adapted-from-methylize" id="id14">Import custom functions adapted from <code class="docutils literal notranslate"><span class="pre">methylize</span></code></a></p></li>
<li><p><a class="reference internal" href="#overall-survival-previous-risk-group" id="id15">Overall survival - previous risk group</a></p></li>
<li><p><a class="reference internal" href="#overall-survival-updated-rr-risk-group" id="id16">Overall survival - updated RR risk group</a></p></li>
<li><p><a class="reference internal" href="#event-free-survival-previous-risk-group" id="id17">Event-free survival - previous risk group</a></p></li>
<li><p><a class="reference internal" href="#event-free-survival-updated-risk-group" id="id18">Event-free survival - updated risk group</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#select-statistically-significant-cpgs" id="id19">Select statistically significant CpGs</a></p>
<ul>
<li><p><a class="reference internal" href="#threshold-based-on-suggested-significance-p-values-1-10e-6" id="id20">Threshold based on suggested-significance p-values (1*10e-6)</a></p></li>
<li><p><a class="reference internal" href="#threshold-based-on-suggested-significance-p-values-1-10e-5" id="id21">Threshold based on suggested-significance p-values (1*10e-5)</a></p>
<ul>
<li><p><a class="reference internal" href="#fetch-genes" id="id22">Fetch genes</a></p></li>
<li><p><a class="reference internal" href="#os-suggested-significance-probes" id="id23">OS suggested-significance probes:</a></p></li>
<li><p><a class="reference internal" href="#efs-suggested-significance-probes" id="id24">EFS suggested-significance probes:</a></p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference internal" href="#save-top-probes" id="id25">Save top probes</a></p></li>
<li><p><a class="reference internal" href="#watermark" id="id26">Watermark</a></p></li>
</ul>
</li>
</ul>
</nav>
<section id="where-the-data-at">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Where the data at?</a><a class="headerlink" href="#where-the-data-at" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;../&#39;</span><span class="p">)</span>

<span class="n">mount</span> <span class="o">=</span> <span class="s1">&#39;/mnt/d/&#39;</span>
<span class="n">input_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;MethylScore/Intermediate_Files/&#39;</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;MethylScore/Processed_Data/&#39;</span>
<span class="n">zhou2016_probes_path</span> <span class="o">=</span> <span class="s1">&#39;/mnt/d/MethylScore/UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="load-data">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Load data</a><a class="headerlink" href="#load-data" title="Permalink to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read df_discovery and df_test</span>
<span class="n">df_methyl</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span>
    <span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;3308samples_333059cpgs_nobatchcorrection_bvalues.pkl&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>

<span class="c1"># Load clinical data</span>
<span class="n">df_labels</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">input_path</span><span class="o">+</span><span class="s1">&#39;discovery_clinical_data.csv&#39;</span><span class="p">,</span>
                                      <span class="n">low_memory</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-samples-based-on-certain-clinical-features">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">Remove samples based on certain clinical features</a><a class="headerlink" href="#remove-samples-based-on-certain-clinical-features" title="Permalink to this heading">#</a></h2>
<section id="select-samples-from-aaml1031-0531-and-03p1-clinical-trials">
<h3><a class="toc-backref" href="#id5" role="doc-backlink">Select samples from AAML1031, 0531, and 03P1 clinical trials</a><a class="headerlink" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df1</span> <span class="o">=</span> <span class="n">df_labels</span><span class="p">[</span><span class="n">df_labels</span><span class="p">[</span><span class="s1">&#39;Clinical Trial&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;AAML0531&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML1031&#39;</span><span class="p">,</span> <span class="s1">&#39;AAML03P1&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df_labels</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2027 samples were removed. 1281 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="select-diagnostic-samples-only">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Select diagnostic samples only</a><a class="headerlink" href="#select-diagnostic-samples-only" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df2</span> <span class="o">=</span> <span class="n">df1</span><span class="p">[</span><span class="n">df1</span><span class="p">[</span><span class="s1">&#39;Sample Type&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span>
    <span class="p">[</span><span class="s1">&#39;Diagnosis&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Bone Marrow&#39;</span><span class="p">,</span> <span class="s1">&#39;Primary Blood Derived Cancer - Peripheral Blood&#39;</span><span class="p">])]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df1</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>332 samples were removed. 949 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="remove-duplicate-samples">
<h3><a class="toc-backref" href="#id7" role="doc-backlink">Remove duplicate samples</a><a class="headerlink" href="#remove-duplicate-samples" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df3</span> <span class="o">=</span> <span class="n">df2</span><span class="p">[</span><span class="o">~</span><span class="n">df2</span><span class="p">[</span><span class="s1">&#39;Patient_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="s1">&#39;last&#39;</span><span class="p">)]</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples were removed. </span><span class="si">{</span><span class="n">df3</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>9 samples were removed. 940 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="match-samples-in-clinical-data-to-samples-in-methylation-data">
<h3><a class="toc-backref" href="#id8" role="doc-backlink">Match samples in clinical data to samples in methylation data</a><a class="headerlink" href="#match-samples-in-clinical-data-to-samples-in-methylation-data" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Match samples in clinical data to samples in methylation data</span>
<span class="n">df_methyl_filtered</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">:]</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Samples in clinical data matched to samples in methylation data.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Samples in clinical data matched to samples in methylation data.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="transform-to-m-values">
<h2><a class="toc-backref" href="#id9" role="doc-backlink">Transform to M-values</a><a class="headerlink" href="#transform-to-m-values" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">math</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>


<span class="k">def</span> <span class="nf">beta2m</span><span class="p">(</span><span class="n">val</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Transfrom beta-values into m-values&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">math</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">val</span><span class="o">/</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">val</span><span class="p">))</span>

<span class="n">x_train_m</span> <span class="o">=</span> <span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">vectorize</span><span class="p">(</span><span class="n">beta2m</span><span class="p">))</span>

<span class="c1"># Get correlation between Beta and M Values</span>
<span class="n">corr</span> <span class="o">=</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">corrwith</span><span class="p">(</span><span class="n">df_methyl_filtered</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Glue them into same dataframe for vizualization purposes</span>
<span class="n">beta_to_m_values</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">x_train_m</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">df_methyl_filtered</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span><span class="n">corr</span><span class="p">],</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                             <span class="n">keys</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="s1">&#39;Correlation&#39;</span><span class="p">],</span>
                             <span class="n">join</span><span class="o">=</span> <span class="s1">&#39;inner&#39;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="n">hue</span><span class="p">,</span><span class="n">s</span><span class="p">):</span>
    
    <span class="c1"># Define theme</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>
    <span class="n">f</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">despine</span><span class="p">(</span><span class="n">f</span><span class="p">,</span> <span class="n">left</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">bottom</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

    <span class="c1"># Define scatterplot</span>

    <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
                    <span class="n">hue</span><span class="o">=</span><span class="n">hue</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;flare&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="n">s</span><span class="p">,</span>
                    <span class="n">sizes</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">8</span><span class="p">),</span> <span class="n">linewidth</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span>
                    <span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1"># Calculate Pearson’s correlation coefficient (r)</span>
    <span class="c1"># and its two-tailed p-value (p)</span>

    <span class="n">r</span><span class="p">,</span><span class="n">p</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">x</span><span class="p">],</span><span class="n">df</span><span class="p">[</span><span class="n">y</span><span class="p">])</span>

    <span class="c1"># Annotate r and p into the plot</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="sa">fr</span><span class="s1">&#39;$\rho$: </span><span class="si">{</span><span class="n">r</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">, p-value: </span><span class="si">{</span><span class="n">p</span><span class="si">:</span><span class="s1">.5f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span>
                <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mf">6.0</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">),</span> <span class="n">xycoords</span><span class="o">=</span><span class="s1">&#39;data&#39;</span><span class="p">,</span>
                <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span>
                <span class="n">bbox</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;boxstyle&#39;</span><span class="p">:</span> <span class="s1">&#39;round&#39;</span><span class="p">,</span> <span class="s1">&#39;facecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;none&#39;</span><span class="p">,</span>
                <span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span> <span class="s1">&#39;lightgray&#39;</span><span class="p">})</span>

    <span class="c1"># Define plot specs</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="n">y</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;Concordance by Pearson&#39;s correlation coefficient ($\rho$) in &quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">))</span> <span class="o">+</span> <span class="s1">&#39; CpG probes&#39;</span><span class="p">,</span>
               <span class="n">fontsize</span> <span class="o">=</span> <span class="mi">12</span><span class="p">)</span> 

    <span class="c1"># Save plot figure</span>
    
    <span class="c1"># plt.savefig(&#39;../Figures/Beta_to_M_Values_&#39; + str(len(df)) + &#39;_scatterplot&#39; + &#39;.png&#39;,</span>
    <span class="c1">#              bbox_inches=&#39;tight&#39;, dpi=300)</span>

    <span class="k">return</span><span class="p">(</span><span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">())</span>

<span class="n">draw_scatterplot</span><span class="p">(</span><span class="n">df</span><span class="o">=</span><span class="n">beta_to_m_values</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;M Values&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;Beta Values&#39;</span><span class="p">,</span><span class="n">hue</span><span class="o">=</span><span class="s1">&#39;Correlation&#39;</span><span class="p">,</span><span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/d0d0b604ca852adcacf35ff441f4da46ac12554d0b89869e82668e3633d70d14.png" src="../_images/d0d0b604ca852adcacf35ff441f4da46ac12554d0b89869e82668e3633d70d14.png" />
</div>
</div>
</section>
<section id="perform-batch-correction">
<h2><a class="toc-backref" href="#id10" role="doc-backlink">Perform batch correction</a><a class="headerlink" href="#perform-batch-correction" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>pyCombat</strong> is a Python tool for batch effects correction in high-throughput molecular data using empirical Bayes methods</p></li>
<li><p><strong>Github</strong>: <a class="reference external" href="https://epigenelabs.github.io/pyComBat/">https://epigenelabs.github.io/pyComBat/</a></p></li>
<li><p><strong>Implementation Preprint</strong>: <a class="reference external" href="https://doi.org/10.1101/2020.03.17.995431">bioRxiv</a></p></li>
<li><p><strong>Original Paper</strong>: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/16632515/">Adjusting batch effects in microarray expression data using empirical Bayes methods</a></p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">combat.pycombat</span> <span class="kn">import</span> <span class="n">pycombat</span>

<span class="c1"># Correct batch effects in the training dataset</span>
<span class="n">df_batchcorrected</span> <span class="o">=</span> <span class="n">pycombat</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">,</span> <span class="n">batch</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully corrected batch effects in the training dataset.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Found 3 batches.
Adjusting for 0 covariate(s) or covariate level(s).
Standardizing Data across genes.
Fitting L/S model and finding priors.
Finding parametric adjustments.
Adjusting the Data
Succesfully corrected batch effects in the training dataset.
</pre></div>
</div>
</div>
</div>
<section id="evaluate-batch-effects">
<h3><a class="toc-backref" href="#id11" role="doc-backlink">Evaluate batch effects</a><a class="headerlink" href="#evaluate-batch-effects" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">sklearn.preprocessing</span> <span class="kn">import</span> <span class="n">StandardScaler</span>
<span class="kn">from</span> <span class="nn">sklearn.decomposition</span> <span class="kn">import</span> <span class="n">PCA</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="k">def</span> <span class="nf">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Standardizes the data.&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">StandardScaler</span><span class="p">()</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">perform_pca</span><span class="p">(</span><span class="n">std_data</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Performs PCA on standardized data.&#39;&#39;&#39;</span>
    <span class="n">pca</span> <span class="o">=</span> <span class="n">PCA</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="mi">42</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">pca</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">std_data</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">create_principal_df</span><span class="p">(</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Creates a DataFrame with the PCA data.&#39;&#39;&#39;</span>
    <span class="n">principal_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="s1">&#39;PC2&#39;</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span>
    <span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;hue&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_methyl</span><span class="p">[</span><span class="n">df_methyl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df3</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">principal_df</span>

<span class="k">def</span> <span class="nf">filter_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">principal_df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Filters the DataFrame by removing outliers.&#39;&#39;&#39;</span>
    <span class="n">outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">threshold</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">outliers</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df_batchcorrected</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_batchcorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df_batchcorrected</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">pcs_uncorrected</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">x_train_m</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Plots PCA data.&#39;&#39;&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">,</span> <span class="n">legend</span> <span class="ow">in</span> <span class="p">[(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA after batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="s1">&#39;full&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="n">non_outliers</span><span class="p">,</span> <span class="n">ax1</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA before batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">non_outliers</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="kc">False</span><span class="p">)]:</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PC2&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;hue&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="n">legend</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;PC 1&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;PC 2&#39;</span><span class="p">)</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.05</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">plot_data</span><span class="p">(</span><span class="n">pcs_batchcorrected</span><span class="p">,</span> <span class="n">pcs_uncorrected</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/f01b743033222b65e9180d507ba7a404a4d086e9f7dabf866ab86c0f4e9befd6.png" src="../_images/f01b743033222b65e9180d507ba7a404a4d086e9f7dabf866ab86c0f4e9befd6.png" />
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ol class="arabic simple">
<li><p>GDC_TARGET-AML contains AAML0531 and 03P1 samples from methylation array 450k.</p></li>
<li><p>GSE124413 contains AAML0531 and 03P1 samples from methylation array EPIC.</p></li>
<li><p>GSE190931 contains AAML1031 samples from methylation array EPIC.</p></li>
</ol>
</div>
</section>
<section id="save-dataframe-for-other-analyses">
<h3><a class="toc-backref" href="#id12" role="doc-backlink">Save dataframe for other analyses</a><a class="headerlink" href="#save-dataframe-for-other-analyses" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># # Save file as csv</span>
<span class="c1"># df_batchcorrected.to_csv(output_path+&#39;methylation_dataset_m-values_COG-924-dx-pts_batch-corrected_FMarchi-9-18-23.csv&#39;)</span>

<span class="c1"># # Save file as .RData</span>

<span class="c1"># import pyreadr</span>

<span class="c1"># pyreadr.write_rdata(output_path+&#39;methylation_dataset_m-values_COG-924-dx-pts_batch-corrected_FMarchi-9-18-23.RData&#39;, df_batchcorrected.reset_index())</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
</section>
<section id="run-coxph-ewas">
<h2><a class="toc-backref" href="#id13" role="doc-backlink">Run CoxPH-EWAS</a><a class="headerlink" href="#run-coxph-ewas" title="Permalink to this heading">#</a></h2>
<section id="import-custom-functions-adapted-from-methylize">
<h3><a class="toc-backref" href="#id14" role="doc-backlink">Import custom functions adapted from <code class="docutils literal notranslate"><span class="pre">methylize</span></code></a><a class="headerlink" href="#import-custom-functions-adapted-from-methylize" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">source.methylize_custom_code.diff_meth_pos_custom</span> <span class="kn">import</span> <span class="o">*</span>
</pre></div>
</div>
</div>
</details>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">new_risk</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">input_path</span> <span class="o">+</span> <span class="s1">&#39;updated_risk_group_COG_RR_1-10-2024/COG_risk_added_v2_cleanedFM.csv&#39;</span><span class="p">,</span>
                       <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,)</span>
                       
<span class="n">df4</span> <span class="o">=</span> <span class="n">df3</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">new_risk</span><span class="p">[[</span><span class="s1">&#39;Cyto/Molecular Risk&#39;</span><span class="p">]])</span>

<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">impute_by_mean</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Impute missing values by computing feature average&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">mean</span><span class="p">())</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="overall-survival-previous-risk-group">
<h3><a class="toc-backref" href="#id15" role="doc-backlink">Overall survival - previous risk group</a><a class="headerlink" href="#overall-survival-previous-risk-group" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_theme</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s2">&quot;white&quot;</span><span class="p">)</span>

<span class="c1"># # ewas_os_cog_unadjusted = diff_meth_pos_custom(meth_data = df_batchcorrected,</span>
<span class="c1"># #                                     duration = df3[&#39;os.time&#39;],</span>
<span class="c1"># #                                     event_observed = df3[&#39;os.evnt&#39;],</span>
<span class="c1"># #                                     pheno_data = df3[&#39;os.time&#39;], # not used</span>
<span class="c1"># #                                     covariates = False,</span>
<span class="c1"># #                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1"># #                                     export = &#39;pkl&#39;,</span>
<span class="c1"># #                                     verbose = True,</span>
<span class="c1"># #                                     filename=output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected&#39;)</span>

<span class="c1"># ewas_os_cog_unadjusted = pd.read_pickle(output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_unadjusted.pkl&#39;)</span>

<span class="c1"># manhattan_plot(stats_results=ewas_os_cog_unadjusted,</span>
<span class="c1">#                         palette=&#39;Paired&#39;,</span>
<span class="c1">#                         save=False,</span>
<span class="c1">#                         suggestive=True,</span>
<span class="c1">#                         significant= False,</span>
<span class="c1">#                         width=12,</span>
<span class="c1">#                         height=4,</span>
<span class="c1">#                         statsmode=False,</span>
<span class="c1">#                         bonferroni=False,</span>
<span class="c1">#                         fontsize=9,</span>
<span class="c1">#                         array_type=&#39;450k&#39;,</span>
<span class="c1">#                         explore=False,</span>
<span class="c1">#                         border = True,</span>
<span class="c1">#                         label_sig_probes=False,</span>
<span class="c1">#                         fwer=0.05,</span>
<span class="c1">#                         cohorts= &quot; COG OS risk unadjusted&quot;,</span>
<span class="c1">#                         filename=&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="overall-survival-updated-rr-risk-group">
<h3><a class="toc-backref" href="#id16" role="doc-backlink">Overall survival - updated RR risk group</a><a class="headerlink" href="#overall-survival-updated-rr-risk-group" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df4</span> <span class="o">=</span> <span class="n">df4</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">Risk_Group_Bin</span><span class="o">=</span><span class="n">df4</span><span class="p">[</span><span class="s1">&#39;Cyto/Molecular Risk&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">({</span><span class="s1">&#39;Low&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;Standard&#39;</span><span class="p">:</span><span class="mf">0.5</span><span class="p">,</span><span class="s1">&#39;High&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">}))</span>

<span class="n">covar_os</span> <span class="o">=</span> <span class="nb">round</span><span class="p">(</span><span class="n">impute_by_mean</span><span class="p">(</span><span class="n">df4</span><span class="p">[[</span><span class="s1">&#39;os.evnt&#39;</span><span class="p">,</span><span class="s1">&#39;os.time&#39;</span><span class="p">,</span><span class="s1">&#39;Risk_Group_Bin&#39;</span><span class="p">]]),</span><span class="mi">3</span><span class="p">)</span>

<span class="c1"># ewas_os_cog_adjusted = diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = covar_os[&#39;os.time&#39;],</span>
<span class="c1">#                                     event_observed = covar_os[&#39;os.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = covar_os[&#39;os.time&#39;], # not used</span>
<span class="c1">#                                     covariates = covar_os[&#39;Risk_Group_Bin&#39;],</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = False,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted&#39;)</span>

<span class="n">ewas_os_cog_adjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_os_batch_corrected_risk_adjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_os_cog_adjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG OS risk adjusted&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_os_unadjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"model_id": "007d82e9353f435798d835c6fc548108", "version_major": 2, "version_minor": 0}</script><img alt="../_images/b795048a7d0252fb6c07fb13cb35d3c758ba93bce55ee2de2c45d62d73b77256.png" src="../_images/b795048a7d0252fb6c07fb13cb35d3c758ba93bce55ee2de2c45d62d73b77256.png" />
</div>
</div>
</section>
<section id="event-free-survival-previous-risk-group">
<h3><a class="toc-backref" href="#id17" role="doc-backlink">Event-free survival - previous risk group</a><a class="headerlink" href="#event-free-survival-previous-risk-group" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># import methylize</span>
<span class="c1"># import seaborn as sns</span>
<span class="c1"># sns.set_theme(style=&quot;white&quot;)</span>

<span class="c1"># # ewas_efs_cog_unadjusted = diff_meth_pos_custom(meth_data = df_batchcorrected,</span>
<span class="c1"># #                                     duration = df3[&#39;efs.time&#39;],</span>
<span class="c1"># #                                     event_observed = df3[&#39;efs.evnt&#39;],</span>
<span class="c1"># #                                     pheno_data = df3[&#39;efs.time&#39;], # not used</span>
<span class="c1"># #                                     covariates = False,</span>
<span class="c1"># #                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1"># #                                     export = &#39;pkl&#39;,</span>
<span class="c1"># #                                     verbose = True,</span>
<span class="c1"># #                                     filename=output_path + &#39;ewas_dmr/ewas_cog_efs_batch_corrected&#39;)</span>

<span class="c1"># ewas_efs_cog_unadjusted = pd.read_pickle(output_path + &#39;ewas_dmr/ewas_cog_efs_batch_corrected_risk_unadjusted.pkl&#39;)</span>

<span class="c1"># manhattan_plot(stats_results=ewas_efs_cog_unadjusted,</span>
<span class="c1">#                         palette=&#39;Paired&#39;,</span>
<span class="c1">#                         save=False,</span>
<span class="c1">#                         suggestive=False,</span>
<span class="c1">#                         significant= False,</span>
<span class="c1">#                         width=12,</span>
<span class="c1">#                         height=4,</span>
<span class="c1">#                         statsmode=False,</span>
<span class="c1">#                         bonferroni=False,</span>
<span class="c1">#                         fontsize=9,</span>
<span class="c1">#                         array_type=&#39;450k&#39;,</span>
<span class="c1">#                         explore=False,</span>
<span class="c1">#                         border = True,</span>
<span class="c1">#                         label_sig_probes=False,</span>
<span class="c1">#                         fwer=0.05,</span>
<span class="c1">#                         cohorts= &quot; COG EFS risk unadjusted&quot;,</span>
<span class="c1">#                         filename=&#39;../Figures/Manhattan_Plots/ewas_efs_unadjusted&#39;)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/9f2fdb73cc3c41624b2d30ad295a051b010232d3a772cd70468c7b26a8482de5.png" src="../_images/9f2fdb73cc3c41624b2d30ad295a051b010232d3a772cd70468c7b26a8482de5.png" />
</div>
</div>
</section>
<section id="event-free-survival-updated-risk-group">
<h3><a class="toc-backref" href="#id18" role="doc-backlink">Event-free survival - updated risk group</a><a class="headerlink" href="#event-free-survival-updated-risk-group" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">covar_efs</span> <span class="o">=</span> <span class="nb">round</span><span class="p">(</span><span class="n">impute_by_mean</span><span class="p">(</span><span class="n">df4</span><span class="p">[[</span><span class="s1">&#39;efs.evnt&#39;</span><span class="p">,</span><span class="s1">&#39;efs.time&#39;</span><span class="p">,</span><span class="s1">&#39;Risk_Group_Bin&#39;</span><span class="p">]]),</span><span class="mi">3</span><span class="p">)</span>

<span class="c1"># ewas_efs_cog_adjusted = diff_meth_pos(meth_data = df_batchcorrected,</span>
<span class="c1">#                                     duration = covar_efs[&#39;efs.time&#39;],</span>
<span class="c1">#                                     event_observed = covar_efs[&#39;efs.evnt&#39;],</span>
<span class="c1">#                                     pheno_data = covar_efs[&#39;efs.time&#39;], # not used</span>
<span class="c1">#                                     covariates = covar_efs[&#39;Risk_Group_Bin&#39;],</span>
<span class="c1">#                                     regression_method = &quot;coxph&quot;,</span>
<span class="c1">#                                     export = &#39;pkl&#39;,</span>
<span class="c1">#                                     verbose = False,</span>
<span class="c1">#                                     filename=output_path + &#39;ewas_dmr/ewas_cog_efs_batch_corrected_risk_adjusted&#39;)</span>

<span class="n">ewas_efs_cog_adjusted</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_cog_efs_batch_corrected_risk_adjusted.pkl&#39;</span><span class="p">)</span>

<span class="n">manhattan_plot</span><span class="p">(</span><span class="n">stats_results</span><span class="o">=</span><span class="n">ewas_efs_cog_adjusted</span><span class="p">,</span>
                        <span class="n">palette</span><span class="o">=</span><span class="s1">&#39;Paired&#39;</span><span class="p">,</span>
                        <span class="n">save</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">suggestive</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">significant</span><span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
                        <span class="n">width</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
                        <span class="n">height</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                        <span class="n">statsmode</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">bonferroni</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fontsize</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span>
                        <span class="n">array_type</span><span class="o">=</span><span class="s1">&#39;450k&#39;</span><span class="p">,</span>
                        <span class="n">explore</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">border</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
                        <span class="n">label_sig_probes</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                        <span class="n">fwer</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                        <span class="n">cohorts</span><span class="o">=</span> <span class="s2">&quot; COG EFS risk adjusted&quot;</span><span class="p">,</span>
                        <span class="n">filename</span><span class="o">=</span><span class="s1">&#39;../Figures/Manhattan_Plots/ewas_efs_adjusted&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"model_id": "e492fedaf1e043c4ba546b62f517f65b", "version_major": 2, "version_minor": 0}</script><img alt="../_images/ef2163c08d8fecc664497eb918dbe1ea492966e69962d84cfd6c38f1edae6342.png" src="../_images/ef2163c08d8fecc664497eb918dbe1ea492966e69962d84cfd6c38f1edae6342.png" />
</div>
</div>
</section>
</section>
<section id="select-statistically-significant-cpgs">
<h2><a class="toc-backref" href="#id19" role="doc-backlink">Select statistically significant CpGs</a><a class="headerlink" href="#select-statistically-significant-cpgs" title="Permalink to this heading">#</a></h2>
<section id="threshold-based-on-suggested-significance-p-values-1-10e-6">
<h3><a class="toc-backref" href="#id20" role="doc-backlink">Threshold based on suggested-significance p-values (1*10e-6)</a><a class="headerlink" href="#threshold-based-on-suggested-significance-p-values-1-10e-6" title="Permalink to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">select_rows</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">threshold</span><span class="p">]</span>

<span class="k">def</span> <span class="nf">get_intersection</span><span class="p">(</span><span class="n">df1</span><span class="p">,</span> <span class="n">df2</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df1</span><span class="p">[</span><span class="n">df1</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df2</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># store the dataframes in a dictionary</span>
<span class="n">data_dict</span> <span class="o">=</span> <span class="p">{</span>
    <span class="c1"># &#39;Previous Risk Group&#39;: {</span>
    <span class="c1">#     &#39;os&#39;: ewas_os_cog_unadjusted,</span>
    <span class="c1">#     &#39;efs&#39;: ewas_efs_cog_unadjusted,</span>
    <span class="c1"># },</span>
    <span class="s1">&#39;Updated Risk Group&#39;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s1">&#39;os&#39;</span><span class="p">:</span> <span class="n">ewas_os_cog_adjusted</span><span class="p">,</span>
        <span class="s1">&#39;efs&#39;</span><span class="p">:</span> <span class="n">ewas_efs_cog_adjusted</span><span class="p">,</span>
    <span class="p">}</span>
<span class="p">}</span>

<span class="c1"># empty dictionary to store the results</span>
<span class="n">results</span> <span class="o">=</span> <span class="p">{}</span>

<span class="c1"># loop through the data_dict</span>
<span class="k">for</span> <span class="n">condition</span><span class="p">,</span> <span class="n">dfs</span> <span class="ow">in</span> <span class="n">data_dict</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">top_dfs</span> <span class="o">=</span> <span class="p">{</span><span class="n">key</span><span class="p">:</span> <span class="n">select_rows</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="mi">1</span><span class="o">*</span><span class="mf">10e-6</span><span class="p">)</span> <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">df</span> <span class="ow">in</span> <span class="n">dfs</span><span class="o">.</span><span class="n">items</span><span class="p">()}</span>
    <span class="n">intersection</span> <span class="o">=</span> <span class="n">get_intersection</span><span class="p">(</span><span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">],</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">])</span>

    <span class="n">results</span><span class="p">[</span><span class="n">condition</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Significant CpG probes in OS&#39;</span><span class="p">:</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="s1">&#39;Significant CpG probes in EFS&#39;</span><span class="p">:</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="s1">&#39;Significant overlapping CpG probes&#39;</span><span class="p">:</span> <span class="n">intersection</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
    <span class="p">}</span>

<span class="c1"># convert results to a pandas DataFrame and display</span>
<span class="n">result_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">results</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>

                    
<span class="c1"># save the results to an excel file</span>
<span class="n">result_df</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/table_significant_cpgs_OS_EFS_summary.xlsx&#39;</span><span class="p">)</span>

<span class="n">result_df</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Significant CpG probes in EFS</th>
      <th>Significant CpG probes in OS</th>
      <th>Significant overlapping CpG probes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Updated Risk Group</th>
      <td>14</td>
      <td>42</td>
      <td>4</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="threshold-based-on-suggested-significance-p-values-1-10e-5">
<h3><a class="toc-backref" href="#id21" role="doc-backlink">Threshold based on suggested-significance p-values (1*10e-5)</a><a class="headerlink" href="#threshold-based-on-suggested-significance-p-values-1-10e-5" title="Permalink to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">select_rows</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;PValue&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">threshold</span><span class="p">]</span>

<span class="k">def</span> <span class="nf">get_intersection</span><span class="p">(</span><span class="n">df1</span><span class="p">,</span> <span class="n">df2</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df1</span><span class="p">[</span><span class="n">df1</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">df2</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>

<span class="c1"># store the dataframes in a dictionary</span>
<span class="n">data_dict</span> <span class="o">=</span> <span class="p">{</span>
    <span class="c1"># &#39;Previous Risk Group&#39;: {</span>
    <span class="c1">#     &#39;os&#39;: ewas_os_cog_unadjusted,</span>
    <span class="c1">#     &#39;efs&#39;: ewas_efs_cog_unadjusted,</span>
    <span class="c1"># },</span>
    <span class="s1">&#39;Updated Risk Group&#39;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s1">&#39;os&#39;</span><span class="p">:</span> <span class="n">ewas_os_cog_adjusted</span><span class="p">,</span>
        <span class="s1">&#39;efs&#39;</span><span class="p">:</span> <span class="n">ewas_efs_cog_adjusted</span><span class="p">,</span>
    <span class="p">}</span>
<span class="p">}</span>

<span class="c1"># empty dictionary to store the results</span>
<span class="n">results</span> <span class="o">=</span> <span class="p">{}</span>

<span class="c1"># loop through the data_dict</span>
<span class="k">for</span> <span class="n">condition</span><span class="p">,</span> <span class="n">dfs</span> <span class="ow">in</span> <span class="n">data_dict</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">top_dfs</span> <span class="o">=</span> <span class="p">{</span><span class="n">key</span><span class="p">:</span> <span class="n">select_rows</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="mi">1</span><span class="o">*</span><span class="mf">10e-5</span><span class="p">)</span> <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">df</span> <span class="ow">in</span> <span class="n">dfs</span><span class="o">.</span><span class="n">items</span><span class="p">()}</span>
    <span class="n">intersection</span> <span class="o">=</span> <span class="n">get_intersection</span><span class="p">(</span><span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">],</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">])</span>

    <span class="n">results</span><span class="p">[</span><span class="n">condition</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;Significant CpG probes in OS&#39;</span><span class="p">:</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="s1">&#39;Significant CpG probes in EFS&#39;</span><span class="p">:</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
        <span class="s1">&#39;Significant overlapping CpG probes&#39;</span><span class="p">:</span> <span class="n">intersection</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
    <span class="p">}</span>

<span class="c1"># convert results to a pandas DataFrame and display</span>
<span class="n">result_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">results</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>

                    
<span class="c1"># save the results to an excel file</span>
<span class="n">result_df</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/table_significant_cpgs_OS_EFS_summary.xlsx&#39;</span><span class="p">)</span>

<span class="n">result_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Significant CpG probes in EFS</th>
      <th>Significant CpG probes in OS</th>
      <th>Significant overlapping CpG probes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Updated Risk Group</th>
      <td>112</td>
      <td>189</td>
      <td>17</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<section id="fetch-genes">
<h4><a class="toc-backref" href="#id22" role="doc-backlink">Fetch genes</a><a class="headerlink" href="#fetch-genes" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Define a function to handle repetitive tasks</span>
<span class="k">def</span> <span class="nf">process_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">other_df</span><span class="p">,</span> <span class="n">column_drop</span><span class="p">):</span>
    <span class="c1"># Join and sort values</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">other_df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">column_drop</span><span class="p">),</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;PValue&#39;</span><span class="p">)</span>

    <span class="c1"># Handle columns with multiple genes separated by `;`</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneClasses&#39;</span><span class="p">]:</span>
        <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">))))</span>

    <span class="k">return</span> <span class="n">df</span>

<span class="c1"># Load manifest</span>
<span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">zhou2016_probes_path</span><span class="p">,</span>
                              <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)[[</span><span class="s1">&#39;chrm&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;GeneNames&#39;</span><span class="p">,</span>
                                                     <span class="s1">&#39;GeneClasses&#39;</span><span class="p">,</span><span class="s1">&#39;CGIPosition&#39;</span><span class="p">]]</span>

<span class="c1"># Get probes from Zhou et al. (2016) that are in ewasOS_un.index</span>
<span class="n">top_efs</span> <span class="o">=</span> <span class="n">process_df</span><span class="p">(</span><span class="n">zhou2016_probes</span><span class="p">,</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;efs&#39;</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;chromosome&#39;</span><span class="p">,</span> <span class="s1">&#39;MAPINFO&#39;</span><span class="p">])</span>
<span class="n">top_os</span> <span class="o">=</span> <span class="n">process_df</span><span class="p">(</span><span class="n">zhou2016_probes</span><span class="p">,</span> <span class="n">top_dfs</span><span class="p">[</span><span class="s1">&#39;os&#39;</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;chromosome&#39;</span><span class="p">,</span> <span class="s1">&#39;MAPINFO&#39;</span><span class="p">])</span>

<span class="c1"># Save top_efs and top_os to excel</span>
<span class="n">top_efs</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/ewas_cpgs_top_EFS.xlsx&#39;</span><span class="p">)</span>
<span class="n">top_os</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/ewas_cpgs_top_OS.xlsx&#39;</span><span class="p">)</span>

<span class="c1"># Get OS probes and add EFS info to it</span>
<span class="n">top_os</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">ewas_efs_cog_adjusted</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span> <span class="n">lsuffix</span><span class="o">=</span><span class="s1">&#39;_OS&#39;</span><span class="p">,</span> <span class="n">rsuffix</span><span class="o">=</span><span class="s1">&#39;_EFS&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/ewas_cpgs_top_OS_with_EFSinfo.xlsx&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="os-suggested-significance-probes">
<h4><a class="toc-backref" href="#id23" role="doc-backlink">OS suggested-significance probes:</a><a class="headerlink" href="#os-suggested-significance-probes" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">itables</span> <span class="kn">import</span> <span class="n">show</span>

<span class="n">show</span><span class="p">(</span><span class="n">top_os</span><span class="p">,</span> <span class="n">classes</span><span class="o">=</span><span class="s2">&quot;display nowrap cell-border&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><style>.itables table td {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables table th {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables thead input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}

.itables tfoot input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}
</style>
<div class="itables">
<table id="a1e1dce7-589d-463b-81d3-eabb3ce1a675" class="display nowrap cell-border"style="table-layout:auto;width:auto;margin:auto;caption-side:bottom"><thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>CGIPosition</th>
      <th>Coefficient</th>
      <th>StandardError</th>
      <th>HazardsRatio</th>
      <th>95%CI_lower</th>
      <th>95%CI_upper</th>
      <th>ZScore</th>
      <th>PValue</th>
      <th>FDR_QValue</th>
      <th>minuslog10value</th>
    </tr>
  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css">
<script type="module">
    // Import jquery and DataTable
    import 'https://code.jquery.com/jquery-3.6.0.min.js';
    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';
    dt($);

    // Define the table data
    const data = [["cg00920892", "chr8", 37874705, 37874707, "RAB11FIP1", "protein_coding", ".", 0.218656, 0.039014, 1.244403, 1.152796, 1.34329, 5.604569, 2.08774e-08, 0.006953, 7.680324], ["cg08900363", "chr7", 597816, 597818, "PRKAR1B", "protein_coding", ".", -0.274044, 0.051303, 0.760299, 0.687566, 0.840724, -5.341641, 9.210906e-08, 0.009702, 7.035698], ["cg08404702", "chr16", 2083386, 2083388, "TSC2", "protein_coding", "N_Shore", 0.128108, 0.0241, 1.136675, 1.084232, 1.191655, 5.315611, 1.063e-07, 0.009702, 6.973467], ["cg01543603", "chr16", 89403132, 89403134, "ANKRD11;RP1-168P16.2", "protein_coding;sense_intronic", ".", 0.192478, 0.036324, 1.21225, 1.128945, 1.301702, 5.298878, 1.165166e-07, 0.009702, 6.933612], ["cg06748884", "chr16", 11138900, 11138902, "CLEC16A", "protein_coding", ".", -0.255997, 0.04932, 0.774144, 0.702815, 0.852713, -5.190543, 2.096812e-07, 0.012407, 6.678441], ["cg00192882", "chr17", 19387806, 19387808, "MFAP4", "protein_coding", ".", 0.254548, 0.049325, 1.289879, 1.171017, 1.420806, 5.160586, 2.461779e-07, 0.012407, 6.608751], ["cg08329113", "chr16", 70737238, 70737240, "VAC14", "protein_coding", ".", 0.187752, 0.036458, 1.206535, 1.123328, 1.295905, 5.149807, 2.607547e-07, 0.012407, 6.583768], ["cg20018344", "chr20", 3223491, 3223493, "ITPA", "protein_coding", ".", 0.199049, 0.040169, 1.220242, 1.127856, 1.320195, 4.955256, 7.223507e-07, 0.029019, 6.141252], ["cg17099306", "chr13", 49549508, 49549510, "RCBTB1", "protein_coding", ".", 0.214141, 0.043355, 1.238798, 1.13788, 1.348665, 4.939269, 7.841584e-07, 0.029019, 6.105596], ["cg06339275", "chr3", 62948996, 62948998, "LINC00698", "lincRNA", ".", -0.167854, 0.034223, 0.845477, 0.790626, 0.904133, -4.904736, 9.355309e-07, 0.031159, 6.028942], ["cg26361535", "chr8", 143494433, 143494435, "ZC3H3", "protein_coding", ".", 0.169164, 0.034876, 1.184314, 1.106065, 1.268099, 4.850456, 1.231779e-06, 0.036171, 5.909467], ["cg01052291", "chr6", 132715433, 132715435, "VNN1", "protein_coding", ".", -0.224235, 0.046337, 0.799127, 0.72975, 0.8751, -4.83926, 1.303237e-06, 0.036171, 5.884977], ["cg00945409", "chr10", 78977907, 78977909, "ZMIZ1-AS1", "antisense", "S_Shelf", -0.280736, 0.058291, 0.755228, 0.673691, 0.846634, -4.816091, 1.463976e-06, 0.03635, 5.834466], ["cg16424082", "chr12", 50929956, 50929958, "METTL7A", "protein_coding", ".", -0.35775, 0.074537, 0.699248, 0.604206, 0.80924, -4.799615, 1.589708e-06, 0.03635, 5.798683], ["cg04839706", "chr16", 46886162, 46886164, "GPT2", "protein_coding", "S_Shore", -0.186375, 0.038982, 0.829963, 0.768912, 0.89586, -4.781027, 1.744018e-06, 0.03635, 5.758449], ["cg14928764", "chr20", 23083970, 23083972, "CD93", "protein_coding", ".", -0.172133, 0.036138, 0.841867, 0.784301, 0.903657, -4.763271, 1.9048e-06, 0.03635, 5.720151], ["cg05430997", "chr13", 109744841, 109744843, ".", ".", ".", 0.135596, 0.02849, 1.14522, 1.083024, 1.210987, 4.759418, 1.94152e-06, 0.03635, 5.711858], ["cg07660635", "chr16", 85902540, 85902542, "IRF8", "protein_coding", "S_Shelf", -0.152823, 0.032126, 0.858281, 0.805906, 0.914061, -4.75704, 1.964526e-06, 0.03635, 5.706742], ["cg10089193", "chr16", 1090938, 1090940, "C1QTNF8", "protein_coding", "S_Shore", 0.167948, 0.035513, 1.182876, 1.103341, 1.268143, 4.729146, 2.254663e-06, 0.039523, 5.646918], ["cg18072388", "chr19", 10323084, 10323086, "RAVER1", "protein_coding", "N_Shore", 0.192287, 0.04081, 1.212018, 1.11885, 1.312945, 4.711779, 2.455633e-06, 0.040493, 5.609837], ["cg04220636", "chr16", 88967679, 88967681, "CBFA2T3", "protein_coding", "Island", 0.161386, 0.034309, 1.175138, 1.098714, 1.256878, 4.703839, 2.553148e-06, 0.040493, 5.592924], ["cg07143043", "chr10", 131970793, 131970795, "RP11-45A17.2;BNIP3", "antisense;protein_coding", ".", 0.202528, 0.043285, 1.224494, 1.124897, 1.33291, 4.67895, 2.883483e-06, 0.042654, 5.540083], ["cg18062721", "chr3", 11601952, 11601954, "VGLL4", "protein_coding", ".", 0.109844, 0.023498, 1.116104, 1.065867, 1.168708, 4.67458, 2.945565e-06, 0.042654, 5.530831], ["cg19432033", "chr17", 7613901, 7613903, "SHBG;FXR2", "protein_coding", "N_Shore", -0.195392, 0.041973, 0.822512, 0.757556, 0.893037, -4.655201, 3.236643e-06, 0.044916, 5.489905], ["cg22410826", "chr13", 109744594, 109744596, ".", ".", ".", 0.154218, 0.033382, 1.166746, 1.092853, 1.245635, 4.619839, 3.840374e-06, 0.051163, 5.415626], ["cg22760287", "chr11", 107705174, 107705176, ".", ".", ".", -0.173874, 0.03771, 0.840403, 0.780529, 0.90487, -4.610849, 4.010283e-06, 0.051371, 5.396825], ["cg25292309", "chr2", 58762291, 58762293, "LINC01122", "lincRNA", ".", -0.218937, 0.047677, 0.803372, 0.731701, 0.882063, -4.592098, 4.388121e-06, 0.05413, 5.357721], ["cg20310759", "chr1", 87152629, 87152631, "RP5-1052I5.2;LINC01140", "lincRNA;protein_coding", "S_Shore", -0.183252, 0.04011, 0.832558, 0.769614, 0.900651, -4.568735, 4.90677e-06, 0.055637, 5.309204], ["cg27049094", "chr2", 74840588, 74840590, "HK2", "protein_coding", ".", 0.126364, 0.027768, 1.134695, 1.074591, 1.198161, 4.55072, 5.346275e-06, 0.055637, 5.271949], ["cg11801959", "chr13", 49549476, 49549478, "RCBTB1", "protein_coding", ".", 0.187774, 0.041303, 1.206561, 1.112736, 1.308296, 4.546296, 5.459829e-06, 0.055637, 5.262821], ["cg06585893", "chr1", 230747805, 230747807, "RP11-99J16__A.2;CAPN9", "antisense;protein_coding", ".", 0.254271, 0.055992, 1.289521, 1.155494, 1.439094, 4.541169, 5.594311e-06, 0.055637, 5.252253], ["cg03801691", "chr3", 11558341, 11558343, "VGLL4", "protein_coding", ".", 0.184033, 0.040531, 1.202055, 1.11026, 1.30144, 4.540574, 5.610139e-06, 0.055637, 5.251026], ["cg27064178", "chr22", 49784571, 49784573, "BRD1", "protein_coding", "Island", 0.156357, 0.034533, 1.169244, 1.092725, 1.251122, 4.527799, 5.960109e-06, 0.055637, 5.224746], ["cg15787807", "chr11", 64741618, 64741620, "RASGRP2", "protein_coding", "N_Shore", -0.185036, 0.040879, 0.831074, 0.767086, 0.900401, -4.52648, 5.997441e-06, 0.055637, 5.222034], ["cg10009497", "chr7", 1055779, 1055781, "C7orf50;GPR146;RP11-449P15.1", "antisense;protein_coding", "N_Shore", 0.145612, 0.032172, 1.156747, 1.08606, 1.232034, 4.526106, 6.008055e-06, 0.055637, 5.221266], ["cg13213755", "chr15", 73634112, 73634114, "NPTN", "protein_coding", "S_Shore", -0.277236, 0.061255, 0.757875, 0.672136, 0.854552, -4.525904, 6.013789e-06, 0.055637, 5.220852], ["cg04205653", "chr21", 44805974, 44805976, "SUMO3", "protein_coding", "S_Shelf", -0.436748, 0.096651, 0.646134, 0.53463, 0.780894, -4.518829, 6.218264e-06, 0.055974, 5.206331], ["cg04663203", "chr17", 47851211, 47851213, "SP6", "protein_coding", "Island", 0.219331, 0.048673, 1.245244, 1.131941, 1.369889, 4.506191, 6.600178e-06, 0.057848, 5.180444], ["cg21229337", "chr6", 36963766, 36963768, "PI16", "protein_coding", "S_Shore", 0.184053, 0.041194, 1.202079, 1.10884, 1.303159, 4.467954, 7.897131e-06, 0.067441, 5.102531], ["cg20703895", "chr1", 157438011, 157438013, ".", ".", ".", -0.37241, 0.083589, 0.689072, 0.584943, 0.811737, -4.455266, 8.378932e-06, 0.069767, 5.076811], ["cg04857395", "chr4", 3514909, 3514911, "LRPAP1", "protein_coding", "Island", 0.099838, 0.022491, 1.104992, 1.057341, 1.154791, 4.439077, 9.034564e-06, 0.073391, 5.044093], ["cg11818438", "chr8", 118904290, 118904292, ".", ".", ".", -0.225902, 0.051111, 0.797796, 0.72175, 0.881855, -4.419872, 9.875956e-06, 0.078316, 5.005421], ["cg13984533", "chr1", 37796839, 37796841, "MANEAL", "protein_coding", "S_Shelf", -0.171951, 0.039084, 0.84202, 0.779927, 0.909057, -4.399497, 1.08502e-05, 0.08104, 4.964562], ["cg18156551", "chr11", 64224444, 64224446, "TRPT1;NUDT22", "protein_coding", "N_Shore", -0.404346, 0.091994, 0.667413, 0.5573, 0.799281, -4.395372, 1.105831e-05, 0.08104, 4.956311], ["cg19708903", "chr10", 131970803, 131970805, "RP11-45A17.2;BNIP3", "antisense;protein_coding", ".", 0.173423, 0.039478, 1.189369, 1.100813, 1.28505, 4.39296, 1.118177e-05, 0.08104, 4.951489], ["cg00532502", "chr1", 246789970, 246789972, "KIF28P;LINC01341", "unitary_pseudogene;processed_transcript", "S_Shore", 0.260906, 0.059487, 1.298106, 1.155246, 1.458633, 4.385905, 1.155047e-05, 0.08104, 4.9374], ["cg00206490", "chr16", 89403320, 89403322, "ANKRD11;RP1-168P16.2", "protein_coding;sense_intronic", ".", 0.128205, 0.029236, 1.136786, 1.073478, 1.203828, 4.385169, 1.158957e-05, 0.08104, 4.935933], ["cg00059652", "chr17", 69609356, 69609358, "LINC01483", "lincRNA", ".", -0.149143, 0.034041, 0.861446, 0.805847, 0.920882, -4.381285, 1.179815e-05, 0.08104, 4.928186], ["cg10187365", "chr21", 33488648, 33488650, "DNAJC28", "protein_coding", "N_Shelf", -0.414492, 0.094726, 0.660676, 0.548728, 0.795462, -4.375697, 1.210452e-05, 0.08104, 4.917052], ["cg18886274", "chr7", 85160554, 85160556, "SEMA3D", "protein_coding", ".", -0.153224, 0.035026, 0.857938, 0.801017, 0.918903, -4.37459, 1.216607e-05, 0.08104, 4.91485], ["cg19706516", "chr1", 151131392, 151131394, "RP11-68I18.10", "sense_overlapping", "Island", 0.188321, 0.043153, 1.207221, 1.109315, 1.313767, 4.364043, 1.276806e-05, 0.082071, 4.893875], ["cg10098541", "chr3", 123066802, 123066804, "PDIA5", "protein_coding", "N_Shore", -0.190075, 0.043633, 0.826897, 0.759121, 0.900724, -4.356235, 1.323186e-05, 0.082071, 4.878379], ["cg05480169", "chr10", 133253903, 133253905, ".", ".", ".", -0.142095, 0.032631, 0.867539, 0.813792, 0.924836, -4.354594, 1.333139e-05, 0.082071, 4.875125], ["cg18437839", "chr1", 201476584, 201476586, ".", ".", ".", -0.249068, 0.057233, 0.779527, 0.69681, 0.872063, -4.351829, 1.350066e-05, 0.082071, 4.869645], ["cg17632028", "chr4", 652752, 652754, "PDE6B", "protein_coding", "N_Shore", -0.173474, 0.03987, 0.840739, 0.777542, 0.909073, -4.350983, 1.355283e-05, 0.082071, 4.86797], ["cg10057841", "chr11", 130126348, 130126350, "APLP2", "protein_coding", "S_Shelf", -0.330465, 0.076353, 0.718589, 0.618712, 0.834589, -4.328106, 1.50397e-05, 0.089448, 4.822761], ["cg16958939", "chr6", 7224989, 7224991, "RREB1", "protein_coding", ".", 0.121111, 0.028049, 1.12875, 1.068372, 1.19254, 4.317869, 1.575428e-05, 0.092054, 4.802601], ["cg05102794", "chr16", 89101072, 89101074, "ACSF3", "protein_coding", "Island", 0.21616, 0.050377, 1.241301, 1.124596, 1.370117, 4.29089, 1.779587e-05, 0.100729, 4.749681], ["cg16429055", "chr17", 17051538, 17051540, "MPRIP", "protein_coding", ".", -0.280883, 0.065534, 0.755117, 0.664097, 0.858612, -4.286068, 1.818636e-05, 0.100729, 4.740254], ["cg21094613", "chr11", 79189641, 79189643, "TENM4", "protein_coding", ".", -0.199219, 0.046495, 0.819371, 0.748003, 0.897547, -4.28472, 1.829692e-05, 0.100729, 4.737622], ["cg00460983", "chr10", 89374183, 89374185, "LIPA", "protein_coding", ".", -0.163177, 0.038116, 0.849441, 0.788294, 0.915331, -4.281003, 1.860532e-05, 0.100729, 4.730363], ["cg01771213", "chr1", 57270164, 57270166, "DAB1", "protein_coding", ".", -0.203368, 0.047536, 0.815978, 0.743388, 0.895656, -4.278159, 1.884457e-05, 0.100729, 4.724814], ["cg20964021", "chr7", 156641708, 156641710, "LINC01006;RNF32", "lincRNA;protein_coding", "S_Shore", -0.311788, 0.072944, 0.732136, 0.634603, 0.84466, -4.274371, 1.916777e-05, 0.100729, 4.717428], ["cg20219911", "chr3", 52779635, 52779637, "ITIH1", "protein_coding", ".", 0.225721, 0.052835, 1.253226, 1.129942, 1.389961, 4.272193, 1.935598e-05, 0.100729, 4.713185], ["cg16822174", "chr12", 19483011, 19483013, "AEBP2", "protein_coding", ".", 0.146181, 0.034427, 1.157405, 1.081886, 1.238196, 4.24614, 2.174849e-05, 0.111439, 4.662571], ["cg05413061", "chr3", 16183799, 16183801, "GALNT15", "protein_coding", ".", -0.29824, 0.070425, 0.742123, 0.646443, 0.851964, -4.234891, 2.286627e-05, 0.113212, 4.640805], ["cg13720395", "chr7", 76328584, 76328586, "YWHAG", "protein_coding", "N_Shore", 0.124718, 0.02946, 1.132829, 1.069272, 1.200164, 4.233524, 2.300571e-05, 0.113212, 4.638164], ["cg08467866", "chr16", 75208215, 75208217, "CTRB2", "protein_coding", ".", 0.167807, 0.039648, 1.182708, 1.094283, 1.278279, 4.232466, 2.311428e-05, 0.113212, 4.63612], ["cg20429981", "chr1", 1678532, 1678534, "SLC35E2B;CDK11B;MMP23B", "protein_coding", "S_Shore", 0.153382, 0.036356, 1.165771, 1.085592, 1.251871, 4.218867, 2.455327e-05, 0.11544, 4.609891], ["cg08480739", "chr11", 67997549, 67997551, "UNC93B1", "protein_coding", "Island", -0.098596, 0.023391, 0.906109, 0.865506, 0.948617, -4.215122, 2.496427e-05, 0.11544, 4.602681], ["cg01312394", "chr2", 46882240, 46882242, ".", ".", ".", -0.182847, 0.043381, 0.832896, 0.765006, 0.906811, -4.214902, 2.49887e-05, 0.11544, 4.602256], ["cg05348324", "chr11", 119797455, 119797457, ".", ".", ".", 0.151364, 0.03592, 1.16342, 1.084331, 1.248279, 4.213952, 2.509403e-05, 0.11544, 4.60043], ["cg02988730", "chr1", 109624886, 109624888, "AMPD2", "protein_coding", ".", 0.260162, 0.061807, 1.29714, 1.149151, 1.464188, 4.209302, 2.561613e-05, 0.11544, 4.591486], ["cg26223555", "chr19", 51824187, 51824189, "FPR3", "protein_coding", ".", -0.264002, 0.062723, 0.767972, 0.679134, 0.868431, -4.209013, 2.564891e-05, 0.11544, 4.590931], ["cg13846675", "chr8", 52168344, 52168346, "ST18", "protein_coding", ".", 0.107861, 0.025708, 1.113893, 1.059157, 1.171456, 4.195589, 2.72163e-05, 0.119891, 4.565171], ["cg00598235", "chr4", 17579056, 17579058, "LAP3", "protein_coding", "S_Shore", -0.291858, 0.069582, 0.746875, 0.651657, 0.856005, -4.194414, 2.735774e-05, 0.119891, 4.56292], ["cg10817441", "chr6", 32855950, 32855952, "PSMB9", "protein_coding", "S_Shore", -0.231392, 0.055222, 0.793428, 0.712038, 0.884122, -4.190253, 2.786432e-05, 0.120525, 4.554952], ["cg21617903", "chr3", 11558238, 11558240, "VGLL4", "protein_coding", ".", 0.173232, 0.041476, 1.189142, 1.096301, 1.289846, 4.176718, 2.957452e-05, 0.123952, 4.529082], ["cg16179125", "chr20", 59007065, 59007067, "CTSZ", "protein_coding", "Island", 0.16758, 0.040137, 1.18244, 1.092987, 1.279215, 4.175253, 2.976547e-05, 0.123952, 4.526287], ["cg14037250", "chr9", 134053771, 134053773, "BRD3;RP11-374P20.4", "antisense;protein_coding", "S_Shore", 0.164605, 0.039424, 1.178927, 1.091261, 1.273635, 4.175196, 2.977298e-05, 0.123952, 4.526178], ["cg19345219", "chr12", 6638238, 6638240, "ACRBP", "protein_coding", ".", 0.201249, 0.048266, 1.22293, 1.112545, 1.344266, 4.169613, 3.051168e-05, 0.125459, 4.515534], ["cg09890699", "chr1", 54356179, 54356181, "SSBP3", "protein_coding", "N_Shore", 0.185604, 0.044578, 1.203945, 1.103218, 1.313868, 4.163534, 3.133589e-05, 0.125728, 4.503958], ["cg18964582", "chr1", 236393881, 236393883, "EDARADD", "protein_coding", "N_Shore", -0.138275, 0.033212, 0.870859, 0.815976, 0.929433, -4.163371, 3.135837e-05, 0.125728, 4.503647], ["cg15576082", "chr2", 241813194, 241813196, "NEU4;AC114730.3", "antisense;protein_coding", "Island", 0.131324, 0.031563, 1.140337, 1.071932, 1.213108, 4.160739, 3.172192e-05, 0.125728, 4.49864], ["cg13360395", "chr2", 123907703, 123907705, ".", ".", ".", -0.155206, 0.037326, 0.856238, 0.795834, 0.921227, -4.158124, 3.20872e-05, 0.125728, 4.493668], ["cg04092699", "chr2", 76748093, 76748095, "LRRTM4", "protein_coding", ".", -0.198189, 0.047756, 0.820215, 0.746927, 0.900694, -4.150065, 3.323802e-05, 0.126094, 4.478365], ["cg17669009", "chr19", 51417630, 51417632, "SIGLEC10", "protein_coding", ".", 0.292408, 0.070471, 1.339649, 1.166825, 1.538071, 4.14932, 3.33464e-05, 0.126094, 4.476951], ["cg26310429", "chr17", 73028309, 73028311, "SLC39A11", "protein_coding", ".", -0.11092, 0.026751, 0.89501, 0.849293, 0.943187, -4.146464, 3.376495e-05, 0.126094, 4.471534], ["cg00549798", "chr22", 40418426, 40418428, "MKL1", "protein_coding", "N_Shore", 0.176426, 0.042567, 1.192946, 1.097457, 1.296743, 4.144653, 3.403287e-05, 0.126094, 4.468101], ["cg11064395", "chr6", 31407526, 31407528, "HCP5;MICA", "sense_overlapping;protein_coding", "S_Shelf", -0.328796, 0.079463, 0.71979, 0.61598, 0.841095, -4.137706, 3.507956e-05, 0.126094, 4.454946], ["cg07080653", "chr4", 53508081, 53508083, "LNX1;LNX1-AS1;FIP1L1", "antisense;protein_coding", ".", -0.118378, 0.028641, 0.88836, 0.839865, 0.939655, -4.133119, 3.578733e-05, 0.126094, 4.446271], ["cg11823637", "chr12", 22729886, 22729888, "RP11-114G22.1", "lincRNA", ".", -0.281394, 0.068084, 0.754731, 0.660448, 0.862473, -4.133031, 3.580104e-05, 0.126094, 4.446104], ["cg25376409", "chr17", 80999487, 80999489, "CHMP6", "protein_coding", "Island", 0.143256, 0.034705, 1.154025, 1.078137, 1.235254, 4.127799, 3.662517e-05, 0.126094, 4.43622], ["cg03092447", "chr16", 89275990, 89275992, "ANKRD11", "protein_coding", "N_Shelf", 0.241868, 0.058605, 1.273626, 1.135423, 1.428652, 4.127106, 3.673564e-05, 0.126094, 4.434912], ["cg18151487", "chr10", 102715582, 102715584, "ARL3;SFXN2", "protein_coding", "S_Shore", -0.165338, 0.040074, 0.847607, 0.78358, 0.916865, -4.125817, 3.694213e-05, 0.126094, 4.432478], ["cg15586054", "chr6", 165658541, 165658543, "PDE10A", "protein_coding", "N_Shore", -0.135463, 0.032834, 0.873311, 0.818882, 0.931359, -4.12576, 3.695118e-05, 0.126094, 4.432372], ["cg21739530", "chr4", 173264916, 173264918, "GALNT7", "protein_coding", ".", 0.147086, 0.035669, 1.158454, 1.080234, 1.242338, 4.1237, 3.728338e-05, 0.126094, 4.428485], ["cg17511968", "chr2", 218856441, 218856443, ".", ".", "N_Shelf", 0.129414, 0.031386, 1.138161, 1.070256, 1.210374, 4.123283, 3.735111e-05, 0.126094, 4.427696], ["cg19137417", "chr17", 81389134, 81389136, "RP11-1055B8.4", "lincRNA", "S_Shore", -0.142328, 0.034573, 0.867337, 0.810512, 0.928146, -4.11676, 3.842363e-05, 0.126094, 4.415402], ["cg16449402", "chr3", 9733040, 9733042, "BRPF1", "protein_coding", "S_Shore", -0.202232, 0.049134, 0.816905, 0.741906, 0.899486, -4.115945, 3.855955e-05, 0.126094, 4.413868], ["cg21750233", "chr11", 6483080, 6483082, "TIMM10B", "protein_coding", "S_Shore", -0.183107, 0.044491, 0.832679, 0.763144, 0.908549, -4.115605, 3.861648e-05, 0.126094, 4.413227], ["cg02678414", "chr16", 2159838, 2159840, "TRAF7", "protein_coding", "S_Shelf", 0.193545, 0.047041, 1.213544, 1.106659, 1.330751, 4.114364, 3.882485e-05, 0.126094, 4.41089], ["cg25035655", "chr11", 68751776, 68751778, "MTL5", "protein_coding", "S_Shore", 0.247481, 0.060165, 1.280795, 1.138327, 1.441095, 4.113353, 3.899538e-05, 0.126094, 4.408987], ["cg14978242", "chr5", 80205311, 80205313, "SERINC5", "protein_coding", ".", 0.166828, 0.040644, 1.181551, 1.091079, 1.279525, 4.104609, 4.049994e-05, 0.129625, 4.392546], ["cg02348449", "chr19", 58119061, 58119063, "ZSCAN18", "protein_coding", "S_Shore", -0.168655, 0.04111, 0.8448, 0.779402, 0.915687, -4.102531, 4.086549e-05, 0.129625, 4.388643], ["cg12762600", "chr15", 89647922, 89647924, "KIF7", "protein_coding", "N_Shore", -0.234866, 0.057362, 0.790677, 0.706598, 0.884761, -4.094427, 4.232137e-05, 0.132634, 4.37344], ["cg23152216", "chr12", 25252150, 25252152, "KRAS", "protein_coding", "S_Shore", -0.16776, 0.041004, 0.845557, 0.780262, 0.916317, -4.091298, 4.289647e-05, 0.132634, 4.367578], ["cg23128949", "chr12", 124501577, 124501579, "NCOR2", "protein_coding", ".", 0.110212, 0.026942, 1.116515, 1.059086, 1.177058, 4.090691, 4.300896e-05, 0.132634, 4.366441], ["cg13857210", "chr9", 136107370, 136107372, ".", ".", ".", 0.129565, 0.031747, 1.138333, 1.069662, 1.211412, 4.081219, 4.480016e-05, 0.135207, 4.34872], ["cg07536737", "chr2", 98013627, 98013629, ".", ".", ".", -0.117585, 0.028826, 0.889065, 0.840227, 0.940742, -4.079092, 4.521198e-05, 0.135207, 4.344747], ["cg16721321", "chr4", 1208907, 1208909, "CTBP1-AS;SPON2", "antisense;protein_coding", "N_Shore", 0.093583, 0.022948, 1.098102, 1.049806, 1.148619, 4.078029, 4.541915e-05, 0.135207, 4.342761], ["cg05800336", "chr10", 131453805, 131453807, ".", ".", ".", 0.173748, 0.042608, 1.189756, 1.094434, 1.29338, 4.077783, 4.546712e-05, 0.135207, 4.342303], ["cg09140274", "chr1", 220896850, 220896852, ".", ".", "S_Shore", -0.188969, 0.046436, 0.827812, 0.755798, 0.906689, -4.069438, 4.712668e-05, 0.138902, 4.326733], ["cg21262032", "chr1", 154465216, 154465218, "IL6R", "protein_coding", ".", 0.144066, 0.035425, 1.15496, 1.077489, 1.238001, 4.066742, 4.767506e-05, 0.139286, 4.321709], ["cg14882966", "chr2", 3651762, 3651764, ".", ".", "Island", 0.110544, 0.027208, 1.116886, 1.058887, 1.178061, 4.062999, 4.844618e-05, 0.140308, 4.31474], ["cg10759293", "chr20", 20018581, 20018583, "NAA20", "protein_coding", "S_Shore", -0.177341, 0.043704, 0.837494, 0.768742, 0.912395, -4.057742, 4.954947e-05, 0.141868, 4.304961], ["cg06605704", "chr16", 4501145, 4501147, "HMOX2", "protein_coding", ".", -0.358837, 0.088486, 0.698488, 0.587272, 0.830766, -4.055276, 5.007515e-05, 0.141868, 4.300378], ["cg04400145", "chr20", 1893035, 1893037, "RP5-858L17.1;SIRPA", "antisense;protein_coding", "N_Shore", -0.179942, 0.044382, 0.835319, 0.765727, 0.911234, -4.054402, 5.026269e-05, 0.141868, 4.298754], ["cg21932513", "chr1", 17425477, 17425479, "RCC2", "protein_coding", ".", 0.261151, 0.064485, 1.298424, 1.144265, 1.473352, 4.049781, 5.126561e-05, 0.142871, 4.290174], ["cg02957270", "chr17", 48430734, 48430736, "SKAP1", "protein_coding", "S_Shore", -0.130508, 0.032234, 0.877649, 0.823917, 0.934885, -4.048822, 5.147609e-05, 0.142871, 4.288394], ["cg10591771", "chr19", 3671119, 3671121, "PIP5K1C;AC004637.1", "antisense;protein_coding", "N_Shore", 0.123066, 0.030422, 1.130959, 1.065495, 1.200444, 4.045299, 5.22563e-05, 0.143838, 4.281861], ["cg17078656", "chr1", 12475057, 12475059, "VPS13D", "protein_coding", ".", -0.300787, 0.074548, 0.740235, 0.639609, 0.856692, -4.034831, 5.464152e-05, 0.148071, 4.262477], ["cg03601372", "chr19", 36776123, 36776125, "CTD-2162K18.3;CTD-2162K18.4", "lincRNA", "Island", -0.168298, 0.041713, 0.845102, 0.778759, 0.917097, -4.034652, 5.468338e-05, 0.148071, 4.262145], ["cg15830431", "chr8", 98940362, 98940364, "STK3", "protein_coding", "Island", 0.175401, 0.043539, 1.191723, 1.094245, 1.297885, 4.028547, 5.612265e-05, 0.150211, 4.250862], ["cg10280339", "chr15", 65519989, 65519991, ".", ".", "S_Shelf", -0.176942, 0.043934, 0.837828, 0.768703, 0.913169, -4.027489, 5.637559e-05, 0.150211, 4.248909], ["cg00223135", "chr15", 43507046, 43507048, "TP53BP1", "protein_coding", "N_Shelf", -0.313299, 0.077892, 0.731031, 0.627529, 0.851605, -4.022195, 5.765819e-05, 0.151241, 4.239139], ["cg17104202", "chr1", 3245839, 3245841, "PRDM16", "protein_coding", "N_Shore", -0.136397, 0.033917, 0.872496, 0.816382, 0.932468, -4.021479, 5.783381e-05, 0.151241, 4.237818], ["cg03071245", "chr13", 36888896, 36888898, "SMAD9", "protein_coding", ".", -0.277777, 0.069155, 0.757466, 0.661451, 0.867418, -4.0167, 5.901869e-05, 0.151241, 4.22901], ["cg19357999", "chr12", 121223958, 121223960, "P2RX4", "protein_coding", ".", 0.163714, 0.040764, 1.177878, 1.087431, 1.275848, 4.016116, 5.916513e-05, 0.151241, 4.227934], ["cg11887376", "chr5", 1118554, 1118556, ".", ".", ".", 0.130132, 0.032405, 1.138979, 1.06889, 1.213664, 4.015856, 5.923054e-05, 0.151241, 4.227454], ["cg19479744", "chr6", 155701180, 155701182, ".", ".", ".", -0.241272, 0.060095, 0.785628, 0.698335, 0.883832, -4.014837, 5.948681e-05, 0.151241, 4.225579], ["cg23458843", "chr12", 57742661, 57742663, "AGAP2;TSPAN31", "protein_coding", "S_Shelf", -0.327282, 0.081729, 0.72088, 0.61418, 0.846117, -4.004502, 6.214826e-05, 0.15681, 4.206571], ["cg18586780", "chr2", 235044659, 235044661, "SH3BP4", "protein_coding", "S_Shore", -0.197176, 0.049293, 0.821046, 0.745434, 0.904328, -4.000087, 6.331911e-05, 0.158563, 4.198465], ["cg21170304", "chr9", 135987756, 135987758, ".", ".", "N_Shelf", -0.241002, 0.060364, 0.78584, 0.698155, 0.884537, -3.992464, 6.539023e-05, 0.162528, 4.184487], ["cg26984593", "chr10", 14009636, 14009638, "FRMD4A", "protein_coding", ".", -0.129546, 0.032467, 0.878494, 0.824334, 0.936213, -3.990128, 6.603778e-05, 0.162922, 4.180208], ["cg13689831", "chr1", 226987375, 226987377, "ADCK3", "protein_coding", ".", 0.170196, 0.042689, 1.185537, 1.090382, 1.288997, 3.986916, 6.693763e-05, 0.163104, 4.17433], ["cg00151914", "chr17", 35731530, 35731532, "RASL10B;TAF15", "protein_coding", "Island", 0.253587, 0.063614, 1.28864, 1.137584, 1.459753, 3.986373, 6.70911e-05, 0.163104, 4.173335], ["cg26373351", "chr12", 11878369, 11878371, "ETV6", "protein_coding", ".", 0.260828, 0.065472, 1.298004, 1.141684, 1.475727, 3.983785, 6.78263e-05, 0.163696, 4.168602], ["cg03596113", "chr1", 3213404, 3213406, "PRDM16", "protein_coding", ".", -0.196022, 0.049236, 0.821994, 0.746379, 0.905271, -3.981275, 6.854648e-05, 0.164244, 4.164015], ["cg23613253", "chr11", 70594241, 70594243, "SHANK2", "protein_coding", ".", -0.09919, 0.024925, 0.905571, 0.862395, 0.950908, -3.979562, 6.904231e-05, 0.164251, 4.160885], ["cg18898209", "chr11", 86889207, 86889209, "PRSS23", "protein_coding", ".", -0.159956, 0.040241, 0.852182, 0.787552, 0.922115, -3.974986, 7.03835e-05, 0.165294, 4.152529], ["cg27658601", "chr17", 82836204, 82836206, "ZNF750;TBCD", "protein_coding", "S_Shelf", -0.134069, 0.03374, 0.87453, 0.818568, 0.934317, -3.973554, 7.080812e-05, 0.165294, 4.149917], ["cg03922381", "chr10", 69217453, 69217455, "RP11-227H15.4", "antisense", ".", -0.179457, 0.045169, 0.835724, 0.764918, 0.913084, -3.973012, 7.096959e-05, 0.165294, 4.148928], ["cg19980929", "chr12", 42239104, 42239106, "YAF2;PPHLN1", "protein_coding", "S_Shore", -0.160118, 0.04036, 0.852043, 0.787239, 0.922182, -3.967209, 7.271932e-05, 0.166301, 4.13835], ["cg03762237", "chr2", 98085052, 98085054, ".", ".", "N_Shore", -0.155049, 0.039097, 0.856373, 0.793202, 0.924575, -3.965791, 7.315288e-05, 0.166301, 4.135769], ["cg01881444", "chr3", 69012423, 69012425, "EOGT", "protein_coding", "N_Shore", -0.373414, 0.094264, 0.68838, 0.572257, 0.828068, -3.961377, 7.451866e-05, 0.166301, 4.127735], ["cg20435097", "chr10", 124632254, 124632256, "FAM53B;RP11-12J10.3", "protein_coding", ".", 0.158451, 0.040002, 1.171694, 1.083339, 1.267256, 3.961067, 7.461571e-05, 0.166301, 4.12717], ["cg05482050", "chr5", 177982332, 177982334, "RP11-1252I4.2", "lincRNA", "N_Shelf", 0.258359, 0.065238, 1.294803, 1.139391, 1.471414, 3.960218, 7.488132e-05, 0.166301, 4.125627], ["cg09632163", "chr16", 2289290, 2289292, "ABCA3", "protein_coding", "Island", 0.122177, 0.030859, 1.129954, 1.063636, 1.200407, 3.959142, 7.521938e-05, 0.166301, 4.12367], ["cg22266824", "chr4", 158996903, 158996905, "C4orf45", "protein_coding", ".", -0.173807, 0.043918, 0.840459, 0.77114, 0.91601, -3.9575, 7.57382e-05, 0.166301, 4.120685], ["cg04665204", "chr17", 81453397, 81453399, "BAHCC1", "protein_coding", "Island", 0.224469, 0.056724, 1.251658, 1.119959, 1.398844, 3.957205, 7.58317e-05, 0.166301, 4.120149], ["cg04516952", "chr19", 50968785, 50968787, "KLK6;CTB-147C22.9", "antisense;protein_coding", ".", 0.222559, 0.056244, 1.24927, 1.118873, 1.394863, 3.957003, 7.589591e-05, 0.166301, 4.119782], ["cg01983248", "chr5", 43482701, 43482703, "RP11-159F24.3;C5orf28", "processed_transcript;protein_coding", "N_Shore", -0.140903, 0.035654, 0.868573, 0.809949, 0.931441, -3.951925, 7.752492e-05, 0.168386, 4.110559], ["cg10526478", "chr16", 85902604, 85902606, "IRF8", "protein_coding", "S_Shelf", -0.266887, 0.067571, 0.76576, 0.670774, 0.874197, -3.949722, 7.824188e-05, 0.168386, 4.106561], ["cg02905663", "chr8", 118111467, 118111469, "EXT1", "protein_coding", "N_Shore", -0.286909, 0.072647, 0.75058, 0.650968, 0.865435, -3.949348, 7.836429e-05, 0.168386, 4.105882], ["cg06142946", "chr11", 17351576, 17351578, "NCR3LG1", "protein_coding", "Island", 0.150097, 0.038024, 1.161947, 1.0785, 1.25185, 3.947442, 7.899053e-05, 0.168644, 4.102425], ["cg17389427", "chr3", 129612142, 129612144, ".", ".", ".", 0.315136, 0.079937, 1.370446, 1.171708, 1.602893, 3.942299, 8.070438e-05, 0.170787, 4.093103], ["cg00550080", "chr6", 30157014, 30157016, "TRIM10", "protein_coding", ".", -0.171602, 0.043539, 0.842314, 0.773417, 0.917349, -3.941363, 8.101992e-05, 0.170787, 4.091408], ["cg19827511", "chr14", 93114298, 93114300, "ITPK1", "protein_coding", "N_Shore", -0.209028, 0.053068, 0.811372, 0.731221, 0.900309, -3.938882, 8.186224e-05, 0.171477, 4.086916], ["cg15346842", "chr2", 59249930, 59249932, "AC007131.1;RP11-444A22.1;AC007131.2", "lincRNA", ".", 0.119315, 0.030316, 1.126725, 1.061726, 1.195703, 3.935661, 8.296802e-05, 0.171879, 4.081089], ["cg03445800", "chr3", 481328, 481330, ".", ".", ".", -0.139529, 0.035456, 0.869768, 0.811377, 0.932361, -3.935222, 8.311978e-05, 0.171879, 4.080296], ["cg15119221", "chr17", 19387441, 19387443, "MFAP4", "protein_coding", ".", 0.174415, 0.04435, 1.190549, 1.091432, 1.298668, 3.932715, 8.399191e-05, 0.171879, 4.075763], ["cg26379258", "chr6", 111567242, 111567244, "TRAF3IP2-AS1;TRAF3IP2", "antisense;protein_coding", ".", 0.183, 0.046551, 1.200814, 1.096103, 1.315528, 3.931164, 8.453547e-05, 0.171879, 4.072961], ["cg13691108", "chr20", 38804603, 38804605, "PPP1R16B", "protein_coding", "N_Shore", 0.176179, 0.044862, 1.192651, 1.092262, 1.302268, 3.927109, 8.597302e-05, 0.171879, 4.065638], ["cg20378520", "chr12", 125219264, 125219266, "TMEM132B", "protein_coding", ".", -0.182693, 0.046577, 0.833024, 0.760345, 0.91265, -3.922367, 8.768313e-05, 0.171879, 4.057084], ["cg02205013", "chr16", 1708796, 1708798, "MAPK8IP3", "protein_coding", "S_Shelf", 0.116715, 0.029757, 1.123799, 1.060131, 1.19129, 3.922308, 8.770469e-05, 0.171879, 4.056977], ["cg21477075", "chr6", 168032755, 168032757, "KIF25", "protein_coding", "N_Shelf", -0.136419, 0.034782, 0.872477, 0.814981, 0.934029, -3.922122, 8.777265e-05, 0.171879, 4.056641], ["cg14134256", "chr11", 94769712, 94769714, "AMOTL1", "protein_coding", "S_Shore", -0.185239, 0.047247, 0.830905, 0.757417, 0.911524, -3.920658, 8.830739e-05, 0.171879, 4.054003], ["cg13696545", "chr3", 55750825, 55750827, "ERC2", "protein_coding", ".", -0.121226, 0.030925, 0.885833, 0.833736, 0.941186, -3.920032, 8.853729e-05, 0.171879, 4.052874], ["cg13844237", "chr9", 125522544, 125522546, "MAPKAP1", "protein_coding", ".", -0.184759, 0.047142, 0.831305, 0.757937, 0.911775, -3.919205, 8.88414e-05, 0.171879, 4.051385], ["cg13440692", "chr1", 1250976, 1250978, "RP5-902P8.12", "lincRNA", "N_Shelf", 0.150818, 0.0385, 1.162785, 1.078271, 1.253923, 3.917311, 8.954205e-05, 0.171879, 4.047973], ["cg18482303", "chr2", 134283808, 134283810, "MGAT5", "protein_coding", ".", -0.089886, 0.022946, 0.914035, 0.873839, 0.956081, -3.917295, 8.954827e-05, 0.171879, 4.047943], ["cg09796602", "chr16", 23144881, 23144883, "USP31", "protein_coding", "N_Shelf", -0.166606, 0.042541, 0.846533, 0.778812, 0.920143, -3.916331, 8.990672e-05, 0.171879, 4.046208], ["cg16451525", "chr16", 89100873, 89100875, "ACSF3", "protein_coding", "Island", 0.119394, 0.030487, 1.126814, 1.061456, 1.196196, 3.916292, 8.992118e-05, 0.171879, 4.046138], ["cg16147063", "chr14", 81221539, 81221541, "CTD-2506P8.6;GTF2A1", "antisense;protein_coding", "Island", 0.310885, 0.079404, 1.364632, 1.167958, 1.594425, 3.915249, 9.031114e-05, 0.171879, 4.044259], ["cg16512727", "chr12", 4539969, 4539971, "C12orf4;RAD51AP1", "protein_coding", "S_Shore", -0.175283, 0.044787, 0.839219, 0.768693, 0.916217, -3.913709, 9.088929e-05, 0.171997, 4.041487], ["cg02642565", "chr16", 4529400, 4529402, "CDIP1", "protein_coding", ".", 0.155468, 0.03974, 1.168204, 1.080668, 1.262831, 3.912157, 9.147543e-05, 0.172128, 4.038696], ["cg09187054", "chr17", 1298861, 1298863, "TUSC5", "protein_coding", ".", 0.139047, 0.035562, 1.149178, 1.071807, 1.232134, 3.909965, 9.230954e-05, 0.172497, 4.034753], ["cg22624391", "chr11", 1916226, 1916228, ".", ".", ".", -0.218484, 0.055927, 0.803736, 0.720292, 0.896847, -3.906623, 9.359514e-05, 0.172497, 4.028747], ["cg22560020", "chr1", 246789611, 246789613, "KIF28P;LINC01341", "unitary_pseudogene;processed_transcript", "S_Shore", 0.132917, 0.034039, 1.142155, 1.068441, 1.220955, 3.904784, 9.430961e-05, 0.172497, 4.025444], ["cg06785302", "chr8", 118111811, 118111813, "EXT1", "protein_coding", "Island", -0.149317, 0.038246, 0.861296, 0.799093, 0.928341, -3.904129, 9.456543e-05, 0.172497, 4.024268], ["cg19839026", "chr14", 50943782, 50943784, "PYGL", "protein_coding", "N_Shore", 0.222213, 0.056919, 1.248838, 1.117008, 1.396225, 3.90402, 9.460796e-05, 0.172497, 4.024072], ["cg04371567", "chr4", 8247121, 8247123, ".", ".", ".", -0.307832, 0.07887, 0.735039, 0.629761, 0.857915, -3.903034, 9.499419e-05, 0.172497, 4.022303], ["cg06721473", "chr3", 50317984, 50317986, "HYAL2", "protein_coding", "N_Shelf", 0.195832, 0.050184, 1.216323, 1.102382, 1.342041, 3.902264, 9.529704e-05, 0.172497, 4.020921], ["cg25737891", "chr2", 11921150, 11921152, "AC096559.1", "lincRNA", ".", -0.158618, 0.040673, 0.853322, 0.787938, 0.924132, -3.899812, 9.626757e-05, 0.173312, 4.01652], ["cg15022671", "chr2", 203637474, 203637476, "AC125238.2", "processed_pseudogene", ".", -0.165816, 0.042546, 0.847202, 0.779421, 0.920878, -3.897344, 9.725358e-05, 0.174146, 4.012094], ["cg21093005", "chr17", 82584038, 82584040, "FOXK2", "protein_coding", "Island", 0.129925, 0.033349, 1.138743, 1.066692, 1.215661, 3.895945, 9.781644e-05, 0.174217, 4.009588], ["cg03272932", "chr16", 87555286, 87555288, ".", ".", ".", -0.131362, 0.033749, 0.8769, 0.820773, 0.936866, -3.892326, 9.928783e-05, 0.175897, 4.003104], ["cg13702536", "chr12", 122731005, 122731007, "HCAR1", "protein_coding", ".", -0.114382, 0.029399, 0.891917, 0.841977, 0.944819, -3.890709, 9.995157e-05, 0.176136, 4.00021]];

    // Define the dt_args
    let dt_args = {"order": []};
    dt_args["data"] = data;

    $(document).ready(function () {
        
        $('#a1e1dce7-589d-463b-81d3-eabb3ce1a675').DataTable(dt_args);
    });
</script>
</div>
</div></div>
</div>
</section>
<section id="efs-suggested-significance-probes">
<h4><a class="toc-backref" href="#id24" role="doc-backlink">EFS suggested-significance probes:</a><a class="headerlink" href="#efs-suggested-significance-probes" title="Permalink to this heading">#</a></h4>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">show</span><span class="p">(</span><span class="n">top_efs</span><span class="p">,</span> <span class="n">classes</span><span class="o">=</span><span class="s2">&quot;display nowrap cell-border&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html"><style>.itables table td {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables table th {
    text-overflow: ellipsis;
    overflow: hidden;
}

.itables thead input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}

.itables tfoot input {
    width: 100%;
    padding: 3px;
    box-sizing: border-box;
}
</style>
<div class="itables">
<table id="27ae280d-7942-4653-ad88-c00f8edbc871" class="display nowrap cell-border"style="table-layout:auto;width:auto;margin:auto;caption-side:bottom"><thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrm</th>
      <th>start</th>
      <th>end</th>
      <th>GeneNames</th>
      <th>GeneClasses</th>
      <th>CGIPosition</th>
      <th>Coefficient</th>
      <th>StandardError</th>
      <th>HazardsRatio</th>
      <th>95%CI_lower</th>
      <th>95%CI_upper</th>
      <th>ZScore</th>
      <th>PValue</th>
      <th>FDR_QValue</th>
      <th>minuslog10value</th>
    </tr>
  </thead><tbody><tr><td>Loading... (need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td></tr></tbody></table>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css">
<script type="module">
    // Import jquery and DataTable
    import 'https://code.jquery.com/jquery-3.6.0.min.js';
    import dt from 'https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs';
    dt($);

    // Define the table data
    const data = [["cg06748884", "chr16", 11138900, 11138902, "CLEC16A", "protein_coding", ".", -0.2504, 0.042514, 0.778489, 0.716249, 0.846138, -5.889765, 3.867454e-09, 0.001288, 8.412575], ["cg00192882", "chr17", 19387806, 19387808, "MFAP4", "protein_coding", ".", 0.216909, 0.040558, 1.242231, 1.147306, 1.345011, 5.348102, 8.88813e-08, 0.014801, 7.05119], ["cg25922105", "chr10", 43359427, 43359429, ".", ".", "N_Shelf", -0.300819, 0.062817, 0.740212, 0.654465, 0.837194, -4.788813, 1.677709e-06, 0.147019, 5.775283], ["cg08900363", "chr7", 597816, 597818, "PRKAR1B", "protein_coding", ".", -0.213428, 0.044664, 0.80781, 0.740102, 0.881713, -4.778544, 1.765691e-06, 0.147019, 5.753085], ["cg18866597", "chr20", 45344681, 45344683, "SDC4", "protein_coding", "N_Shelf", -0.391579, 0.083773, 0.675989, 0.573629, 0.796613, -4.67427, 2.950007e-06, 0.181021, 5.530177], ["cg08729279", "chr2", 215843654, 215843656, "LINC00607", "lincRNA", ".", -0.274204, 0.059302, 0.760177, 0.676763, 0.853872, -4.623832, 3.767151e-06, 0.181021, 5.423987], ["cg06806210", "chr1", 228215425, 228215427, "OBSCN", "protein_coding", "N_Shore", -0.234227, 0.051466, 0.791182, 0.715267, 0.875155, -4.551062, 5.337574e-06, 0.181021, 5.272656], ["cg18112681", "chr11", 63535926, 63535928, "RARRES3", "protein_coding", ".", -0.158064, 0.034741, 0.853795, 0.797594, 0.913957, -4.54974, 5.371216e-06, 0.181021, 5.269927], ["cg06400255", "chr16", 30051724, 30051726, "ALDOA;FAM57B", "protein_coding", "N_Shore", -0.14927, 0.033147, 0.861337, 0.807158, 0.919152, -4.503308, 6.690384e-06, 0.181021, 5.174549], ["cg12449049", "chr6", 25088770, 25088772, "CMAHP", "unitary_pseudogene", ".", -0.148384, 0.033226, 0.8621, 0.807747, 0.920109, -4.465914, 7.972754e-06, 0.181021, 5.098392], ["cg00945409", "chr10", 78977907, 78977909, "ZMIZ1-AS1", "antisense", "S_Shelf", -0.237576, 0.053217, 0.788537, 0.710434, 0.875227, -4.464251, 8.034937e-06, 0.181021, 5.095018], ["cg08069883", "chr3", 72032668, 72032670, ".", ".", ".", -0.185829, 0.041721, 0.830415, 0.765214, 0.901173, -4.454141, 8.42297e-06, 0.181021, 5.074535], ["cg05197036", "chr5", 141444469, 141444471, "PCDHGA12;PCDHGB1;PCDHGA5;PCDHGB2;PCDHGA8;PCDHGB3;PCDHGA1;PCDHGB6;PCDHGA6;PCDHGB5;PCDHGA9;PCDHGA2;PCDHGB7;PCDHGA10;PCDHGA11;PCDHGA7;PCDHGA4;PCDHGA3;PCDHGB4", "protein_coding", "S_Shelf", -0.316889, 0.07129, 0.728411, 0.633424, 0.837642, -4.445085, 8.785714e-06, 0.181021, 5.056223], ["cg17875657", "chr2", 209496087, 209496089, "MAP2", "protein_coding", ".", -0.161335, 0.036314, 0.851007, 0.792543, 0.913784, -4.442791, 8.879958e-06, 0.181021, 5.051589], ["cg13958583", "chr8", 100761836, 100761838, ".", ".", ".", -0.158977, 0.03605, 0.853016, 0.794825, 0.915467, -4.409946, 1.033962e-05, 0.181021, 4.985495], ["cg12026095", "chr19", 48965203, 48965205, "FTL", "protein_coding", "Island", 1.412138, 0.320607, 4.104722, 2.18969, 7.694577, 4.404578, 1.059899e-05, 0.181021, 4.974736], ["cg01048810", "chr1", 44677142, 44677144, "C1orf228", "protein_coding", "S_Shelf", -0.163449, 0.037179, 0.84921, 0.789529, 0.913402, -4.396259, 1.101327e-05, 0.181021, 4.958084], ["cg11975397", "chr1", 159470190, 159470192, "RP11-180D21.3;RP11-550P17.5", "antisense;lincRNA", ".", -0.121121, 0.027576, 0.885927, 0.839315, 0.935127, -4.392235, 1.121914e-05, 0.181021, 4.95004], ["cg17540192", "chr7", 98245946, 98245948, "TECPR1", "protein_coding", ".", -0.093, 0.021186, 0.911194, 0.874133, 0.949826, -4.389736, 1.134883e-05, 0.181021, 4.945049], ["cg20310759", "chr1", 87152629, 87152631, "RP5-1052I5.2;LINC01140", "lincRNA;protein_coding", "S_Shore", -0.146858, 0.033467, 0.863416, 0.808599, 0.92195, -4.388149, 1.143194e-05, 0.181021, 4.94188], ["cg02767093", "chr13", 98478400, 98478402, "STK24", "protein_coding", "S_Shelf", -0.225478, 0.051526, 0.798134, 0.721468, 0.882947, -4.376055, 1.208466e-05, 0.181021, 4.917765], ["cg15215077", "chr6", 119240673, 119240675, "MAN1A1", "protein_coding", ".", -0.220803, 0.050471, 0.801875, 0.726349, 0.885254, -4.374824, 1.215308e-05, 0.181021, 4.915314], ["cg08079170", "chr11", 128180711, 128180713, "RP11-702B10.2", "lincRNA", ".", -0.185218, 0.042412, 0.830923, 0.764644, 0.902947, -4.367091, 1.259121e-05, 0.181021, 4.899933], ["cg14928764", "chr20", 23083970, 23083972, "CD93", "protein_coding", ".", -0.134004, 0.030739, 0.874586, 0.82345, 0.928898, -4.359361, 1.304429e-05, 0.181021, 4.884579], ["cg14722693", "chr8", 19578939, 19578941, "CSGALNACT1", "protein_coding", ".", -0.167258, 0.038574, 0.845982, 0.784381, 0.91242, -4.336077, 1.450481e-05, 0.193238, 4.838488], ["cg19599045", "chr14", 22820528, 22820530, "SLC7A7", "protein_coding", "N_Shore", -0.194969, 0.045132, 0.822861, 0.753199, 0.898965, -4.319973, 1.560485e-05, 0.199897, 4.80674], ["cg26971423", "chr15", 63492831, 63492833, ".", ".", ".", -0.181044, 0.042249, 0.834398, 0.768088, 0.906433, -4.28521, 1.825665e-05, 0.215763, 4.738579], ["cg00598235", "chr4", 17579056, 17579058, "LAP3", "protein_coding", "S_Shore", -0.242658, 0.056674, 0.784539, 0.702059, 0.87671, -4.281632, 1.855273e-05, 0.215763, 4.731592], ["cg13213755", "chr15", 73634112, 73634114, "NPTN", "protein_coding", "S_Shore", -0.220436, 0.051518, 0.802169, 0.725127, 0.887397, -4.278841, 1.878688e-05, 0.215763, 4.726145], ["cg24422027", "chr17", 2756010, 2756012, ".", ".", "Island", 0.09458, 0.022154, 1.099197, 1.05249, 1.147977, 4.269177, 1.961958e-05, 0.217815, 4.70731], ["cg01052291", "chr6", 132715433, 132715435, "VNN1", "protein_coding", ".", -0.170296, 0.040017, 0.843415, 0.779793, 0.912229, -4.255614, 2.084759e-05, 0.217973, 4.680944], ["cg05075308", "chr4", 184408209, 184408211, "IRF2", "protein_coding", ".", -0.137155, 0.032324, 0.871835, 0.818314, 0.928857, -4.243102, 2.204515e-05, 0.217973, 4.656687], ["cg09372525", "chr1", 109493047, 109493049, "CYB561D1", "protein_coding", "N_Shore", -0.146611, 0.034594, 0.863629, 0.807013, 0.924218, -4.23801, 2.255097e-05, 0.217973, 4.646835], ["cg06130562", "chr16", 8849048, 8849050, "PMM2;RP11-77H9.2;RP11-152P23.2", "antisense;protein_coding", ".", -0.282035, 0.066642, 0.754247, 0.661893, 0.859487, -4.232111, 2.315083e-05, 0.217973, 4.635433], ["cg26112639", "chr1", 247416803, 247416805, "NLRP3", "protein_coding", ".", -0.256176, 0.060777, 0.774006, 0.687085, 0.871922, -4.214996, 2.497826e-05, 0.217973, 4.602438], ["cg14910406", "chr2", 219280584, 219280586, "DNAJB2", "protein_coding", "S_Shore", 0.367492, 0.087233, 1.444108, 1.217158, 1.713375, 4.212764, 2.522645e-05, 0.217973, 4.598144], ["cg11269166", "chr2", 171347336, 171347338, "METTL8", "protein_coding", ".", -0.138182, 0.032819, 0.87094, 0.816682, 0.928803, -4.210466, 2.548447e-05, 0.217973, 4.593724], ["cg06845655", "chr11", 85651326, 85651328, "TMEM126A", "protein_coding", "S_Shelf", -0.131365, 0.031221, 0.876898, 0.824847, 0.932234, -4.207512, 2.58198e-05, 0.217973, 4.588047], ["cg26124526", "chr5", 61323104, 61323106, ".", ".", ".", -0.1601, 0.03808, 0.852059, 0.79078, 0.918086, -4.204288, 2.619056e-05, 0.217973, 4.581855], ["cg08314781", "chr2", 54903891, 54903893, "EML6", "protein_coding", ".", -0.146562, 0.034877, 0.863672, 0.806607, 0.924775, -4.202273, 2.642478e-05, 0.217973, 4.577989], ["cg25535982", "chr11", 74319354, 74319356, "RP11-632K5.3", "lincRNA", ".", -0.264611, 0.063185, 0.767505, 0.678106, 0.868689, -4.187867, 2.815889e-05, 0.217973, 4.550385], ["cg24320673", "chr17", 74466094, 74466096, "CD300A", "protein_coding", ".", -0.205268, 0.049074, 0.814429, 0.739744, 0.896654, -4.182853, 2.878736e-05, 0.217973, 4.540798], ["cg08998192", "chr6", 32837792, 32837794, "TAP2;XXbac-BPG246D15.9", "protein_coding", "N_Shore", -0.120624, 0.028845, 0.886367, 0.837647, 0.937921, -4.181851, 2.891453e-05, 0.217973, 4.538884], ["cg09432758", "chr19", 1649123, 1649125, "TCF3", "protein_coding", "S_Shelf", 0.221366, 0.052974, 1.24778, 1.124725, 1.384299, 4.178741, 2.931279e-05, 0.217973, 4.532943], ["cg07499292", "chr1", 46436715, 46436717, "AL137797.1;FAAHP1", "miRNA;transcribed_unprocessed_pseudogene", ".", -0.299987, 0.071807, 0.740828, 0.643569, 0.852785, -4.177673, 2.945067e-05, 0.217973, 4.530905], ["cg04578362", "chr2", 144835191, 144835193, "TEX41", "lincRNA", ".", 0.111496, 0.026831, 1.117949, 1.060677, 1.178313, 4.155434, 3.246702e-05, 0.233704, 4.488558], ["cg14399183", "chr9", 121286029, 121286031, "GSN;GSN-AS1", "antisense;protein_coding", ".", -0.207405, 0.050008, 0.81269, 0.736814, 0.89638, -4.147425, 3.362349e-05, 0.233704, 4.473357], ["cg06529643", "chr10", 71892529, 71892531, ".", ".", "S_Shelf", -0.15769, 0.038046, 0.854115, 0.792741, 0.920239, -4.144738, 3.402027e-05, 0.233704, 4.468262], ["cg18564075", "chr12", 116267257, 116267259, "MED13L", "protein_coding", ".", -0.153817, 0.037146, 0.857429, 0.797222, 0.922183, -4.140864, 3.459998e-05, 0.233704, 4.460924], ["cg03116016", "chr16", 15489748, 15489750, "RP11-1021N1.1;C16orf45", "protein_coding", ".", -0.112707, 0.027269, 0.893413, 0.846917, 0.942461, -4.133163, 3.578047e-05, 0.233704, 4.446354], ["cg13033858", "chr12", 108854549, 108854551, "SSH1", "protein_coding", "N_Shelf", -0.104998, 0.025404, 0.900326, 0.856595, 0.946289, -4.133126, 3.578622e-05, 0.233704, 4.446284], ["cg02474542", "chr11", 72589846, 72589848, "PDE2A", "protein_coding", "N_Shore", 0.2289, 0.055567, 1.257216, 1.127485, 1.401874, 4.119324, 3.799862e-05, 0.237866, 4.420232], ["cg05299836", "chr16", 31107745, 31107747, "BCKDK;AC135050.1", "miRNA;protein_coding", "N_Shore", 0.317558, 0.077181, 1.373769, 1.180912, 1.598123, 4.114477, 3.880585e-05, 0.237866, 4.411103], ["cg12708336", "chr17", 43368914, 43368916, ".", ".", ".", 0.097498, 0.02371, 1.102409, 1.052352, 1.154848, 4.112142, 3.920054e-05, 0.237866, 4.406708], ["cg00375146", "chr11", 1200585, 1200587, "MUC5AC", "protein_coding", "Island", 0.135314, 0.033083, 1.144896, 1.073016, 1.221591, 4.090176, 4.310457e-05, 0.237866, 4.365477], ["cg06605704", "chr16", 4501145, 4501147, "HMOX2", "protein_coding", ".", -0.297984, 0.072869, 0.742314, 0.643518, 0.856276, -4.089294, 4.326882e-05, 0.237866, 4.363825], ["cg23443627", "chr17", 74466024, 74466026, "CD300A", "protein_coding", ".", -0.210763, 0.051557, 0.809966, 0.732118, 0.896091, -4.087946, 4.352099e-05, 0.237866, 4.361301], ["cg00041989", "chr17", 57750676, 57750678, ".", ".", ".", -0.139964, 0.034251, 0.869389, 0.812942, 0.929756, -4.086412, 4.380958e-05, 0.237866, 4.358431], ["cg16270222", "chr17", 43369027, 43369029, ".", ".", ".", 0.107061, 0.026222, 1.113002, 1.057246, 1.171699, 4.08292, 4.447325e-05, 0.237866, 4.351901], ["cg06024557", "chr5", 150052438, 150052440, "HMGXB3", "protein_coding", ".", -0.290521, 0.071253, 0.747874, 0.650395, 0.859962, -4.077296, 4.55624e-05, 0.237866, 4.341393], ["cg23361265", "chr17", 16014345, 16014347, "TTC19", "protein_coding", ".", -0.143968, 0.035316, 0.865915, 0.808006, 0.927975, -4.07659, 4.570097e-05, 0.237866, 4.340075], ["cg06650260", "chr1", 26913986, 26913988, "NUDC;NR0B2", "protein_coding", ".", -0.226598, 0.055604, 0.797241, 0.714924, 0.889037, -4.075227, 4.596951e-05, 0.237866, 4.33753], ["cg16317961", "chr18", 35041783, 35041785, "MAPRE2", "protein_coding", "S_Shore", -1.194256, 0.293423, 0.302929, 0.170443, 0.538398, -4.070086, 4.699577e-05, 0.237866, 4.327941], ["cg06985664", "chr7", 116727544, 116727546, "MET", "protein_coding", ".", -0.140271, 0.034478, 0.869122, 0.812331, 0.929884, -4.068441, 4.732883e-05, 0.237866, 4.324874], ["cg08855705", "chr8", 93923489, 93923491, "PDP1", "protein_coding", ".", -0.140275, 0.034487, 0.869119, 0.812314, 0.929896, -4.067494, 4.752142e-05, 0.237866, 4.323111], ["cg07049515", "chr19", 49597729, 49597731, "PRR12", "protein_coding", "Island", -0.306118, 0.075281, 0.7363, 0.635295, 0.853362, -4.066365, 4.775211e-05, 0.237866, 4.321007], ["cg11823637", "chr12", 22729886, 22729888, "RP11-114G22.1", "lincRNA", ".", -0.237141, 0.058334, 0.78888, 0.703651, 0.884433, -4.065226, 4.798586e-05, 0.237866, 4.318887], ["cg25152348", "chr22", 50508282, 50508284, "LMF2;NCAPH2", "protein_coding", "Island", 1.163158, 0.286321, 3.200022, 1.825729, 5.608796, 4.062429, 4.856482e-05, 0.237866, 4.313678], ["cg27018309", "chr16", 8849264, 8849266, "PMM2;RP11-77H9.2;RP11-152P23.2", "antisense;protein_coding", ".", -0.15224, 0.037547, 0.858783, 0.797853, 0.924365, -4.05463, 5.021373e-05, 0.239776, 4.299178], ["cg03258927", "chr1", 113981967, 113981969, "OLFML3", "protein_coding", ".", -0.097179, 0.023973, 0.907393, 0.865745, 0.951045, -4.053789, 5.03946e-05, 0.239776, 4.297616], ["cg10601624", "chr12", 6295210, 6295212, ".", ".", "S_Shelf", -0.137035, 0.033838, 0.871939, 0.815988, 0.931727, -4.049809, 5.125955e-05, 0.240456, 4.290225], ["cg24607181", "chr17", 43368834, 43368836, ".", ".", ".", 0.15066, 0.037303, 1.162601, 1.080633, 1.250787, 4.038793, 5.372685e-05, 0.246566, 4.269809], ["cg21769117", "chr6", 31737495, 31737497, "CLIC1", "protein_coding", "N_Shelf", -0.221442, 0.054863, 0.801363, 0.719664, 0.892336, -4.036288, 5.430363e-05, 0.246566, 4.265171], ["cg21611262", "chr5", 145872879, 145872881, "GRXCR2", "protein_coding", ".", -0.247114, 0.061254, 0.781052, 0.692691, 0.880683, -4.034224, 5.4783e-05, 0.246566, 4.261354], ["cg12586594", "chr19", 53869454, 53869456, "AC008753.6;MYADM", "antisense;protein_coding", "Island", 0.212185, 0.052709, 1.236376, 1.115025, 1.370935, 4.025578, 5.683564e-05, 0.252394, 4.245379], ["cg03376089", "chr17", 17936094, 17936096, "TOM1L2", "protein_coding", ".", -0.146892, 0.03664, 0.863387, 0.803558, 0.92767, -4.009047, 6.096418e-05, 0.258549, 4.214925], ["cg25828093", "chr8", 26183486, 26183488, ".", ".", ".", -0.281073, 0.070112, 0.754974, 0.65804, 0.866186, -4.008918, 6.099751e-05, 0.258549, 4.214688], ["cg14978242", "chr5", 80205311, 80205313, "SERINC5", "protein_coding", ".", 0.130242, 0.03252, 1.139104, 1.068765, 1.214072, 4.004991, 6.201993e-05, 0.258549, 4.207469], ["cg11578066", "chr3", 129315049, 129315051, "H1FX;H1FX-AS1", "antisense;protein_coding", "Island", 0.942142, 0.235295, 2.56547, 1.617645, 4.068653, 4.004082, 6.225876e-05, 0.258549, 4.2058], ["cg17413194", "chr17", 65057877, 65057879, "GNA13", "protein_coding", "S_Shore", -0.2416, 0.060369, 0.78537, 0.697732, 0.884016, -4.002086, 6.278643e-05, 0.258549, 4.202134], ["cg19380581", "chr18", 46031990, 46031992, "PSTPIP2", "protein_coding", "S_Shelf", -0.122714, 0.030665, 0.884516, 0.83292, 0.939308, -4.001737, 6.287929e-05, 0.258549, 4.201492], ["cg02960777", "chr7", 77933530, 77933532, "PHTF2", "protein_coding", ".", -0.154554, 0.038657, 0.856797, 0.79428, 0.924236, -3.99812, 6.384767e-05, 0.259329, 4.194855], ["cg02678414", "chr16", 2159838, 2159840, "TRAF7", "protein_coding", "S_Shelf", 0.140415, 0.03516, 1.150751, 1.07412, 1.232849, 3.993555, 6.508991e-05, 0.260522, 4.186486], ["cg14397690", "chr15", 60396748, 60396750, "ANXA2", "protein_coding", "N_Shore", -0.142374, 0.035682, 0.867297, 0.808714, 0.930123, -3.990059, 6.605673e-05, 0.260522, 4.180083], ["cg15830431", "chr8", 98940362, 98940364, "STK3", "protein_coding", "Island", 0.150475, 0.03773, 1.162387, 1.07953, 1.251603, 3.988208, 6.657432e-05, 0.260522, 4.176693], ["cg04314978", "chr11", 36488359, 36488361, "TRAF6", "protein_coding", ".", -0.167311, 0.041978, 0.845936, 0.779123, 0.918478, -3.985739, 6.727031e-05, 0.260522, 4.172177], ["cg24180227", "chr19", 45366841, 45366843, "ERCC2", "protein_coding", "S_Shelf", -0.167804, 0.042154, 0.845519, 0.77847, 0.918344, -3.980731, 6.870376e-05, 0.263015, 4.16302], ["cg22162847", "chr1", 150562302, 150562304, "ADAMTSL4-AS1", "processed_transcript", "S_Shore", 0.193739, 0.048818, 1.213779, 1.103025, 1.335654, 3.968556, 7.230932e-05, 0.26441, 4.140806], ["cg17078656", "chr1", 12475057, 12475059, "VPS13D", "protein_coding", ".", -0.253593, 0.063937, 0.776008, 0.684609, 0.879609, -3.966278, 7.300374e-05, 0.26441, 4.136655], ["cg26888941", "chr6", 33430744, 33430746, "SYNGAP1", "protein_coding", "S_Shelf", 0.389509, 0.098291, 1.476255, 1.217575, 1.789894, 3.962792, 7.40782e-05, 0.26441, 4.13031], ["cg20867674", "chr6", 163347378, 163347380, ".", ".", ".", -0.105029, 0.026511, 0.900298, 0.854712, 0.948315, -3.961715, 7.441342e-05, 0.26441, 4.128349], ["cg23409168", "chr8", 116089464, 116089466, "LINC00536", "lincRNA", ".", -0.11224, 0.028341, 0.89383, 0.845534, 0.944885, -3.960309, 7.485275e-05, 0.26441, 4.125792], ["cg02898412", "chr8", 11766014, 11766016, ".", ".", "N_Shelf", -0.230611, 0.058233, 0.794048, 0.708401, 0.89005, -3.96018, 7.489327e-05, 0.26441, 4.125557], ["cg15507449", "chr7", 156002132, 156002134, ".", ".", ".", -0.155077, 0.03916, 0.856349, 0.793081, 0.924665, -3.96005, 7.493421e-05, 0.26441, 4.12532], ["cg24747122", "chr17", 65057810, 65057812, "GNA13", "protein_coding", "S_Shore", -0.200476, 0.050644, 0.818341, 0.741013, 0.903739, -3.958509, 7.541902e-05, 0.26441, 4.122519], ["cg11851174", "chr17", 17809294, 17809296, "RAI1", "protein_coding", "N_Shore", 0.157074, 0.039737, 1.170082, 1.082411, 1.264854, 3.952853, 7.722503e-05, 0.266438, 4.112242], ["cg00795277", "chr17", 7439785, 7439787, "RP11-104H15.10;FGF11;RP11-104H15.7;RP11-104H15.8", "antisense;processed_transcript;protein_coding", "Island", 0.128485, 0.032539, 1.137104, 1.06685, 1.211985, 3.948705, 7.857511e-05, 0.266438, 4.104715], ["cg06935361", "chr13", 32390316, 32390318, "BRCA2", "protein_coding", ".", -0.113824, 0.028831, 0.892415, 0.843385, 0.944296, -3.947947, 7.882413e-05, 0.266438, 4.103341], ["cg05413061", "chr3", 16183799, 16183801, "GALNT15", "protein_coding", ".", -0.243789, 0.061769, 0.783653, 0.694299, 0.884507, -3.946815, 7.919756e-05, 0.266438, 4.101288], ["cg11937508", "chr2", 206218878, 206218880, "GPR1-AS;GPR1", "processed_transcript;protein_coding", ".", -0.160802, 0.040843, 0.851461, 0.785957, 0.922423, -3.937055, 8.248756e-05, 0.274731, 4.083612], ["cg12044599", "chr11", 67438836, 67438838, "PTPRCAP;CORO1B", "protein_coding", "S_Shelf", 0.103272, 0.026291, 1.108793, 1.053105, 1.167426, 3.928037, 8.564213e-05, 0.277376, 4.067313], ["cg06666376", "chr19", 3480597, 3480599, "SMIM24", "protein_coding", "N_Shore", 0.121346, 0.030934, 1.129016, 1.062598, 1.199585, 3.922767, 8.753788e-05, 0.277376, 4.057804], ["cg12172163", "chr13", 113171639, 113171641, "PROZ", "protein_coding", ".", 0.140732, 0.035889, 1.151116, 1.072927, 1.235003, 3.92128, 8.80799e-05, 0.277376, 4.055123], ["cg27627381", "chr17", 77456017, 77456019, "SEPT9", "protein_coding", ".", 0.215902, 0.05506, 1.240981, 1.114033, 1.382396, 3.921207, 8.810653e-05, 0.277376, 4.054992], ["cg24397815", "chr1", 50333271, 50333273, ".", ".", "Island", 0.112966, 0.028811, 1.119594, 1.058124, 1.184636, 3.9209, 8.821893e-05, 0.277376, 4.054438], ["cg02320937", "chr11", 582010, 582012, "PHRF1", "protein_coding", ".", 0.204485, 0.052192, 1.226893, 1.107594, 1.359042, 3.917922, 8.931561e-05, 0.277376, 4.049073], ["cg10196861", "chr16", 23182669, 23182671, "SCNN1G", "protein_coding", "Island", 0.189218, 0.048329, 1.208305, 1.099104, 1.328355, 3.915221, 9.032136e-05, 0.277376, 4.04421], ["cg01312394", "chr2", 46882240, 46882242, ".", ".", ".", -0.146166, 0.037335, 0.864014, 0.803047, 0.92961, -3.914961, 9.041882e-05, 0.277376, 4.043741], ["cg03572680", "chr11", 124898657, 124898659, "ROBO4", "protein_coding", ".", -0.144881, 0.037016, 0.865125, 0.804583, 0.930223, -3.914007, 9.077703e-05, 0.277376, 4.042024], ["cg21637653", "chr17", 17213452, 17213454, "RP11-45M22.4;MPRIP;FLCN", "protein_coding", ".", -0.280788, 0.071853, 0.755189, 0.655985, 0.869394, -3.907809, 9.313687e-05, 0.282, 4.030878], ["cg23886101", "chr7", 44398616, 44398618, "NUDCD3", "protein_coding", ".", -0.100986, 0.025861, 0.903946, 0.859269, 0.950945, -3.904896, 9.426578e-05, 0.282612, 4.025646], ["cg04544475", "chr17", 58358093, 58358095, "RNF43;BZRAP1-AS1", "antisense;protein_coding", ".", -0.37826, 0.096917, 0.685052, 0.566536, 0.828361, -3.902928, 9.503609e-05, 0.282612, 4.022111]];

    // Define the dt_args
    let dt_args = {"order": []};
    dt_args["data"] = data;

    $(document).ready(function () {
        
        $('#27ae280d-7942-4653-ad88-c00f8edbc871').DataTable(dt_args);
    });
</script>
</div>
</div></div>
</div>
</section>
</section>
</section>
<section id="save-top-probes">
<h2><a class="toc-backref" href="#id25" role="doc-backlink">Save top probes</a><a class="headerlink" href="#save-top-probes" title="Permalink to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Select only significant probes</span>
<span class="n">ewas_top_cpgs_os</span> <span class="o">=</span> <span class="n">df_batchcorrected</span><span class="p">[</span><span class="n">top_os</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_list</span><span class="p">()]</span>
<span class="n">ewas_top_cpgs_efs</span> <span class="o">=</span> <span class="n">df_batchcorrected</span><span class="p">[</span><span class="n">top_efs</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_list</span><span class="p">()]</span>

<span class="c1"># Save x to pickle in output_path</span>
<span class="n">ewas_top_cpgs_os</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_cpgs_os.csv&#39;</span><span class="p">)</span>
<span class="n">ewas_top_cpgs_efs</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_cpgs_efs&#39;</span><span class="p">)</span>

<span class="c1"># Print a statement saying that you have saved the data, the shape and the path</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully saved the data in &#39;</span> <span class="o">+</span> <span class="n">output_path</span> <span class="o">+</span> <span class="s1">&#39;ewas_dmr/ewas_top_x.csv&#39;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;The shape of the data is &#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">ewas_top_cpgs_os</span><span class="o">.</span><span class="n">shape</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Succesfully saved the data in /mnt/d/MethylScore/Processed_Data/ewas_dmr/ewas_top_x.csv
The shape of the data is (940, 189)
</pre></div>
</div>
</div>
</div>
</section>
<section id="watermark">
<h2><a class="toc-backref" href="#id26" role="doc-backlink">Watermark</a><a class="headerlink" href="#watermark" title="Permalink to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Python implementation: CPython
Python version       : 3.8.16
IPython version      : 8.12.2

methylize: 1.1.1
pandas   : 2.0.2

Compiler    : GCC 11.3.0
OS          : Linux
Release     : 5.15.90.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 20
Architecture: 64bit
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./models"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="../data-processing/reclassification_strategy.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Reclassification to Current Clinical Guidelines</p>
      </div>
    </a>
    <a class="right-next"
       href="3.1_MethylScoreAML_Px_Development_v5.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">MethylScoreAML Px Development</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-data">Load data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-samples-based-on-certain-clinical-features">Remove samples based on certain clinical features</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-samples-from-aaml1031-0531-and-03p1-clinical-trials">Select samples from AAML1031, 0531, and 03P1 clinical trials</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#select-diagnostic-samples-only">Select diagnostic samples only</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#remove-duplicate-samples">Remove duplicate samples</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#match-samples-in-clinical-data-to-samples-in-methylation-data">Match samples in clinical data to samples in methylation data</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#transform-to-m-values">Transform to M-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#perform-batch-correction">Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-dataframe-for-other-analyses">Save dataframe for other analyses</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-coxph-ewas">Run CoxPH-EWAS</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#import-custom-functions-adapted-from-methylize">Import custom functions adapted from <code class="docutils literal notranslate"><span class="pre">methylize</span></code></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-survival-previous-risk-group">Overall survival - previous risk group</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-survival-updated-rr-risk-group">Overall survival - updated RR risk group</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#event-free-survival-previous-risk-group">Event-free survival - previous risk group</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#event-free-survival-updated-risk-group">Event-free survival - updated risk group</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#select-statistically-significant-cpgs">Select statistically significant CpGs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#threshold-based-on-suggested-significance-p-values-1-10e-6">Threshold based on suggested-significance p-values (1*10e-6)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#threshold-based-on-suggested-significance-p-values-1-10e-5">Threshold based on suggested-significance p-values (1*10e-5)</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#fetch-genes">Fetch genes</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#os-suggested-significance-probes">OS suggested-significance probes:</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#efs-suggested-significance-probes">EFS suggested-significance probes:</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#save-top-probes">Save top probes</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Marchi et al., 2023. Lamba Lab @ University of Florida. Please do not share this notebook publicly.
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>