# Specific Aims

Pediatric Acute Myeloid Leukemia (AML) is a devastating disease associated with high morbidity and mortality. Though the development of precision medicine efforts such as small molecule inhibitors and immunotherapy have improved survivorship, the results have been underwhelming, with 5-year survival rates for young patients at 67%. Decades of fundamental research have posed leukemic stem cells as highly heterogeneous, elusive, and ever evolving, so translating that many layers of complexity into clinical practice has been a monumental challenge. Disease identification and risk stratification are foundational pillars of cancer treatment regimen and systematically inform therapy decisions. In AML, studies in adults have shown that recurrent mutations, cytogenetic lesions, and deregulated DNA methylation not only predict treatment outcome but also hold potential value in refining patient classification, in turn informing treatment decisions. Unlike adult AML, however, pediatric AML shows an intriguing genomic stability with lower abundance of oncogenic mutations that are characteristic of adult disease. Despite these differences, compromised epigenomic integrity still hallmarks the disease in the young and the old, which begs for more in-depth scrutiny.
Seminal work in recent years utilizing mRNA-driven machine learning approaches have resulted in the development of leukemic stem cell predictive scores named LSC17 and LSC6, which hold substantial prognostic and clinical value. These have highlighted the need to classify leukemia by its stemness, as well as pointed to DNMT3B (de-novo methyltransferase) as the most prominent gene in defining lethality of disease when highly expressed. Previous work from our group have also shown the prognostic value of DNMT3B as well genome-wide methylation burden in predicting outcome in AML. This is surprising given that DNMT3A mutations that are common in adults are nonexistent in pediatric AML, suggesting that DNA-methylation landscape of pediatric patients may be driven by different mechanisms than that of adults. Such results synergize with the genomic stability, and epigenomic instability seen in many pediatric cancers. Given the success of transcriptomics-based scores in AML and the potential evidence of deregulated DNA methylation in pediatric AML, we propose to leverage machine learning tools to develop DNA methylation-based signatures of clinical utility in pediatric AML.

- __Aim 1: Develop and validate a robust methylation-based prognostic risk-score predictive of outcome in pediatric AML using machine learning approaches.__
  - Aim 1A: We will use genome-wide methylation data (Infinium MethylationEPIC) from Children’s Oncology Group’s (COG) (AAML0531, AAML1031, n=1024) to develop a prognostic score (MethylScore) predictive of EFS using learning algorithms. The score will be utilized for patient classification and will be further tested for association with other outcome endpoints, such as EFS, RR, MRD1.
  - Aim 1B: We will validate MethylScore scores in independent cohorts, such as the Multi-Site St. Jude AML02 (n=164), AML08 (n=42), and the Japanese AML-05 trial (n=55).

- __Aim 2: Integrate DNA methylation score with other known prognostic features for holistic and comprehensive evaluation of AML patients.__
  - Aim 2A: Integrate MethylScore with cytogenetic and molecular classifications as well as with the newly established transcriptomic based pLSC6 score to provide a robust classification system and enhance the predictive power by combing all the known key players in AML 02 trial.
  - Aim 2B: Validate the sensitivity and specificity of the integrated me-Cytomol-LSC6 scores by testing it in independent clinical trials listed in Aim 2A.

- __Aim 3: Develop methods for rapid clinical translation of the score for moving precision medicine in AML.__
  - Aim 3A:  Validate score through a next-generation sequencing platform, such as the Oxford Nanopore’s MinION device, for establishing a rapid turnaround time to accelerate its clinical translation. We will use samples form UF Leukemia Bank and from AML08 trial for this.
  - Aim 3B: Develop a web-based tool for easy classification of patients with respect to different prognostic scores.

We anticipate that our proposed model will reach clinical-grade sensitivity and specificity, overcoming current limitations and unveiling a novel methodology for AML characterization and risk-stratification.
