
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Discovery Dataset &#8212; Acute Leukemia Methylome Atlas Study</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css?v=564be945" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'v1/1_discovery_dataset';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Validation Dataset" href="2_validation_dataset_v2.html" />
    <link rel="prev" title="Data Visualization" href="../source/6_plots.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../index.html">
  
  
  
  
  
  
    <p class="title logo__title">Acute Leukemia Methylome Atlas Study</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../index.html">
                    Overview
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Latest</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../source/1_discovery.html">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../source/2_validation.html">Validation Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../source/3_test.html">Test Dataset (Work in Progress)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../source/4_clinical_data.html">Clinical Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../source/5_signature.html">37-CpG Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="../source/5_classifiers.html">Classifiers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../source/6_plots.html">Data Visualization</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Previous 2022-2024</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Discovery Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_validation_dataset_v2.html">Validation Dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_reclassification_strategy.html">Dx and Px Reclassification</a></li>
<li class="toctree-l1"><a class="reference internal" href="4_models.html">Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_plots.html">Data Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="6_test_dataset.html">Preprocess long reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="7_test_models.html">Run epigenomic models</a></li>
<li class="toctree-l1"><a class="reference internal" href="8_test_atlas_1852.html">UF HemBank 1852 case study</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/ewasCox_vs_pacmapLGBM.html">Px Models Benchmarks</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/5_plots_6-17-24.html">AML-only map with new interactive prognostic and race histograms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/2_Univariate_CoxPH-EWAS_v4.html">EWAS Runs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v5.html">924pts-OS-35CpGs-ASH23</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v6.html">940pts-OS-48CpGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.1_MethylScoreAML_Px_Development_v7.html">940pts-EFS-55CpGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v2.html">OS-35CpGs-Validation-ASH23</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v3.html">OS-48CpGs-Validation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tests/3.2_MethylScoreAML_Px_Validation_Peds_v4.html">EFS-55CpGs-Validation</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/v1/1_discovery_dataset.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Discovery Dataset</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#discovery-train-dataset-sources">Discovery (train) dataset sources</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-1-process-raw-data-to-get-methylation-beta-values">Step 1. Process raw data to get methylation beta values</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-merge-dataframes">Load and Merge Dataframes</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-2-remove-suboptimal-probes">Step 2. Remove suboptimal probes</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-3-remove-sex-linked-probes">Step 3. Remove sex-linked probes</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-missing-values">Evaluate Missing Values</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-4-remove-non-hematopoietic-samples">Step 4. Remove non-hematopoietic samples</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-5-exclude-samples-in-which-20-of-probes-have-failed-p-values">Step 5. Exclude samples in which &gt;20% of probes have failed p-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-6-exclude-cpg-probes-that-are-missing-in-5-of-samples">Step 6. Exclude CpG probes that are missing in &gt;5% of samples</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Evaluate Missing Values</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-7-impute-remaining-missing-values">Step 7. Impute remaining missing values</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Evaluate Missing Values</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-8-remove-outliers-by-pca">Step 8. Remove outliers by PCA</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-dataset-without-batch-correction">Save dataset without batch correction</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-9-perform-batch-correction">Step 9. Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-batch-corrected-dataset">Save batch-corrected dataset</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-final-sample-size-by-batch">Evaluate final sample size by batch</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#appendix-1-how-to-download-data-from-geo-or-gdc">Appendix 1. How to download data from GEO or GDC</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-expression-omnibus-geo">Gene Expression Omnibus (GEO)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomic-data-commons-gdc">Genomic Data Commons (GDC)</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#appendix-2-how-to-create-this-electronic-notebook">Appendix 2. How to create this electronic notebook</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="discovery-dataset">
<h1>Discovery Dataset<a class="headerlink" href="#discovery-dataset" title="Link to this heading">#</a></h1>
<section id="discovery-train-dataset-sources">
<h2>Discovery (train) dataset sources<a class="headerlink" href="#discovery-train-dataset-sources" title="Link to this heading">#</a></h2>
<div class="pst-scrollable-table-container"><table class="table">
<thead>
<tr class="row-odd"><th class="head"><p>Dataset</p></th>
<th class="head"><p>Reference</p></th>
<th class="head"><p>Disease</p></th>
<th class="head"><p>Raw Data Source</p></th>
<th class="head"><p>Download Sample Size</p></th>
<th class="head"><p>Population</p></th>
<th class="head"><p>Platform</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>COG_AAML1031</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/36815378/">Bertrums et al., 2023</a></p></td>
<td><p>AML</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE190931">GSE190931</a></p></td>
<td><p>1048</p></td>
<td><p>Pediatric</p></td>
<td><p>EPIC</p></td>
</tr>
<tr class="row-odd"><td><p>COG_AAML0531</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/34793513/">Bolouri et al., 2021</a></p></td>
<td><p>AML</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE124413">GSE124413</a></p></td>
<td><p>500</p></td>
<td><p>Pediatric</p></td>
<td><p>EPIC</p></td>
</tr>
<tr class="row-even"><td><p>Japanese AML-05</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/35008106/">Yamato et al., 2022</a></p></td>
<td><p>AML</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133986">GSE133986</a></p></td>
<td><p>64</p></td>
<td><p>Pediatric</p></td>
<td><p>EPIC</p></td>
</tr>
<tr class="row-odd"><td><p>AML_TARGET-450k</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/29227476/">Bolouri et al., 2018</a></p></td>
<td><p>AML</p></td>
<td><p><a class="reference external" href="https://portal.gdc.cancer.gov/repository?facetTab=cases&amp;amp;filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.program.name%22%2C%22value%22%3A%5B%22TARGET%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22TARGET-AML%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_format%22%2C%22value%22%3A%5B%22txt%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.platform%22%2C%22value%22%3A%5B%22illumina%20human%20methylation%20450%22%2C%22illumina%20methylation%20epic%22%5D%7D%7D%5D%7D">GDC_TARGET-AML</a></p></td>
<td><p>317</p></td>
<td><p>Pediatric</p></td>
<td><p>450k</p></td>
</tr>
<tr class="row-even"><td><p>AML_TCGA</p></td>
<td><p><a class="reference external" href="https://www.nejm.org/doi/full/10.1056/nejmoa1301689">TCGA, 2013</a></p></td>
<td><p>AML</p></td>
<td><p><a class="reference external" href="https://portal.gdc.cancer.gov/repository?facetTab=cases&amp;amp;filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.program.name%22%2C%22value%22%3A%5B%22TCGA%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22TCGA-LAML%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_format%22%2C%22value%22%3A%5B%22txt%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.platform%22%2C%22value%22%3A%5B%22illumina%20human%20methylation%20450%22%2C%22illumina%20methylation%20epic%22%5D%7D%7D%5D%7D">GDC_TCGA-AML</a></p></td>
<td><p>194</p></td>
<td><p>Adult</p></td>
<td><p>450k</p></td>
</tr>
<tr class="row-odd"><td><p>BeatAML</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/33707228/">Giacopelli et al., 2021</a></p></td>
<td><p>AML</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE159907">GSE159907</a></p></td>
<td><p>316</p></td>
<td><p>Adult</p></td>
<td><p>EPIC</p></td>
</tr>
<tr class="row-even"><td><p>MDS_tAML</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/33446256/">Cabezon et al., 2021</a></p></td>
<td><p>tAML, MDS</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE152710">GSE152710</a></p></td>
<td><p>166</p></td>
<td><p>Adult</p></td>
<td><p>450k</p></td>
</tr>
<tr class="row-odd"><td><p>Nordic_ALL</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/24063430/">Nordlund et al., 2013</a></p></td>
<td><p>ALL</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi">GSE49031</a></p></td>
<td><p>945</p></td>
<td><p>Pediatric</p></td>
<td><p>450k</p></td>
</tr>
<tr class="row-even"><td><p>ALL_TARGET</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30209392/">Alexander et al., 2018</a></p></td>
<td><p>ALL</p></td>
<td><p><a class="reference external" href="https://portal.gdc.cancer.gov/repository?facetTab=files&amp;amp;filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.program.name%22%2C%22value%22%3A%5B%22TARGET%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22TARGET-ALL-P3%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_format%22%2C%22value%22%3A%5B%22txt%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.platform%22%2C%22value%22%3A%5B%22illumina%20human%20methylation%20450%22%2C%22illumina%20methylation%20epic%22%5D%7D%7D%5D%7D&amp;amp;searchTableTab=cases">GDC-TARGET-ALL</a></p></td>
<td><p>102</p></td>
<td><p>Pediatric</p></td>
<td><p>EPIC</p></td>
</tr>
<tr class="row-odd"><td><p>Tcell_ALL_GRAAL</p></td>
<td><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/34039737/">Touzart et al., 2021</a></p></td>
<td><p>ALL</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE147667">GSE147667</a></p></td>
<td><p>155</p></td>
<td><p>Both</p></td>
<td><p>EPIC</p></td>
</tr>
</tbody>
</table>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>These publicly available datasets were selected from GEO and GDC in 2023 based on the following criteria:</p>
<ol class="arabic simple">
<li><p>It must stem from a high-quality clinical trial or a large cohort study.</p></li>
<li><p>It must contain clearly-labeled diagnostic leukemia samples from bone marrow or peripheral blood.</p></li>
<li><p>It must be available as raw methylation data (<em>.idat</em> files) from methylation arrays (450k or EPIC).</p></li>
</ol>
</div>
</section>
<section id="step-1-process-raw-data-to-get-methylation-beta-values">
<h2>Step 1. Process raw data to get methylation beta values<a class="headerlink" href="#step-1-process-raw-data-to-get-methylation-beta-values" title="Link to this heading">#</a></h2>
<p>This step will use SeSAMe (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30085201/">Zhou et al. , 2018</a>) to take signal intensity data (red and green .idat files) and convert them to beta values, which represent the % of methylation for each CpG site.</p>
<p>Process command:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>python -m methylprep -v process -d &lt;directory&gt; --all --batch_size 199
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>From this point on we will move from using <code class="docutils literal notranslate"><span class="pre">python3.7</span></code> to using <code class="docutils literal notranslate"><span class="pre">python3.8</span></code>.</p>
</div>
<section id="where-the-data-at">
<h3>Where the data at?<a class="headerlink" href="#where-the-data-at" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">mount</span> <span class="o">=</span> <span class="s1">&#39;/mnt/e/ALMA/&#39;</span>
<span class="n">input_path_450k</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Raw_Data/Methyl_Array_450k/&#39;</span>
<span class="n">input_path_EPIC</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Raw_Data/Methyl_Array_EPIC/&#39;</span>

<span class="n">zhou2016_probes_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;UnreliableProbesList_Zhou2016/EPIC.anno.GRCh38.tsv&#39;</span>

<span class="n">output_path</span> <span class="o">=</span> <span class="n">mount</span> <span class="o">+</span> <span class="s1">&#39;Intermediate_Files/&#39;</span>
</pre></div>
</div>
</div>
</details>
</div>
</section>
<section id="load-and-merge-dataframes">
<h3>Load and Merge Dataframes<a class="headerlink" href="#load-and-merge-dataframes" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">methylcheck</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">input_paths</span> <span class="o">=</span> <span class="p">[</span>
    <span class="p">(</span><span class="n">input_path_EPIC</span><span class="p">,</span> <span class="s1">&#39;GSE190931&#39;</span><span class="p">),</span>      <span class="c1"># COG_AAML1031</span>
    <span class="p">(</span><span class="n">input_path_EPIC</span><span class="p">,</span> <span class="s1">&#39;GSE124413&#39;</span><span class="p">),</span>      <span class="c1"># COG_AAML0531_03P1</span>
    <span class="p">(</span><span class="n">input_path_EPIC</span><span class="p">,</span> <span class="s1">&#39;GSE133986&#39;</span><span class="p">),</span>      <span class="c1"># Japanese AML-05</span>
    <span class="p">(</span><span class="n">input_path_450k</span><span class="p">,</span> <span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">),</span> <span class="c1"># AML_TARGET-450k</span>
    <span class="p">(</span><span class="n">input_path_450k</span><span class="p">,</span> <span class="s1">&#39;GDC_TCGA-AML&#39;</span><span class="p">),</span>   <span class="c1"># AML_TCGA</span>
    <span class="p">(</span><span class="n">input_path_EPIC</span><span class="p">,</span> <span class="s1">&#39;GSE159907&#39;</span><span class="p">),</span>      <span class="c1"># BeatAML</span>
    <span class="p">(</span><span class="n">input_path_450k</span><span class="p">,</span> <span class="s1">&#39;GSE152710&#39;</span><span class="p">),</span>      <span class="c1"># MDS_tAML</span>
    <span class="p">(</span><span class="n">input_path_450k</span><span class="p">,</span> <span class="s1">&#39;GSE49031&#39;</span><span class="p">),</span>       <span class="c1"># Nordic_ALL</span>
    <span class="p">(</span><span class="n">input_path_EPIC</span><span class="p">,</span> <span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">),</span> <span class="c1"># ALL_TARGET</span>
    <span class="p">(</span><span class="n">input_path_EPIC</span><span class="p">,</span> <span class="s1">&#39;GSE147667&#39;</span><span class="p">)</span>       <span class="c1"># Tcell_ALL_GRAAL</span>
<span class="p">]</span>

<span class="c1"># Load the dataframes from the input paths</span>
<span class="n">df</span> <span class="o">=</span> <span class="p">[</span><span class="n">methylcheck</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">path</span> <span class="o">+</span> <span class="n">name</span><span class="p">,</span> <span class="n">silent</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span> <span class="k">for</span> <span class="n">path</span><span class="p">,</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">input_paths</span><span class="p">]</span>

<span class="c1"># Concatenate the dataframes in the df list</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">keys</span><span class="o">=</span> <span class="p">[</span><span class="n">name</span> <span class="k">for</span> <span class="n">_</span><span class="p">,</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">input_paths</span><span class="p">]</span> <span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39; Dataset (df) contains </span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> rows (5mC sites/probes) and </span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> columns (samples).&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> Dataset (df) contains 452453 rows (5mC sites/probes) and 3845 columns (samples).
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="step-2-remove-suboptimal-probes">
<h2>Step 2. Remove suboptimal probes<a class="headerlink" href="#step-2-remove-suboptimal-probes" title="Link to this heading">#</a></h2>
<p>There are several critera for exclusion of probes: Areas that have polymorphisms, cross-hybridization, repeat sequence elements, or base color changes can affect probe quality. Below are publications that have benchmarked probe quality and have provided lists of probes to exclude:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3592906/">Chen et al., 2013</a></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3740789/">Price et al., 2013</a></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3943510/">Naeem et al., 2014</a></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4659175/">DacaRoszak et al., 2015</a></p></li>
<li><p><a class="reference external" href="https://academic.oup.com/nar/article/45/4/e22/2290930">Zhou et al., 2016</a></p></li>
</ul>
<p>This function removes proves listed as sub-optimal according to <a class="reference external" href="https://academic.oup.com/nar/article/45/4/e22/2290930">Zhou et al., 2016</a>. For the .tsv file containing the annotated probes, download the paper’s supplementary material. See figure 5 of their paper for detailed description.</p>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">exclude_suboptimal_probes</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;This function removes proves listed as sub-optimal according to:</span>
<span class="sd">    </span>
<span class="sd">    Zhou, W., Laird, P. W. &amp; Shen, H.. Comprehensive characterization,</span>
<span class="sd">    annotation and innovative use of Infinium DNA methylation BeadChip probes.</span>
<span class="sd">    Nucleic Acids Research gkw967 (2016).</span>
<span class="sd">    doi:10.1093/nar/gkw967</span>

<span class="sd">    For the .tsv file containing the annotated probes, download the paper&#39;s</span>
<span class="sd">    supplementary material.</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="c1"># Load the list of suboptimal probes</span>
    <span class="n">zhou2016_probes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">zhou2016_probes_path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

    <span class="c1"># Select the probes that are listed as suboptimal</span>
    <span class="n">unreliable_probes</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">zhou2016_probes</span><span class="p">[</span><span class="n">zhou2016_probes</span><span class="p">[</span><span class="s1">&#39;MASK.general&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="kc">True</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>

    <span class="c1"># Remove the unreliable probes from the dataframe</span>
    <span class="n">df_</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="o">~</span><span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">unreliable_probes</span><span class="p">)]</span>
    
    <span class="nb">print</span><span class="p">(</span>
        <span class="sa">f</span><span class="s1">&#39;Removed </span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df_</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> suboptimal probes. </span><span class="si">{</span><span class="n">df_</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> probes remaining.&#39;</span><span class="p">)</span>

    <span class="k">return</span><span class="p">(</span><span class="n">df_</span><span class="p">)</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">exclude_suboptimal_probes</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Removed 47382 suboptimal probes. 405071 probes remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="step-3-remove-sex-linked-probes">
<h2>Step 3. Remove sex-linked probes<a class="headerlink" href="#step-3-remove-sex-linked-probes" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span> <span class="o">=</span> <span class="n">methylcheck</span><span class="o">.</span><span class="n">exclude_sex_control_probes</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s1">&#39;450k&#39;</span><span class="p">,</span> <span class="n">no_sex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">no_control</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>450k: Removed 9570 sex-linked probes from 3845 samples. 395501 probes remaining.
</pre></div>
</div>
</div>
</div>
<section id="evaluate-missing-values">
<h3>Evaluate Missing Values<a class="headerlink" href="#evaluate-missing-values" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">missingno</span> <span class="k">as</span> <span class="nn">msno</span>

<span class="c1"># Plot the missing values</span>
<span class="n">msno</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">20000</span><span class="p">:</span><span class="mi">20050</span><span class="p">,:]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;Axes: &gt;
</pre></div>
</div>
<img alt="../_images/909828ecc78f929f2aa4065d3f2b7083c03e5b38e14b3c1c91324ce5212da7f5.png" src="../_images/909828ecc78f929f2aa4065d3f2b7083c03e5b38e14b3c1c91324ce5212da7f5.png" />
</div>
</div>
</section>
</section>
<section id="step-4-remove-non-hematopoietic-samples">
<h2>Step 4. Remove non-hematopoietic samples<a class="headerlink" href="#step-4-remove-non-hematopoietic-samples" title="Link to this heading">#</a></h2>
<p>These are:</p>
<ul class="simple">
<li><p>GSE49031_colon_cancer_cell_line_samples</p></li>
<li><p>GSE49031_Human_HCT116_DKO_cells</p></li>
<li><p>GSE49031_Amplified_gDNA_controls</p></li>
<li><p>GDC_TARGET_AML_fibroblast_samples</p></li>
<li><p>GSE147667_thymic_cells</p></li>
<li><p>GDC_TARGET_ALL_lymphoid_normal</p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">GSE49031_colon_cancer_cell_line_samples</span> <span class="o">=</span>  <span class="p">[(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5723662002_R06C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5723662025_R06C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5854945004_R04C01&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5854945004_R06C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5859501021_R04C01&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5859501021_R06C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5975819012_R05C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5975819012_R06C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;6005574026_R05C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;6005574026_R06C02&#39;</span><span class="p">),</span>
                                            <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;6929726092_R06C02&#39;</span><span class="p">)]</span>

<span class="n">GSE49031_Human_HCT116_DKO_cells</span> <span class="o">=</span> <span class="p">[(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5723662002_R06C02&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5723662025_R06C02&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5854945004_R06C02&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5859501021_R06C02&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5975819012_R06C02&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;6005574026_R06C02&#39;</span><span class="p">)]</span>

<span class="n">GSE49031_Amplified_gDNA_controls</span> <span class="o">=</span> <span class="p">[(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5854945004_R04C01&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5859501021_R04C01&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;5975819012_R05C02&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;6005574026_R05C02&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GSE49031&#39;</span><span class="p">,</span> <span class="s1">&#39;6929726092_R06C02&#39;</span><span class="p">)]</span>

<span class="n">GDC_TARGET_AML_fibroblast_samples</span> <span class="o">=</span> <span class="p">[(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;1a602aab-e9ca-424d-af25-700cc2b27ae5_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;945418a6-95eb-41ee-b8a7-d5aa06fc955e_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;c396b6bc-37d4-41bc-811c-1fc7c2e74f92_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;9e5e437c-e6e0-48ad-8dc4-0b796060f959_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;c2d806c0-787b-41df-a6f7-818002a2cb4f_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;5faf4ea1-953b-4832-81d3-9baa6e19518d_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;55022a0a-77e0-4d6f-a3d9-7298bf77d8f0_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;afc86206-9551-4736-ac8d-b046a2bc3723_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;4a8d583d-5bec-42be-8dd4-125046f6930c_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;019ed3aa-08d5-460e-9acb-6077e14afe94_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;be873ff1-60fd-4baf-8233-02b263888459_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;e8249779-ed07-433b-bdfe-372ab353c265_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;1ce5d77b-d738-4660-ad20-350fbad6bf59_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;a3199fe8-d674-4eae-ba87-22676ec2fea3_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;d7c8504b-18b8-4c95-ae31-da8fc69a977a_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;030a10f8-be6c-4396-9159-8bda6eacbc7d_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;1c9a556f-e53d-4e2f-b6ca-8843e281319a_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;3a9977a3-030b-4ae9-9252-42cee8ab0a89_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;76121d4f-1b40-48ce-87ea-e81950072fbe_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;1d0deb92-a781-4e25-bbb9-94f21962d1bc_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;21722d63-93c1-4afa-b7d2-b16147e9615e_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;47a8e28f-18b9-4263-95d5-64d255d216cd_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;3da55883-a2bf-45c8-be26-2340625241ad_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;da60f5d8-7e39-4061-85b6-b2842e0d47e6_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;f9af494b-a87a-4b60-975b-48372e93962c_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;feb61d16-7262-4825-8c3d-6516959315ab_noid&#39;</span><span class="p">),</span>
                                    <span class="p">(</span><span class="s1">&#39;GDC_TARGET-AML&#39;</span><span class="p">,</span> <span class="s1">&#39;f7cb4d3d-63d5-4edb-babe-c2aadd22e5de_noid&#39;</span><span class="p">)]</span>

<span class="n">GSE147667_thymic_cells</span> <span class="o">=</span> <span class="p">[(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;201233650075_R07C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;203724130020_R01C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;201233650075_R08C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;203724130020_R02C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;201233650023_R05C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;203724130020_R05C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;201233650023_R06C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;203724130020_R06C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;201233650023_R01C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;203724130020_R03C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;201233650023_R02C01&#39;</span><span class="p">),</span>
                          <span class="p">(</span><span class="s1">&#39;GSE147667&#39;</span><span class="p">,</span> <span class="s1">&#39;203724130020_R04C01&#39;</span><span class="p">)]</span>

<span class="n">GDC_TARGET_ALL_lymphoid_normal</span> <span class="o">=</span> <span class="p">[(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;baa6556c-5041-40f7-b478-733e25bbcfae_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;e4889d0c-96dd-470f-b2c2-8fc120b4bed7_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;8ab6e84b-143e-4dc3-8339-f8ad30d646d1_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;3c73ab5f-1840-4dac-a678-959d5f57c92c_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;f21ea1c9-128c-4b59-ae4f-4d0be6e67759_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;6d1f3396-0dd7-45dc-adcb-405ca4ec8a78_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;57de5be1-4200-48dd-a31f-b38a32c23cda_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;ad7cf92d-95ec-4f76-8377-619ad796b492_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;97af751d-0ef9-4d14-ae67-14319985ca77_noid&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="s1">&#39;GDC_TARGET-ALL&#39;</span><span class="p">,</span> <span class="s1">&#39;dd96e3d0-a4b5-4769-a65a-00e9996f4526_noid&#39;</span><span class="p">),]</span>

<span class="c1"># Drop all samples selected above</span>
<span class="n">columns_to_drop</span> <span class="o">=</span> <span class="n">GSE49031_colon_cancer_cell_line_samples</span> <span class="o">+</span>\
                  <span class="n">GSE49031_Human_HCT116_DKO_cells</span> <span class="o">+</span>\
                  <span class="n">GSE49031_Amplified_gDNA_controls</span> <span class="o">+</span>\
                  <span class="n">GDC_TARGET_AML_fibroblast_samples</span> <span class="o">+</span>\
                  <span class="n">GSE147667_thymic_cells</span> <span class="o">+</span>\
                  <span class="n">GDC_TARGET_ALL_lymphoid_normal</span>

<span class="k">def</span> <span class="nf">remove_duplicates</span><span class="p">(</span><span class="n">input_list</span><span class="p">):</span>
    <span class="k">return</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">input_list</span><span class="p">))</span>

<span class="n">columns_to_drop</span> <span class="o">=</span> <span class="n">remove_duplicates</span><span class="p">(</span><span class="n">columns_to_drop</span><span class="p">)</span> 

<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">columns_to_drop</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">columns_to_drop</span><span class="p">)</span><span class="si">}</span><span class="s1"> non-hematopoietic samples removed. </span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>60 non-hematopoietic samples removed. 3785 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="step-5-exclude-samples-in-which-20-of-probes-have-failed-p-values">
<h2>Step 5. Exclude samples in which &gt;20% of probes have failed p-values<a class="headerlink" href="#step-5-exclude-samples-in-which-20-of-probes-have-failed-p-values" title="Link to this heading">#</a></h2>
<p>A strict metric implemented by Illumina quality control process is <code class="docutils literal notranslate"><span class="pre">FAIL</span> <span class="pre">by</span> <span class="pre">pval</span></code>, which happens if, for a given sample, the detection p-value is &gt;0.05 in &gt;20% of probes.</p>
<p>Recall that detection p-values measure how likely it is that signals are background fluorescence. There are a few methods for calculating these:  <code class="docutils literal notranslate"><span class="pre">SeSAMe</span></code> and <code class="docutils literal notranslate"><span class="pre">methylprep</span></code> implement pOOBAH, which stands for P-value Out Of Band (OOB) probes for Array Hybridization. For more, see SeSAMe’s paper in <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30085201/">Zhou et al. , 2018</a>.</p>
<p>In other words, we will exclude samples that Illumina QC categorizes as FAIL(pval) for meeting the condition: (pOOBAH &gt; 0.05) &gt;20% probes. Here, failed probes are listed as <code class="docutils literal notranslate"><span class="pre">NaN</span></code>, so we will count the number of <code class="docutils literal notranslate"><span class="pre">NaN</span></code> values in each sample and exclude samples that have more than 20% <code class="docutils literal notranslate"><span class="pre">NaN</span></code> values.</p>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">exclude_failed_samples</span><span class="p">(</span><span class="n">df2</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;This function removes samples that have more than 20% NaN values.&#39;&#39;&#39;</span>

    <span class="c1"># Calculate the number of NaN values in each sample</span>
    <span class="n">nan_count</span> <span class="o">=</span> <span class="n">df2</span><span class="o">.</span><span class="n">isna</span><span class="p">()</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>

    <span class="c1"># Calculate the total number of probes (rows) in the DataFrame</span>
    <span class="n">total_probes</span> <span class="o">=</span> <span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>

    <span class="c1"># Calculate the percentage of NaN values for each sample</span>
    <span class="n">nan_percentage</span> <span class="o">=</span> <span class="p">(</span><span class="n">nan_count</span> <span class="o">/</span> <span class="n">total_probes</span><span class="p">)</span> <span class="o">*</span> <span class="mi">100</span>

    <span class="c1"># Identify samples that meet the condition of having more than 20% NaN values</span>
    <span class="n">samples_to_exclude</span> <span class="o">=</span> <span class="n">nan_percentage</span><span class="p">[</span><span class="n">nan_percentage</span> <span class="o">&gt;</span> <span class="mi">20</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>

    <span class="c1"># Exclude samples that meet the condition from the DataFrame</span>
    <span class="n">filtered_df2</span> <span class="o">=</span> <span class="n">df2</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">samples_to_exclude</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
    
    <span class="c1"># Print the number of samples before and after filtering</span>
    <span class="nb">print</span><span class="p">(</span>
    <span class="sa">f</span><span class="s2">&quot;Removed </span><span class="si">{</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="n">filtered_df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2"> samples (</span><span class="si">{</span><span class="nb">round</span><span class="p">((</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="n">filtered_df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span><span class="o">/</span><span class="n">df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">*</span><span class="mi">100</span><span class="p">,</span><span class="mi">2</span><span class="p">)</span><span class="si">}</span><span class="s2">%). </span><span class="si">{</span><span class="n">filtered_df2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2"> samples remaining.&quot;</span>
    <span class="p">)</span>
    
    <span class="k">return</span> <span class="n">filtered_df2</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">exclude_failed_samples</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Removed 466 samples (12.31%). 3319 samples remaining.
</pre></div>
</div>
</div>
</div>
</section>
<section id="step-6-exclude-cpg-probes-that-are-missing-in-5-of-samples">
<h2>Step 6. Exclude CpG probes that are missing in &gt;5% of samples<a class="headerlink" href="#step-6-exclude-cpg-probes-that-are-missing-in-5-of-samples" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">probe_cutoff</span><span class="p">(</span><span class="n">qc_betas</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
    <span class="n">qc_betas2</span> <span class="o">=</span> <span class="n">qc_betas</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">thresh</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">threshold</span><span class="o">*</span><span class="n">qc_betas</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">qc_betas</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">qc_betas2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> probes removed. </span><span class="si">{</span><span class="n">qc_betas2</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1"> probes remaining.&#39;</span><span class="p">)</span>

    <span class="k">return</span><span class="p">(</span><span class="n">qc_betas2</span><span class="p">)</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">probe_cutoff</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">threshold</span><span class="o">=</span><span class="mf">0.95</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>62443 probes removed. 333058 probes remaining.
</pre></div>
</div>
</div>
</div>
<section id="id1">
<h3>Evaluate Missing Values<a class="headerlink" href="#id1" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Plot the missing values</span>
<span class="n">msno</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">20000</span><span class="p">:</span><span class="mi">20050</span><span class="p">,:]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;Axes: &gt;
</pre></div>
</div>
<img alt="../_images/44615fd915f351ea516545c6c1194c45d2a910f9fd1c2ce1b2d7a3d9c020bf57.png" src="../_images/44615fd915f351ea516545c6c1194c45d2a910f9fd1c2ce1b2d7a3d9c020bf57.png" />
</div>
</div>
</section>
</section>
<section id="step-7-impute-remaining-missing-values">
<h2>Step 7. Impute remaining missing values<a class="headerlink" href="#step-7-impute-remaining-missing-values" title="Link to this heading">#</a></h2>
<ul>
<li><p><strong>Method</strong>: Mean by batch (it replaces each missing value by averaging all the known values for that CpG across samples within their batch).</p></li>
<li><p><strong>Simple mean is recommended for methylation β-values</strong>:</p>
<p><em>In conclusion, the consolidated and manufacturer encouraged practice to use β-value seems appropriate for DNA methylation data imputation. The choice of the best imputation method is somewhat more subtle and depends essentially on the available computational resources and the amount of missing values. Independently of the expected missingness mechanisms, regression-based methods provide on average more accurate estimates of the missing values. However, imputations with regression methods in the presence of limited computational resources can be a rather challenging task. In such cases, the simple mean approach can surprisingly be a better choice than more sophisticated methods</em> (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/32600298/">Lena et al., 2020</a>).</p>
</li>
<li><p><strong>This step inevitably introduces bias</strong>: More sophisticated methods, however, also do. Benchmarks done by the paper above and others reported that the increase in error across methods is roughly similar (<a class="reference external" href="https://scikit-learn.org/stable/auto_examples/impute/plot_missing_values.html#sphx-glr-auto-examples-impute-plot-missing-values-py">see here</a>), which places <code class="docutils literal notranslate"><span class="pre">imputation</span> <span class="pre">by</span> <span class="pre">mean</span></code> as an equivalently error-prone method compared to others but vastly simpler and faster.</p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="k">def</span> <span class="nf">impute_with_mean_of_batch</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;This function imputes missing values with the mean of the batch the sample belongs to.&#39;&#39;&#39;</span>

    <span class="c1"># Separate samples based on first level of the multi-index and fill NaNs with mean</span>
    <span class="n">df_dict</span> <span class="o">=</span> <span class="p">{</span><span class="n">key</span><span class="p">:</span> <span class="n">df</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">))</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span> <span class="k">for</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">levels</span><span class="p">[</span><span class="mi">0</span><span class="p">]}</span>
    
    <span class="c1"># Merge the dataframes in multi-index format</span>
    <span class="n">df2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">df_dict</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">join</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">)</span>
    
    <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;NaN values filled by the mean probe value of the batch.&#39;</span><span class="p">)</span>
    
    <span class="k">return</span> <span class="n">df2</span>

<span class="c1"># Impute missing values</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">impute_with_mean_of_batch</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>NaN values filled by the mean probe value of the batch.
</pre></div>
</div>
</div>
</div>
<section id="id2">
<h3>Evaluate Missing Values<a class="headerlink" href="#id2" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Plot the missing values</span>
<span class="n">msno</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">20000</span><span class="p">:</span><span class="mi">20050</span><span class="p">,:]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">))</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;Axes: &gt;
</pre></div>
</div>
<img alt="../_images/ad6a068b2109174ef9209bad5aa1c978919ae5ecdfc307d0b8220113bae58372.png" src="../_images/ad6a068b2109174ef9209bad5aa1c978919ae5ecdfc307d0b8220113bae58372.png" />
</div>
</div>
</section>
</section>
<section id="step-8-remove-outliers-by-pca">
<h2>Step 8. Remove outliers by PCA<a class="headerlink" href="#step-8-remove-outliers-by-pca" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">sklearn.preprocessing</span> <span class="kn">import</span> <span class="n">StandardScaler</span>
<span class="kn">from</span> <span class="nn">sklearn.decomposition</span> <span class="kn">import</span> <span class="n">PCA</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>

<span class="c1"># Define functions</span>

<span class="k">def</span> <span class="nf">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Standardizes the data.&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="n">StandardScaler</span><span class="p">()</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">perform_pca</span><span class="p">(</span><span class="n">std_data</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Performs PCA on standardized data.&#39;&#39;&#39;</span>
    <span class="n">pca</span> <span class="o">=</span> <span class="n">PCA</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="mi">42</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">pca</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">std_data</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">create_principal_df</span><span class="p">(</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">df</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Creates a DataFrame with the PCA data.&#39;&#39;&#39;</span>
    <span class="n">principal_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">pca_data</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="s1">&#39;PC2&#39;</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span>
    <span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;hue&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">principal_df</span>

<span class="k">def</span> <span class="nf">filter_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">principal_df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Filters the DataFrame by removing outliers.&#39;&#39;&#39;</span>
    <span class="n">outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">threshold</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">outliers</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="c1"># Execute functions</span>


<span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">principal_df</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df</span><span class="p">)</span>

<span class="c1"># Define threshold to be &gt; 5 standard deviations from the mean</span>
<span class="n">threshold</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">std</span><span class="p">()</span> <span class="o">*</span> <span class="mi">5</span>

<span class="n">non_outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">threshold</span><span class="p">]</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">filter_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">principal_df</span><span class="p">,</span> <span class="n">threshold</span><span class="p">)</span>

<span class="c1"># Plot the PCA data</span>

<span class="k">def</span> <span class="nf">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">,</span> <span class="n">threshold</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Plots PCA data.&#39;&#39;&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">9</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span> <span class="ow">in</span> <span class="p">[(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">ax1</span><span class="p">,</span>
                                     <span class="s1">&#39;PCA containing outliers, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])),</span>
                                    <span class="p">(</span><span class="n">non_outliers</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span>
                                     <span class="s1">&#39;PCA without outliers, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">non_outliers</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]))]:</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PC2&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;hue&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">threshold</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">1.5</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;PC 1&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;PC 2&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">,</span> <span class="n">threshold</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Removed </span><span class="si">{</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> outliers. </span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1"> samples remaining.&#39;</span><span class="p">)</span>

<span class="c1"># List outliers</span>

<span class="n">outliers</span> <span class="o">=</span> <span class="n">principal_df</span><span class="p">[</span><span class="n">principal_df</span><span class="p">[</span><span class="s1">&#39;PC1&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">threshold</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_list</span><span class="p">()</span>
<span class="n">outliers</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/d18a5afc64dad4079b555ba7960d4840a7fafe73f699b333dd367a0465e37c42.png" src="../_images/d18a5afc64dad4079b555ba7960d4840a7fafe73f699b333dd367a0465e37c42.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Removed 11 outliers. 3308 samples remaining.
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[(&#39;GSE190931&#39;, &#39;202905580033_R06C01&#39;),
 (&#39;GSE190931&#39;, &#39;202905580033_R07C01&#39;),
 (&#39;GSE190931&#39;, &#39;202905580033_R08C01&#39;),
 (&#39;GSE190931&#39;, &#39;202908420004_R02C01&#39;),
 (&#39;GSE190931&#39;, &#39;202908420004_R04C01&#39;),
 (&#39;GSE190931&#39;, &#39;202908420004_R05C01&#39;),
 (&#39;GSE190931&#39;, &#39;202908420004_R06C01&#39;),
 (&#39;GSE190931&#39;, &#39;202908420004_R07C01&#39;),
 (&#39;GSE190931&#39;, &#39;202908420004_R08C01&#39;),
 (&#39;GDC_TARGET-AML&#39;, &#39;22d1e9ae-b8e2-4f58-bd06-e8f80484530c_noid&#39;),
 (&#39;GDC_TARGET-AML&#39;, &#39;e77c3816-d002-46ac-a030-49a2060fc54a_noid&#39;)]
</pre></div>
</div>
</div>
</div>
<section id="save-dataset-without-batch-correction">
<h3>Save dataset without batch correction<a class="headerlink" href="#save-dataset-without-batch-correction" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_nobatch</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">level</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>

<span class="n">dataset_title</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df_nobatch</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">samples_</span><span class="si">{</span><span class="n">df_nobatch</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="mi">1</span><span class="si">}</span><span class="s1">cpgs_nobatchcorrection_bvalues.pkl&#39;</span>

<span class="n">df_nobatch</span><span class="o">.</span><span class="n">to_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="n">dataset_title</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Successfuly saved at: </span><span class="si">{</span><span class="n">output_path</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">dataset_title</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Successfuly saved at: /mnt/e/ALMA/Intermediate_Files/3308samples_333058cpgs_nobatchcorrection_bvalues.pkl
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="step-9-perform-batch-correction">
<h2>Step 9. Perform batch correction<a class="headerlink" href="#step-9-perform-batch-correction" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>pyCombat</strong>: a Python tool for batch effects correction in high-throughput molecular data using empirical Bayes methods</p></li>
<li><p><strong>Github</strong>: <a class="reference external" href="https://epigenelabs.github.io/pyComBat/">https://epigenelabs.github.io/pyComBat/</a></p></li>
<li><p><strong>Implementation Preprint</strong>: <a class="reference external" href="https://doi.org/10.1101/2020.03.17.995431">bioRxiv</a></p></li>
<li><p><strong>Original Paper</strong>: <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/16632515/">Adjusting batch effects in microarray expression data using empirical Bayes methods</a></p></li>
</ul>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">combat.pycombat</span> <span class="kn">import</span> <span class="n">pycombat</span>

<span class="c1"># Correct batch effects in the training dataset</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pycombat</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="p">,</span> <span class="n">batch</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">droplevel</span><span class="p">(</span><span class="mi">1</span><span class="p">))</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Succesfully corrected batch effects in the training dataset.&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Found 10 batches.
Adjusting for 0 covariate(s) or covariate level(s).
Standardizing Data across genes.
Fitting L/S model and finding priors.
Finding parametric adjustments.
Adjusting the Data
Succesfully corrected batch effects in the training dataset.
</pre></div>
</div>
</div>
</div>
<section id="evaluate-batch-effects">
<h3>Evaluate batch effects<a class="headerlink" href="#evaluate-batch-effects" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">X_std</span> <span class="o">=</span> <span class="n">standardize_data</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="n">principal_components</span> <span class="o">=</span> <span class="n">perform_pca</span><span class="p">(</span><span class="n">X_std</span><span class="p">)</span>
<span class="n">principal_df</span> <span class="o">=</span> <span class="n">create_principal_df</span><span class="p">(</span><span class="n">principal_components</span><span class="p">,</span> <span class="n">df</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&#39;&#39;&#39;Plots PCA data.&#39;&#39;&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="p">(</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">df</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">,</span> <span class="n">legend</span> <span class="ow">in</span> <span class="p">[(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA after batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">principal_df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="s1">&#39;full&#39;</span><span class="p">),</span>
                                  <span class="p">(</span><span class="n">non_outliers</span><span class="p">,</span> <span class="n">ax1</span><span class="p">,</span>
                                   <span class="s1">&#39;PCA before batch correction, n=&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">non_outliers</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="kc">False</span><span class="p">)]:</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s1">&#39;PC1&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;PC2&#39;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s1">&#39;hue&#39;</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="n">legend</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;PC 1&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;PC 2&#39;</span><span class="p">)</span>

    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.05</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">plot_data</span><span class="p">(</span><span class="n">principal_df</span><span class="p">,</span> <span class="n">non_outliers</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<img alt="../_images/66aa330ce4bdb6e3b84a9c984d88530ad4d4d53c805b396c366a3dcecd88fb1c.png" src="../_images/66aa330ce4bdb6e3b84a9c984d88530ad4d4d53c805b396c366a3dcecd88fb1c.png" />
</div>
</div>
</section>
<section id="save-batch-corrected-dataset">
<h3>Save batch-corrected dataset<a class="headerlink" href="#save-batch-corrected-dataset" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">level</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="s1">&#39;Batch&#39;</span><span class="p">)</span>

<span class="n">dataset_title</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">samples_</span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="mi">1</span><span class="si">}</span><span class="s1">cpgs_withbatchcorrection_bvalues.pkl&#39;</span>

<span class="n">df</span><span class="o">.</span><span class="n">to_pickle</span><span class="p">(</span><span class="n">output_path</span> <span class="o">+</span> <span class="n">dataset_title</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Successfuly saved at: </span><span class="si">{</span><span class="n">output_path</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">dataset_title</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Successfuly saved at: /mnt/e/ALMA/Intermediate_Files/3308samples_333058cpgs_withbatchcorrection_bvalues.pkl
</pre></div>
</div>
</div>
</div>
</section>
<section id="evaluate-final-sample-size-by-batch">
<h3>Evaluate final sample size by batch<a class="headerlink" href="#evaluate-final-sample-size-by-batch" title="Link to this heading">#</a></h3>
<div class="cell tag_hide-input docutils container">
<details class="hide above-input">
<summary aria-label="Toggle hidden content">
<span class="collapsed">Show code cell source</span>
<span class="expanded">Hide code cell source</span>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;Batch&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Batch
GSE49031          933
GSE190931         581
GSE124413         495
GSE159907         316
GDC_TARGET-AML    287
GDC_TCGA-AML      194
GSE152710         166
GSE147667         141
GDC_TARGET-ALL    131
GSE133986          64
Name: count, dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="appendix-1-how-to-download-data-from-geo-or-gdc">
<h2>Appendix 1. How to download data from GEO or GDC<a class="headerlink" href="#appendix-1-how-to-download-data-from-geo-or-gdc" title="Link to this heading">#</a></h2>
<section id="gene-expression-omnibus-geo">
<h3>Gene Expression Omnibus (GEO)<a class="headerlink" href="#gene-expression-omnibus-geo" title="Link to this heading">#</a></h3>
<ol class="arabic">
<li><p>Install Python3.7</p>
<ul class="simple">
<li><p>Go to <a class="reference external" href="https://www.python.org/downloads/release/python-379/">python.org</a> and download python according to your operating system</p></li>
<li><p>During install, click on the box to enable it on PATH</p></li>
</ul>
</li>
<li><p>Open a terminal (Windows: git bash, Linux/Mac: terminal)</p>
<ul class="simple">
<li><p>Write the following commands to install methylprep:</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip<span class="w"> </span>install<span class="w"> </span>--upgrade<span class="w"> </span>pip
pip<span class="w"> </span>install<span class="w"> </span><span class="nv">methylprep</span><span class="o">==</span><span class="m">1</span>.7.1
</pre></div>
</div>
</li>
<li><p>Download dataset of interest directly from GEO:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python<span class="w"> </span>-m<span class="w"> </span>methylprep<span class="w"> </span>beta_bake<span class="w"> </span>-z<span class="w"> </span>-i<span class="w"> </span>&lt;GEO<span class="w"> </span>series<span class="w"> </span>ID&gt;<span class="w"> </span>-d<span class="w"> </span>&lt;directory&gt;<span class="w"> </span>
</pre></div>
</div>
</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><code class="docutils literal notranslate"><span class="pre">methylprep</span></code> API can be found <a class="reference external" href="https://www.life-epigenetics-methylprep.readthedocs-hosted.com/en/latest/docs/cli.html">here</a>.</p>
</div>
</section>
<section id="genomic-data-commons-gdc">
<h3>Genomic Data Commons (GDC)<a class="headerlink" href="#genomic-data-commons-gdc" title="Link to this heading">#</a></h3>
<ol class="arabic">
<li><p>Go to <a class="reference external" href="https://www.portal.gdc.cancer.gov/repository">GDC Repository</a>.</p></li>
<li><p>Select the right filtering criteria for you (or click on one of the links from the table above).</p></li>
<li><p>Once you have a selection of your samples, download the <em>Manisfest</em> file by clicking on the <em>Manifest</em> button.</p></li>
<li><p>Load them samples into your cart, go to cart page and download the <em>Sample Sheet</em>. This step is crucial to link sample IDs to the .idat files.</p></li>
<li><p>To download the data, you will need the GDC Data Transfer Tool (command-line) that you can find <a class="reference external" href="https://www.gdc.cancer.gov/access-data/gdc-data-transfer-tool">here</a>.</p></li>
<li><p>Follow the instructions on the link about how to download the file and use it.</p></li>
<li><p>As an example, on Windows, download the file and extract it to your data folder, then run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">.</span>\<span class="n">gdc</span><span class="o">-</span><span class="n">client</span> <span class="n">download</span> <span class="o">-</span><span class="n">m</span> <span class="o">&lt;</span><span class="n">path</span> <span class="n">to</span> <span class="n">the</span> <span class="n">manisfest</span> <span class="n">file</span> <span class="n">you</span> <span class="n">downloaded</span><span class="o">&gt;.</span><span class="n">txt</span> <span class="o">--</span><span class="n">no</span><span class="o">-</span><span class="n">related</span><span class="o">-</span><span class="n">files</span> <span class="o">--</span><span class="n">no</span><span class="o">-</span><span class="n">annotations</span> <span class="o">--</span><span class="n">latest</span>
</pre></div>
</div>
</li>
<li><p>When the .idat files are downloaded, move them to Linux file system (to be used with WSL). If working on a Mac or native Linux, ignore this step.</p></li>
<li><p>Save the sample sheet file as <em>samplesheet.csv</em> in the directory where the data are located.</p></li>
<li><p>Extract .idat files from folders and remove .parcel files and empty folders:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">find</span> <span class="o">.</span> <span class="o">-</span><span class="n">mindepth</span> <span class="mi">2</span> <span class="o">-</span><span class="nb">type</span> <span class="n">f</span> <span class="o">-</span><span class="n">exec</span> <span class="n">mv</span> <span class="o">-</span><span class="n">t</span> <span class="o">.</span> <span class="o">-</span><span class="n">i</span> <span class="s1">&#39;</span><span class="si">{}</span><span class="s1">&#39;</span> <span class="o">+</span><span class="p">;</span>
<span class="n">find</span> <span class="o">.</span> <span class="o">-</span><span class="nb">type</span> <span class="n">f</span> <span class="o">-</span><span class="n">name</span> <span class="s2">&quot;*.parcel&quot;</span> <span class="o">-</span><span class="n">exec</span> <span class="n">rm</span> <span class="o">-</span><span class="n">f</span> <span class="p">{}</span> \<span class="p">;</span>
<span class="n">find</span> <span class="o">.</span> <span class="o">-</span><span class="n">mindepth</span> <span class="mi">1</span> <span class="o">-</span><span class="nb">type</span> <span class="n">d</span> <span class="o">-</span><span class="n">exec</span> <span class="n">rm</span> <span class="o">-</span><span class="n">rf</span> <span class="p">{}</span> \<span class="p">;</span>
</pre></div>
</div>
</li>
</ol>
<p>Congrats! You are ready to process the files using <code class="docutils literal notranslate"><span class="pre">methylprep</span></code>.</p>
</section>
</section>
<section id="appendix-2-how-to-create-this-electronic-notebook">
<h2>Appendix 2. How to create this electronic notebook<a class="headerlink" href="#appendix-2-how-to-create-this-electronic-notebook" title="Link to this heading">#</a></h2>
<ol class="arabic">
<li><p>Use Jupyter Book</p>
<ul class="simple">
<li><p>Go to <a class="reference external" href="https://jupyterbook.org">jupyterbook.org</a>.</p></li>
<li><p>Follow the steps on their page.</p></li>
</ul>
</li>
<li><p>Set up <code class="docutils literal notranslate"><span class="pre">toc.yml</span></code> and <code class="docutils literal notranslate"><span class="pre">_config.yml</span></code> files</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">toc.yml</span></code> is the table of contents file. It contains the structure of the book.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">_config.yml</span></code> is the configuration file. It contains the configuration of the book.</p></li>
</ul>
</li>
<li><p>Build the book</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>jb<span class="w"> </span>build<span class="w"> </span>.
</pre></div>
</div>
</li>
<li><p>Push to Github</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>git<span class="w"> </span>add<span class="w"> </span>.
git<span class="w"> </span>commit<span class="w"> </span>-m<span class="w"> </span><span class="s2">&quot;message&quot;</span>
git<span class="w"> </span>push
</pre></div>
</div>
<p>or do it through VSCode GUI.</p>
</li>
<li><p>Push to Github pages using <code class="docutils literal notranslate"><span class="pre">ghp-import</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ghp-import<span class="w"> </span>-n<span class="w"> </span>-p<span class="w"> </span>-f<span class="w"> </span>-o<span class="w"> </span>_build/html
</pre></div>
</div>
</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><code class="docutils literal notranslate"><span class="pre">jupyterbook</span></code> works best on Linux/Mac. If on Windows, use WSL2.</p>
</div>
</section>
<section id="watermark">
<h2>Watermark<a class="headerlink" href="#watermark" title="Link to this heading">#</a></h2>
<div class="cell tag_remove-input docutils container">
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Author: Francisco_Marchi@Lamba_Lab_UF

Python implementation: CPython
Python version       : 3.8.16
IPython version      : 8.12.2

methylcheck: 0.8.5
pandas     : 2.0.2

Compiler    : GCC 11.3.0
OS          : Linux
Release     : 5.15.133.1-microsoft-standard-WSL2
Machine     : x86_64
Processor   : x86_64
CPU cores   : 20
Architecture: 64bit
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./v1"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="../source/6_plots.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Data Visualization</p>
      </div>
    </a>
    <a class="right-next"
       href="2_validation_dataset_v2.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Validation Dataset</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#discovery-train-dataset-sources">Discovery (train) dataset sources</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-1-process-raw-data-to-get-methylation-beta-values">Step 1. Process raw data to get methylation beta values</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#where-the-data-at">Where the data at?</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-and-merge-dataframes">Load and Merge Dataframes</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-2-remove-suboptimal-probes">Step 2. Remove suboptimal probes</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-3-remove-sex-linked-probes">Step 3. Remove sex-linked probes</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-missing-values">Evaluate Missing Values</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-4-remove-non-hematopoietic-samples">Step 4. Remove non-hematopoietic samples</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-5-exclude-samples-in-which-20-of-probes-have-failed-p-values">Step 5. Exclude samples in which &gt;20% of probes have failed p-values</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-6-exclude-cpg-probes-that-are-missing-in-5-of-samples">Step 6. Exclude CpG probes that are missing in &gt;5% of samples</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Evaluate Missing Values</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-7-impute-remaining-missing-values">Step 7. Impute remaining missing values</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Evaluate Missing Values</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-8-remove-outliers-by-pca">Step 8. Remove outliers by PCA</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-dataset-without-batch-correction">Save dataset without batch correction</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-9-perform-batch-correction">Step 9. Perform batch correction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-batch-effects">Evaluate batch effects</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#save-batch-corrected-dataset">Save batch-corrected dataset</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#evaluate-final-sample-size-by-batch">Evaluate final sample size by batch</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#appendix-1-how-to-download-data-from-geo-or-gdc">Appendix 1. How to download data from GEO or GDC</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-expression-omnibus-geo">Gene Expression Omnibus (GEO)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomic-data-commons-gdc">Genomic Data Commons (GDC)</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#appendix-2-how-to-create-this-electronic-notebook">Appendix 2. How to create this electronic notebook</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#watermark">Watermark</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Marchi et al. @ Lamba Lab @ University of Florida.
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>